<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>ASC new sample gene set analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ASC new sample gene set analysis</h1>

</div>


<div id="genome-wide-analysis" class="section level2">
<h2>Genome-wide analysis</h2>
<div id="variant-count-in-ptv-with-pli0.995" class="section level3">
<h3>Variant count in PTV with <code>PLI&gt;=0.995</code></h3>
<table>
<thead>
<tr class="header">
<th align="left">AF cutoff</th>
<th align="center">Trans</th>
<th align="right">Untrans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">&lt;0.000001</td>
<td align="center">659</td>
<td align="right">655</td>
</tr>
<tr class="even">
<td align="left">&lt;0.00001</td>
<td align="center">779</td>
<td align="right">805</td>
</tr>
<tr class="odd">
<td align="left">&lt;0.0001</td>
<td align="center">876</td>
<td align="right">966</td>
</tr>
<tr class="even">
<td align="left">&lt;0.001</td>
<td align="center">1164</td>
<td align="right">1413</td>
</tr>
</tbody>
</table>
<p>When AF &lt;0.1%, there are 1164 transmitted variants and 1413
untransmitted, not consistent with numbers in cell paper 695 vs 557.</p>
<ul>
<li><code>pLI&gt;=0.995</code></li>
<li><code>isPTV==T</code></li>
<li><code>isIndel=="FALSE"</code></li>
<li><code>Gnomad_non_neuro_AF&lt;AF_cutoff</code></li>
<li><code>num_alleles&lt;5</code></li>
</ul>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">filter</th>
<th align="center">Trans</th>
<th align="right">Untrans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">VQSLOD&lt;3.75</td>
<td align="center">255</td>
<td align="right">546</td>
</tr>
<tr class="even">
<td align="left">VQSLOD&lt;5.13</td>
<td align="center">300</td>
<td align="right">624</td>
</tr>
<tr class="odd">
<td align="left">VQSLOD&lt;9</td>
<td align="center">1151</td>
<td align="right">1415</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">818</td>
<td align="right">785</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 10.5 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">714</td>
<td align="right">567</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 11 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">678</td>
<td align="right">526</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -2</td>
<td align="center">956</td>
<td align="right">946</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 5 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">821</td>
<td align="right">804</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 10 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">822</td>
<td align="right">861</td>
</tr>
</tbody>
</table>
<ul>
<li>Kyle suggested: QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</li>
<li>cutoff for <code>AS_ReadPosRankSum</code> could decrease to, for
instance -2, or even lower, while maintaining balanced T/UT, but the OR
consistently stay around 1.01, a bit away from 1.2</li>
<li>OR will increase with increasing cutoff for <code>QD</code>. When
cutoff =10.5, both OR and variant counts are close to numbers in paper,
but when cutoff &gt;10.5, the variant count will become smaller than
than in paper, seems not appropriate.<br />
</li>
<li>With balanced T/UT, increase of cutoff for <code>AS_SOR</code> will
decrease OR, even below 1, becoming depleted<br />
</li>
<li><code>pLI&gt;=0.995</code></li>
<li><code>isPTV==T</code></li>
<li><code>isIndel=="FALSE"</code></li>
<li><code>Gnomad_non_neuro_AF&lt;0.001</code></li>
</ul>
</div>
<div id="plot-of-qd-value" class="section level3">
<h3>plot of QD value</h3>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   0.323  11.300  12.480  12.440  13.560  63.520</code></pre>
<p>use filters</p>
<ul>
<li><code>isIndel=="FALSE"</code></li>
<li><code>AS_SOR &lt;= 3</code></li>
<li><code>AS_ReadPosRankSum &gt;= -0.8</code></li>
</ul>
</div>
<div
id="run-mirage-vs-on-6-variant-groups-in-kyle-data-without-gene-level-information"
class="section level3">
<h3>Run mirage-vs on 6 variant groups in Kyle data without gene level
information</h3>
<ul>
<li>Use mirage-vs to test if there are signals in six variant
groups<br />
</li>
<li>filter <code>Gnomad_non_neuro_AF&lt;0.05</code> was applied</li>
</ul>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.
<span class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for another 3 MPC
groups when running mirage.</p>
<pre><code>## # A tibble: 6 x 7
##   X                      No_case No_contr    OR  pval eta_est  eta_pval
##   &lt;chr&gt;                    &lt;int&gt;    &lt;int&gt; &lt;dbl&gt; &lt;int&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 PLI&gt;=0.995                4436     7317 0.606     1 0.00662 7.98e- 64
## 2 pLI within (0.5,0.995)   10523    14050 0.749     1 0.00342 2.38e- 40
## 3 pLI&lt;0.5                 241695   258839 0.934     1 0.00331 9.61e-122
## 4 MPC&gt;=2                   62501    70221 0.890     1 0.00383 3.47e- 34
## 5 MPC within (1,2)        372944   389959 0.956     1 0.00310 7.05e- 45
## 6 MPC&lt;1                  5085602  5154878 0.987     1 0.00127 1.14e-160</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div id="rare-variants-with-pli0.995" class="section level4">
<h4>rare variants with PLI&gt;=0.995</h4>
</div>
</div>
<div id="gene-set-analysis-taking-into-account-gene-level-information"
class="section level3">
<h3>gene set analysis taking into account gene level information</h3>
<ul>
<li><p>Run mirage on all 17988 genes and <span
class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups and
<span class="math inline">\(\gamma=3\)</span> for 3 MPC groups.</p></li>
<li><p>apply filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
</ul>
<table>
<thead>
<tr class="header">
<th>variant category</th>
<th>parameter</th>
<th>estimate</th>
<th>p value</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>\</td>
<td><span class="math inline">\(\hat{\delta}\)</span></td>
<td>0.0201</td>
<td>0</td>
<td></td>
</tr>
<tr class="even">
<td>(1):PLI&gt;=0.995</td>
<td><span class="math inline">\(\hat{\eta}_1\)</span></td>
<td>0.17</td>
<td><span class="math inline">\(1.72\times 10^{-66}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>(2):0.5&lt;=pLI&lt;0.995</td>
<td><span class="math inline">\(\hat{\eta}_2\)</span></td>
<td>0.04</td>
<td><span class="math inline">\(3.90\times 10^{-39}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>(3):pLI&lt;0.5</td>
<td><span class="math inline">\(\hat{\eta}_3\)</span></td>
<td>0.06</td>
<td><span class="math inline">\(1.55\times 10^{-120}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>(4):MPC&gt;=2</td>
<td><span class="math inline">\(\hat{\eta}_4\)</span></td>
<td>0.06</td>
<td><span class="math inline">\(1.55\times 10^{-36}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>(5):1&lt;=MPC&lt;2</td>
<td><span class="math inline">\(\hat{\eta}_5\)</span></td>
<td>0.12</td>
<td><span class="math inline">\(6.95\times 10^{-57}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>(6):0&lt;=MPC&lt;1</td>
<td><span class="math inline">\(\hat{\eta}_6\)</span></td>
<td>0.05</td>
<td><span class="math inline">\(2.69\times 10^{-182}\)</span></td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Below are top ranked genes.</p>
<pre><code>## # A tibble: 17,987 x 4
##        X Gene          BF post.prob
##    &lt;int&gt; &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;
##  1   536 FHOD3    1.16e39         1
##  2   933 ADCY2    1.24e29         1
##  3  1839 NABP2    7.66e25         1
##  4  4978 IL23R    9.38e18         1
##  5  6465 PRB1     8.70e20         1
##  6  7904 NPIPA1   3.13e45         1
##  7  9990 KIR2DL3  4.34e17         1
##  8 10155 PRAM1    2.51e34         1
##  9 11473 NEFH     3.64e35         1
## 10 13705 HLA-DRB5 1.38e20         1
## # ... with 17,977 more rows</code></pre>
</div>
</div>
<div id="asd-genes-in-cell-paper" class="section level2">
<h2>102 ASD genes in cell paper</h2>
<div id="variant-set-vs-analysis-without-gene-level-information"
class="section level3">
<h3>variant set (VS) analysis without gene level information</h3>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.
<span class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for 3 MPC groups when
running mirage.</p>
<pre><code>## # A tibble: 6 x 7
##   X                  No_case No_contr    OR  pval   eta_est eta_pval
##   &lt;chr&gt;                &lt;int&gt;    &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;
## 1 PLI&gt;=0.995             132      381 0.346 1     0.0000375    1    
## 2 pLI&gt;=0.5&amp;pLI&lt;0.995      45       40 1.12  0.332 0.166        0.324
## 3 pLI&lt;0.5                 32       22 1.45  0.110 0.230        0.269
## 4 MPC&gt;=2                 788      955 0.825 1.00  0.000301     1    
## 5 MPC&gt;=1&amp;MPC&lt;2          5481     5762 0.951 0.996 0.000653     1    
## 6 MPC&lt;1                30680    30991 0.990 0.896 0.000134     1</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="gene-set-analysis-taking-into-account-gene-level-information-1"
class="section level3">
<h3>gene set analysis taking into account gene level information</h3>
<ul>
<li><p><span class="math inline">\(\gamma=6\)</span> for LoF variant and
<span class="math inline">\(\gamma=3\)</span> for missense when running
mirage.</p></li>
<li><p>apply filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
</ul>
<table>
<caption>Parameter estimate by MIRAGE for 102 ASD genes.</caption>
<thead>
<tr class="header">
<th>variant categories</th>
<th>parameter</th>
<th>estimate</th>
<th>p value</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>\</td>
<td><span class="math inline">\(\hat{\delta}\)</span></td>
<td>0.40</td>
<td><span class="math inline">\(1.10\times 10^{-6}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>PLI&gt;=0.995</td>
<td><span class="math inline">\(\hat{\eta}_1\)</span></td>
<td>0.1452</td>
<td><span class="math inline">\(1.13\times 10^{-6}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>0.5&lt;=pLI&lt;0.995</td>
<td><span class="math inline">\(\hat{\eta}_2\)</span></td>
<td>1</td>
<td>0.1546</td>
<td></td>
</tr>
<tr class="even">
<td>pLI&lt;0.5</td>
<td><span class="math inline">\(\hat{\eta}_3\)</span></td>
<td>0.7379</td>
<td>0.2145</td>
<td></td>
</tr>
<tr class="odd">
<td>MPC&gt;=2</td>
<td><span class="math inline">\(\hat{\eta}_4\)</span></td>
<td>0.1789</td>
<td><span class="math inline">\(5.20\times 10^{-4}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>1&lt;=MPC&lt;2</td>
<td><span class="math inline">\(\hat{\eta}_5\)</span></td>
<td><span class="math inline">\(8.13\times 10^{-5}\)</span></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>0&lt;=MPC&lt;1</td>
<td><span class="math inline">\(\hat{\eta}_6\)</span></td>
<td><span class="math inline">\(6.64\times 10^{-12}\)</span></td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
</div>
<div id="top-5-constraint-genes" class="section level2">
<h2>Top 5% constraint genes</h2>
<div id="burden-analysis" class="section level3">
<h3>Burden analysis</h3>
<div id="look-at-all-variants-with-af10e-4" class="section level4">
<h4>Look at all variants with AF&lt;10E-4</h4>
<p>In old sample, we observe signal in constraint genes with <span
class="math inline">\(AF&lt;10^{-4}\)</span>.</p>
<p>There are 199395 transmitted variants in proband and 204963
untransmitted variants.</p>
<pre class="r"><code>poisson.test(c(sum(new_sample_trans_data_Constraint_Gene$Transmitted_proband), sum(new_sample_trans_data_Constraint_Gene$Untransmitted_proband)), c(num.family, num.family), r=1, alternative=&quot;greater&quot;)</code></pre>
<pre><code>## 
##  Comparison of Poisson rates
## 
## data:  c(sum(new_sample_trans_data_Constraint_Gene$Transmitted_proband), sum(new_sample_trans_data_Constraint_Gene$Untransmitted_proband)) time base: c(num.family, num.family)
## count1 = 162760, expected count1 = 159218, p-value &lt; 2.2e-16
## alternative hypothesis: true rate ratio is greater than 1
## 95 percent confidence interval:
##  1.03942     Inf
## sample estimates:
## rate ratio 
##   1.045505</code></pre>
</div>
</div>
<div id="run-mirage-vs-for-all-variants-together"
class="section level3">
<h3>Run MIRAGE-VS for all variants together</h3>
<p>Put all variants with AF&lt;10E-4 in one category and run MIRAGE-VS.
<span class="math inline">\(\widehat{\eta}=0.0041\)</span>, p value is
<span class="math inline">\(1.6\times 10^{-9}\)</span>.</p>
</div>
<div
id="run-mirage-vs-on-constraint-genes-with-6-variant-groups-in-cell-paper-without-gene-level-information"
class="section level3">
<h3>Run MIRAGE-VS on constraint genes with 6 variant groups in cell
paper without gene level information</h3>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.</p>
<ul>
<li><p><span class="math inline">\(\gamma=6\)</span> for 3 PLI variant
groups and <span class="math inline">\(\gamma=3\)</span> for 3 MPC
groups when running mirage.</p></li>
<li><p>apply filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
</ul>
<pre><code>## # A tibble: 6 x 7
##   X                      No_case No_contr    OR  pval   eta_est eta_pval
##   &lt;chr&gt;                    &lt;int&gt;    &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;
## 1 (1):PLI&gt;=0.995             712     1603 0.444  1    0.00568   1.77e-29
## 2 (2):pLI&gt;=0.5&amp;pLI&lt;0.995     645      898 0.718  1.00 0.0000572 1   e+ 0
## 3 (3):pLI&lt;0.5               1976     2694 0.733  1    0.000204  1   e+ 0
## 4 (4):MPC&gt;=2               10068    11577 0.870  1    0.000192  1   e+ 0
## 5 (5):MPC&gt;=1&amp;MPC&lt;2         54752    57131 0.958  1.00 0.00128   1.29e- 6
## 6 (6):MPC&lt;1               157653   160577 0.982  1.00 0.000326  1   e+ 0</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div
id="gene-set-analysis-using-mirage-taking-into-account-gene-level-information"
class="section level3">
<h3>gene set analysis using mirage taking into account gene level
information</h3>
<ul>
<li><span class="math inline">\(\gamma=6\)</span> for PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for MPC groups when
running mirage.</li>
<li>apply filter <code>Gnomad_non_neuro_AF&lt;0.05</code></li>
</ul>
<table>
<thead>
<tr class="header">
<th>variant category</th>
<th>parameter</th>
<th>estimate</th>
<th>p value</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>\</td>
<td><span class="math inline">\(\hat{\delta}\)</span></td>
<td>0.0911</td>
<td>0</td>
<td></td>
</tr>
<tr class="even">
<td>(1):PLI&gt;=0.995</td>
<td><span class="math inline">\(\hat{\eta}_1\)</span></td>
<td>0.1509</td>
<td>0</td>
<td></td>
</tr>
<tr class="odd">
<td>(2):0.5&lt;=pLI&lt;0.995</td>
<td><span class="math inline">\(\hat{\eta}_2\)</span></td>
<td><span class="math inline">\(1.14\times 10^{-48}\)</span></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>(3):pLI&lt;0.5</td>
<td><span class="math inline">\(\hat{\eta}_3\)</span></td>
<td>0.5452</td>
<td><span class="math inline">\(8.72\times 10^{-11}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>(4):MPC&gt;=2</td>
<td><span class="math inline">\(\hat{\eta}_4\)</span></td>
<td><span class="math inline">\(1.03\times 10^{-8}\)</span></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>(5):1&lt;=MPC&lt;2</td>
<td><span class="math inline">\(\hat{\eta}_5\)</span></td>
<td>0.0258</td>
<td><span class="math inline">\(1.75\times 10^{-16}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>(6):0&lt;=MPC&lt;1</td>
<td><span class="math inline">\(\hat{\eta}_6\)</span></td>
<td>0.0263</td>
<td><span class="math inline">\(1.41\times 10^{-10}\)</span></td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>Below are top ranked genes.</p>
<pre><code>## # A tibble: 857 x 4
##        X Gene       BF post.prob
##    &lt;int&gt; &lt;chr&gt;   &lt;dbl&gt;     &lt;dbl&gt;
##  1   267 ADCY2 6.64e38     1    
##  2   123 NF1   1.03e16     1.00 
##  3   334 FMN2  4.37e 9     1.00 
##  4   639 HLA-C 9.33e 7     1.00 
##  5   560 DHX38 1.16e 4     0.999
##  6   640 HLA-B 3.85e 3     0.997
##  7   128 ACLY  1.66e 2     0.943
##  8    84 FRY   4.52e 0     0.312
##  9   565 KAT2A 3.93e 0     0.283
## 10   581 NKPD1 3.92e 0     0.282
## # ... with 847 more rows</code></pre>
<div id="top-genes" class="section level4">
<h4>Top genes</h4>
<p>Those top genes with large BF do not have burden.</p>
<pre><code>## # A tibble: 10 x 5
##    Gene       BF Transmitted_proband Untransmitted_proband    OR
##    &lt;chr&gt;   &lt;dbl&gt;               &lt;dbl&gt;                 &lt;dbl&gt; &lt;dbl&gt;
##  1 ADCY2 6.64e38                6584                  6408 1.03 
##  2 NF1   1.03e16                3807                  3810 0.999
##  3 FMN2  4.37e 9               12902                 13084 0.986
##  4 HLA-C 9.33e 7                  48                   153 0.314
##  5 DHX38 1.16e 4               15778                 15437 1.02 
##  6 HLA-B 3.85e 3                1613                  1911 0.844
##  7 ACLY  1.66e 2                1775                  1761 1.01 
##  8 FRY   4.52e 0               13202                 13351 0.989
##  9 KAT2A 3.93e 0                3572                  3522 1.01 
## 10 NKPD1 3.92e 0                1680                  1617 1.04</code></pre>
</div>
<div id="vairants-driving-the-signal-in-top-gene"
class="section level4">
<h4>Vairants driving the signal in top gene</h4>
<p>variants that drive the BF of top genes. Those variants surviving
after filter <code>AF&lt;5%</code> has close to 10 times untransmitted
variants than transmitted, thereby drive the BF to a high number.</p>
<pre><code>* 148 5:7757570:TGGCCGTGTTCAACATGGTAAGTCCCAGAGCAC:T                                 ADCY2        0.00882                  15              219           1

&gt; BF.var.inte(15, 219, 3, 3, num.family, num.family)
[1] 6.321461e+36

In gene ADCY2, there is one variant with `15      219        1 6.015290e+39              1` driving the signal.

* 17:31232075:A:T  NF1      11      123        5 1.022707e+18              5

* 1:240207775:C:T FMN2      23      130        6 202948206699  6 

* 6:31271622:C:CGGAG HLA-C       4       43        3 3.113611e+02    
6:31271643:A:T HLA-C       3       43        6 1.513167e+03              6

* 16:72109460:GAGCT:G DHX38      35      122        3 4.678945e+04              3
</code></pre>
</div>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
