<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>ASC new sample gene set analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ASC new sample gene set analysis</h1>

</div>


<div id="variant-count-in-ptv-with-pli0.995-at-different-af-cutoffs"
class="section level2">
<h2>Variant count in PTV with <code>PLI&gt;=0.995</code> at different AF
cutoffs</h2>
<table>
<thead>
<tr class="header">
<th align="left">AF cutoff</th>
<th align="center">Trans</th>
<th align="right">Untrans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">&lt;0.000001</td>
<td align="center">659</td>
<td align="right">655</td>
</tr>
<tr class="even">
<td align="left">&lt;0.00001</td>
<td align="center">779</td>
<td align="right">805</td>
</tr>
<tr class="odd">
<td align="left">&lt;0.0001</td>
<td align="center">876</td>
<td align="right">966</td>
</tr>
<tr class="even">
<td align="left">&lt;0.001</td>
<td align="center">1164</td>
<td align="right">1413</td>
</tr>
</tbody>
</table>
<p>When AF &lt;0.1%, there are 1164 transmitted variants and 1413
untransmitted, not consistent with numbers in cell paper 695 vs 557.</p>
<ul>
<li><code>pLI&gt;=0.995</code></li>
<li><code>isPTV==T</code></li>
<li><code>isIndel=="FALSE"</code></li>
<li><code>Gnomad_non_neuro_AF&lt;AF_cutoff</code></li>
<li><code>num_alleles&lt;5</code></li>
</ul>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">filter</th>
<th align="center">Trans</th>
<th align="right">Untrans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">VQSLOD&lt;3.75</td>
<td align="center">255</td>
<td align="right">546</td>
</tr>
<tr class="even">
<td align="left">VQSLOD&lt;5.13</td>
<td align="center">300</td>
<td align="right">624</td>
</tr>
<tr class="odd">
<td align="left">VQSLOD&lt;9</td>
<td align="center">1151</td>
<td align="right">1415</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">818</td>
<td align="right">785</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 6 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">816</td>
<td align="right">726</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">738</td>
<td align="right">604</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 10.5 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">714</td>
<td align="right">567</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 11 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">678</td>
<td align="right">526</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -2</td>
<td align="center">956</td>
<td align="right">946</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 5 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">821</td>
<td align="right">804</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 10 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">822</td>
<td align="right">861</td>
</tr>
</tbody>
</table>
<ul>
<li>Kyle suggested: QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</li>
<li>with filter
<code>QD &gt;= 6 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code>,
0.11 transmitted (816/7291) vs 0.10 untransmitted vps</li>
<li>cutoff for <code>AS_ReadPosRankSum</code> could decrease to, for
instance -2, or even lower, while maintaining balanced T/UT, but the OR
consistently stay around 1.01, a bit away from 1.2</li>
<li>OR will increase with increasing cutoff for <code>QD</code>. When
cutoff =10.5, both OR and variant counts are close to numbers in paper,
but when cutoff &gt;10.5, the variant count will become smaller than
than in paper, seems not appropriate.<br />
</li>
<li>With balanced T/UT, increase of cutoff for <code>AS_SOR</code> will
decrease OR, even below 1, becoming depleted<br />
</li>
<li><code>pLI&gt;=0.995</code></li>
<li><code>isPTV==T</code></li>
<li><code>isIndel=="FALSE"</code></li>
<li><code>Gnomad_non_neuro_AF&lt;0.001</code></li>
</ul>
</div>
<div id="plot-of-qd-value" class="section level2">
<h2>plot of QD value</h2>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   0.323  11.270  12.480  12.418  13.580  46.520</code></pre>
<p>use filters</p>
<ul>
<li><code>isIndel=="FALSE"</code></li>
<li><code>AS_SOR &lt;= 3</code></li>
<li><code>AS_ReadPosRankSum &gt;= -0.8</code></li>
</ul>
</div>
<div id="run-mirage-vs-on-6-variant-groups-in-cell-paper"
class="section level2">
<h2>Run mirage-vs on 6 variant groups in cell paper</h2>
<ul>
<li>Use mirage-vs to test if there are signals in six variant
groups<br />
</li>
<li>filter <code>Gnomad_non_neuro_AF&lt;0.05</code> was applied</li>
<li><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></li>
<li><code>isSyn=="FALSE"</code></li>
</ul>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.
<span class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for another 3 MPC
groups when running mirage_vs. </p>
<pre><code>## # A tibble: 6 x 7
##   X                      No_case No_contr    OR     pval eta_est eta_pval
##   &lt;chr&gt;                    &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;
## 1 PLI&gt;=0.995                1351     1220  1.11 5.17e- 3 0.130   1.62e- 3
## 2 pLI within (0.5,0.995)    2939     2764  1.06 1.06e- 2 0.0643  2.11e- 2
## 3 pLI&lt;0.5                 109275   108331  1.01 2.16e- 2 0.00390 8.92e- 1
## 4 MPC&gt;=2                   48429    47071  1.03 5.64e- 6 0.0345  2.72e- 3
## 5 MPC within (1,2)        325625   319079  1.02 1.80e-16 0.0101  1.72e-11
## 6 MPC&lt;1                  4768875  4745827  1.00 3.96e-14 0.00131 2.49e-28</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="run-mirage-vs-on-180-variant-sets" class="section level2">
<h2>Run MIRAGE-VS on 180 variant sets</h2>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<ul>
<li><p>exclude syn variants <code>isSyn=="FALSE"</code></p></li>
<li><p>6 variant groups X 3 AF categories (0.001&lt;AF&lt;0.01,
0.0001&lt;AF&lt;0.001, AF&lt;0.0001) X 10 gene sets=180 variant
groups</p></li>
<li><p>use filter
<code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
<li><p>solid blue line is FDR 0.05 and dashed red line is Bonferroni
correction p = 0.05/total number of tests.</p></li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<ul>
<li>one outlier with extremely small MIRAGE_Vs p value is gene set
<code>AutismKB_MPC&lt;1_0.0001&lt;AF&lt;0.001</code> with 6802
(transmitted) vs 6914 (untransmitted) on 1901 loci. Burden p value:
0.9839; MIRAGE_VS p value: 5.195326e-15.</li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<ul>
<li><p>variant set
<code>ID gene_PLI&gt;=0.995_0.001&lt;AF&lt;0.01</code>, two loci, 13
(Trans) vs 15 (Nontrans) and 21 vs 8=&gt; 34 vs 23 in total.</p></li>
<li><p>varying <span class="math inline">\(\bar{\gamma}\)</span>
produces quite different p values for MIRAGE_VS.</p></li>
<li><p>blue dashed line: burden test p value of 0.09242524</p></li>
</ul>
</div>
<div id="run-mirage-genomewide" class="section level2">
<h2>Run MIRAGE genomewide</h2>
<ul>
<li><p>Run mirage on all 17850 genes and <span
class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups and
<span class="math inline">\(\gamma=3\)</span> for 3 MPC groups.</p></li>
<li><p>AF&lt;0.05, <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
</ul>
<table>
<thead>
<tr class="header">
<th align="left">variant category</th>
<th align="center">parameter</th>
<th align="center">estimate</th>
<th align="right">p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">\</td>
<td align="center"><span
class="math inline">\(\hat{\delta}\)</span></td>
<td align="center">0.0677</td>
<td align="right"><span class="math inline">\(1.383766e-42\)</span></td>
</tr>
<tr class="even">
<td align="left">(1):PLI&gt;=0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_1\)</span></td>
<td align="center">0.8341</td>
<td align="right"><span class="math inline">\(1.887287e-02\)</span></td>
</tr>
<tr class="odd">
<td align="left">(2):0.5&lt;=pLI&lt;0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_2\)</span></td>
<td align="center">0.8510</td>
<td align="right"><span class="math inline">\(1.236803e-02\)</span></td>
</tr>
<tr class="even">
<td align="left">(3):pLI&lt;0.5</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_3\)</span></td>
<td align="center">0.0292</td>
<td align="right"><span class="math inline">\(6.837139e-01\)</span></td>
</tr>
<tr class="odd">
<td align="left">(4):MPC&gt;=2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_4\)</span></td>
<td align="center">0.4683</td>
<td align="right"><span class="math inline">\(1.576707e-03\)</span></td>
</tr>
<tr class="even">
<td align="left">(5):1&lt;=MPC&lt;2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_5\)</span></td>
<td align="center">0.1136</td>
<td align="right"><span class="math inline">\(1.401804e-12\)</span></td>
</tr>
<tr class="odd">
<td align="left">(6):0&lt;=MPC&lt;1</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_6\)</span></td>
<td align="center">0.0120</td>
<td align="right"><span class="math inline">\(1.471682e-32\)</span></td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>Below are top ranked genes.</p>
<pre><code>## # A tibble: 17,849 x 4
##        X Gene          BF post.prob
##    &lt;int&gt; &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;
##  1 10093 HLA-DRB1 2.13e15     1.00 
##  2  7049 SSC5D    5.48e12     1.00 
##  3 13643 GSTM1    3.10e11     1.00 
##  4  6420 TXNDC2   5.32e 6     1.00 
##  5  7169 PRAM1    1.81e 3     0.992
##  6  6234 ENPP7    2.86e 1     0.675
##  7   828 PTEN     1.47e 1     0.516
##  8   315 SUPT5H   1.42e 1     0.507
##  9   408 MICAL3   1.38e 1     0.500
## 10  5778 DPH1     1.32e 1     0.489
## # ... with 17,839 more rows</code></pre>
<ul>
<li><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5955937/">HLA-DRB1</a></li>
<li><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7697115/">SSC5D</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/33546147/">GSTM1</a></li>
<li><a
href="https://www.nature.com/articles/s41598-021-00123-x">TXNDC2</a></li>
</ul>
</div>
<div id="asd-genes-in-cell-paper" class="section level2">
<h2>102 ASD genes in cell paper</h2>
<div id="variant-set-vs-analysis-by-mirage_vs" class="section level3">
<h3>variant set (VS) analysis by MIRAGE_VS</h3>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.
<span class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for 3 MPC groups when
running mirage_vs.</p>
<pre><code>## # A tibble: 6 x 7
##   X                  No_case No_contr    OR    pval  eta_est eta_pval
##   &lt;chr&gt;                &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 PLI&gt;=0.995              31       20  1.55 0.0804  0.227     0.342  
## 2 pLI&gt;=0.5&amp;pLI&lt;0.995      12        1 12    0.00171 1.00      0.00135
## 3 pLI&lt;0.5                 18       11  1.64 0.132   0.299     0.294  
## 4 MPC&gt;=2                 607      546  1.11 0.0386  0.0681    0.222  
## 5 MPC&gt;=1&amp;MPC&lt;2          4740     4685  1.01 0.289   0.0335    0.0797 
## 6 MPC&lt;1                28977    28532  1.02 0.0321  0.000207  1</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="gene-set-analysis-by-mirage" class="section level3">
<h3>gene set analysis by MIRAGE</h3>
<ul>
<li><p><span class="math inline">\(\gamma=6\)</span> for LoF variant and
<span class="math inline">\(\gamma=3\)</span> for missense when running
mirage.</p></li>
<li><p>filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
</ul>
<table>
<caption>Parameter estimate by MIRAGE for 102 ASD genes.</caption>
<thead>
<tr class="header">
<th align="left">variant category</th>
<th align="center">parameter</th>
<th align="center">estimate</th>
<th align="right">p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">\</td>
<td align="center"><span
class="math inline">\(\hat{\delta}\)</span></td>
<td align="center">0.9998</td>
<td align="right"><span class="math inline">\(0.0295\)</span></td>
</tr>
<tr class="even">
<td align="left">PLI&gt;=0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_1\)</span></td>
<td align="center">0.4321</td>
<td align="right"><span class="math inline">\(0.2334\)</span></td>
</tr>
<tr class="odd">
<td align="left">0.5&lt;=pLI&lt;0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_2\)</span></td>
<td align="center">1</td>
<td align="right">0.0056</td>
</tr>
<tr class="even">
<td align="left">pLI&lt;0.5</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_3\)</span></td>
<td align="center">0.5156</td>
<td align="right">0.2508</td>
</tr>
<tr class="odd">
<td align="left">MPC&gt;=2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_4\)</span></td>
<td align="center">0.1176</td>
<td align="right">0.1685</td>
</tr>
<tr class="even">
<td align="left">1&lt;=MPC&lt;2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_5\)</span></td>
<td align="center">0.0539</td>
<td align="right">0.0729</td>
</tr>
<tr class="odd">
<td align="left">0&lt;=MPC&lt;1</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_6\)</span></td>
<td align="center"><span
class="math inline">\(7.814873e-09\)</span></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
</div>
<div id="top-5-constraint-genes" class="section level2">
<h2>Top 5% constraint genes</h2>
<p>Use below filters and stated otherwise</p>
<ul>
<li><p>filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
</ul>
<div id="burden-analysis-for-variants-in-constrant-genes-with-af10e-4"
class="section level3">
<h3>Burden analysis for variants in constrant genes with
AF&lt;10E-4</h3>
<ul>
<li><p>In old sample, we observe signal in constraint genes with <span
class="math inline">\(AF&lt;10^{-4}\)</span>.</p></li>
<li><p>in new sample, there are 99333 transmitted variants and 94575
untransmitted variants.</p></li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div
id="run-mirage_vs-on-all-variants-in-constraint-genes-with-af10e-4-in-one-group"
class="section level3">
<h3>Run MIRAGE_VS on all variants in constraint genes with AF&lt;10E-4
in one group</h3>
<ul>
<li><p>Put all variants with AF&lt;10E-4 in one category and run
MIRAGE_VS. <span class="math inline">\(\widehat{\eta}=0.0882\)</span>, p
value is <span class="math inline">\(2.585515e-34\)</span>.</p></li>
<li><p>Burden p value vs MIRAGE_VS p value</p></li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
<div id="run-mirage_vs-on-6-variant-groups-in-cell-paper"
class="section level3">
<h3>Run MIRAGE_VS on 6 variant groups in cell paper</h3>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.</p>
<ul>
<li><span class="math inline">\(\gamma=6\)</span> for 3 PLI variant
groups and <span class="math inline">\(\gamma=3\)</span> for 3 MPC
groups when running mirage.</li>
</ul>
<pre><code>## # A tibble: 6 x 7
##   X                      No_case No_contr    OR       pval  eta_est  eta_pval
##   &lt;chr&gt;                    &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;      &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;
## 1 (1):PLI&gt;=0.995             225      178 1.26  0.0109     0.200    0.0150   
## 2 (2):pLI&gt;=0.5&amp;pLI&lt;0.995     174      182 0.956 0.683      0.000342 1        
## 3 (3):pLI&lt;0.5                854      775 1.10  0.0266     0.212    0.00105  
## 4 (4):MPC&gt;=2                7968     7638 1.04  0.00422    0.0730   0.0000137
## 5 (5):MPC&gt;=1&amp;MPC&lt;2         48412    47060 1.03  0.00000614 0.00926  0.0918   
## 6 (6):MPC&lt;1               146671   145348 1.01  0.00721    0.000871 1</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
</div>
<div id="gene-set-analysis-by-mirage-1" class="section level3">
<h3>gene set analysis by MIRAGE</h3>
<ul>
<li><p><span class="math inline">\(\gamma=6\)</span> for PLI variant
groups and <span class="math inline">\(\gamma=3\)</span> for MPC groups
when running mirage.</p></li>
<li><p>836 genes in total</p></li>
</ul>
<table>
<thead>
<tr class="header">
<th align="left">variant category</th>
<th align="center">parameter</th>
<th align="center">estimate</th>
<th align="right">p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">\</td>
<td align="center"><span
class="math inline">\(\hat{\delta}\)</span></td>
<td align="center">0.9832</td>
<td align="right"><span class="math inline">\(2.014043e-06\)</span></td>
</tr>
<tr class="even">
<td align="left">(1):PLI&gt;=0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_1\)</span></td>
<td align="center">0.2036</td>
<td align="right">0.0150</td>
</tr>
<tr class="odd">
<td align="left">(2):0.5&lt;=pLI&lt;0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_2\)</span></td>
<td align="center"><span
class="math inline">\(1.118248e-35\)</span></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">(3):pLI&lt;0.5</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_3\)</span></td>
<td align="center">0.2145</td>
<td align="right">0.0011$</td>
</tr>
<tr class="odd">
<td align="left">(4):MPC&gt;=2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_4\)</span></td>
<td align="center">0.1115</td>
<td align="right">0.0000152777</td>
</tr>
<tr class="even">
<td align="left">(5):1&lt;=MPC&lt;2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_5\)</span></td>
<td align="center">0.0175</td>
<td align="right">0.0559135877</td>
</tr>
<tr class="odd">
<td align="left">(6):0&lt;=MPC&lt;1</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_6\)</span></td>
<td align="center">2.259378e-06</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<div id="top-10-genes-with-burden" class="section level4">
<h4>Top 10 genes with burden</h4>
<ul>
<li>top genes with large BF don’t have heavy burden.</li>
</ul>
<pre><code>## # A tibble: 10 x 5
##    Gene        BF Transmitted_proband Untransmitted_proband    OR
##    &lt;chr&gt;    &lt;dbl&gt;               &lt;dbl&gt;                 &lt;dbl&gt; &lt;dbl&gt;
##  1 LPA       6.23               12960                 13000 0.997
##  2 KIAA1109  3.76                3805                  3710 1.03 
##  3 DNM2      2.91                 167                   142 1.18 
##  4 EIF3I     2.87                  18                    13 1.38 
##  5 ACACA     2.73                 440                   424 1.04 
##  6 NRXN2     2.71                 993                  1057 0.939
##  7 CTNNA2    2.45                 152                   130 1.17 
##  8 ZMIZ1     2.11                1638                  1695 0.966
##  9 SUPT5H    2.09                 294                   288 1.02 
## 10 F12       2.09                 156                   169 0.923</code></pre>
<ul>
<li><p><a
href="https://www.nature.com/articles/s41398-021-01578-2.pdf?proof=t">KIAA1109</a></p></li>
<li><p><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5785978/pdf/krnb-15-01-1391437.pdf">EIF3I</a></p></li>
<li><p><a
href="https://gene.sfari.org/database/human-gene/CTNNA2">CTNNA2</a></p></li>
</ul>
</div>
<div id="vairants-driving-the-signal-in-top-gene"
class="section level4">
<h4>Vairants driving the signal in top gene</h4>
</div>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
