<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>ASC new sample gene set analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.25/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.11.3/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ASC new sample gene set analysis</h1>

</div>


<div id="variant-count-in-ptv-with-pli0.995-at-different-af-cutoffs"
class="section level2">
<h2>Variant count in PTV with <code>PLI&gt;=0.995</code> at different AF
cutoffs</h2>
<table>
<thead>
<tr class="header">
<th align="left">AF cutoff</th>
<th align="center">Trans</th>
<th align="right">Untrans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">&lt;0.000001</td>
<td align="center">659</td>
<td align="right">655</td>
</tr>
<tr class="even">
<td align="left">&lt;0.00001</td>
<td align="center">779</td>
<td align="right">805</td>
</tr>
<tr class="odd">
<td align="left">&lt;0.0001</td>
<td align="center">876</td>
<td align="right">966</td>
</tr>
<tr class="even">
<td align="left">&lt;0.001</td>
<td align="center">1164</td>
<td align="right">1413</td>
</tr>
</tbody>
</table>
<p>When AF &lt;0.1%, there are 1164 transmitted variants and 1413
untransmitted, not consistent with numbers in cell paper 695 vs 557.</p>
<ul>
<li><code>pLI&gt;=0.995</code></li>
<li><code>isPTV==T</code></li>
<li><code>isIndel=="FALSE"</code></li>
<li><code>Gnomad_non_neuro_AF&lt;AF_cutoff</code></li>
<li><code>num_alleles&lt;5</code></li>
</ul>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">filter</th>
<th align="center">Trans</th>
<th align="right">Untrans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">VQSLOD&lt;3.75</td>
<td align="center">255</td>
<td align="right">546</td>
</tr>
<tr class="even">
<td align="left">VQSLOD&lt;5.13</td>
<td align="center">300</td>
<td align="right">624</td>
</tr>
<tr class="odd">
<td align="left">VQSLOD&lt;9</td>
<td align="center">1151</td>
<td align="right">1415</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">818</td>
<td align="right">785</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 6 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">816</td>
<td align="right">726</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">738</td>
<td align="right">604</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 10.5 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">714</td>
<td align="right">567</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 11 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">678</td>
<td align="right">526</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum
&gt;= -2</td>
<td align="center">956</td>
<td align="right">946</td>
</tr>
<tr class="even">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 5 &amp; AS_ReadPosRankSum
&gt;= -0.8</td>
<td align="center">821</td>
<td align="right">804</td>
</tr>
<tr class="odd">
<td align="left">QD &gt;= 1 &amp; AS_SOR &lt;= 10 &amp;
AS_ReadPosRankSum &gt;= -0.8</td>
<td align="center">822</td>
<td align="right">861</td>
</tr>
</tbody>
</table>
<ul>
<li>Kyle suggested: QD &gt;= 1 &amp; AS_SOR &lt;= 3 &amp;
AS_ReadPosRankSum &gt;= -0.8</li>
<li>with filter
<code>QD &gt;= 6 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code>,
0.11 transmitted (816/7291) vs 0.10 untransmitted vps</li>
<li>cutoff for <code>AS_ReadPosRankSum</code> could decrease to, for
instance -2, or even lower, while maintaining balanced T/UT, but the OR
consistently stay around 1.01, a bit away from 1.2</li>
<li>OR will increase with increasing cutoff for <code>QD</code>. When
cutoff =10.5, both OR and variant counts are close to numbers in paper,
but when cutoff &gt;10.5, the variant count will become smaller than
than in paper, seems not appropriate.<br />
</li>
<li>With balanced T/UT, increase of cutoff for <code>AS_SOR</code> will
decrease OR, even below 1, becoming depleted<br />
</li>
<li><code>pLI&gt;=0.995</code></li>
<li><code>isPTV==T</code></li>
<li><code>isIndel=="FALSE"</code></li>
<li><code>Gnomad_non_neuro_AF&lt;0.001</code></li>
</ul>
</div>
<div id="plot-of-qd-value" class="section level2">
<h2>plot of QD value</h2>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   0.323  11.270  12.480  12.418  13.580  46.520</code></pre>
<p>use filters</p>
<ul>
<li><code>isIndel=="FALSE"</code></li>
<li><code>AS_SOR &lt;= 3</code></li>
<li><code>AS_ReadPosRankSum &gt;= -0.8</code></li>
</ul>
</div>
<div id="run-mirage-vs-on-6-variant-groups-in-cell-paper"
class="section level2">
<h2>Run mirage-vs on 6 variant groups in cell paper</h2>
<ul>
<li>Use mirage-vs to test if there are signals in six variant
groups<br />
</li>
<li>filter <code>Gnomad_non_neuro_AF&lt;0.05</code> was applied</li>
<li><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></li>
<li><code>isSyn=="FALSE"</code></li>
</ul>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.
<span class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for another 3 MPC
groups when running mirage_vs. </p>
<pre><code>## # A tibble: 6 x 7
##   X                      No_case No_contr    OR     pval eta_est eta_pval
##   &lt;chr&gt;                    &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;
## 1 PLI&gt;=0.995                1351     1220  1.11 5.17e- 3 0.130   1.62e- 3
## 2 pLI within (0.5,0.995)    2939     2764  1.06 1.06e- 2 0.0643  2.11e- 2
## 3 pLI&lt;0.5                 109275   108331  1.01 2.16e- 2 0.00390 8.92e- 1
## 4 MPC&gt;=2                   48429    47071  1.03 5.64e- 6 0.0345  2.72e- 3
## 5 MPC within (1,2)        325625   319079  1.02 1.80e-16 0.0101  1.72e-11
## 6 MPC&lt;1                  4768875  4745827  1.00 3.96e-14 0.00131 2.49e-28</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="run-mirage-vs-on-180-variant-sets" class="section level2">
<h2>Run MIRAGE-VS on 180 variant sets</h2>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<ul>
<li><p>exclude syn variants <code>isSyn=="FALSE"</code></p></li>
<li><p>6 variant groups X 3 AF categories (0.001&lt;AF&lt;0.01,
0.0001&lt;AF&lt;0.001, AF&lt;0.0001) X 10 gene sets=180 variant
groups</p></li>
<li><p>use filter
<code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
<li><p>solid blue line is FDR 0.05 and dashed red line is Bonferroni
correction p = 0.05/total number of tests.</p></li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<ul>
<li><p>one outlier with extremely small MIRAGE_Vs p value is variant set
<code>AutismKB_MPC&lt;1_0.0001&lt;AF&lt;0.001</code> with 6802
(transmitted) vs 6914 (untransmitted) on 1901 loci. Burden p value:
0.9839; MIRAGE_VS p value: 5.195326e-15.</p></li>
<li><p>this variant set has several loci driving the BF</p></li>
</ul>
<pre><code>1500 1500      76      262        1 1.185888e+12
1499 1499      56      152        1 1.008106e+05
 
 BF.var.inte(trans=72, nontrans=262, bar.gamma=3, sig=3, num.family, num.family)
[1] 403764608704</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<ul>
<li><p>variant set
<code>ID gene_PLI&gt;=0.995_0.001&lt;AF&lt;0.01</code>, two loci, 13
(Trans) vs 15 (Nontrans) and 21 vs 8=&gt; 34 vs 23 in total.</p></li>
<li><p>varying <span class="math inline">\(\bar{\gamma}\)</span>
produces quite different p values for MIRAGE_VS.</p></li>
<li><p>blue dashed line: burden test p value of 0.09242524</p></li>
</ul>
</div>
<div id="run-mirage-genomewide" class="section level2">
<h2>Run MIRAGE genomewide</h2>
<ul>
<li><p><code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
<li><p><code>!str_detect(Consequence, 'intron_variant')</code> exclude
introns</p></li>
<li><p><code>!str_detect(Consequence, 'UTR')</code> exclude
UTRs</p></li>
</ul>
<div id="variant-types" class="section level3">
<h3>variant types</h3>
<div id="htmlwidget-9eceb20dc2a931752ae4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9eceb20dc2a931752ae4">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["missense_variant","missense_variant,splice_region_variant","stop_gained","downstream_gene_variant","upstream_gene_variant","splice_donor_variant","splice_acceptor_variant","start_lost","splice_region_variant,non_coding_transcript_exon_variant","non_coding_transcript_exon_variant","stop_lost","stop_gained,splice_region_variant","splice_donor_variant,non_coding_transcript_variant","missense_variant,NMD_transcript_variant","splice_acceptor_variant,non_coding_transcript_variant","start_lost,splice_region_variant","stop_gained,NMD_transcript_variant","missense_variant,splice_region_variant,NMD_transcript_variant","incomplete_terminal_codon_variant,coding_sequence_variant","splice_donor_variant,NMD_transcript_variant","splice_acceptor_variant,NMD_transcript_variant","stop_lost,splice_region_variant","coding_sequence_variant","start_lost,NMD_transcript_variant","mature_miRNA_variant","splice_acceptor_variant,missense_variant","stop_lost,NMD_transcript_variant"],[1050575,28324,25268,14895,10046,6778,5122,1961,1369,984,850,807,434,344,327,52,20,14,10,11,9,9,5,2,1,1,1],[0.894524,0.024117,0.021515,0.012683,0.008554,0.005771,0.004361,0.00167,0.001166,0.000838,0.000724,0.000687,0.00037,0.000293,0.000278,4.4e-05,1.7e-05,1.2e-05,9e-06,9e-06,8e-06,8e-06,4e-06,2e-06,1e-06,1e-06,1e-06]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Consequence<\/th>\n      <th>n<\/th>\n      <th>prop<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="af-distribution" class="section level3">
<h3>AF distribution</h3>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre><code>##      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
## 0.000e+00 0.000e+00 1.455e-05 4.037e-04 7.197e-05 5.000e-02</code></pre>
</div>
<div id="parameter-estimate" class="section level3">
<h3>parameter estimate</h3>
<ul>
<li><p>Run mirage on all 17850 genes and <span
class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups and
<span class="math inline">\(\gamma=3\)</span> for 3 MPC groups.</p></li>
<li><p><code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
</ul>
<table>
<thead>
<tr class="header">
<th align="left">variant category</th>
<th align="center">parameter</th>
<th align="center">estimate</th>
<th align="right">p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">\</td>
<td align="center"><span
class="math inline">\(\hat{\delta}\)</span></td>
<td align="center">0.0677</td>
<td align="right"><span class="math inline">\(1.383766e-42\)</span></td>
</tr>
<tr class="even">
<td align="left">(1):PLI&gt;=0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_1\)</span></td>
<td align="center">0.8341</td>
<td align="right"><span class="math inline">\(1.887287e-02\)</span></td>
</tr>
<tr class="odd">
<td align="left">(2):0.5&lt;=pLI&lt;0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_2\)</span></td>
<td align="center">0.8510</td>
<td align="right"><span class="math inline">\(1.236803e-02\)</span></td>
</tr>
<tr class="even">
<td align="left">(3):pLI&lt;0.5</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_3\)</span></td>
<td align="center">0.0292</td>
<td align="right"><span class="math inline">\(6.837139e-01\)</span></td>
</tr>
<tr class="odd">
<td align="left">(4):MPC&gt;=2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_4\)</span></td>
<td align="center">0.4683</td>
<td align="right"><span class="math inline">\(1.576707e-03\)</span></td>
</tr>
<tr class="even">
<td align="left">(5):1&lt;=MPC&lt;2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_5\)</span></td>
<td align="center">0.1136</td>
<td align="right"><span class="math inline">\(1.401804e-12\)</span></td>
</tr>
<tr class="odd">
<td align="left">(6):0&lt;=MPC&lt;1</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_6\)</span></td>
<td align="center">0.0120</td>
<td align="right"><span class="math inline">\(1.471682e-32\)</span></td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Below are top ranked genes.</p>
<pre><code>## # A tibble: 17,849 x 4
##        X Gene          BF post.prob
##    &lt;int&gt; &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;
##  1 10093 HLA-DRB1 2.13e15     1.00 
##  2  7049 SSC5D    5.48e12     1.00 
##  3 13643 GSTM1    3.10e11     1.00 
##  4  6420 TXNDC2   5.32e 6     1.00 
##  5  7169 PRAM1    1.81e 3     0.992
##  6  6234 ENPP7    2.86e 1     0.675
##  7   828 PTEN     1.47e 1     0.516
##  8   315 SUPT5H   1.42e 1     0.507
##  9   408 MICAL3   1.38e 1     0.500
## 10  5778 DPH1     1.32e 1     0.489
## # ... with 17,839 more rows</code></pre>
<ul>
<li><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5955937/">HLA-DRB1</a></li>
<li><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7697115/">SSC5D</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/33546147/">GSTM1</a></li>
<li><a
href="https://www.nature.com/articles/s41598-021-00123-x">TXNDC2</a></li>
</ul>
</div>
</div>
<div id="asd-genes-in-cell-paper" class="section level2">
<h2>102 ASD genes in cell paper</h2>
<div id="variant-set-vs-analysis-by-mirage_vs" class="section level3">
<h3>variant set (VS) analysis by MIRAGE_VS</h3>
<p>This figure shows the comparison between burden test (one side two
sample poisson test) and mirage-vs for proband in 6 variant groups.
<span class="math inline">\(\gamma=6\)</span> for 3 PLI variant groups
and <span class="math inline">\(\gamma=3\)</span> for 3 MPC groups when
running mirage_vs.</p>
<pre><code>## # A tibble: 6 x 7
##   X                  No_case No_contr    OR    pval  eta_est eta_pval
##   &lt;chr&gt;                &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 PLI&gt;=0.995              31       20  1.55 0.0804  0.227     0.342  
## 2 pLI&gt;=0.5&amp;pLI&lt;0.995      12        1 12    0.00171 1.00      0.00135
## 3 pLI&lt;0.5                 18       11  1.64 0.132   0.299     0.294  
## 4 MPC&gt;=2                 607      546  1.11 0.0386  0.0681    0.222  
## 5 MPC&gt;=1&amp;MPC&lt;2          4740     4685  1.01 0.289   0.0335    0.0797 
## 6 MPC&lt;1                28977    28532  1.02 0.0321  0.000207  1</code></pre>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div id="gene-set-analysis-by-mirage" class="section level3">
<h3>gene set analysis by MIRAGE</h3>
<ul>
<li><p><span class="math inline">\(\gamma=6\)</span> for LoF variant and
<span class="math inline">\(\gamma=3\)</span> for missense when running
mirage.</p></li>
<li><p>filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
</ul>
<table>
<caption>Parameter estimate by MIRAGE for 102 ASD genes.</caption>
<thead>
<tr class="header">
<th align="left">variant category</th>
<th align="center">parameter</th>
<th align="center">estimate</th>
<th align="right">p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">\</td>
<td align="center"><span
class="math inline">\(\hat{\delta}\)</span></td>
<td align="center">0.9998</td>
<td align="right"><span class="math inline">\(0.0295\)</span></td>
</tr>
<tr class="even">
<td align="left">PLI&gt;=0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_1\)</span></td>
<td align="center">0.4321</td>
<td align="right"><span class="math inline">\(0.2334\)</span></td>
</tr>
<tr class="odd">
<td align="left">0.5&lt;=pLI&lt;0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_2\)</span></td>
<td align="center">1</td>
<td align="right">0.0056</td>
</tr>
<tr class="even">
<td align="left">pLI&lt;0.5</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_3\)</span></td>
<td align="center">0.5156</td>
<td align="right">0.2508</td>
</tr>
<tr class="odd">
<td align="left">MPC&gt;=2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_4\)</span></td>
<td align="center">0.1176</td>
<td align="right">0.1685</td>
</tr>
<tr class="even">
<td align="left">1&lt;=MPC&lt;2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_5\)</span></td>
<td align="center">0.0539</td>
<td align="right">0.0729</td>
</tr>
<tr class="odd">
<td align="left">0&lt;=MPC&lt;1</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_6\)</span></td>
<td align="center"><span
class="math inline">\(7.814873e-09\)</span></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
</div>
<div id="top-5-constraint-genes" class="section level2">
<h2>Top 5% constraint genes</h2>
<p>Use below filters and stated otherwise</p>
<ul>
<li><p>filter <code>Gnomad_non_neuro_AF&lt;0.05</code></p></li>
<li><p><code>isSyn=="FALSE"</code></p></li>
<li><p><code>isIndel=="FALSE" &amp; QD &gt;= 10 &amp; AS_SOR &lt;= 3 &amp; AS_ReadPosRankSum &gt;= -0.8</code></p></li>
<li><p><code>filter(!str_detect(Consequence, 'intron_variant'))</code>
exclude intron</p></li>
</ul>
<div id="one-variant-set-constrant-genes-af10e-4"
class="section level3">
<h3>One variant set: constrant genes+ AF&lt;10E-4</h3>
<ul>
<li><p>In old sample, we observe signal in constraint genes with <span
class="math inline">\(AF&lt;10^{-4}\)</span>.</p></li>
<li><p>in new sample, there are 99333 transmitted variants and 94575
untransmitted variants.</p></li>
<li><p>Put all variants with AF&lt;10E-4 in one category and run
MIRAGE_VS. <span class="math inline">\(\widehat{\eta}=0.0882\)</span>, p
value is <span class="math inline">\(2.585515e-34\)</span>.</p></li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
</div>
<div id="run-mirage_vs-on-6-variant-groups-in-cell-paper"
class="section level3">
<h3>Run MIRAGE_VS on 6 variant groups in cell paper</h3>
<ul>
<li><span class="math inline">\(\gamma=6\)</span> for 3 PLI variant
groups and <span class="math inline">\(\gamma=3\)</span> for 3 MPC
groups when running mirage_vs.</li>
</ul>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-26-1.png" width="672" /><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-26-2.png" width="672" /></p>
</div>
<div id="gene-set-analysis-by-mirage-1" class="section level3">
<h3>gene set analysis by MIRAGE</h3>
<ul>
<li><p><span class="math inline">\(\gamma=6\)</span> for PLI variant
groups and <span class="math inline">\(\gamma=3\)</span> for MPC groups
when running mirage.</p></li>
<li><p>836 genes in total</p></li>
</ul>
<table>
<caption>Summary of top 5% constraint genes</caption>
<colgroup>
<col width="8%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">variant category</th>
<th align="center">parameter</th>
<th align="center">MIRAGE: estimate</th>
<th align="center">MIRAGE:p value</th>
<th align="center">Trans</th>
<th align="center">non-Trans</th>
<th align="center">OR</th>
<th align="center">MIRAGE_VS: estimate</th>
<th align="right">MIRAGE_VS: p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">\</td>
<td align="center"><span
class="math inline">\(\hat{\delta}\)</span></td>
<td align="center">0.9832</td>
<td align="center"><span
class="math inline">\(2.014043e-06\)</span></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">(1):PLI&gt;=0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_1\)</span></td>
<td align="center">0.2036</td>
<td align="center">0.0150</td>
<td align="center">225</td>
<td align="center">178</td>
<td align="center">1.26</td>
<td align="center">0.200</td>
<td align="right">0.0150</td>
</tr>
<tr class="odd">
<td align="left">(2):0.5&lt;=pLI&lt;0.995</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_2\)</span></td>
<td align="center"><span
class="math inline">\(1.118248e-35\)</span></td>
<td align="center">1</td>
<td align="center">174</td>
<td align="center">182</td>
<td align="center">0.956</td>
<td align="center">0.000342</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">(3):pLI&lt;0.5</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_3\)</span></td>
<td align="center">0.2145</td>
<td align="center">0.0011</td>
<td align="center">854</td>
<td align="center">775</td>
<td align="center">1.10</td>
<td align="center">0.212</td>
<td align="right">0.00105</td>
</tr>
<tr class="odd">
<td align="left">(4):MPC&gt;=2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_4\)</span></td>
<td align="center">0.1115</td>
<td align="center">0.0000152777</td>
<td align="center">7968</td>
<td align="center">7638</td>
<td align="center">1.04</td>
<td align="center">0.0730</td>
<td align="right">0.0000137</td>
</tr>
<tr class="even">
<td align="left">(5):1&lt;=MPC&lt;2</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_5\)</span></td>
<td align="center">0.0175</td>
<td align="center">0.0559135877</td>
<td align="center">48412</td>
<td align="center">47060</td>
<td align="center">1.03</td>
<td align="center">0.00926</td>
<td align="right">0.0918</td>
</tr>
<tr class="odd">
<td align="left">(6):0&lt;=MPC&lt;1</td>
<td align="center"><span
class="math inline">\(\hat{\eta}_6\)</span></td>
<td align="center">2.259378e-06</td>
<td align="center">1</td>
<td align="center">146671</td>
<td align="center">145348</td>
<td align="center">1.01</td>
<td align="center">0.000871</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-28-1.png" width="672" /><img src="ASC_new_sample_gene_set_files/figure-html/unnamed-chunk-28-2.png" width="672" /></p>
<div id="top-genes-with-burden" class="section level4">
<h4>Top genes with burden</h4>
<ul>
<li>top genes with large BF don’t have heavy burden.</li>
</ul>
<div id="htmlwidget-f30d7598595affc4f25a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f30d7598595affc4f25a">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100"],["LPA","KIAA1109","DNM2","EIF3I","ACACA","NRXN2","CTNNA2","ZMIZ1","SUPT5H","F12","MYH10","DYNC1H1","PLXNA1","ZBTB17","KBTBD6","PLXNC1","INTS3","SF3B3","ATP2B2","GPAT2","CAPRIN1","CACNA1D","ADCY5","DDB1","PRDM10","TAOK1","DDX41","DNM1","PI4KA","TRIM3","KAT2A","ROCK2","MIB1","DHX38","SYNGAP1","MED12L","GRIA1","ZNF282","TCHH","SMARCA2","NAGS","UNC13A","RARA","SCN1A","KCNV1","AP3D1","USP11","INO80","PRPF8","TMEM30B","EML5","MYH7","DOCK4","STAT5B","PDE2A","CUL3","CACNA1C","CLIP3","MED23","NSD1","MEF2C","FLNA","KDM2B","PANK1","PITPNM2","FLNC","WDFY3","WDTC1","CACNG2","CPNE9","EP400","PACS1","FUT4","DSCAM","SLC4A8","ZNF835","DNM3","STXBP1","PDS5B","ZNF777","MTA1","SMG1","LTBP3","ADAMTS5","KCNQ2","OGDH","CHD4","COPA","PRKD2","SZT2","FAM84B","FRYL","RUVBL2","CREBBP","HIRA","RTF1","CYLD","CHD3","HNRNPH1","AP2B1"],[6.233,3.7629,2.9091,2.8671,2.729,2.7051,2.4525,2.1127,2.0877,2.0873,2.0384,1.9847,1.9707,1.8696,1.8379,1.8373,1.8204,1.7418,1.7234,1.7036,1.6954,1.6407,1.6393,1.6351,1.5697,1.5543,1.5512,1.5462,1.5443,1.5296,1.5245,1.5046,1.4973,1.4941,1.4909,1.4855,1.4817,1.4743,1.4725,1.4698,1.4628,1.4526,1.4475,1.4465,1.4444,1.438,1.4324,1.431,1.423,1.4039,1.4014,1.4014,1.3829,1.3826,1.3806,1.3799,1.3754,1.3681,1.3669,1.366,1.3651,1.3649,1.3622,1.3588,1.3583,1.3538,1.3495,1.3426,1.3416,1.3363,1.3329,1.3318,1.3287,1.3252,1.3228,1.321,1.3209,1.3207,1.32,1.3101,1.3068,1.3047,1.3035,1.2985,1.2931,1.2851,1.2805,1.2793,1.2791,1.2762,1.2745,1.2744,1.2738,1.2691,1.2683,1.2668,1.2622,1.2596,1.2585,1.2582],[12741,3760,94,16,128,943,90,90,46,142,255,1112,280,75,86,1228,542,258,298,466,40,284,213,55,3749,40,71,113,1490,31,2346,2564,58,963,129,5134,152,598,3023,1813,79,1316,60,223,12,3269,57,627,47,393,2636,314,1510,49,861,1387,325,35,79,2454,14,214,1397,513,503,1842,372,63,15,170,3230,64,112,1560,49,7993,83,54,73,1298,74,1027,1114,4063,107,942,1850,94,279,3098,65,700,28,609,91,15,38,123,3233,35],[12785,3669,83,13,124,996,72,90,40,153,215,1095,270,65,62,1162,594,260,265,423,40,256,168,29,3698,34,63,103,1469,27,2271,2696,53,941,114,4941,143,624,3081,1699,71,1230,35,247,2,3274,54,505,52,355,2575,326,1511,34,859,1333,324,45,53,2345,8,213,1453,484,479,1914,392,49,20,157,2935,65,96,1532,35,7954,69,49,77,1307,62,949,1080,4105,82,949,1965,65,288,2996,54,692,26,633,66,14,32,99,3149,25],[0.9966,1.0248,1.1325,1.2307,1.0323,0.9468,1.25,1,1.15,0.9281,1.186,1.0155,1.037,1.1538,1.3871,1.0568,0.9125,0.9923,1.1245,1.1016,1,1.1094,1.2678,1.8965,1.0138,1.1765,1.127,1.0971,1.0143,1.1481,1.033,0.951,1.0943,1.0234,1.1316,1.0391,1.063,0.9583,0.9812,1.0671,1.1127,1.0699,1.7142,0.9028,5.9993,0.9985,1.0556,1.2416,0.9039,1.107,1.0237,0.9632,0.9993,1.4411,1.0023,1.0405,1.0031,0.7778,1.4905,1.0465,1.7499,1.0047,0.9614,1.0599,1.0501,0.9624,0.949,1.2857,0.75,1.0828,1.1005,0.9846,1.1667,1.0183,1.4,1.0049,1.2029,1.102,0.948,0.9931,1.1935,1.0822,1.0315,0.9898,1.3049,0.9926,0.9415,1.4461,0.9687,1.0341,1.2037,1.0116,1.0769,0.9621,1.3788,1.0715,1.1875,1.2424,1.0267,1.4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>BF<\/th>\n      <th>Transmitted_proband<\/th>\n      <th>Untransmitted_proband<\/th>\n      <th>OR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li><p><a
href="https://www.nature.com/articles/s41398-021-01578-2.pdf?proof=t">KIAA1109</a></p></li>
<li><p><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5785978/pdf/krnb-15-01-1391437.pdf">EIF3I</a></p></li>
<li><p><a
href="https://gene.sfari.org/database/human-gene/CTNNA2">CTNNA2</a></p></li>
</ul>
</div>
<div id="vairants-driving-the-signal-in-top-gene"
class="section level4">
<h4>Vairants driving the signal in top gene</h4>
</div>
</div>
<div id="use-fixed-widehateta-to-compute-bf" class="section level3">
<h3>use fixed <span class="math inline">\(\widehat{\eta}\)</span> to
compute BF</h3>
<div id="htmlwidget-b85bf888dc62471906a9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b85bf888dc62471906a9">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["318","378","100","543","570","367","212","413","33","508","22","124","334","297","563","161","136","130","311","578","696","650","772","787","405","678","160","822","606","485","306","391","748","714","219","204","326","655","554","568","322","722","462","670","474","246","167","107","140","723","327","247","159","214","148","430","141","771","454","154","555","166","252","364","7","549","427","658","630","316","458","531","293","126","179","332","495","196","406","286","632","448","441","372","509","587","611","677","702","133","288","475","344","58","790","122","229","639","786","28","35","674","625","806","760","389","712","237","498","542","830","660","561","268","190","775","53","599","123","379","32","299","183","315","201","261","479","648","812","115","529","649","304","146","287","333","137","490","347","616","267","187","289","345","387","426","428","489","538","592","753","816","354","491","523","833","74","250","669","455","566","769","732","9","785","231","301","581","823","744","18","24","143","264","452","438","516","168","99","692","796","152","10","185","257","284","337","443","447","459","477","493","499","573","666","727","738","750","751","761","768","408","56","89","260","392","631","163","620","820","83","497","528","203","709","662","525","202","369","291","189","469","657","703","377","511","266","409","396","480","789","263","8","29","259","797","5","20","75","104","111","128","155","175","192","199","239","265","298","340","351","401","588","607","637","711","717","782","774","117","157","470","169","622","486","505","49","188","244","245","303","339","450","467","487","488","524","615","668","724","805","819","23","51","831","255","504","664","725","737","373","102","643","116","97","236","541","585","142","323","777","634","522","653","400","270","283","269","135","586","718","757","746","375","353","552","726","11","343","38","439","461","481","355","596","91","803","319","603","667","282","398","565","393","629","12","17","21","39","48","50","54","55","57","59","64","65","69","79","86","87","88","98","101","112","120","125","151","164","171","172","177","186","208","210","211","220","221","226","227","232","233","238","254","258","274","279","292","295","296","300","320","342","350","356","360","386","390","397","407","411","412","414","417","422","429","433","437","444","446","456","463","464","468","471","506","514","530","535","539","540","548","557","560","562","567","571","577","584","589","593","600","602","605","608","612","613","621","644","647","654","663","665","672","682","683","685","686","687","688","689","690","691","694","695","735","741","742","749","754","756","759","764","766","778","779","781","792","793","802","815","818","828","19","704","209","500","734","181","118","338","4","90","178","230","482","519","532","590","673","700","758","825","715","824","457","63","145","194","205","251","349","380","381","503","645","661","693","799","721","432","76","478","103","144","174","335","635","628","619","2","156","85","382","31","336","814","617","150","66","110","431","507","62","242","275","361","420","705","80","484","421","604","638","544","801","119","384","213","423","595","235","36","193","794","71","453","30","736","92","113","184","191","234","308","362","374","402","419","424","434","466","575","609","640","679","684","745","755","762","807","834","368","67","108","521","810","6","73","106","222","224","272","410","513","579","582","624","642","114","404","95","277","302","314","553","614","747","813","47","729","556","598","770","78","280","675","162","576","370","131","223","132","460","710","601","626","16","225","127","395","701","733","121","720","34","740","182","418","697","276","656","798","465","527","440","659","731","26","359","271","681","256","81","518","330","37","520","42","70","294","358","365","502","767","780","788","652","149","105","307","324","483","512","791","835","545","170","817","310","52","200","707","317","325","228","559","449","739","285","510","45","763","680","436","61","305","3","517","804","836","394","551","533","623","403","60","84","352","716","636","618","501","218","476","134","591","198","138","496","800","348","273","442","574","706","827","328","72","473","534","366","564","278","248","385","68","795","206","1","376","809","388","526","290","371","832","240","425","671","558","783","43","207","776","153","281","515","633","713","808","341","158","472","547","494","321","180","13","25","597","129","569","109","197","651","572","215","195","147","743","329","77","583","752","253","594","730","139","610","44","82","641","176","708","728","383","96","492","435","217","445","821","309","811","41","363","216","765","399","415","93","699","27","331","537","826","173","536","357","719","40","165","46","451","241","627","416","646","312","262","249","14","546","550","580","15","313","698","829","243","676","346","773","784","94"],["ACACA","DNM2","ZMIZ1","PI4KA","PLXNA1","MYH10","DYNC1H1","SUPT5H","ZBTB17","CTNNA2","INTS3","CAPRIN1","NAGS","SF3B3","ATP2B2","PITPNM2","TRIM3","DDB1","TAOK1","MED12L","SYNGAP1","GRIA1","DNM1","SMARCA2","CLIP3","LPA","KDM2B","USP11","CACNA1D","CUL3","PRPF8","UNC13A","KCNV1","FLNC","MYH7","PDS5B","KAT2A","NSD1","CACNG2","ADCY5","RARA","ZNF282","ROCK2","MEF2C","SCN1A","INO80","CACNA1C","SLIT1","LTBP3","ZNF777","STAT5B","RTF1","CIT","MTA1","PDE2A","RUVBL2","PACS1","STXBP1","PNPLA6","ATP2A2","CSNK1E","FBXL14","FOXB1","CHD3","RBM15","SF3A1","PRKD2","HNRNPH1","ANKRD17","AP2B1","MAP2K7","KCNQ2","EDC4","CHST1","SLC4A8","EZH1","BIRC6","PLXNC1","PIP5K1C","KAT8","WDFY3","HCN2","CNOT3","CXXC1","GPAT2","TRA2B","KIAA1109","GRM1","SRF","MARK2","CYLD","TLK1","KAT7","WDTC1","TLN1","PAX6","ZFP36L1","KDM3B","CACNA1B","ASH1L","COPA","MED23","FRYL","THOC2","SOX17","SIN3B","CUX1","CYP46A1","PRKCE","HIRA","MED12","NIPBL","BRD1","SMG1","CHD4","LRRC8A","RYR2","DHX30","CSTF3","SMARCA4","SPEN","GLG1","MAP3K12","ASIC2","XPO4","LINGO1","SF3B1","CSNK1A1","FLNA","PRDM10","GNAS","CAMK2A","ABR","PPP1CA","CREBBP","DHX8","NRXN2","KIF1A","DHX40","MAB21L2","MAPK8IP3","LRP1","CHD9","COL1A1","WIZ","SYMPK","ARHGAP35","ATAD2B","DSCAM","CLASP2","TIGD5","MED14","DLG4","HDLBP","CSNK2A1","TAF1","LRRC7","FBN1","PDE8B","CAMSAP3","ZBTB20","LHX2","OGDH","KCND3","GRIN1","RBM25","SLC7A5","PIK3CA","KDM5C","TRRAP","SETDB1","UBAP2L","SPTBN2","CHD2","VAV1","PTPRS","SRC","ITPR2","KCNMA1","EHMT2","TLE4","FUT4","RSBN1","SMARCC2","ARIH1","SETD1A","NSF","U2AF2","ZBTB45","ELAVL1","NCKAP1","DNMT3A","KCNK12","STAG1","TNPO1","AGAP3","AUTS2","FAM84B","KCNK9","XKR4","ZNF462","MAP4K1","ZNF672","EBF3","SIN3A","KCNN1","CNOT6L","EP400","HTT","RBM10","CELF2","SIX3","RBM38","FRY","ACTL6B","HCN1","KCNG1","CDK8","ROCK1","CNOT1","AGAP2","MYCN","DDX41","LRRC73","ILF3","ZAP70","GRIN2A","ACTN4","TSSK6","SATB2","VCP","SLCO3A1","MTOR","ARHGEF2","CSK","HNRNPK","AMPD2","POGZ","PTGER3","PANK1","DSCAML1","LRP4","CUX2","TUBA1B","CPSF6","ARHGEF7","UBE3A","NR2F2","DHX38","SP6","PSMC5","GNA11","EIF4A2","FEZF2","TRPC7","VGF","PLXNA4","NACC2","ZER1","IGSF9B","NOS1","SLC4A10","KCNA6","PDS5A","TRIP12","PCBP1","PROX1","KIF5A","MEIS2","RHOV","TUBB3","KPNB1","TUBB4A","KCNJ3","PSMD1","INPP5D","KCNB1","ABCE1","HMGCR","REPIN1","GRIA3","CASK","DENND4B","KDM1A","NLGN3","FEM1B","GFPT1","KIF2A","ZNF775","FSCN1","TCF4","GLUD1","IK","ADAMTS15","PPP3CB","BCL11B","ADARB1","DVL3","RBM14","TOP2A","GTF3C4","TERT","SLC12A5","CPEB4","ZNF536","PRKCB","FBXL19","CASKIN1","RPS6KA4","AP2M1","TRIM24","ARHGAP39","UBR5","ZNF236","BPTF","MYH9","KCNH2","ATP1A1","IGF2BP1","DNM3","PPP2R1A","KCNF1","FZD7","SOX9","SNRK","DIP2C","TRPC5","SRCIN1","RBM5","ADCY2","ZSCAN10","AP3D1","ATP6V1A","MAST3","TADA2B","NOTCH2","GPR89A","TCHH","GNB1","TMCC2","ACTA1","FMN2","HNRNPU","RPS6KA1","AHDC1","DLGAP3","ZMYM4","SZT2","PITX3","PDZD8","PRLHR","EIF3A","ZNF503","GATA3","BCL9L","MYOD1","B4GALNT4","IPO7","ZNF605","WNT1","DDN","FMNL3","ANKRD52","PCDH17","IPO5","YY1","NYNRIN","HECTD1","TMEM30B","KCNH5","ACOT1","IRF2BPL","GOLGA6L1","PIAS1","HCN4","SLX1B","ZNF768","CTCF","CLEC18A","CLEC18C","KLHL36","MLLT6","CHRNE","MRC2","NEURL4","GALR2","BRD4","ANO8","ATP13A1","RYR1","LRFN1","SAMD4B","HNRNPUL1","MEGF8","PPP1R13L","ZC3H4","PRR12","SHANK1","SAFB2","ZNF835","XAB2","WDR33","LCT","NR4A2","PXDN","EGR4","KIF3B","LRRN4","ADAMTS5","PRDM15","ABCG1","ZNRF3","CBX6","ACO2","SCO2","CASR","SEC61A1","TSC22D2","ABCC5","TP63","EPM2AIP1","PLXNB1","QRICH1","DOCK3","CADPS","PCDH10","TBC1D9","ADRA2C","PCDHGC3","TCERG1","UNC5A","IL6ST","MAST4","BEND3","DHX16","MDC1","VARS2","HLA-C","HLA-B","DDX39B","LTB","PRRC2A","VARS","CFB","RXRB","FBXL18","NCF1","YWHAG","HAS2","FAM83H","CPSF1","EGR3","VCPIP1","ZFHX4","TSC1","ADAMTSL2","KCNT1","CNTNAP3","APBA1","IRS4","EIF2S3","DDX3X","FAM120C","PI4KB","GTPBP2","MYCBP2","NRXN1","IKZF1","ACVR1B","B3GAT1","CDC27","AMY1C","ZMYND11","ASIC1","ACTN1","RHOB","TOP1","EEF1A2","PAK2","CDK19","BRPF3","XPO7","SMC1A","NRF1","IQSEC2","EVI5L","EIF3I","KDM2A","PPFIA2","NBEA","MYO5A","TLK2","TNPO2","NFIX","XPO1","HDAC3","RICTOR","ZBTB12","GABBR2","EZH2","SLC17A7","RERE","STAT1","ACTA2","PC","KCNA1","UBTF","AFF4","PPP2R2C","DHX15","KIF1B","MED13L","TCF7L2","NACC1","LMNA","EFTUD2","KLHL15","RAPGEF2","GRM5","NCDN","ZBTB16","PPFIA3","HTRA2","KPNA6","ACTC1","SULT1A4","UBE2O","NKPD1","TMEM63B","FBXL15","SPEG","MARK4","SEMA3F","HNRNPA0","TOP3B","CLCN4","BRSK2","TECR","AKT1","GPR4","CTNNB1","DICER1","CDK11B","WNK1","TRPM3","CHD5","INSR","MEX3A","ACTB","ANXA8L1","ABCG4","GDF11","CAND1","EML5","KIAA0100","SRSF2","ONECUT2","CHST8","CLASRP","FBXO46","PRMT1","KIF5C","U2SURP","CPNE9","ETF1","SOX4","TUBB","PABPC1","PUF60","CHD7","STAG2","OGT","TUBB6","MACF1","CTR9","CHD6","PDZD4","AHCYL1","SERBP1","TM9SF3","BAZ1A","SSTR1","PDPK1","HNRNPL","STK35","PLCL2","SOX2","PHOX2B","PURA","STT3A","LRFN3","EGR2","PPP4C","USP7","CDK5R1","EIF3D","SMARCA5","XKR6","RPS6KA3","NAV1","ELMO1","KCNJ4","SMARCC1","HSPA5","WNT8B","ZNF689","GFOD1","UBC","CHST2","MIB1","DAGLA","RALGAPA1","EML3","MAP4K4","SRRT","CELSR3","NOA1","NBPF15","TRIM9","DGKZ","MAU2","USP49","ADCY1","KCNC1","CUL1","ATP1A2","GTF2I","ESPL1","CADM4","ITPR3","TAOK2","F12","DAPK1","ZEB2","ZFP64","PRKCG","DROSHA","GCK","KCNN3","UNK","CACNG3","TRIM39","TLE3","TRIM8","PPP1R16B","TUBG1","POGK","PLCG1","TPR","JUN","FBXL16","EIF4A1","CBX4","SOX11","RNF20","BRD3","SMU1","EBF1","CREBZF","BTAF1","USP22","KLHL11","TUBA4A","INPP4A","NPR2","ATRX","GNAZ","DDX23","USP9X","MYO18A","HNRNPR","PSPC1","EEF1A1","LHX1","ACLY","GPHN","CACNA1I","KHSRP","BAZ1B","ZNF668","SNRNP200","GPR25","BHLHE22","GABBR1","LRRC4B","PUM1","NCOR1","AMY1A","CENPB","WDR44","BRWD3","UPF1","PATZ1","PRPF6","RBM47","ATP4A","OPRD1","SMC3","HELZ","MKLN1","DDX46","CLCN3","SPTBN1","METTL3","MYT1L","PLCB3","TRIM71","ARGLU1","SF1","KCNG3","PCDH19","CLTC","GTF3C1","BRSK1","PAQR9","KCNQ5","PHF8","STAT5A","JAK1","SCN2A","PLCB4","RPTOR","SLC6A1","CORO1A","CKMT1B","NOTCH3","KCNQ4","PCSK5","DCLK1","GPR88","DNMT1","SLC9A6","CHERP","ATP9A","ZNF319","DLGAP1","ZMYM3","HERC2","DMWD","TFAP2A","MGAT3","NOTCH1","UBR4","KBTBD6","NTNG2","HCFC2","SRCAP","RBM39","MARCH6","DOCK4","SMARCA1","HOXB8","KSR2","SCN3A","SMARCB1","CAD","THRA","SCN8A","PRAMEF6","ADAR","ITPR1","RTN4RL2","KALRN","CUL5","NALCN","CYFIP2","PLXND1","ZNF219","ATP2B1","PITPNM1","GRM3","STAT3","PIK3CD","SATB1","ARC","HERC1","SCN5A","HECW1","MEN1","BRPF1","CDC73","INA","CTNNA1","KCNH3","PHIP","CARD11","APC2","HK1","ADCY3","AP2A1","CHD8","SAFB","UBA1","SUPT6H","HCFC1","DHX9","KDM6B","SUPT16H","ARFGEF1","DOT1L","ATP1A3","TUBB8","DEF6","TRIM46","CNTNAP1","DYRK1A","HUWE1","DHH","ITSN1","NLGN2","BRAF","CACNA1E","GRIN2B","KIF21B","C3","SLC12A6","POLR2B","GRIK5","TRIO","GIT1","RASGRF1","SEMA6D","PRAMEF14","BCR","MORC2","TNIK","PRAMEF19","NF1","SCUBE3","NLGN4X","AQR","HIVEP2","CACNA1G","SPTAN1","ABCA2","ASAH2"],[6.0172,4.2618,4.1739,3.4846,3.4765,3.3026,3.1496,3.0171,2.9004,2.8635,2.7795,2.762,2.6371,2.4677,2.296,2.2692,2.2289,2.2049,2.1798,2.1238,2.1088,2.0983,2.0515,2.0404,2.0208,1.9817,1.9392,1.9341,1.9308,1.9222,1.9167,1.8936,1.8801,1.8702,1.8533,1.8082,1.7962,1.7917,1.7863,1.775,1.7719,1.7683,1.7528,1.7509,1.7389,1.718,1.6749,1.6523,1.6523,1.6466,1.6394,1.6371,1.6235,1.6179,1.6121,1.6097,1.6086,1.6027,1.5784,1.5727,1.5698,1.5698,1.5601,1.5475,1.5395,1.5381,1.5289,1.526,1.4966,1.4965,1.496,1.4921,1.4834,1.4828,1.4828,1.4828,1.4749,1.4723,1.4648,1.4619,1.4614,1.4569,1.4344,1.4344,1.4344,1.4344,1.4344,1.4344,1.4344,1.4337,1.4219,1.4219,1.4211,1.4196,1.416,1.4067,1.4013,1.4007,1.3932,1.3852,1.3837,1.38,1.3733,1.3669,1.3632,1.3568,1.3555,1.355,1.3549,1.3549,1.3549,1.3495,1.3433,1.3395,1.3319,1.3313,1.3306,1.3242,1.3216,1.3214,1.3211,1.3199,1.3171,1.3107,1.3107,1.3107,1.3107,1.3107,1.3107,1.3068,1.306,1.3056,1.301,1.2992,1.2992,1.2992,1.2965,1.2952,1.2878,1.2878,1.2854,1.2854,1.2854,1.2854,1.2854,1.2854,1.2854,1.2854,1.2854,1.2854,1.2843,1.2843,1.2797,1.2741,1.2741,1.273,1.2706,1.2635,1.2573,1.2557,1.2501,1.2499,1.2398,1.2314,1.2303,1.2272,1.2272,1.2272,1.2272,1.2227,1.2207,1.2207,1.2207,1.2207,1.2112,1.2109,1.21,1.2099,1.2077,1.2056,1.202,1.1999,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1977,1.1919,1.1902,1.1872,1.1872,1.1872,1.1872,1.1872,1.1807,1.1807,1.1768,1.1735,1.1629,1.1608,1.1601,1.1556,1.1545,1.1428,1.1313,1.1305,1.1293,1.1258,1.1252,1.1252,1.1249,1.1214,1.1208,1.12,1.1195,1.1188,1.1154,1.115,1.1061,1.1019,1.0997,1.0954,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0944,1.0943,1.0934,1.0893,1.0868,1.0863,1.0861,1.086,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0848,1.0801,1.0753,1.0753,1.0753,1.0753,1.0753,1.0753,1.0682,1.0666,1.0659,1.0659,1.0626,1.0607,1.0565,1.0564,1.0545,1.053,1.0477,1.0443,1.0421,1.0416,1.0374,1.0353,1.0352,1.0322,1.0312,1.0293,1.0282,1.0282,1.0282,1.0275,1.0268,1.0262,1.0235,1.0225,1.0206,1.0192,1.0192,1.0192,1.0192,1.0192,1.0171,1.0168,1.0167,1.0105,1.0102,1.0102,1.0097,1.0094,1.0068,1.0059,1.0047,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.9993,0.998,0.998,0.9974,0.9965,0.9926,0.9924,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9912,0.9847,0.9847,0.9839,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9825,0.9778,0.9761,0.9759,0.9739,0.9739,0.9739,0.9739,0.9739,0.9739,0.9739,0.9697,0.966,0.9657,0.9654,0.964,0.9612,0.9543,0.9543,0.951,0.9506,0.9482,0.9459,0.9459,0.9417,0.9408,0.9395,0.9395,0.9395,0.9395,0.9395,0.9382,0.9376,0.9318,0.9313,0.9313,0.9313,0.9302,0.927,0.9254,0.9231,0.9231,0.9231,0.9231,0.9229,0.9208,0.9185,0.9183,0.9161,0.9159,0.9119,0.9058,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9057,0.9052,0.9026,0.9017,0.9007,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8978,0.8934,0.8899,0.8899,0.8899,0.8899,0.8899,0.8899,0.8899,0.8888,0.8854,0.8836,0.8831,0.8821,0.8817,0.8808,0.88,0.8754,0.875,0.8744,0.8744,0.8719,0.8706,0.8689,0.8673,0.8673,0.8661,0.8653,0.8644,0.8643,0.8643,0.863,0.8585,0.8585,0.8585,0.8579,0.8575,0.8565,0.8559,0.8541,0.8513,0.8499,0.8484,0.8463,0.8437,0.8435,0.8435,0.8425,0.8424,0.8417,0.8392,0.8361,0.8361,0.8346,0.8326,0.831,0.8236,0.822,0.8213,0.8204,0.8204,0.8204,0.8204,0.8204,0.8204,0.8204,0.8204,0.8204,0.8204,0.8193,0.8192,0.8132,0.8132,0.8132,0.8132,0.8132,0.8132,0.8132,0.812,0.8103,0.8093,0.8068,0.8061,0.8061,0.8061,0.8016,0.8012,0.7995,0.799,0.799,0.7988,0.7968,0.7967,0.7962,0.7947,0.7943,0.7925,0.792,0.792,0.7897,0.7897,0.7897,0.7897,0.7877,0.7845,0.7844,0.7841,0.7837,0.7828,0.7824,0.782,0.782,0.7797,0.779,0.7779,0.776,0.776,0.7744,0.7722,0.7721,0.7708,0.7708,0.7708,0.7701,0.7699,0.7694,0.7692,0.7684,0.7684,0.7683,0.7673,0.7662,0.7657,0.7622,0.7615,0.7604,0.7573,0.7542,0.7522,0.7501,0.7485,0.747,0.7407,0.7375,0.7369,0.7365,0.7341,0.7324,0.729,0.7255,0.7241,0.7237,0.7218,0.7216,0.7212,0.7205,0.7205,0.7153,0.7146,0.7146,0.7146,0.7146,0.7146,0.709,0.7083,0.7083,0.7083,0.7069,0.7053,0.7008,0.6991,0.692,0.6911,0.6892,0.6869,0.6859,0.6857,0.6844,0.684,0.6802,0.6799,0.673,0.673,0.6658,0.6613,0.6613,0.6595,0.6564,0.6564,0.6564,0.6488,0.6445,0.6423,0.6418,0.6409,0.6366,0.636,0.6356,0.6304,0.6268,0.6261,0.6259,0.6258,0.6232,0.6098,0.6096,0.6091,0.6085,0.6074,0.6042,0.5962,0.5811,0.5787,0.5742,0.5731,0.5558,0.555,0.5513,0.5473,0.5437,0.5437,0.5284,0.5217,0.5214,0.5187,0.5107,0.5107,0.5052,0.5004,0.4928,0.4903,0.4891,0.4805,0.4625,0.4573,0.4464,0.4312,0.4242,0.4146,0.4123,0.3931,0.3915,0.3878,0.3649,0.3505,0.2897,0.2572,0.1676]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>BF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li><span class="math inline">\(\eta&#39;s\)</span> are fixed at values
in old results, i.e. <span class="math inline">\(\eta_1 (PTV,
pLI&gt;0.995)=0.41, \eta_4 (MPC&gt;2)=0.16\)</span>, all others are set
0,</li>
</ul>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
