<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Cell data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Cell data</h1>

</div>


<pre><code>## Warning: package &#39;tidyverse&#39; was built under R version 3.6.3</code></pre>
<pre><code>## -- Attaching packages ------------------------------- tidyverse 1.3.0 --</code></pre>
<pre><code>## v ggplot2 3.3.1     v purrr   0.3.4
## v tibble  3.0.1     v dplyr   1.0.0
## v tidyr   1.1.0     v stringr 1.4.0
## v readr   1.3.1     v forcats 0.4.0</code></pre>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 3.6.3</code></pre>
<pre><code>## Warning: package &#39;tibble&#39; was built under R version 3.6.3</code></pre>
<pre><code>## Warning: package &#39;tidyr&#39; was built under R version 3.6.3</code></pre>
<pre><code>## Warning: package &#39;purrr&#39; was built under R version 3.6.3</code></pre>
<pre><code>## Warning: package &#39;dplyr&#39; was built under R version 3.6.3</code></pre>
<pre><code>## -- Conflicts ---------------------------------- tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()</code></pre>
<p>This is the transmitted data from <a href="https://www.sciencedirect.com/science/article/abs/pii/S0092867419313984">cell paper</a></p>
<pre><code>## Joining, by = c(&quot;Variant&quot;, &quot;FILTER&quot;, &quot;VQSLOD&quot;, &quot;GENE_NAME&quot;, &quot;Consequence&quot;, &quot;PCC_MAC&quot;, &quot;MAC&quot;, &quot;AN&quot;, &quot;exac_AC&quot;, &quot;exac_AN&quot;, &quot;gnomad_AC&quot;, &quot;gnomad_AN&quot;, &quot;Transmitted_proband&quot;, &quot;Untransmitted_proband&quot;, &quot;Transmitted_sibling&quot;, &quot;Untransmitted_sibling&quot;, &quot;Transmitted_proband_male&quot;, &quot;Untransmitted_proband_male&quot;, &quot;Transmitted_proband_female&quot;, &quot;Untransmitted_proband_female&quot;, &quot;Transmitted_male_sibling&quot;, &quot;Untransmitted_male_sibling&quot;, &quot;Transmitted_female_sibling&quot;, &quot;Untransmitted_female_sibling&quot;, &quot;Missense&quot;, &quot;PTV&quot;, &quot;Inframe&quot;, &quot;Indel&quot;, &quot;pLI&quot;, &quot;loftee&quot;, &quot;loftee_flag&quot;, &quot;MPC&quot;)</code></pre>
<pre class="r"><code>cell_trans_data</code></pre>
<pre><code>## # A tibble: 994,532 x 32
##    Variant FILTER  VQSLOD GENE_NAME Consequence PCC_MAC   MAC    AN exac_AC
##    &lt;fct&gt;   &lt;fct&gt;    &lt;dbl&gt; &lt;fct&gt;     &lt;fct&gt;         &lt;int&gt; &lt;int&gt; &lt;int&gt;   &lt;int&gt;
##  1 1:8612~ [PASS]  1.23   AL645608~ [missense_~       1     2 52584       4
##  2 1:8612~ [PASS] -1.90   AL645608~ [missense_~       1     1 52616      NA
##  3 1:8612~ [PASS] -0.043  AL645608~ [missense_~       4     4 52602      NA
##  4 1:8613~ [PASS] -0.745  SAMD11    [missense_~       1     2 52710      NA
##  5 1:8613~ [PASS]  2.41   SAMD11    [missense_~       2     2 52652      NA
##  6 1:8613~ [PASS] -1.42   AL645608~ [missense_~       1     1 52606      NA
##  7 1:8656~ [PASS] -0.856  SAMD11    [missense_~       1     2 49990      NA
##  8 1:8656~ [PASS] -0.287  SAMD11    [missense_~       1     2 50314       1
##  9 1:8656~ [PASS]  0.203  SAMD11    [missense_~       2     2 50628       3
## 10 1:8656~ [PASS] -0.0374 SAMD11    [missense_~       1     2 50456       1
## # ... with 994,522 more rows, and 23 more variables: exac_AN &lt;int&gt;,
## #   gnomad_AC &lt;int&gt;, gnomad_AN &lt;int&gt;, Transmitted_proband &lt;int&gt;,
## #   Untransmitted_proband &lt;int&gt;, Transmitted_sibling &lt;int&gt;,
## #   Untransmitted_sibling &lt;int&gt;, Transmitted_proband_male &lt;int&gt;,
## #   Untransmitted_proband_male &lt;int&gt;, Transmitted_proband_female &lt;int&gt;,
## #   Untransmitted_proband_female &lt;int&gt;, Transmitted_male_sibling &lt;int&gt;,
## #   Untransmitted_male_sibling &lt;int&gt;, Transmitted_female_sibling &lt;int&gt;,
## #   Untransmitted_female_sibling &lt;int&gt;, Missense &lt;lgl&gt;, PTV &lt;lgl&gt;,
## #   Inframe &lt;lgl&gt;, Indel &lt;lgl&gt;, pLI &lt;dbl&gt;, loftee &lt;fct&gt;,
## #   loftee_flag &lt;fct&gt;, MPC &lt;dbl&gt;</code></pre>
<pre class="r"><code>colnames(cell_trans_data)</code></pre>
<pre><code>##  [1] &quot;Variant&quot;                      &quot;FILTER&quot;                      
##  [3] &quot;VQSLOD&quot;                       &quot;GENE_NAME&quot;                   
##  [5] &quot;Consequence&quot;                  &quot;PCC_MAC&quot;                     
##  [7] &quot;MAC&quot;                          &quot;AN&quot;                          
##  [9] &quot;exac_AC&quot;                      &quot;exac_AN&quot;                     
## [11] &quot;gnomad_AC&quot;                    &quot;gnomad_AN&quot;                   
## [13] &quot;Transmitted_proband&quot;          &quot;Untransmitted_proband&quot;       
## [15] &quot;Transmitted_sibling&quot;          &quot;Untransmitted_sibling&quot;       
## [17] &quot;Transmitted_proband_male&quot;     &quot;Untransmitted_proband_male&quot;  
## [19] &quot;Transmitted_proband_female&quot;   &quot;Untransmitted_proband_female&quot;
## [21] &quot;Transmitted_male_sibling&quot;     &quot;Untransmitted_male_sibling&quot;  
## [23] &quot;Transmitted_female_sibling&quot;   &quot;Untransmitted_female_sibling&quot;
## [25] &quot;Missense&quot;                     &quot;PTV&quot;                         
## [27] &quot;Inframe&quot;                      &quot;Indel&quot;                       
## [29] &quot;pLI&quot;                          &quot;loftee&quot;                      
## [31] &quot;loftee_flag&quot;                  &quot;MPC&quot;</code></pre>
<div id="annotations-in-the-data" class="section level2">
<h2>Annotations in the data</h2>
<ul>
<li>PTV: protein-truncating variants including nonsense, frameshift, and essential splice sites variants</li>
<li>Missense</li>
<li>pLI: probability of loss of function intolerance</li>
<li>MPC: missense badness, PolyPhen-2, constraint score</li>
</ul>
</div>
<div id="replicate-burden-analysis-for-transmitted-variants-in-figure-1b" class="section level2">
<h2>Replicate Burden analysis for transmitted variants in Figure 1B</h2>
<table>
<caption>variant count of transmitted PTV and MPC</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pLI&gt;=0.995</th>
<th align="right">0.5&lt;pLI&lt;0.995</th>
<th align="right">MPC&gt;=2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>transmitted</td>
<td align="right">704</td>
<td align="right">1273</td>
<td align="right">6341</td>
</tr>
<tr class="even">
<td>untransmitted</td>
<td align="right">564</td>
<td align="right">1191</td>
<td align="right">6159</td>
</tr>
</tbody>
</table>
<table>
<caption>Enrichment of transmited PTV and MPC in proband using binomial exact test</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">pLI&gt;=0.995</th>
<th align="right">0.5&lt;pLI&lt;0.995</th>
<th align="right">MPC&gt;=2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>OR</td>
<td align="right">1.2820608</td>
<td align="right">1.087920</td>
<td align="right">0.6325627</td>
</tr>
<tr class="even">
<td>RR</td>
<td align="right">1.2482270</td>
<td align="right">1.068850</td>
<td align="right">1.0295503</td>
</tr>
<tr class="odd">
<td>p.value</td>
<td align="right">0.0000467</td>
<td align="right">0.051352</td>
<td align="right">0.0527311</td>
</tr>
</tbody>
</table>
<p>In the paper, when <span class="math inline">\(pLI&gt;=0.995\)</span> for PTV, there are 695 transmitted vs 557 untransmitted from 5869 parents, leading to RR 1.2 and p value 0.07?, and these numbers are close to that in the above table.</p>
</div>
<div id="annotate-the-data-by-annovar" class="section level2">
<h2>Annotate the data by ANNOVAR</h2>
<p>Since the data doesn’t have allele frequency, use ANNOVAR to annotate it</p>
<pre class="r"><code>head(cell_trans_data)</code></pre>
<pre><code>## # A tibble: 6 x 32
##   Variant FILTER VQSLOD GENE_NAME Consequence PCC_MAC   MAC    AN exac_AC
##   &lt;fct&gt;   &lt;fct&gt;   &lt;dbl&gt; &lt;fct&gt;     &lt;fct&gt;         &lt;int&gt; &lt;int&gt; &lt;int&gt;   &lt;int&gt;
## 1 1:8612~ [PASS]  1.23  AL645608~ [missense_~       1     2 52584       4
## 2 1:8612~ [PASS] -1.90  AL645608~ [missense_~       1     1 52616      NA
## 3 1:8612~ [PASS] -0.043 AL645608~ [missense_~       4     4 52602      NA
## 4 1:8613~ [PASS] -0.745 SAMD11    [missense_~       1     2 52710      NA
## 5 1:8613~ [PASS]  2.41  SAMD11    [missense_~       2     2 52652      NA
## 6 1:8613~ [PASS] -1.42  AL645608~ [missense_~       1     1 52606      NA
## # ... with 23 more variables: exac_AN &lt;int&gt;, gnomad_AC &lt;int&gt;,
## #   gnomad_AN &lt;int&gt;, Transmitted_proband &lt;int&gt;,
## #   Untransmitted_proband &lt;int&gt;, Transmitted_sibling &lt;int&gt;,
## #   Untransmitted_sibling &lt;int&gt;, Transmitted_proband_male &lt;int&gt;,
## #   Untransmitted_proband_male &lt;int&gt;, Transmitted_proband_female &lt;int&gt;,
## #   Untransmitted_proband_female &lt;int&gt;, Transmitted_male_sibling &lt;int&gt;,
## #   Untransmitted_male_sibling &lt;int&gt;, Transmitted_female_sibling &lt;int&gt;,
## #   Untransmitted_female_sibling &lt;int&gt;, Missense &lt;lgl&gt;, PTV &lt;lgl&gt;,
## #   Inframe &lt;lgl&gt;, Indel &lt;lgl&gt;, pLI &lt;dbl&gt;, loftee &lt;fct&gt;,
## #   loftee_flag &lt;fct&gt;, MPC &lt;dbl&gt;</code></pre>
<div id="prepare-the-data-in-the-appropriate-format-for-annovar" class="section level3">
<h3>prepare the data in the appropriate format for annovar</h3>
<p>To run annovar, the simplist format has 5 columns chr, star, end, ref, alt and each row is one variant.</p>
<ol style="list-style-type: decimal">
<li>use R to extract 5 required columns</li>
</ol>
<pre class="r"><code>cell_data=read.table(&quot;cell_trans_data.txt&quot;, header=T, fill=T) # note &quot;fill&quot; some lines with less than 38 elements
locus=matrix(cell_data$Variant)
split_locus=strsplit(locus, &quot;:&quot;) # &quot;:&quot;  separates chr, start and end 
avinput=matrix(nrow=length(split_locus), ncol=5)  # format of avinput; chr, start, end, ref, alt
for (i in 1:length(split_locus))
  { 
   avinput[i,1:2]=split_locus[[i]][1:2]
   avinput[i,3]=split_locus[[i]][2]
   avinput[i,4:5]=split_locus[[i]][3:4]
}

#write.table(avinput, file=&quot;avinput.txt&quot;,row.names=F, col.names=F, quote=F)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>then transfer into tab-delimited columns</li>
</ol>
<pre class="r"><code>head avinput.txt
1 861283 861283 G C
1 861287 861287 C T
1 861289 861289 G A
1 861341 861341 A G
1 861389 861389 C T
1 861398 861398 G A
1 865635 865635 G A
1 865645 865645 T G
1 865655 865655 T G
1 865686 865686 A G</code></pre>
<pre class="r"><code>awk &#39;{ for(i=1;i&lt;=NF;i++){if(i==NF){printf(&quot;%s\n&quot;,$NF);}else {printf(&quot;%s\t&quot;,$i)}}}&#39;
input.data &gt; output.bed</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>download relevant database</li>
</ol>
<pre class="r"><code>perl annotate_variation.pl -buildver hg19 -downdb -webfrom annovar exac03 humandb/</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>run annovar function</li>
</ol>
<pre class="r"><code>perl annotate_variation.pl -filter -build hg19 -dbtype exac03 avinput humandb/
  # the output file with AF annotated is avinput.hg19_exac03_dropped</code></pre>
</div>
</div>
<div id="refine-annotations" class="section level2">
<h2>Refine annotations</h2>
<p>Stratify six variant groups PTV (PLI&gt;=0.995, 0.5-0.995, &lt;0.5) and missense (MPC&gt;=2, 1-2, 0-1) by AF (0.01-0.05, 0.001-0.01, &lt;0.001), resulting in 18 categories.</p>
<pre><code>## Joining, by = &quot;Variant&quot;</code></pre>
<p><img src="ASC_cell_data_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre><code>##      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
## 0.000e+00 0.000e+00 8.245e-06 1.294e-05 1.696e-05 1.200e-03</code></pre>
<pre><code>##                          transmitted untransmitted
## PLI&gt;=0.995&amp;AF&lt;=0.001             704           564
## 0.5&lt;=pLI&lt;0.995&amp;AF&lt;=0.001        1273          1191
## pLI&lt;0.5&amp;AF&lt;=0.001              18959         18591
## MPC&gt;=2&amp;AF&lt;=0.001                6341          6159
## 1&lt;=MPC&lt;2&amp;AF&lt;=0.001             50355         50369
## 0&lt;=MPC&lt;1&amp;0.001&lt;AF&lt;=0.01            0             0
## 0&lt;=MPC&lt;1&amp;AF&lt;=0.001            339394        340489</code></pre>
<p>It looks like all variants in the data are filtered with cutoff, like <span class="math inline">\(0.1%\)</span>.</p>
</div>
<div id="genome-wide-analysis" class="section level2">
<h2>Genome-wide analysis</h2>
<div id="run-mirage-vs-on-6-variant-groups-in-cell-paper" class="section level3">
<h3>Run mirage-vs on 6 variant groups in cell paper</h3>
<p>Use mirage-vs to test if there are signals in six variant groups</p>
<p>This figure shows the comparison between burden test (one side two sample poisson test) and mirage-vs for proband in 6 variant groups. <span class="math inline">\(\gamma=6\)</span> for LoF variant and <span class="math inline">\(\gamma=3\)</span> for missense when running mirage.</p>
<pre><code>## # A tibble: 6 x 7
##   X              No_case No_contr    OR      pval eta_est  eta_pval
##   &lt;fct&gt;            &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 PLI&gt;=0.995         704      564 1.25  0.0000467 0.161   0.0000663
## 2 0.5&lt;=pLI&lt;0.995    1273     1191 1.07  0.0514    0.0519  0.0698   
## 3 pLI&lt;0.5          18956    18589 1.02  0.0295    0.0186  0.0126   
## 4 MPC&gt;=2            6341     6159 1.03  0.0527    0.0391  0.0483   
## 5 1&lt;=MPC&lt;2         50355    50369 1.00  0.519     0.0107  0.794    
## 6 0&lt;=MPC&lt;1        339394   340489 0.997 0.908     0.00602 1</code></pre>
<p><img src="ASC_cell_data_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="gene-set-analysis" class="section level3">
<h3>gene set analysis</h3>
<p>Run mirage on all 17653 genes and <span class="math inline">\(\gamma=6\)</span> for LoF variants and <span class="math inline">\(\gamma=3\)</span> for missense.</p>
<table>
<thead>
<tr class="header">
<th>parameter</th>
<th>estimate</th>
<th>p value</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\hat{\delta}\)</span></td>
<td>0.15</td>
<td><span class="math inline">\(1.64\times 10^{-8}\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(\hat{\eta}_1\)</span></td>
<td>1</td>
<td><span class="math inline">\(3.52\times 10^{-6}\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\hat{\eta}_2\)</span></td>
<td>0.48</td>
<td>0.03</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\hat{\eta}_3\)</span></td>
<td>0.12</td>
<td>0.01</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\hat{\eta}_4\)</span></td>
<td>0.20</td>
<td>0.05</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\hat{\eta}_5\)</span></td>
<td>0.02</td>
<td>0.46</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\hat{\eta}_6\)</span></td>
<td>0.0014</td>
<td>1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="run-mirage-on-102-asd-genes-in-cell-paper" class="section level2">
<h2>Run mirage on 102 ASD genes in cell paper</h2>
<div id="vs-analysis" class="section level3">
<h3>VS analysis</h3>
<p>This figure shows the comparison between burden test (one side two sample poisson test) and mirage-vs for proband in 6 variant groups. <span class="math inline">\(\gamma=6\)</span> for LoF variant and <span class="math inline">\(\gamma=3\)</span> for missense when running mirage.</p>
<pre><code>## # A tibble: 6 x 7
##   X              No_case No_contr    OR    pval eta_est eta_pval
##   &lt;fct&gt;            &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;
## 1 PLI&gt;=0.995          20       14  1.43 0.196   0.278    0.243  
## 2 0.5&lt;=pLI&lt;0.995      17        5  3.4  0.00845 0.819    0.0109 
## 3 pLI&lt;0.5             21        7  3    0.00627 0.732    0.00896
## 4 MPC&gt;=2             267      232  1.15 0.0640  0.105    0.101  
## 5 1&lt;=MPC&lt;2           934      920  1.02 0.381   0.0214   0.517  
## 6 0&lt;=MPC&lt;1          2697     2640  1.02 0.222   0.00618  1</code></pre>
<p><img src="ASC_cell_data_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="gene-set-analysis-1" class="section level3">
<h3>gene set analysis</h3>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
