<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>ASD WES data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ASD WES data</h1>
<h4 class="author"><em>Shengtong Han</em></h4>
<h4 class="date"><em>YYYY-MM-DD</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-07-26</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4475309</p>
<!-- Add your analysis here -->
<div id="quality-control-qc" class="section level2">
<h2>Quality Control (QC)</h2>
<p>N1=N0=4315 trios.</p>
<div id="annotated-data-quality." class="section level3">
<h3>Annotated data quality.</h3>
<pre class="r"><code>N1=4315; N0=4315
All.Anno.Data[All.Anno.Data ==&quot;.&quot;] &lt;- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF&lt;1e-2 &amp; All.Anno.Data$Annotation!=&quot;synonymous SNV&quot;),] # use AF cutoff and exclude synonumous SNV
var.data=data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr)</code></pre>
<ul>
<li>sample size is of 4315</li>
<li>set ExacAF of NA to be 0</li>
<li>use <span class="math inline">\(10^{-2}\)</span> as AF cutoff and exclude synonymous SNV</li>
</ul>
<table>
<caption>Quality of annotated data</caption>
<thead>
<tr class="header">
<th align="left">item</th>
<th align="right">value.AllVar</th>
<th align="right">value.withoutsyn</th>
<th align="right">value.nonsyn</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">#{Variants}</td>
<td align="right">7.300520e+05</td>
<td align="right">5.313820e+05</td>
<td align="right">3.464720e+05</td>
</tr>
<tr class="even">
<td align="left">#{Gene}</td>
<td align="right">1.562400e+04</td>
<td align="right">1.538100e+04</td>
<td align="right">1.530900e+04</td>
</tr>
<tr class="odd">
<td align="left">rate.ca</td>
<td align="right">5.291810e+03</td>
<td align="right">3.206519e+03</td>
<td align="right">1.698942e+03</td>
</tr>
<tr class="even">
<td align="left">rate.co</td>
<td align="right">5.300939e+03</td>
<td align="right">3.213075e+03</td>
<td align="right">1.703369e+03</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Polyphen2.HDIV.score</td>
<td align="right">4.756415e-01</td>
<td align="right">6.525946e-01</td>
<td align="right">9.946836e-01</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_SIFT.score</td>
<td align="right">4.741964e-01</td>
<td align="right">6.468040e-01</td>
<td align="right">9.600256e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_CADD.raw</td>
<td align="right">5.144099e-01</td>
<td align="right">6.785194e-01</td>
<td align="right">9.970993e-01</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_Exon.Brain.Exp</td>
<td align="right">4.545169e-01</td>
<td align="right">4.022887e-01</td>
<td align="right">5.789876e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Annotation</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_Exon.Exac.Cons</td>
<td align="right">7.188022e-01</td>
<td align="right">6.429856e-01</td>
<td align="right">9.177076e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_ExacAF</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
</tbody>
</table>
<p>The table summaries the ratio of non-missing values in every annotation feature from different types of variants</p>
<ul>
<li>first column: all variants, i.e. everything; second column: all variants excluding synonymous variants; third column: non-synonymous variants only</li>
<li>“Annotation” row means nonsynonymous SNV, intronic, etc.</li>
</ul>
</div>
<div id="variant-rate-at-different-af-cutoffs" class="section level3">
<h3>Variant rate at different AF cutoffs</h3>
<p>“private” variants is defined as the variants whose ExAC AF is NA or 0.</p>
<table>
<caption>Rate of variants in cases and controls at varying AF cutoffs</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">No.rows</th>
<th align="right">No.var.ca</th>
<th align="right">No.var.co</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td align="right">730052</td>
<td align="right">22834161</td>
<td align="right">22873551</td>
<td align="right">5291.81020</td>
<td align="right">5300.93882</td>
</tr>
<tr class="even">
<td>MAF&lt;0.01</td>
<td align="right">702842</td>
<td align="right">1709362</td>
<td align="right">1718204</td>
<td align="right">396.14415</td>
<td align="right">398.19328</td>
</tr>
<tr class="odd">
<td>MAF&lt;0.001</td>
<td align="right">650546</td>
<td align="right">674891</td>
<td align="right">673540</td>
<td align="right">156.40579</td>
<td align="right">156.09270</td>
</tr>
<tr class="even">
<td>MAF&lt;1e-04</td>
<td align="right">513457</td>
<td align="right">353965</td>
<td align="right">350677</td>
<td align="right">82.03129</td>
<td align="right">81.26929</td>
</tr>
<tr class="odd">
<td>MAF&lt;1e-05</td>
<td align="right">280151</td>
<td align="right">166399</td>
<td align="right">164355</td>
<td align="right">38.56292</td>
<td align="right">38.08922</td>
</tr>
<tr class="even">
<td>private</td>
<td align="right">186581</td>
<td align="right">108035</td>
<td align="right">106789</td>
<td align="right">25.03708</td>
<td align="right">24.74832</td>
</tr>
</tbody>
</table>
</div>
<div id="burden-of-synonymous-mutations" class="section level3">
<h3>Burden of synonymous mutations</h3>
<p>Test: one side (“greater”) poisson test to see if the rate in cases is signifcantly greater than that in controls.</p>
<table>
<caption>Burden analysis of synonymous mutations at verying AF cutoffs</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td align="right">0.9987678</td>
<td align="right">0.9955549</td>
<td align="right">2085.291309</td>
<td align="right">2087.863963</td>
</tr>
<tr class="even">
<td>MAF&lt;0.01</td>
<td align="right">0.9957651</td>
<td align="right">0.9857511</td>
<td align="right">123.098030</td>
<td align="right">123.621553</td>
</tr>
<tr class="odd">
<td>MAF&lt;0.001</td>
<td align="right">1.0057524</td>
<td align="right">0.0392774</td>
<td align="right">43.800927</td>
<td align="right">43.550406</td>
</tr>
<tr class="even">
<td>MAF&lt;1e-04</td>
<td align="right">1.0164539</td>
<td align="right">0.0002454</td>
<td align="right">21.346002</td>
<td align="right">21.000463</td>
</tr>
<tr class="odd">
<td>MAF&lt;1e-05</td>
<td align="right">1.0232528</td>
<td align="right">0.0006142</td>
<td align="right">9.290614</td>
<td align="right">9.079490</td>
</tr>
<tr class="even">
<td>private</td>
<td align="right">1.0106303</td>
<td align="right">0.1182712</td>
<td align="right">5.882735</td>
<td align="right">5.820857</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="variant-level-analysis" class="section level2">
<h2>Variant level analysis</h2>
<div id="single-variant" class="section level3">
<h3>single variant</h3>
<pre class="r"><code>kable(summy, caption=&quot;Burden analysis of single variant features&quot;)</code></pre>
<table>
<caption>Burden analysis of single variant features</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MAF&lt;1e-2</td>
<td align="right">0.9944437</td>
<td align="right">0.9999908</td>
<td align="right">273.046118</td>
<td align="right">274.571727</td>
</tr>
<tr class="even">
<td>MAF&lt;1e-3</td>
<td align="right">1.0005560</td>
<td align="right">0.3924590</td>
<td align="right">112.604867</td>
<td align="right">112.542294</td>
</tr>
<tr class="odd">
<td>MAF&lt;1e-4</td>
<td align="right">1.0069099</td>
<td align="right">0.0064589</td>
<td align="right">60.685284</td>
<td align="right">60.268830</td>
</tr>
<tr class="even">
<td>Prby Damaging</td>
<td align="right">0.9950484</td>
<td align="right">0.9651177</td>
<td align="right">61.614368</td>
<td align="right">61.920973</td>
</tr>
<tr class="odd">
<td>Psbl Damaging</td>
<td align="right">0.9908945</td>
<td align="right">0.9893700</td>
<td align="right">29.204635</td>
<td align="right">29.473001</td>
</tr>
<tr class="even">
<td>SIFT&lt;0.05</td>
<td align="right">0.9984344</td>
<td align="right">0.7043431</td>
<td align="right">54.092932</td>
<td align="right">54.177752</td>
</tr>
<tr class="odd">
<td>CADD top10%</td>
<td align="right">1.0029500</td>
<td align="right">0.3174970</td>
<td align="right">12.212746</td>
<td align="right">12.176825</td>
</tr>
<tr class="even">
<td>BrainExp top10%</td>
<td align="right">0.9865076</td>
<td align="right">0.9653803</td>
<td align="right">8.201159</td>
<td align="right">8.313326</td>
</tr>
<tr class="odd">
<td>Consensus</td>
<td align="right">0.9975912</td>
<td align="right">0.8447574</td>
<td align="right">81.675782</td>
<td align="right">81.873001</td>
</tr>
<tr class="even">
<td>LoF</td>
<td align="right">0.9860321</td>
<td align="right">0.9596908</td>
<td align="right">7.067439</td>
<td align="right">7.167555</td>
</tr>
<tr class="odd">
<td>Private</td>
<td align="right">1.0119870</td>
<td align="right">0.0079189</td>
<td align="right">19.154345</td>
<td align="right">18.927462</td>
</tr>
</tbody>
</table>
<p>Test: use two sample one side poisson test (“greater”)</p>
<ul>
<li>probably damaging:Polyphen2.HDIV.score &gt;=0.957</li>
<li>possibly damaging: &lt;0.957 &amp; &gt;=0.453;</li>
<li>consensus: “Prby Damaging”, or “SIFT&lt;0.05”, or “CADD top10%” (take union)</li>
<li>LoF:stopgain+ frameshift substitution+splicing+stoploss (take union)</li>
</ul>
</div>
<div id="lof-in-different-gene-set" class="section level3">
<h3>LoF in different gene set</h3>
</div>
</div>
<div id="exon-level-analysis" class="section level2">
<h2>Exon level analysis</h2>
<p>Test: use two sample one side poisson test</p>
<ul>
<li>high values means more constraint</li>
<li>critical exon=brian expression exon top 25%+exon constraint score top 25%</li>
</ul>
<pre class="r"><code>kable(exon.summ, caption=&quot;Burden analysis of exons&quot;)</code></pre>
<table>
<caption>Burden analysis of exons</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Top%90</td>
<td align="right">1.0013548</td>
<td align="right">0.4181860</td>
<td align="right">11.13395</td>
<td align="right">11.11889</td>
</tr>
<tr class="even">
<td>Top%80</td>
<td align="right">0.9972590</td>
<td align="right">0.7387544</td>
<td align="right">24.95805</td>
<td align="right">25.02665</td>
</tr>
<tr class="odd">
<td>Top%70</td>
<td align="right">0.9961925</td>
<td align="right">0.8685243</td>
<td align="right">39.71587</td>
<td align="right">39.86767</td>
</tr>
<tr class="even">
<td>Top%60</td>
<td align="right">0.9969765</td>
<td align="right">0.8539354</td>
<td align="right">55.86188</td>
<td align="right">56.03129</td>
</tr>
<tr class="odd">
<td>Top%50</td>
<td align="right">0.9978200</td>
<td align="right">0.8065358</td>
<td align="right">72.55411</td>
<td align="right">72.71263</td>
</tr>
<tr class="even">
<td>critical exon</td>
<td align="right">0.9927310</td>
<td align="right">0.9892038</td>
<td align="right">45.73441</td>
<td align="right">46.06929</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-level-analysis" class="section level2">
<h2>Gene level analysis</h2>
<p><strong>Gene level: two sample one side (“greater”) test; high confidence gene: q&lt;0.1, SFARI cate1&amp;2; Moder confi: 0.3&gt;q&gt;0.1 SFARI cat3 &amp; S</strong></p>
<pre class="r"><code>kable(gene.summy, caption=&quot;Burden analysis of gene set&quot;)</code></pre>
<table>
<caption>Burden analysis of gene set</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cate1</td>
<td align="right">1.0433640</td>
<td align="right">0.2085456</td>
<td align="right">0.1840093</td>
<td align="right">0.1763615</td>
</tr>
<tr class="even">
<td>cate2</td>
<td align="right">0.9930499</td>
<td align="right">0.6017183</td>
<td align="right">0.5629200</td>
<td align="right">0.5668598</td>
</tr>
<tr class="odd">
<td>cate3</td>
<td align="right">1.0087867</td>
<td align="right">0.3286067</td>
<td align="right">1.2505214</td>
<td align="right">1.2396292</td>
</tr>
<tr class="even">
<td>cate4</td>
<td align="right">1.0067047</td>
<td align="right">0.2896337</td>
<td align="right">3.2709154</td>
<td align="right">3.2491309</td>
</tr>
<tr class="odd">
<td>cate5</td>
<td align="right">1.0174530</td>
<td align="right">0.2319454</td>
<td align="right">0.8646582</td>
<td align="right">0.8498262</td>
</tr>
<tr class="even">
<td>cate6</td>
<td align="right">0.9539171</td>
<td align="right">0.8402698</td>
<td align="right">0.1918888</td>
<td align="right">0.2011587</td>
</tr>
<tr class="odd">
<td>cateS</td>
<td align="right">0.9971525</td>
<td align="right">0.5543337</td>
<td align="right">0.8926999</td>
<td align="right">0.8952491</td>
</tr>
<tr class="even">
<td>TADAq&lt;5%</td>
<td align="right">1.0482375</td>
<td align="right">0.0904492</td>
<td align="right">0.3928158</td>
<td align="right">0.3747393</td>
</tr>
<tr class="odd">
<td>TADAq&lt;20%</td>
<td align="right">1.0183183</td>
<td align="right">0.1955915</td>
<td align="right">1.0692932</td>
<td align="right">1.0500579</td>
</tr>
<tr class="even">
<td>TADAq&lt;30%</td>
<td align="right">1.0091811</td>
<td align="right">0.2831035</td>
<td align="right">1.8850521</td>
<td align="right">1.8679027</td>
</tr>
<tr class="odd">
<td>TADAq&lt;40%</td>
<td align="right">1.0028665</td>
<td align="right">0.4152328</td>
<td align="right">2.7566628</td>
<td align="right">2.7487833</td>
</tr>
<tr class="even">
<td>TADAq&lt;50%</td>
<td align="right">0.9818818</td>
<td align="right">0.9629683</td>
<td align="right">4.3580533</td>
<td align="right">4.4384705</td>
</tr>
<tr class="odd">
<td>TADAq&gt;90%</td>
<td align="right">0.9928361</td>
<td align="right">0.9996247</td>
<td align="right">101.4289687</td>
<td align="right">102.1608343</td>
</tr>
<tr class="even">
<td>ID gene</td>
<td align="right">1.0086483</td>
<td align="right">0.2702458</td>
<td align="right">2.4055620</td>
<td align="right">2.3849363</td>
</tr>
<tr class="odd">
<td>High</td>
<td align="right">1.0056386</td>
<td align="right">0.4108869</td>
<td align="right">0.8266512</td>
<td align="right">0.8220162</td>
</tr>
<tr class="even">
<td>Mod</td>
<td align="right">0.9950643</td>
<td align="right">0.6591709</td>
<td align="right">3.0836616</td>
<td align="right">3.0989571</td>
</tr>
<tr class="odd">
<td>PSD</td>
<td align="right">0.9944588</td>
<td align="right">0.7106721</td>
<td align="right">4.5334878</td>
<td align="right">4.5587486</td>
</tr>
<tr class="even">
<td>FMRP</td>
<td align="right">0.9955732</td>
<td align="right">0.7322074</td>
<td align="right">8.9123986</td>
<td align="right">8.9520278</td>
</tr>
<tr class="odd">
<td>AutismKB</td>
<td align="right">1.0112301</td>
<td align="right">0.2219197</td>
<td align="right">2.2329085</td>
<td align="right">2.2081112</td>
</tr>
<tr class="even">
<td>constraint gene</td>
<td align="right">0.9848970</td>
<td align="right">0.9501030</td>
<td align="right">5.3499421</td>
<td align="right">5.4319815</td>
</tr>
<tr class="odd">
<td>RVIS</td>
<td align="right">1.0036775</td>
<td align="right">0.2707991</td>
<td align="right">12.9663963</td>
<td align="right">12.9188876</td>
</tr>
<tr class="even">
<td>Haploinsuff gene</td>
<td align="right">0.9975098</td>
<td align="right">0.7491371</td>
<td align="right">33.4203940</td>
<td align="right">33.5038239</td>
</tr>
<tr class="odd">
<td>SCZ gene</td>
<td align="right">0.9986123</td>
<td align="right">0.6154038</td>
<td align="right">20.3462341</td>
<td align="right">20.3745075</td>
</tr>
<tr class="even">
<td>Olfac.gene</td>
<td align="right">0.9806059</td>
<td align="right">0.9825049</td>
<td align="right">5.2964079</td>
<td align="right">5.4011587</td>
</tr>
<tr class="odd">
<td>67SCZriskgene</td>
<td align="right">0.9976476</td>
<td align="right">0.5475076</td>
<td align="right">0.9828505</td>
<td align="right">0.9851680</td>
</tr>
</tbody>
</table>
</div>
<div id="combined-feature-analysis" class="section level2">
<h2>Combined feature analysis</h2>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.2 (2015-08-14)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.15.1  dplyr_0.4.3   RSQLite_1.0.0 DBI_0.3.1    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.1     digest_0.6.8    rprojroot_1.2   assertthat_0.1 
 [5] R6_2.1.1        backports_1.0.5 git2r_0.18.0    magrittr_1.5   
 [9] evaluate_0.10   highr_0.5.1     stringi_0.5-5   rmarkdown_1.3  
[13] tools_3.2.2     stringr_1.2.0   parallel_3.2.2  yaml_2.1.14    
[17] htmltools_0.3.5</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
