<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>ASD_dbGaP</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ASD_dbGaP</h1>

</div>


<p>Four cohort studies</p>
<table>
<thead>
<tr class="header">
<th>Group</th>
<th>No.subject</th>
<th>No.var called</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DS-ASD</td>
<td>10008</td>
<td>3804448</td>
</tr>
<tr class="even">
<td>GRU</td>
<td>2509</td>
</tr>
<tr class="odd">
<td>DS-AOND-MDS</td>
<td>162</td>
</tr>
<tr class="even">
<td>HMB-MDS</td>
<td>93</td>
</tr>
<tr class="odd">
<td>Total</td>
<td>12772</td>
</tr>
</tbody>
</table>
<div id="ds-asd-10008-individuals" class="section level2">
<h2><a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000298.v4.p3">DS-ASD</a> 10008 individuals</h2>
<pre class="r"><code> bcftools stats pseudocc.tucc.vcf
# This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
# The command line was: bcftools stats  pseudocc.tucc.vcf
#
# Definition of sets:
# ID    [2]id   [3]tab-separated file names
ID  0   pseudocc.tucc.vcf
# SN, Summary numbers:
#   number of records   .. number of data rows in the VCF
#   number of no-ALTs   .. reference-only sites, ALT is either &quot;.&quot; or identical to REF
#   number of SNPs      .. number of rows with a SNP
#   number of MNPs      .. number of rows with a MNP, such as CC&gt;TT
#   number of indels    .. number of rows with an indel
#   number of others    .. number of rows with other type, for example a symbolic allele or
#                          a complex substitution, such as ACT&gt;TCGA
#   number of multiallelic sites     .. number of rows with multiple alternate alleles
#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
#
#   Note that rows containing multiple types will be counted multiple times, in each
#   counter. For example, a row with a SNP and an indel increments both the SNP and
#   the indel counter.
#
# SN    [2]id   [3]key  [4]value
SN  0   number of samples:  5512
SN  0   number of records:  3452066
SN  0   number of no-ALTs:  35521
SN  0   number of SNPs: 3193922
SN  0   number of MNPs: 0
SN  0   number of indels:   212812
SN  0   number of others:   9811
SN  0   number of multiallelic sites:   0
SN  0   number of multiallelic SNP sites:   0
# TSTV, transitions/transversions:
# TSTV  [2]id   [3]ts   [4]tv   [5]ts/tv    [6]ts (1st ALT) [7]tv (1st ALT) [8]ts/tv (1st ALT)
TSTV    0   2250696 942762  2.39    2250696 942762  2.39
# SiS, Singleton stats:
# SiS   [2]id   [3]allele count [4]number of SNPs   [5]number of transitions    [6]number of transversions  [7]number of indels [8]repeat-consistent    [9]repeat-inconsistent  [10]not applicable
SiS 0   1   3193458 2250696 942762  212812  0   0   212812
# AF, Stats by non-reference allele frequency:
# AF    [2]id   [3]allele frequency [4]number of SNPs   [5]number of transitions    [6]number of transversions  [7]number of indels [8]repeat-consistent    [9]repeat-inconsistent  [10]not applicable
AF  0   0.000000    3193458 2250696 942762  212812  0   0   212812
# QUAL, Stats by quality:
# QUAL  [2]id   [3]Quality  [4]number of SNPs   [5]number of transitions (1st ALT)  [6]number of transversions (1st ALT)    [7]number of indels
QUAL    0   998 3193458 2250696 942762  212812
# IDD, InDel distribution:
# IDD   [2]id   [3]length (deletions negative)  [4]count
IDD 0   -60 838
IDD 0   -59 10
IDD 0   -58 12
IDD 0   -57 32
IDD 0   -56 14
IDD 0   -55 10
IDD 0   -54 35
IDD 0   -53 12
IDD 0   -52 16
IDD 0   -51 28
IDD 0   -50 24
IDD 0   -49 21
IDD 0   -48 94
IDD 0   -47 19
IDD 0   -46 35
IDD 0   -45 57
IDD 0   -44 35
IDD 0   -43 21
IDD 0   -42 82
IDD 0   -41 40
IDD 0   -40 42
IDD 0   -39 79
IDD 0   -38 48
IDD 0   -37 41
IDD 0   -36 106
IDD 0   -35 58
IDD 0   -34 74
IDD 0   -33 106
IDD 0   -32 69
IDD 0   -31 79
IDD 0   -30 232
IDD 0   -29 112
IDD 0   -28 104
IDD 0   -27 285
IDD 0   -26 159
IDD 0   -25 167
IDD 0   -24 432
IDD 0   -23 219
IDD 0   -22 245
IDD 0   -21 589
IDD 0   -20 341
IDD 0   -19 335
IDD 0   -18 663
IDD 0   -17 398
IDD 0   -16 489
IDD 0   -15 908
IDD 0   -14 652
IDD 0   -13 722
IDD 0   -12 1611
IDD 0   -11 932
IDD 0   -10 1144
IDD 0   -9  1895
IDD 0   -8  1347
IDD 0   -7  1649
IDD 0   -6  3548
IDD 0   -5  4679
IDD 0   -4  12231
IDD 0   -3  23718
IDD 0   -2  23919
IDD 0   -1  54129
IDD 0   1   41404
IDD 0   2   6596
IDD 0   3   5461
IDD 0   4   4154
IDD 0   5   1968
IDD 0   6   2319
IDD 0   7   1039
IDD 0   8   979
IDD 0   9   1093
IDD 0   10  609
IDD 0   11  489
IDD 0   12  664
IDD 0   13  345
IDD 0   14  343
IDD 0   15  489
IDD 0   16  318
IDD 0   17  285
IDD 0   18  523
IDD 0   19  234
IDD 0   20  254
IDD 0   21  370
IDD 0   22  210
IDD 0   23  157
IDD 0   24  236
IDD 0   25  123
IDD 0   26  109
IDD 0   27  150
IDD 0   28  101
IDD 0   29  89
IDD 0   30  128
IDD 0   31  66
IDD 0   32  75
IDD 0   33  85
IDD 0   34  51
IDD 0   35  51
IDD 0   36  68
IDD 0   37  55
IDD 0   38  48
IDD 0   39  58
IDD 0   40  55
IDD 0   41  39
IDD 0   42  71
IDD 0   43  50
IDD 0   44  33
IDD 0   45  36
IDD 0   46  49
IDD 0   47  29
IDD 0   48  63
IDD 0   49  37
IDD 0   50  28
IDD 0   51  35
IDD 0   52  18
IDD 0   53  16
IDD 0   54  15
IDD 0   55  13
IDD 0   56  9
IDD 0   57  28
IDD 0   58  8
IDD 0   59  8
IDD 0   60  385
# ST, Substitution types:
# ST    [2]id   [3]type [4]count
ST  0   A&gt;C 93692
ST  0   A&gt;G 338794
ST  0   A&gt;T 74903
ST  0   C&gt;A 142266
ST  0   C&gt;G 162908
ST  0   C&gt;T 787530
ST  0   G&gt;A 784925
ST  0   G&gt;C 162209
ST  0   G&gt;T 142017
ST  0   T&gt;A 72872
ST  0   T&gt;C 339447
ST  0   T&gt;G 91895
# DP, Depth distribution
# DP    [2]id   [3]bin  [4]number of genotypes  [5]fraction of genotypes (%)    [6]number of sites  [7]fraction of sites (%)</code></pre>
</div>
<div id="hmb-mds-93-individuals" class="section level2">
<h2><a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000298.v4.p3">HMB-MDS</a>: 93 individuals</h2>
<div id="phenotype-.txt-file" class="section level3">
<h3>Phenotype: .txt file</h3>
<pre class="r"><code># Study accession: phs000298.v4.p3
# Table accession: pht002283.v3.p3.c4
# Consent group: Health/Medical/Biomedical (MDS)
# Citation instructions: The study accession (phs000298.v4.p3) is used to cite the study and its data tables and documents. The data in this file should be cited using the accession pht002283.v3.p3.c4.
# To cite columns of data within this file, please use the variable (phv#) accessions below:
#
# 1) the table name and the variable (phv#) accessions below; or
# 2) you may cite a variable as phv#.v3.p3.c4.

##      phv00160418.v3.p3.c4    phv00167339.v3.p3.c4    phv00167341.v3.p3.c4    phv00167342.v3.p3.c4    phv00369219.v1.p3.c4
dbGaP_Subject_ID        SUBJID  SEX     AGE     RACE    DIAGNOSIS
2533851 19-952879-11    M                       CONTROL</code></pre>
</div>
<div id="genotype-vcf-file" class="section level3">
<h3>Genotype: vcf file</h3>
<pre class="r"><code>##fileformat=VCFv4.1
##ALT=&lt;ID=NON_REF,Description=&quot;Represents any possible alternative allele at this location&quot;&gt;
##FILTER=&lt;ID=InbreedingCoeff,Description=&quot;InbreedingCoeff &lt; -0.3&quot;&gt;
##FILTER=&lt;ID=LowQual,Description=&quot;Low quality&quot;&gt;
##FILTER=&lt;ID=VQSRTrancheINDEL95.00to96.00,Description=&quot;Truth sensitivity tranche level for INDEL model at VQS Lod: 0.6951 &lt;= x &lt; 0.8703&quot;&gt;
##FILTER=&lt;ID=VQSRTrancheINDEL96.00to97.00,Description=&quot;Truth sensitivity tranche level for INDEL model at VQS Lod: 0.5424 &lt;= x &lt; 0.6951&quot;&gt;
##FILTER=&lt;ID=VQSRTrancheINDEL97.00to99.00,Description=&quot;Truth sensitivity tranche level for INDEL model at VQS Lod: -0.2106 &lt;= x &lt; 0.5424&quot;&gt;
##FILTER=&lt;ID=VQSRTrancheINDEL99.00to99.50,Description=&quot;Truth sensitivity tranche level for INDEL model at VQS Lod: -0.9744 &lt;= x &lt; -0.2106&quot;&gt;
##FILTER=&lt;ID=VQSRTrancheINDEL99.50to99.90,Description=&quot;Truth sensitivity tranche level for INDEL model at VQS Lod: -9.6841 &lt;= x &lt; -0.9744&quot;&gt;
##FILTER=&lt;ID=VQSRTrancheINDEL99.90to99.95,Description=&quot;Truth sensitivity tranche level for INDEL model at VQS Lod: -26.8213 &lt;= x &lt; -9.6841&quot;&gt;</code></pre>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
