<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>ASD_dbGaP</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ASD_dbGaP</h1>

</div>


<p>Four cohort studies</p>
<table>
<thead>
<tr class="header">
<th>Group</th>
<th>No.subject</th>
<th>No.var called</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DS-ASD</td>
<td>10008</td>
<td>3452056 (locus with observed alleles)</td>
</tr>
<tr class="even">
<td>GRU</td>
<td>2509</td>
</tr>
<tr class="odd">
<td>DS-AOND-MDS</td>
<td>162</td>
</tr>
<tr class="even">
<td>HMB-MDS</td>
<td>93</td>
</tr>
<tr class="odd">
<td>Total</td>
<td>12772</td>
</tr>
</tbody>
</table>
<div id="ds-asd-sample" class="section level2">
<h2>DS-ASD sample</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000298.v4.p3">Sample data</a> has 10008 individuals</p>
</div>
<div id="old-sample-vs-ds-asd-sample" class="section level2">
<h2>old sample vs DS-ASD sample</h2>
<div id="sample-decomposition-by-af" class="section level3">
<h3>sample decomposition by AF</h3>
<p><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre><code>## # A tibble: 10 x 3
##    cutoff     group   counts
##    &lt;chr&gt;      &lt;chr&gt;    &lt;dbl&gt;
##  1 All        old     730052
##  2 AF&lt;0.1%    old     464077
##  3 0.1%&lt;AF&lt;1% old      52143
##  4 1%&lt;AF&lt;5%   old      11145
##  5 AF&gt;5%      old      15379
##  6 All        DS-ASD 3451246
##  7 AF&lt;0.1%    DS-ASD 1706806
##  8 0.1%&lt;AF&lt;1% DS-ASD  222402
##  9 1%&lt;AF&lt;5%   DS-ASD   59489
## 10 AF&gt;5%      DS-ASD   75381</code></pre>
<p><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-3-1.png" width="672" /><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>
</div>
<div id="several-top-genes" class="section level3">
<h3>several top genes</h3>
<pre><code>##                    No.case No.control
## DS.ASD CYFIP1          179        198
## DS.ASD EP400          1082       1881
## DS.ASD FBN1            195        234
## DS.ASD SRCAP           339        458
## DS.ASD DYNC1H1          80        124
## DS.ASD ABCA2           226        246
## DS.ASD DOCK4           503        516
## DS.ASD CACNA1D         124        187
## DS.ASD CHD8            136        326
## DS.ASD TRIP12           33         34
## old.sample CYFIP1      138        147
## old.sample EP400        51         48
## old.sample FBN1        271        227
## old.sample SRCAP       182        148
## old.sample DYNC1H1      45         45
## old.sample ABCA2        16          3
## old.sample DOCK4       325        340
## old.sample CACNA1D     131        101
## old.sample CHD8         62         52
## old.sample TRIP12       26         19</code></pre>
<p><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="gene-cyfip1" class="section level3">
<h3>Gene CYFIP1</h3>
<pre class="r"><code>All.Anno.rare.var%&gt;%filter(Gene==&quot;CYFIP1&quot;, ID==&quot;15:22963816-22963816_A_G&quot;)</code></pre>
<pre><code>## # A tibble: 1 x 11
##   ID    Gene  No.case No.contr Polyphen2.HDIV.~ SIFT.score CADD.raw
##   &lt;fct&gt; &lt;fct&gt;   &lt;int&gt;    &lt;int&gt; &lt;fct&gt;            &lt;fct&gt;      &lt;fct&gt;   
## 1 15:2~ CYFI~       8        0 0.998            0.19       3.585   
## # ... with 4 more variables: Exon.Brain.Exp &lt;dbl&gt;, Annotation &lt;fct&gt;,
## #   Exon.Exac.Cons &lt;dbl&gt;, ExacAF &lt;dbl&gt;</code></pre>
<pre class="r"><code>DS.ASD.rare.var%&gt;%filter(Gene==&quot;CYFIP1&quot;, ID==&quot;chr15:22963816_22963817_A_G&quot;)</code></pre>
<pre><code>## # A tibble: 1 x 9
##   ID    No.case No.contr Gene  Annotation ExacAF Polyphen2.HDIV CADD.raw
##   &lt;fct&gt;   &lt;int&gt;    &lt;int&gt; &lt;fct&gt; &lt;fct&gt;       &lt;dbl&gt; &lt;fct&gt;          &lt;fct&gt;   
## 1 chr1~       6        2 CYFI~ nonsynony~ 0.0002 .              .       
## # ... with 1 more variable: SIFT.score &lt;fct&gt;</code></pre>
<table>
<caption>chr15:22963816-22963816_A~G</caption>
<thead>
<tr class="header">
<th></th>
<th>No.case</th>
<th>No.control</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>old</td>
<td>8</td>
<td>0</td>
</tr>
<tr class="even">
<td>DS-ASD</td>
<td>6</td>
<td>2</td>
</tr>
</tbody>
</table>
</div>
<div id="qc-for-annotations" class="section level3">
<h3>QC for annotations</h3>
<table>
<caption>Old sample</caption>
<thead>
<tr class="header">
<th align="left">item</th>
<th align="right">value.AllVar</th>
<th align="right">value.withoutsyn</th>
<th align="right">value.nonsyn</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">#{Variants}</td>
<td align="right">7.300520e+05</td>
<td align="right">5.313820e+05</td>
<td align="right">3.464720e+05</td>
</tr>
<tr class="even">
<td align="left">#{Gene}</td>
<td align="right">1.562400e+04</td>
<td align="right">1.538100e+04</td>
<td align="right">1.530900e+04</td>
</tr>
<tr class="odd">
<td align="left">rate.ca</td>
<td align="right">5.291810e+03</td>
<td align="right">3.206519e+03</td>
<td align="right">1.698942e+03</td>
</tr>
<tr class="even">
<td align="left">rate.co</td>
<td align="right">5.300939e+03</td>
<td align="right">3.213075e+03</td>
<td align="right">1.703369e+03</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Polyphen2.HDIV.score</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_SIFT.score</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_CADD.raw</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_Exon.Brain.Exp</td>
<td align="right">4.545169e-01</td>
<td align="right">4.022887e-01</td>
<td align="right">5.789876e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Annotation</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_Exon.Exac.Cons</td>
<td align="right">7.188022e-01</td>
<td align="right">6.429856e-01</td>
<td align="right">9.177076e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_ExacAF</td>
<td align="right">7.445812e-01</td>
<td align="right">7.305629e-01</td>
<td align="right">7.281512e-01</td>
</tr>
</tbody>
</table>
<table>
<caption>DS-ASD sample</caption>
<thead>
<tr class="header">
<th align="left">item</th>
<th align="right">value.AllVar</th>
<th align="right">value.withoutsyn</th>
<th align="right">value.nonsyn</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">#{Variants}</td>
<td align="right">3.451246e+06</td>
<td align="right">1.157408e+06</td>
<td align="right">1.108935e+06</td>
</tr>
<tr class="even">
<td align="left">#{Gene}</td>
<td align="right">1.827400e+04</td>
<td align="right">1.825300e+04</td>
<td align="right">1.824500e+04</td>
</tr>
<tr class="odd">
<td align="left">rate.ca</td>
<td align="right">5.143263e+04</td>
<td align="right">7.781897e+03</td>
<td align="right">7.365552e+03</td>
</tr>
<tr class="even">
<td align="left">rate.co</td>
<td align="right">5.529460e+04</td>
<td align="right">8.226857e+03</td>
<td align="right">7.777235e+03</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Annotation</td>
<td align="right">5.151873e-01</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_ExacAF</td>
<td align="right">6.026377e-01</td>
<td align="right">6.538861e-01</td>
<td align="right">6.565371e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Polyphen2.HDIV</td>
<td align="right">9.153086e-01</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_CADD.raw</td>
<td align="right">9.153086e-01</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_SIFT.score</td>
<td align="right">9.153086e-01</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="gene-set-analysis" class="section level2">
<h2>Gene set analysis</h2>
<pre class="r"><code>N1=2403; N0=2403
DS.ASD.sample[DS.ASD.sample ==&quot;.&quot;] &lt;- NA
DS.ASD.sample$ExacAF[is.na(DS.ASD.sample$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=DS.ASD.sample[which(DS.ASD.sample$ExacAF&lt;0.05 &amp; DS.ASD.sample$Annotation!=&quot;synonymous SNV&quot;),] # use AF cutoff and exclude synonumous SNV
var.data=as_tibble(data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr))
LoF.def=c(&quot;stopgain&quot;, &quot;frameshift substitution&quot;, &quot;splicing&quot;, &quot;stoploss&quot;)
LoF.var=as.character(Anno.Data$ID[which(Anno.Data$Annotation %in% LoF.def==T)]) </code></pre>
<div id="simple-burden-test" class="section level3">
<h3>Simple burden test</h3>
<pre><code>## Warning: `src_sqlite()` is deprecated as of dplyr 1.0.0.
## Please use `tbl()` directly with a database connection
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre><code>## 
## Attaching package: &#39;kableExtra&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     group_rows</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Burden analysis of gene set
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rate.ca
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rate.co
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cate1
</td>
<td style="text-align:right;">
0.7709677
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.8951311
</td>
<td style="text-align:right;">
1.161049
</td>
</tr>
<tr>
<td style="text-align:left;">
cate2
</td>
<td style="text-align:right;">
0.7211256
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3.7856846
</td>
<td style="text-align:right;">
5.249688
</td>
</tr>
<tr>
<td style="text-align:left;">
cate3
</td>
<td style="text-align:right;">
0.8416703
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
8.0944653
</td>
<td style="text-align:right;">
9.617145
</td>
</tr>
<tr>
<td style="text-align:left;">
cate4
</td>
<td style="text-align:right;">
0.8854299
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
20.2517686
</td>
<td style="text-align:right;">
22.872243
</td>
</tr>
<tr>
<td style="text-align:left;">
cate5
</td>
<td style="text-align:right;">
0.9326654
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
6.2367873
</td>
<td style="text-align:right;">
6.687058
</td>
</tr>
<tr>
<td style="text-align:left;">
cate6
</td>
<td style="text-align:right;">
0.8429837
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1.0299625
</td>
<td style="text-align:right;">
1.221806
</td>
</tr>
<tr>
<td style="text-align:left;">
cateS
</td>
<td style="text-align:right;">
0.8387536
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.6346234
</td>
<td style="text-align:right;">
6.717853
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;5%
</td>
<td style="text-align:right;">
0.7734742
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2.7424053
</td>
<td style="text-align:right;">
3.545568
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;20%
</td>
<td style="text-align:right;">
0.8283586
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
7.4490221
</td>
<td style="text-align:right;">
8.992509
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;30%
</td>
<td style="text-align:right;">
0.7921760
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
11.8651685
</td>
<td style="text-align:right;">
14.977944
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;40%
</td>
<td style="text-align:right;">
0.8030515
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
18.8805660
</td>
<td style="text-align:right;">
23.511028
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;50%
</td>
<td style="text-align:right;">
0.8246703
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
30.0083229
</td>
<td style="text-align:right;">
36.388265
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&gt;90%
</td>
<td style="text-align:right;">
0.8519021
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
773.1598002
</td>
<td style="text-align:right;">
907.568872
</td>
</tr>
<tr>
<td style="text-align:left;">
ID gene
</td>
<td style="text-align:right;">
0.9093362
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
12.6217228
</td>
<td style="text-align:right;">
13.880150
</td>
</tr>
<tr>
<td style="text-align:left;">
High conf
</td>
<td style="text-align:right;">
0.7316017
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5.6966292
</td>
<td style="text-align:right;">
7.786517
</td>
</tr>
<tr>
<td style="text-align:left;">
Mod conf
</td>
<td style="text-align:right;">
0.8271188
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
18.5359967
</td>
<td style="text-align:right;">
22.410320
</td>
</tr>
<tr>
<td style="text-align:left;">
PSD
</td>
<td style="text-align:right;">
0.8428174
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
30.9783604
</td>
<td style="text-align:right;">
36.755722
</td>
</tr>
<tr>
<td style="text-align:left;">
FMRP
</td>
<td style="text-align:right;">
0.8319357
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
65.2721598
</td>
<td style="text-align:right;">
78.458177
</td>
</tr>
<tr>
<td style="text-align:left;">
AutismKB
</td>
<td style="text-align:right;">
0.7948590
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
13.9492301
</td>
<td style="text-align:right;">
17.549313
</td>
</tr>
<tr>
<td style="text-align:left;">
constraint gene
</td>
<td style="text-align:right;">
0.7439750
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
52.2983770
</td>
<td style="text-align:right;">
70.295880
</td>
</tr>
<tr>
<td style="text-align:left;">
RVIS
</td>
<td style="text-align:right;">
0.8444988
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
81.9662921
</td>
<td style="text-align:right;">
97.059093
</td>
</tr>
<tr>
<td style="text-align:left;">
Haploinsuff gene
</td>
<td style="text-align:right;">
0.8815135
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
212.9300874
</td>
<td style="text-align:right;">
241.550562
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ gene
</td>
<td style="text-align:right;">
0.8249196
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
161.8160633
</td>
<td style="text-align:right;">
196.159800
</td>
</tr>
<tr>
<td style="text-align:left;">
Olfac.gene
</td>
<td style="text-align:right;">
0.8491957
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
34.9958385
</td>
<td style="text-align:right;">
41.210570
</td>
</tr>
<tr>
<td style="text-align:left;">
67SCZriskgene
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Burden analysis of LoF in different gene sets
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.ca
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.co
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cate1
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.9804688
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
cate2
</td>
<td style="text-align:right;">
0.2706245
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
351
</td>
<td style="text-align:right;">
1297
</td>
</tr>
<tr>
<td style="text-align:left;">
cate3
</td>
<td style="text-align:right;">
0.8475000
</td>
<td style="text-align:right;">
0.9887491
</td>
<td style="text-align:right;">
339
</td>
<td style="text-align:right;">
400
</td>
</tr>
<tr>
<td style="text-align:left;">
cate4
</td>
<td style="text-align:right;">
0.6938202
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
494
</td>
<td style="text-align:right;">
712
</td>
</tr>
<tr>
<td style="text-align:left;">
cate5
</td>
<td style="text-align:right;">
0.7238095
</td>
<td style="text-align:right;">
0.9872584
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
105
</td>
</tr>
<tr>
<td style="text-align:left;">
cate6
</td>
<td style="text-align:right;">
1.1428571
</td>
<td style="text-align:right;">
0.5000000
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
cateS
</td>
<td style="text-align:right;">
0.7815315
</td>
<td style="text-align:right;">
0.9997572
</td>
<td style="text-align:right;">
347
</td>
<td style="text-align:right;">
444
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;5%
</td>
<td style="text-align:right;">
0.3663366
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
101
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;20%
</td>
<td style="text-align:right;">
0.4980315
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
253
</td>
<td style="text-align:right;">
508
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;30%
</td>
<td style="text-align:right;">
0.5644372
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
346
</td>
<td style="text-align:right;">
613
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;40%
</td>
<td style="text-align:right;">
0.6133225
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
755
</td>
<td style="text-align:right;">
1231
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;50%
</td>
<td style="text-align:right;">
0.6635007
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
978
</td>
<td style="text-align:right;">
1474
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&gt;90%
</td>
<td style="text-align:right;">
0.7490092
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
24569
</td>
<td style="text-align:right;">
32802
</td>
</tr>
<tr>
<td style="text-align:left;">
ID gene
</td>
<td style="text-align:right;">
0.5024272
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
207
</td>
<td style="text-align:right;">
412
</td>
</tr>
<tr>
<td style="text-align:left;">
High conf
</td>
<td style="text-align:right;">
0.2818387
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
374
</td>
<td style="text-align:right;">
1327
</td>
</tr>
<tr>
<td style="text-align:left;">
Mod conf
</td>
<td style="text-align:right;">
0.6680851
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
628
</td>
<td style="text-align:right;">
940
</td>
</tr>
<tr>
<td style="text-align:left;">
PSD
</td>
<td style="text-align:right;">
0.5138785
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
685
</td>
<td style="text-align:right;">
1333
</td>
</tr>
<tr>
<td style="text-align:left;">
FMRP
</td>
<td style="text-align:right;">
0.5527370
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
828
</td>
<td style="text-align:right;">
1498
</td>
</tr>
<tr>
<td style="text-align:left;">
AutismKB
</td>
<td style="text-align:right;">
0.6336032
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
313
</td>
<td style="text-align:right;">
494
</td>
</tr>
<tr>
<td style="text-align:left;">
constraint gene
</td>
<td style="text-align:right;">
0.3537683
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
629
</td>
<td style="text-align:right;">
1778
</td>
</tr>
<tr>
<td style="text-align:left;">
RVIS
</td>
<td style="text-align:right;">
0.6853180
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
1433
</td>
<td style="text-align:right;">
2091
</td>
</tr>
<tr>
<td style="text-align:left;">
Haploinsuff gene
</td>
<td style="text-align:right;">
0.5026760
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
2442
</td>
<td style="text-align:right;">
4858
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ gene
</td>
<td style="text-align:right;">
0.6490760
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
3161
</td>
<td style="text-align:right;">
4870
</td>
</tr>
<tr>
<td style="text-align:left;">
Olfac.gene
</td>
<td style="text-align:right;">
0.7984705
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
1775
</td>
<td style="text-align:right;">
2223
</td>
</tr>
<tr>
<td style="text-align:left;">
67SCZriskgene
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<pre><code>## Loading required package: viridisLite</code></pre>
<pre><code>## 
## Attaching package: &#39;viridis&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:scales&#39;:
## 
##     viridis_pal</code></pre>
<pre><code>## Loading required package: magrittr</code></pre>
<pre><code>## 
## Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     set_names</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     extract</code></pre>
<pre><code>## The following object is masked from &#39;package:rlang&#39;:
## 
##     set_names</code></pre>
<p><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>
<div id="constraint-genes" class="section level2">
<h2>Constraint genes</h2>
<div id="cons-simple-burden-analysis" class="section level3">
<h3>Cons: Simple burden analysis</h3>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
The burden of 8 categories is
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
The burden of eight categories
</caption>
<thead>
<tr>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.ca
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.co
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.2669903
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
55
</td>
<td style="text-align:right;">
206
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3651399
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
574
</td>
<td style="text-align:right;">
1572
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7118617
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
13215
</td>
<td style="text-align:right;">
18564
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6773831
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
8428
</td>
<td style="text-align:right;">
12442
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6890022
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
14190
</td>
<td style="text-align:right;">
20595
</td>
</tr>
<tr>
<td style="text-align:right;">
0.8186472
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
43656
</td>
<td style="text-align:right;">
53327
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7567757
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
22645
</td>
<td style="text-align:right;">
29923
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7035852
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
22431
</td>
<td style="text-align:right;">
31881
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="ASD_dbGaP_files/figure-html/unnamed-chunk-13-1.png" width="672" /> ### Apply mirage to constraint gene set</p>
<p><a href="DS-ASD-constraint-gene.html">Here</a> it is.</p>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
