<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>Gene set analysis for SCZ data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene set analysis for SCZ data</h1>
<h4 class="author">Shengtong Han</h4>
<h4 class="date">YYYY-MM-DD</h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<pre class="r"><code>N1=2536; N0=2543
All.Anno.Data[All.Anno.Data ==&quot;.&quot;] &lt;- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF&lt;0.05 &amp; All.Anno.Data$Annotation!=&quot;synonymous SNV&quot;),] # use AF cutoff and exclude synonumous SNV
var.data=data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr)
LoF.def=c(&quot;stopgain&quot;, &quot;frameshift substitution&quot;, &quot;splicing&quot;, &quot;stoploss&quot;)
LoF.var=as.character(Anno.Data$ID[which(Anno.Data$Annotation %in% LoF.def==T)]) </code></pre>
<div id="simple-burden-test" class="section level2">
<h2>Simple burden test</h2>
<pre><code>## Warning: `src_sqlite()` is deprecated as of dplyr 1.0.0.
## Please use `tbl()` directly with a database connection
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<ul>
<li>Two sample one side (“greater”) test</li>
<li>high confidence gene: TADA q&lt;0.05, <a href="https://sfari.org/">SFARI</a> cate1&amp;2</li>
<li>Moder confi: 0.3&gt;TADA q&gt;0.05 SFARI cat3 &amp; S</li>
<li>ID gene: 252 genes from AJHG <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4067558/">paper</a></li>
<li>842 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232425/">FMRP</a> gene</li>
<li>171 <a href="http://autismkb.cbi.pku.edu.cn/">AutismKB</a> genes</li>
<li>1003 <a href="http://www.nature.com/ng/journal/v46/n9/abs/ng.3050.html?foxtrotcallback=true">constraint</a> genes</li>
<li><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709">RVIS</a> gene</li>
<li><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709">Haploinsufficient</a> genes</li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/24463508">Schizophrenia</a>, SCZ gene</li>
<li>861 <a href="http://www.genenames.org/genefamilies/OR">Olfactory</a> gene</li>
<li>67SCZ genes are from SCZ genes with cutoff 0.3.</li>
<li>de novo gene: TADA_SNV_CNV_combined_Feb7, look at column qvalue.combined, the smaller q value, the more likely to be risk gene</li>
<li>Constraint gene set: use Exac Constraint data ConstraintMat.RDS, look at the last column lof.Z, the smaller value the less constraint</li>
<li>ASD prior list: ASD_Priors, look at the column-post, the larger, the more likely to be risk.</li>
</ul>
<table>
<caption>Burden analysis of gene set</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cate1</td>
<td align="right">1.0695491</td>
<td align="right">0.0585134</td>
<td align="right">0.4546530</td>
<td align="right">0.4250885</td>
</tr>
<tr class="even">
<td>cate2</td>
<td align="right">0.9985422</td>
<td align="right">0.5292488</td>
<td align="right">1.4002366</td>
<td align="right">1.4022808</td>
</tr>
<tr class="odd">
<td>cate3</td>
<td align="right">1.0266947</td>
<td align="right">0.0331565</td>
<td align="right">3.9073344</td>
<td align="right">3.8057413</td>
</tr>
<tr class="even">
<td>cate4</td>
<td align="right">0.9844201</td>
<td align="right">0.9638093</td>
<td align="right">10.1805994</td>
<td align="right">10.3417224</td>
</tr>
<tr class="odd">
<td>cate5</td>
<td align="right">0.9875734</td>
<td align="right">0.7719781</td>
<td align="right">2.7180599</td>
<td align="right">2.7522611</td>
</tr>
<tr class="even">
<td>cate6</td>
<td align="right">1.0364296</td>
<td align="right">0.2608020</td>
<td align="right">0.2791798</td>
<td align="right">0.2693669</td>
</tr>
<tr class="odd">
<td>cateS</td>
<td align="right">1.0388224</td>
<td align="right">0.0122912</td>
<td align="right">2.8170347</td>
<td align="right">2.7117578</td>
</tr>
<tr class="even">
<td>TADAq&lt;5%</td>
<td align="right">1.0439491</td>
<td align="right">0.0574849</td>
<td align="right">1.0993691</td>
<td align="right">1.0530869</td>
</tr>
<tr class="odd">
<td>TADAq&lt;20%</td>
<td align="right">1.0136952</td>
<td align="right">0.2008975</td>
<td align="right">3.0705836</td>
<td align="right">3.0290995</td>
</tr>
<tr class="even">
<td>TADAq&lt;30%</td>
<td align="right">1.0157384</td>
<td align="right">0.1059530</td>
<td align="right">5.1230284</td>
<td align="right">5.0436492</td>
</tr>
<tr class="odd">
<td>TADAq&lt;40%</td>
<td align="right">1.0097134</td>
<td align="right">0.1674085</td>
<td align="right">7.9593849</td>
<td align="right">7.8828156</td>
</tr>
<tr class="even">
<td>TADAq&lt;50%</td>
<td align="right">1.0058483</td>
<td align="right">0.2274969</td>
<td align="right">13.1009464</td>
<td align="right">13.0247739</td>
</tr>
<tr class="odd">
<td>TADAq&gt;90%</td>
<td align="right">1.0000912</td>
<td align="right">0.4759251</td>
<td align="right">354.1565457</td>
<td align="right">354.1242627</td>
</tr>
<tr class="even">
<td>ID gene</td>
<td align="right">1.0340983</td>
<td align="right">0.0008008</td>
<td align="right">7.1175079</td>
<td align="right">6.8828156</td>
</tr>
<tr class="odd">
<td>High</td>
<td align="right">1.0301681</td>
<td align="right">0.0596191</td>
<td align="right">2.2231861</td>
<td align="right">2.1580810</td>
</tr>
<tr class="even">
<td>Mod</td>
<td align="right">1.0200271</td>
<td align="right">0.0188176</td>
<td align="right">8.7819401</td>
<td align="right">8.6095163</td>
</tr>
<tr class="odd">
<td>PSD</td>
<td align="right">1.0106383</td>
<td align="right">0.0683577</td>
<td align="right">15.7322555</td>
<td align="right">15.5666536</td>
</tr>
<tr class="even">
<td>FMRP</td>
<td align="right">1.0099078</td>
<td align="right">0.0241821</td>
<td align="right">31.8134858</td>
<td align="right">31.5013763</td>
</tr>
<tr class="odd">
<td>AutismKB</td>
<td align="right">1.0017832</td>
<td align="right">0.4360438</td>
<td align="right">6.8730284</td>
<td align="right">6.8607943</td>
</tr>
<tr class="even">
<td>constraint gene</td>
<td align="right">1.0060751</td>
<td align="right">0.1560072</td>
<td align="right">22.1403785</td>
<td align="right">22.0066850</td>
</tr>
<tr class="odd">
<td>RVIS</td>
<td align="right">1.0081863</td>
<td align="right">0.0328905</td>
<td align="right">40.3702681</td>
<td align="right">40.0424695</td>
</tr>
<tr class="even">
<td>Haploinsuff gene</td>
<td align="right">1.0004430</td>
<td align="right">0.4345452</td>
<td align="right">110.8304416</td>
<td align="right">110.7813606</td>
</tr>
<tr class="odd">
<td>SCZ gene</td>
<td align="right">1.0058876</td>
<td align="right">0.0367401</td>
<td align="right">73.5611199</td>
<td align="right">73.1305545</td>
</tr>
<tr class="even">
<td>Olfac.gene</td>
<td align="right">0.9833342</td>
<td align="right">0.9884215</td>
<td align="right">14.2117508</td>
<td align="right">14.4526150</td>
</tr>
<tr class="odd">
<td>67SCZriskgene</td>
<td align="right">1.0247803</td>
<td align="right">0.0410306</td>
<td align="right">4.0555994</td>
<td align="right">3.9575305</td>
</tr>
</tbody>
</table>
<table>
<caption>Burden analysis of LoF in different gene sets</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">No.ca</th>
<th align="right">No.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cate1</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td>cate2</td>
<td align="right">1.5041404</td>
<td align="right">0.4987079</td>
<td align="right">3</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td>cate3</td>
<td align="right">1.0027603</td>
<td align="right">0.5078977</td>
<td align="right">214</td>
<td align="right">214</td>
</tr>
<tr class="even">
<td>cate4</td>
<td align="right">1.0402466</td>
<td align="right">0.4115798</td>
<td align="right">111</td>
<td align="right">107</td>
</tr>
<tr class="odd">
<td>cate5</td>
<td align="right">1.4728041</td>
<td align="right">0.0559349</td>
<td align="right">47</td>
<td align="right">32</td>
</tr>
<tr class="even">
<td>cate6</td>
<td align="right">0.5013801</td>
<td align="right">0.8744825</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td>cateS</td>
<td align="right">1.0027603</td>
<td align="right">0.5118616</td>
<td align="right">166</td>
<td align="right">166</td>
</tr>
<tr class="even">
<td>TADAq&lt;5%</td>
<td align="right">1.0027603</td>
<td align="right">0.5960236</td>
<td align="right">8</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td>TADAq&lt;20%</td>
<td align="right">1.0197562</td>
<td align="right">0.4939897</td>
<td align="right">60</td>
<td align="right">59</td>
</tr>
<tr class="even">
<td>TADAq&lt;30%</td>
<td align="right">1.0116342</td>
<td align="right">0.4917074</td>
<td align="right">114</td>
<td align="right">113</td>
</tr>
<tr class="odd">
<td>TADAq&lt;40%</td>
<td align="right">0.9750979</td>
<td align="right">0.6343082</td>
<td align="right">282</td>
<td align="right">290</td>
</tr>
<tr class="even">
<td>TADAq&lt;50%</td>
<td align="right">1.1126518</td>
<td align="right">0.0743893</td>
<td align="right">405</td>
<td align="right">365</td>
</tr>
<tr class="odd">
<td>TADAq&gt;90%</td>
<td align="right">1.0048765</td>
<td align="right">0.3592773</td>
<td align="right">11396</td>
<td align="right">11372</td>
</tr>
<tr class="even">
<td>ID gene</td>
<td align="right">0.8508269</td>
<td align="right">0.7754651</td>
<td align="right">28</td>
<td align="right">33</td>
</tr>
<tr class="odd">
<td>High</td>
<td align="right">1.0027603</td>
<td align="right">0.5960236</td>
<td align="right">8</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td>Mod</td>
<td align="right">0.9919000</td>
<td align="right">0.5548855</td>
<td align="right">274</td>
<td align="right">277</td>
</tr>
<tr class="odd">
<td>PSD</td>
<td align="right">1.3787953</td>
<td align="right">0.1422610</td>
<td align="right">33</td>
<td align="right">24</td>
</tr>
<tr class="even">
<td>FMRP</td>
<td align="right">0.9830983</td>
<td align="right">0.5733708</td>
<td align="right">50</td>
<td align="right">51</td>
</tr>
<tr class="odd">
<td>AutismKB</td>
<td align="right">1.0027603</td>
<td align="right">0.5747118</td>
<td align="right">13</td>
<td align="right">13</td>
</tr>
<tr class="even">
<td>constraint gene</td>
<td align="right">0.8736921</td>
<td align="right">0.8411995</td>
<td align="right">88</td>
<td align="right">101</td>
</tr>
<tr class="odd">
<td>RVIS</td>
<td align="right">1.0545460</td>
<td align="right">0.3060351</td>
<td align="right">224</td>
<td align="right">213</td>
</tr>
<tr class="even">
<td>Haploinsuff gene</td>
<td align="right">0.9892094</td>
<td align="right">0.5685699</td>
<td align="right">292</td>
<td align="right">296</td>
</tr>
<tr class="odd">
<td>SCZ gene</td>
<td align="right">1.0214535</td>
<td align="right">0.3491793</td>
<td align="right">765</td>
<td align="right">751</td>
</tr>
<tr class="even">
<td>Olfac.gene</td>
<td align="right">1.0662467</td>
<td align="right">0.1324173</td>
<td align="right">655</td>
<td align="right">616</td>
</tr>
<tr class="odd">
<td>67SCZriskgene</td>
<td align="right">1.2074052</td>
<td align="right">0.0945265</td>
<td align="right">118</td>
<td align="right">98</td>
</tr>
</tbody>
</table>
</div>
<div id="variant-partitions" class="section level2">
<h2>Variant partitions</h2>
<p>A gene usually has many variants such as missense, LoF, etc which contribute to the disease at different degree of deleteriousness. So we partition variants from all genes into different variant categories based on their annotation priors.</p>
<div id="eight-variant-groups" class="section level3">
<h3>Eight variant groups</h3>
<table>
<caption>Fixed parameters for eight variant categories.</caption>
<thead>
<tr class="header">
<th>Category</th>
<th>AF</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>C1:LoF</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
</tr>
<tr class="even">
<td>C2:LoF</td>
<td><span class="math inline">\(AF &lt; 0.01\)</span></td>
</tr>
<tr class="odd">
<td>C3:Damaging</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
</tr>
<tr class="even">
<td>C4:Damaging</td>
<td><span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
</tr>
<tr class="odd">
<td>C5:Damaging</td>
<td><span class="math inline">\(AF &lt; 0.001\)</span></td>
</tr>
<tr class="even">
<td>C6:Non-Damaging</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
</tr>
<tr class="odd">
<td>C7:Non-Damaging</td>
<td><span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
</tr>
<tr class="even">
<td>C8:Non-Damaging</td>
<td><span class="math inline">\(AF &lt; 0.001\)</span></td>
</tr>
</tbody>
</table>
<ul>
<li>Damaging (probably damaging):Polyphen2.HDIV.score &gt;=0.957</li>
</ul>
</div>
</div>
<div id="constraint-top-5-1003-gene-set" class="section level2">
<h2>Constraint (top 5%-1003) gene set</h2>
<p>Constraint gene set is from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">this paper</a>.</p>
<div id="cons-simple-burden-analysis" class="section level3">
<h3>Cons: Simple burden analysis</h3>
<p>The burden of 8 categories is</p>
<pre class="r"><code>colnames(burden.summy)=c(&quot;OR&quot;, &quot;p.value&quot;, &quot;No.ca&quot;, &quot;No.co&quot;)
kable(burden.summy, caption=&quot;The burden of eight categories&quot;)</code></pre>
<table>
<caption>The burden of eight categories</caption>
<thead>
<tr class="header">
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">No.ca</th>
<th align="right">No.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NaN</td>
<td align="right">1.0000000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0.8736921</td>
<td align="right">0.8411995</td>
<td align="right">88</td>
<td align="right">101</td>
</tr>
<tr class="odd">
<td align="right">1.0269332</td>
<td align="right">0.0725397</td>
<td align="right">6160</td>
<td align="right">6015</td>
</tr>
<tr class="even">
<td align="right">1.0381518</td>
<td align="right">0.0347048</td>
<td align="right">4840</td>
<td align="right">4675</td>
</tr>
<tr class="odd">
<td align="right">0.9724785</td>
<td align="right">0.8783505</td>
<td align="right">3372</td>
<td align="right">3477</td>
</tr>
<tr class="even">
<td align="right">1.0110971</td>
<td align="right">0.1371398</td>
<td align="right">19890</td>
<td align="right">19726</td>
</tr>
<tr class="odd">
<td align="right">0.9916250</td>
<td align="right">0.7431442</td>
<td align="right">11755</td>
<td align="right">11887</td>
</tr>
<tr class="even">
<td align="right">0.9978114</td>
<td align="right">0.5634710</td>
<td align="right">9678</td>
<td align="right">9726</td>
</tr>
</tbody>
</table>
</div>
<div id="cons-run-mirage" class="section level3">
<h3>Cons: run mirage</h3>
<div id="parameter-estimate-when-af-5" class="section level4">
<h4>parameter estimate when AF &lt; 5%</h4>
<p><span class="math inline">\(\widehat{\delta}=0.09\)</span> with p value of 0.56.</p>
<pre><code>## -- Attaching packages -- tidyverse 1.3.0 --</code></pre>
<pre><code>## v ggplot2 3.3.1     v purrr   0.3.4
## v tibble  3.0.1     v stringr 1.4.0
## v tidyr   1.1.0     v forcats 0.4.0
## v readr   1.3.1</code></pre>
<pre><code>## -- Conflicts ----- tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()</code></pre>
<p><img src="Geneset-SCZ_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<table>
<caption>parameter estimates and p values in variant groups. 0’s estimates are not exactly estimated as 0 but they are extremely close to 0, like <span class="math inline">\(e^{-10}\)</span>.</caption>
<thead>
<tr class="header">
<th>Category</th>
<th>AF</th>
<th>eta.est</th>
<th>p value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>C1:LoF</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>C2:LoF</td>
<td><span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0</td>
<td>1</td>
</tr>
<tr class="odd">
<td>C3:Damaging</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>1</td>
</tr>
<tr class="even">
<td>C4:Damaging</td>
<td><span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.16</td>
<td>0.14</td>
</tr>
<tr class="odd">
<td>C5:Damaging</td>
<td><span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0</td>
<td>1</td>
</tr>
<tr class="even">
<td>C6:Non-Damaging</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>1</td>
</tr>
<tr class="odd">
<td>C7:Non-Damaging</td>
<td><span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>1</td>
</tr>
<tr class="even">
<td>C8:Non-Damaging</td>
<td><span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.26</td>
<td>0.12</td>
</tr>
</tbody>
</table>
</div>
<div id="top-genes" class="section level4">
<h4>Top genes</h4>
<pre><code>## # A tibble: 777 x 3
##    Gene        BF post.prob
##    &lt;fct&gt;    &lt;dbl&gt;     &lt;dbl&gt;
##  1 DNMT3A   73.8      0.882
##  2 MACF1    19.4      0.664
##  3 TMEM102  13.1      0.571
##  4 NOTCH2   10.2      0.509
##  5 ARHGAP35  8.28     0.457
##  6 CSMD2     7.80     0.442
##  7 UBR5      4.45     0.312
##  8 SRRT      4.32     0.305
##  9 GRIK5     3.67     0.272
## 10 RAPGEF2   3.54     0.265
## # ... with 767 more rows</code></pre>
</div>
<div id="highlight-gene-dnmt3a" class="section level4">
<h4>highlight gene DNMT3A</h4>
<pre class="r"><code>Gene: DNMT3A briven by damaging variants              
             ID   Gene No.case No.contr category      var.BF original.index
949 chr2:25523096-25523096_T_G DNMT3A      39       13        3 566.1746060              4
943 chr2:25457243-25457243_G_A DNMT3A       1        0        4   1.4530292              5
944 chr2:25467078-25467078_C_A DNMT3A       0        1        7   0.5482178              8
945 chr2:25468174-25468174_T_C DNMT3A       1        0        7   1.4530292              8
946 chr2:25469073-25469073_G_A DNMT3A      10        8        7   0.3861151              8
947 chr2:25469142-25469142_A_G DNMT3A       0        1        7   0.5482178              8
948 chr2:25469614-25469614_G_A DNMT3A       1        0        7   1.4530292              8</code></pre>
<pre class="r"><code>Gene: MACF1, signal driven by rare damaing and non-damaging variants 

                            ID  Gene No.case No.contr category      var.BF original.index
379 chr1:39847730-39847730_G_A MACF1      45       47        2  0.04355588              3
353 chr1:39775303-39775303_A_G MACF1      19       13        3  0.58277783              4
359 chr1:39783013-39783013_G_T MACF1      90       74        3  0.15193265              4
362 chr1:39788295-39788295_C_A MACF1       2        4        3  0.23939249              4
367 chr1:39816547-39816547_C_G MACF1      19        7        3 13.10494586              4
386 chr1:39853640-39853640_A_C MACF1      45       25        3  6.09805554              4
387 chr1:39853641-39853641_G_T MACF1      90       74        3  0.15193265              4
394 chr1:39853959-39853959_G_A MACF1      19       12        3  0.87925645              4
403 chr1:39900230-39900230_C_T MACF1       2        3        3  0.36622226              4
420 chr1:39951304-39951304_T_G MACF1      24       23        3  0.10043482              4
345 chr1:39549925-39549925_G_A MACF1       1        1        4  0.76755484              5
346 chr1:39549942-39549942_C_T MACF1       0        2        4  0.32948455              5
348 chr1:39752993-39752993_G_A MACF1       1        0        4  1.45302921              5
349 chr1:39753145-39753145_C_T MACF1       0        1        4  0.54821782              5
356 chr1:39776562-39776562_C_T MACF1       1        2        4  0.44240738              5
357 chr1:39781210-39781210_T_G MACF1       1        1        4  0.76755484              5
358 chr1:39781321-39781321_A_G MACF1       1        2        4  0.44240738              5
360 chr1:39784189-39784189_A_T MACF1       1        0        4  1.45302921              5
361 chr1:39785384-39785384_C_G MACF1       1        2        4  0.44240738              5
363 chr1:39788346-39788346_C_T MACF1       0        1        4  0.54821782              5
364 chr1:39788635-39788635_C_G MACF1       1        0        4  1.45302921              5
375 chr1:39827287-39827287_C_T MACF1       0        1        4  0.54821782              5
376 chr1:39835745-39835745_C_G MACF1       1        1        4  0.76755484              5
377 chr1:39835746-39835746_G_A MACF1       0        1        4  0.54821782              5
380 chr1:39851427-39851427_G_A MACF1       2        1        4  1.09359980              5
381 chr1:39851451-39851451_G_C MACF1       1        0        4  1.45302921              5
382 chr1:39851518-39851518_T_C MACF1       0        1        4  0.54821782              5
383 chr1:39851554-39851554_G_A MACF1       1        0        4  1.45302921              5
384 chr1:39851563-39851563_C_T MACF1       0        2        4  0.32948455              5
389 chr1:39853740-39853740_C_G MACF1       0        5        4  0.12191241              5
391 chr1:39853818-39853818_G_A MACF1       0        1        4  0.54821782              5
392 chr1:39853893-39853893_C_T MACF1       0        1        4  0.54821782              5
393 chr1:39853900-39853900_G_A MACF1       1        0        4  1.45302921              5
396 chr1:39854170-39854170_G_A MACF1       0        1        4  0.54821782              5
397 chr1:39889792-39889792_T_G MACF1       0        3        4  0.21687257              5
398 chr1:39893177-39893177_C_T MACF1       0        1        4  0.54821782              5
399 chr1:39893669-39893669_G_A MACF1       0        2        4  0.32948455              5
401 chr1:39895535-39895535_G_A MACF1       0        1        4  0.54821782              5
402 chr1:39896485-39896485_C_T MACF1       0        1        4  0.54821782              5
404 chr1:39903464-39903464_G_A MACF1       3        9        4  0.08613987              5
405 chr1:39904981-39904981_T_A MACF1       0        1        4  0.54821782              5
406 chr1:39908490-39908490_A_C MACF1       3        3        4  0.49340212              5
408 chr1:39910413-39910413_G_A MACF1       1        2        4  0.44240738              5
409 chr1:39910418-39910418_G_A MACF1       5        5        4  0.34806276              5
410 chr1:39913252-39913252_A_T MACF1       1        0        4  1.45302921              5
412 chr1:39916854-39916854_C_A MACF1       0        1        4  0.54821782              5
413 chr1:39917864-39917864_G_A MACF1       1        0        4  1.45302921              5
414 chr1:39917931-39917931_A_C MACF1       0        1        4  0.54821782              5
418 chr1:39951230-39951230_C_T MACF1       1        0        4  1.45302921              5
419 chr1:39951243-39951243_G_A MACF1       5        5        4  0.34806276              5
350 chr1:39757595-39757595_G_A MACF1      14       16        6  0.09148606              7
351 chr1:39763365-39763365_G_T MACF1      54       44        6  0.16436446              7
378 chr1:39838200-39838200_A_G MACF1       6       10        6  0.09719669              7
347 chr1:39715702-39715702_A_G MACF1       2        0        7  2.14039567              8
352 chr1:39765991-39765991_A_G MACF1       1        1        7  0.76755484              8
354 chr1:39775999-39775999_G_A MACF1       1        0        7  1.45302921              8
355 chr1:39776527-39776527_A_G MACF1       4        0        7  4.80333231              8
365 chr1:39790337-39790337_C_T MACF1       0        1        7  0.54821782              8
366 chr1:39790348-39790348_G_C MACF1       0        1        7  0.54821782              8
368 chr1:39823159-39823159_T_C MACF1      12        3        7 13.43065361              8
369 chr1:39823306-39823306_G_A MACF1       0        1        7  0.54821782              8
370 chr1:39823349-39823349_C_A MACF1       1        0        7  1.45302921              8
371 chr1:39823363-39823363_C_T MACF1       0        1        7  0.54821782              8
372 chr1:39823365-39823365_T_C MACF1       1        3        7  0.27799691              8
373 chr1:39824468-39824468_G_A MACF1       0        3        7  0.21687257              8
374 chr1:39826587-39826587_A_G MACF1       2        2        7  0.60727166              8
385 chr1:39853356-39853356_A_G MACF1       1        1        7  0.76755484              8
388 chr1:39853687-39853687_T_A MACF1       4        9        7  0.09258787              8
390 chr1:39853797-39853797_G_T MACF1       1        0        7  1.45302921              8
395 chr1:39854002-39854002_G_A MACF1       7        2        7  3.71945900              8
400 chr1:39893676-39893676_A_G MACF1       6        5        7  0.46235530              8
407 chr1:39908511-39908511_G_T MACF1       2        1        7  1.09359980              8
411 chr1:39914356-39914356_G_A MACF1       1        1        7  0.76755484              8
415 chr1:39919407-39919407_G_A MACF1       4        1        7  2.31557551              8
416 chr1:39920758-39920758_G_A MACF1       2        1        7  1.09359980              8
417 chr1:39924788-39924788_C_T MACF1       1        0        7  1.45302921              8</code></pre>
</div>
</div>
<div id="cons-estimate-beta_i-asd-gene-specific-delta_i" class="section level3">
<h3>Cons: Estimate <span class="math inline">\(\beta_i\)</span>, ASD gene specific <span class="math inline">\(\delta_i\)</span></h3>
<div id="eight-categories" class="section level4">
<h4>(2) Eight categories</h4>
<table>
<caption>Parameter estimates of 1003 constraint genes by EM with eight partitions with gene specific <span class="math inline">\(\delta\)</span>. NA means no variants.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0292</td>
<td>2.3766</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0391</td>
<td>1.1027</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0862</td>
<td>1.6528</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="rvis-gene" class="section level2">
<h2>RVIS gene</h2>
<div id="rvisestimate-beta_i-asd-gene-specific-delta_i" class="section level3">
<h3>RVIS:Estimate <span class="math inline">\(\beta_i\)</span>, ASD gene specific <span class="math inline">\(\delta_i\)</span></h3>
<pre class="r"><code>beta=matrix(nrow=5, ncol=8)
rownames(beta)=c(&quot;Quantile score&lt;10&quot;, &quot;10-20&quot;, &quot;20-30&quot;, &quot;30-40&quot;, &quot;40-50&quot;)
beta[1,]=c(0, 0.0977,  0, 0.0179,  0.0132, 0,  0.0203,  0.0415)
beta[2,]=c(0, 0.1240,  0, 0.0206,   0, 0, 0,  0.0004)
beta[3,]=c(0,  0, 0,   0,   0.0940,0.0615,  0,  0.0639)
beta[4,]=c(0,  0,  0, 0.0285,   0.0813,0,  0.0889,   0.0005)
beta[5,]=c(0, 0.0500,   0, 0,  0,  0.0370, 0.0315,    0.0005)
plot(beta[,1], ylim=c(0,0.5), type=&quot;o&quot;, xaxt=&quot;n&quot;, xlab=&quot;&quot;, ylab=expression(paste(hat(beta))), main=&quot;Parameter estimate of five non-nested gene sets&quot;)
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type=&quot;o&quot;, col=i)
legend(4, 0.5, c(paste(&quot;beta&quot;, seq(1,8), sep=&quot;&quot;)), col=seq(1,8), lty=rep(1,8))</code></pre>
<p><img src="Geneset-SCZ_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
</div>
<div id="constraint-genes" class="section level2">
<h2>Constraint genes</h2>
<p>There are 17,416 genes.</p>
<div id="con-estimate-beta_i-asd-gene-specific-delta_i" class="section level3">
<h3>Con: Estimate <span class="math inline">\(\beta_i\)</span>, ASD gene specific <span class="math inline">\(\delta_i\)</span></h3>
<div id="non-nested-gene-sets" class="section level4">
<h4>Non-nested gene sets</h4>
<p>With their constraint scores, focus on top 10% genes, top 10-20% genes, top 20-30% genes, top 30-40% genes, top 40-50% genes.</p>
<pre class="r"><code>beta=matrix(nrow=5, ncol=8)
rownames(beta)=c(&quot;top10%&quot;, &quot;10-20%&quot;, &quot;20-30%&quot;, &quot;30-40%&quot;, &quot;40-50%&quot;)
beta[1,]=c(0.9999, 0,  0, 0.0065,  0.0002, 0,  0.0250,  0.0029)
beta[2,]=c(0, 0.1189,  0, 0.0087,   0, 0, 0,  0.00005)
beta[3,]=c(0,  0, 0,   0.0097,   0,0,  0,  0.0046)
beta[4,]=c(0,  0,  0, 0,   0.0046,0,  0.0003,   0.0601)
beta[5,]=c(0, 0,  0,  0.0017, 0.0323,   0,0.0074,   0.0207)
plot(beta[,1], ylim=c(0,1), type=&quot;o&quot;, xaxt=&quot;n&quot;, xlab=&quot;&quot;, ylab=expression(paste(hat(beta))), main=&quot;Parameter estimate of five non-nested gene sets&quot;)
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type=&quot;o&quot;, col=i)
legend(4, 0.8, c(paste(&quot;beta&quot;, seq(1,8), sep=&quot;&quot;)), col=seq(1,8), lty=rep(1,8))</code></pre>
<p><img src="Geneset-SCZ_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="whole-genome" class="section level2">
<h2>Whole genome</h2>
<p>consider all 15624 genes in ASD data.</p>
<div id="whole-estimate-beta_i-asd-gene-specific-delta_i" class="section level3">
<h3>Whole: Estimate <span class="math inline">\(\beta_i\)</span>, ASD gene specific <span class="math inline">\(\delta_i\)</span></h3>
<table>
<caption>Parameter estimates of 15624 genes by EM with eight partitions with gene specific <span class="math inline">\(\delta_i\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0063</td>
<td>2.2903</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0238</td>
<td>10.8729</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0.0029</td>
<td>0.6222</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0166</td>
<td>4.1023</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0043</td>
<td>0</td>
</tr>
</tbody>
</table>
</div>
<div id="whole-estimate-beta_i-common-delta" class="section level3">
<h3>Whole: Estimate <span class="math inline">\(\beta_i\)</span>, common <span class="math inline">\(\delta\)</span></h3>
<p><span class="math inline">\(\widehat{\delta}=0.2\)</span> with test statistics of 24.2446.</p>
<table>
<caption>Parameter estimates of all genes by EM with eight partitions with commom <span class="math inline">\(\delta\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0024</td>
<td>0.5010</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0228</td>
<td>17.4871</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0.0010</td>
<td>0.3323</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0134</td>
<td>5.4018</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0103</td>
<td>0.5224</td>
</tr>
</tbody>
</table>
</div>
<div id="whole-estimate-beta_i-scz-de-novo-prior-delta_i" class="section level3">
<h3>Whole: Estimate <span class="math inline">\(\beta_i\)</span>, SCZ de novo prior <span class="math inline">\(\delta_i\)</span></h3>
<p>Use posterior probability PP as priors</p>
<ul>
<li>prior by 2 categoris</li>
</ul>
<table>
<caption>Parameter estimates by EM with eight partitions with gene specific <span class="math inline">\(\delta_i\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0012</td>
<td>0.03</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0707</td>
<td>11.67</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0.0090</td>
<td>0.57</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0689</td>
<td>5.27</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0029</td>
<td>0.05</td>
</tr>
</tbody>
</table>
<ul>
<li>prior by 3 categoris</li>
</ul>
<table>
<caption>Parameter estimates by EM with eight partitions with gene specific <span class="math inline">\(\delta_i\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0774</td>
<td>9.95</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0.0132</td>
<td>0.65</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0.0871</td>
<td>5.11</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0016</td>
<td>0.02</td>
</tr>
</tbody>
</table>
<!-- Add your analysis here -->
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 18363)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] forcats_0.4.0   stringr_1.4.0   purrr_0.3.4     readr_1.3.1    
##  [5] tidyr_1.1.0     tibble_3.0.1    ggplot2_3.3.1   tidyverse_1.3.0
##  [9] knitr_1.23      dplyr_1.0.0     RSQLite_2.1.1  
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_1.1.0 xfun_0.7         haven_2.3.1      lattice_0.20-38 
##  [5] colorspace_1.4-1 vctrs_0.3.0      generics_0.0.2   htmltools_0.3.6 
##  [9] yaml_2.2.0       utf8_1.1.4       blob_1.2.1       rlang_0.4.6     
## [13] pillar_1.4.4     withr_2.1.2      glue_1.4.1       DBI_1.0.0       
## [17] bit64_0.9-7      dbplyr_1.4.4     modelr_0.1.8     readxl_1.3.1    
## [21] lifecycle_0.2.0  munsell_0.5.0    gtable_0.3.0     cellranger_1.1.0
## [25] rvest_0.3.5      memoise_1.1.0    evaluate_0.13    labeling_0.3    
## [29] fansi_0.4.0      highr_0.8        broom_0.5.2      Rcpp_1.0.1      
## [33] backports_1.1.4  scales_1.0.0     jsonlite_1.6     fs_1.3.1        
## [37] bit_1.1-14       hms_0.5.3        digest_0.6.19    stringi_1.4.3   
## [41] grid_3.6.0       cli_1.1.0        tools_3.6.0      magrittr_1.5    
## [45] crayon_1.3.4     pkgconfig_2.0.3  ellipsis_0.3.0   xml2_1.3.2      
## [49] reprex_0.3.0     lubridate_1.7.4  assertthat_0.2.1 rmarkdown_1.13  
## [53] httr_1.4.1       rstudioapi_0.10  R6_2.4.1         nlme_3.1-139    
## [57] compiler_3.6.0</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
