<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>Gene Set Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene Set Analysis</h1>
<h4 class="author">Shengtong Han</h4>
<h4 class="date">YYYY-MM-DD</h4>

</div>


<!-- Add your analysis here -->
<pre class="r"><code>N1=4315; N0=4315
All.Anno.Data[All.Anno.Data ==&quot;.&quot;] &lt;- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF&lt;0.05 &amp; All.Anno.Data$Annotation!=&quot;synonymous SNV&quot;),] # use AF cutoff and exclude synonumous SNV
var.data=as_tibble(data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr))
LoF.def=c(&quot;stopgain&quot;, &quot;frameshift substitution&quot;, &quot;splicing&quot;, &quot;stoploss&quot;)
LoF.var=as.character(Anno.Data$ID[which(Anno.Data$Annotation %in% LoF.def==T)]) </code></pre>
<div id="simple-burden-test" class="section level2">
<h2>Simple burden test</h2>
<ul>
<li>Two sample one side (“greater”) test</li>
<li>high confidence gene: TADA q&lt;0.05, <a href="https://sfari.org/">SFARI</a> cate1&amp;2</li>
<li>Moder confi: 0.3&gt;TADA q&gt;0.05 SFARI cat3 &amp; S</li>
<li>ID gene: 252 genes from AJHG <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4067558/">paper</a></li>
<li>842 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232425/">FMRP</a> gene</li>
<li>171 <a href="http://autismkb.cbi.pku.edu.cn/">AutismKB</a> genes</li>
<li>1003 <a href="http://www.nature.com/ng/journal/v46/n9/abs/ng.3050.html?foxtrotcallback=true">constraint</a> genes</li>
<li><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709">RVIS</a> gene</li>
<li><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709">Haploinsufficient</a> genes</li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/24463508">Schizophrenia</a>, SCZ gene</li>
<li>861 <a href="http://www.genenames.org/genefamilies/OR">Olfactory</a> gene</li>
<li>67SCZ genes are from SCZ genes with cutoff 0.3.</li>
<li>de novo gene: TADA_SNV_CNV_combined_Feb7, look at column qvalue.combined, the smaller q value, the more likely to be risk gene</li>
<li>Constraint gene set: use Exac Constraint data ConstraintMat.RDS, look at the last column lof.Z, the smaller value the less constraint</li>
<li>ASD prior list: ASD_Priors, look at the column-post, the larger, the more likely to be risk.</li>
</ul>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Burden analysis of gene set
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rate.ca
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rate.co
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cate1
</td>
<td style="text-align:right;">
0.9976341
</td>
<td style="text-align:right;">
0.5316724
</td>
<td style="text-align:right;">
0.2931634
</td>
<td style="text-align:right;">
0.2938586
</td>
</tr>
<tr>
<td style="text-align:left;">
cate2
</td>
<td style="text-align:right;">
0.9934686
</td>
<td style="text-align:right;">
0.6231251
</td>
<td style="text-align:right;">
0.9870220
</td>
<td style="text-align:right;">
0.9935110
</td>
</tr>
<tr>
<td style="text-align:left;">
cate3
</td>
<td style="text-align:right;">
0.9976905
</td>
<td style="text-align:right;">
0.5675548
</td>
<td style="text-align:right;">
2.3026651
</td>
<td style="text-align:right;">
2.3079954
</td>
</tr>
<tr>
<td style="text-align:left;">
cate4
</td>
<td style="text-align:right;">
1.0008482
</td>
<td style="text-align:right;">
0.4606364
</td>
<td style="text-align:right;">
6.8361530
</td>
<td style="text-align:right;">
6.8303592
</td>
</tr>
<tr>
<td style="text-align:left;">
cate5
</td>
<td style="text-align:right;">
1.0023832
</td>
<td style="text-align:right;">
0.4450279
</td>
<td style="text-align:right;">
1.7545771
</td>
<td style="text-align:right;">
1.7504056
</td>
</tr>
<tr>
<td style="text-align:left;">
cate6
</td>
<td style="text-align:right;">
0.9479090
</td>
<td style="text-align:right;">
0.9188497
</td>
<td style="text-align:right;">
0.2994206
</td>
<td style="text-align:right;">
0.3158749
</td>
</tr>
<tr>
<td style="text-align:left;">
cateS
</td>
<td style="text-align:right;">
0.9940770
</td>
<td style="text-align:right;">
0.6500498
</td>
<td style="text-align:right;">
1.8669757
</td>
<td style="text-align:right;">
1.8780997
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;5%
</td>
<td style="text-align:right;">
1.0420070
</td>
<td style="text-align:right;">
0.0698689
</td>
<td style="text-align:right;">
0.6208575
</td>
<td style="text-align:right;">
0.5958285
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;20%
</td>
<td style="text-align:right;">
1.0066310
</td>
<td style="text-align:right;">
0.3421668
</td>
<td style="text-align:right;">
1.8294322
</td>
<td style="text-align:right;">
1.8173812
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;30%
</td>
<td style="text-align:right;">
1.0043203
</td>
<td style="text-align:right;">
0.3618086
</td>
<td style="text-align:right;">
3.2324450
</td>
<td style="text-align:right;">
3.2185400
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;40%
</td>
<td style="text-align:right;">
0.9981514
</td>
<td style="text-align:right;">
0.5762986
</td>
<td style="text-align:right;">
4.7550406
</td>
<td style="text-align:right;">
4.7638470
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;50%
</td>
<td style="text-align:right;">
0.9861335
</td>
<td style="text-align:right;">
0.9668467
</td>
<td style="text-align:right;">
7.9274623
</td>
<td style="text-align:right;">
8.0389340
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&gt;90%
</td>
<td style="text-align:right;">
0.9910332
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
208.1110081
</td>
<td style="text-align:right;">
209.9939745
</td>
</tr>
<tr>
<td style="text-align:left;">
ID gene
</td>
<td style="text-align:right;">
0.9853493
</td>
<td style="text-align:right;">
0.9299548
</td>
<td style="text-align:right;">
4.5668598
</td>
<td style="text-align:right;">
4.6347625
</td>
</tr>
<tr>
<td style="text-align:left;">
High conf
</td>
<td style="text-align:right;">
0.9981478
</td>
<td style="text-align:right;">
0.5438571
</td>
<td style="text-align:right;">
1.3738123
</td>
<td style="text-align:right;">
1.3763615
</td>
</tr>
<tr>
<td style="text-align:left;">
Mod conf
</td>
<td style="text-align:right;">
0.9963161
</td>
<td style="text-align:right;">
0.6622900
</td>
<td style="text-align:right;">
5.8289687
</td>
<td style="text-align:right;">
5.8505214
</td>
</tr>
<tr>
<td style="text-align:left;">
PSD
</td>
<td style="text-align:right;">
0.9970598
</td>
<td style="text-align:right;">
0.6628794
</td>
<td style="text-align:right;">
9.2736964
</td>
<td style="text-align:right;">
9.3010429
</td>
</tr>
<tr>
<td style="text-align:left;">
FMRP
</td>
<td style="text-align:right;">
0.9958167
</td>
<td style="text-align:right;">
0.7849446
</td>
<td style="text-align:right;">
16.2743917
</td>
<td style="text-align:right;">
16.3427578
</td>
</tr>
<tr>
<td style="text-align:left;">
AutismKB
</td>
<td style="text-align:right;">
0.9994632
</td>
<td style="text-align:right;">
0.5236361
</td>
<td style="text-align:right;">
4.7466976
</td>
<td style="text-align:right;">
4.7492468
</td>
</tr>
<tr>
<td style="text-align:left;">
constraint gene
</td>
<td style="text-align:right;">
0.9951003
</td>
<td style="text-align:right;">
0.7667974
</td>
<td style="text-align:right;">
10.0723059
</td>
<td style="text-align:right;">
10.1219003
</td>
</tr>
<tr>
<td style="text-align:left;">
RVIS
</td>
<td style="text-align:right;">
1.0013601
</td>
<td style="text-align:right;">
0.3805547
</td>
<td style="text-align:right;">
23.5455388
</td>
<td style="text-align:right;">
23.5135574
</td>
</tr>
<tr>
<td style="text-align:left;">
Haploinsuff gene
</td>
<td style="text-align:right;">
0.9921820
</td>
<td style="text-align:right;">
0.9986197
</td>
<td style="text-align:right;">
67.0871379
</td>
<td style="text-align:right;">
67.6157590
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ gene
</td>
<td style="text-align:right;">
0.9944114
</td>
<td style="text-align:right;">
0.9527662
</td>
<td style="text-align:right;">
41.0713789
</td>
<td style="text-align:right;">
41.3022016
</td>
</tr>
<tr>
<td style="text-align:left;">
Olfac.gene
</td>
<td style="text-align:right;">
0.9763183
</td>
<td style="text-align:right;">
0.9999808
</td>
<td style="text-align:right;">
13.4906141
</td>
<td style="text-align:right;">
13.8178447
</td>
</tr>
<tr>
<td style="text-align:left;">
67SCZriskgene
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Burden analysis of LoF in different gene sets
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.ca
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.co
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cate1
</td>
<td style="text-align:right;">
8.0000000
</td>
<td style="text-align:right;">
0.0195312
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cate2
</td>
<td style="text-align:right;">
1.5500000
</td>
<td style="text-align:right;">
0.0803898
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
20
</td>
</tr>
<tr>
<td style="text-align:left;">
cate3
</td>
<td style="text-align:right;">
1.0395604
</td>
<td style="text-align:right;">
0.2884161
</td>
<td style="text-align:right;">
473
</td>
<td style="text-align:right;">
455
</td>
</tr>
<tr>
<td style="text-align:left;">
cate4
</td>
<td style="text-align:right;">
1.0369004
</td>
<td style="text-align:right;">
0.3508540
</td>
<td style="text-align:right;">
281
</td>
<td style="text-align:right;">
271
</td>
</tr>
<tr>
<td style="text-align:left;">
cate5
</td>
<td style="text-align:right;">
1.3469388
</td>
<td style="text-align:right;">
0.0676752
</td>
<td style="text-align:right;">
66
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
cate6
</td>
<td style="text-align:right;">
1.7500000
</td>
<td style="text-align:right;">
0.1431394
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
cateS
</td>
<td style="text-align:right;">
0.9693878
</td>
<td style="text-align:right;">
0.6800507
</td>
<td style="text-align:right;">
380
</td>
<td style="text-align:right;">
392
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;5%
</td>
<td style="text-align:right;">
2.0370370
</td>
<td style="text-align:right;">
0.0013212
</td>
<td style="text-align:right;">
55
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;20%
</td>
<td style="text-align:right;">
1.3884298
</td>
<td style="text-align:right;">
0.0033555
</td>
<td style="text-align:right;">
168
</td>
<td style="text-align:right;">
121
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;30%
</td>
<td style="text-align:right;">
1.3111111
</td>
<td style="text-align:right;">
0.0034669
</td>
<td style="text-align:right;">
236
</td>
<td style="text-align:right;">
180
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;40%
</td>
<td style="text-align:right;">
1.1257143
</td>
<td style="text-align:right;">
0.0574301
</td>
<td style="text-align:right;">
394
</td>
<td style="text-align:right;">
350
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&lt;50%
</td>
<td style="text-align:right;">
1.0612245
</td>
<td style="text-align:right;">
0.1685170
</td>
<td style="text-align:right;">
572
</td>
<td style="text-align:right;">
539
</td>
</tr>
<tr>
<td style="text-align:left;">
TADAq&gt;90%
</td>
<td style="text-align:right;">
0.9608535
</td>
<td style="text-align:right;">
0.9999889
</td>
<td style="text-align:right;">
22066
</td>
<td style="text-align:right;">
22965
</td>
</tr>
<tr>
<td style="text-align:left;">
ID gene
</td>
<td style="text-align:right;">
0.9067358
</td>
<td style="text-align:right;">
0.8390201
</td>
<td style="text-align:right;">
175
</td>
<td style="text-align:right;">
193
</td>
</tr>
<tr>
<td style="text-align:left;">
High conf
</td>
<td style="text-align:right;">
1.7222222
</td>
<td style="text-align:right;">
0.0055873
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
36
</td>
</tr>
<tr>
<td style="text-align:left;">
Mod conf
</td>
<td style="text-align:right;">
1.0386643
</td>
<td style="text-align:right;">
0.2687646
</td>
<td style="text-align:right;">
591
</td>
<td style="text-align:right;">
569
</td>
</tr>
<tr>
<td style="text-align:left;">
PSD
</td>
<td style="text-align:right;">
0.9911504
</td>
<td style="text-align:right;">
0.5562213
</td>
<td style="text-align:right;">
224
</td>
<td style="text-align:right;">
226
</td>
</tr>
<tr>
<td style="text-align:left;">
FMRP
</td>
<td style="text-align:right;">
1.2926829
</td>
<td style="text-align:right;">
0.0013805
</td>
<td style="text-align:right;">
318
</td>
<td style="text-align:right;">
246
</td>
</tr>
<tr>
<td style="text-align:left;">
AutismKB
</td>
<td style="text-align:right;">
1.1666667
</td>
<td style="text-align:right;">
0.1408879
</td>
<td style="text-align:right;">
119
</td>
<td style="text-align:right;">
102
</td>
</tr>
<tr>
<td style="text-align:left;">
constraint gene
</td>
<td style="text-align:right;">
1.0668954
</td>
<td style="text-align:right;">
0.1368236
</td>
<td style="text-align:right;">
622
</td>
<td style="text-align:right;">
583
</td>
</tr>
<tr>
<td style="text-align:left;">
RVIS
</td>
<td style="text-align:right;">
1.2034976
</td>
<td style="text-align:right;">
0.0003207
</td>
<td style="text-align:right;">
757
</td>
<td style="text-align:right;">
629
</td>
</tr>
<tr>
<td style="text-align:left;">
Haploinsuff gene
</td>
<td style="text-align:right;">
1.0897995
</td>
<td style="text-align:right;">
0.0185978
</td>
<td style="text-align:right;">
1250
</td>
<td style="text-align:right;">
1147
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ gene
</td>
<td style="text-align:right;">
1.0552965
</td>
<td style="text-align:right;">
0.0699201
</td>
<td style="text-align:right;">
1584
</td>
<td style="text-align:right;">
1501
</td>
</tr>
<tr>
<td style="text-align:left;">
Olfac.gene
</td>
<td style="text-align:right;">
0.9916487
</td>
<td style="text-align:right;">
0.6451150
</td>
<td style="text-align:right;">
3681
</td>
<td style="text-align:right;">
3712
</td>
</tr>
<tr>
<td style="text-align:left;">
67SCZriskgene
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="variant-partitions" class="section level2">
<h2>Variant partitions</h2>
<p>A gene usually has many variants such as missense, LoF, etc which contribute to the disease at different degree of deleteriousness. So we partition variants from all genes into different variant categories based on their annotation priors.</p>
<div id="eight-variant-groups" class="section level3">
<h3>Eight variant groups</h3>
<table>
<caption>Fixed parameters for eight variant categories.</caption>
<thead>
<tr class="header">
<th>Category</th>
<th>AF</th>
<th>fixed <span class="math inline">\(\eta\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>C1:LoF</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td><span class="math inline">\(\eta_1= 0.5\)</span></td>
</tr>
<tr class="even">
<td>C2:LoF</td>
<td><span class="math inline">\(AF &lt; 0.01\)</span></td>
<td><span class="math inline">\(\eta_2=0.9\)</span></td>
</tr>
<tr class="odd">
<td>C3:Damaging</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td><span class="math inline">\(\eta_3= 0.1\)</span></td>
</tr>
<tr class="even">
<td>C4:Damaging</td>
<td><span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td><span class="math inline">\(\eta_4= 0.2\)</span></td>
</tr>
<tr class="odd">
<td>C5:Damaging</td>
<td><span class="math inline">\(AF &lt; 0.001\)</span></td>
<td><span class="math inline">\(\eta_5= 0.4\)</span></td>
</tr>
<tr class="even">
<td>C6:Non-Damaging</td>
<td><span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td><span class="math inline">\(\eta_6= 0.05\)</span></td>
</tr>
<tr class="odd">
<td>C7:Non-Damaging</td>
<td><span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td><span class="math inline">\(\eta_7= 0.1\)</span></td>
</tr>
<tr class="even">
<td>C8:Non-Damaging</td>
<td><span class="math inline">\(AF &lt; 0.001\)</span></td>
<td><span class="math inline">\(\eta_8=0.2\)</span></td>
</tr>
</tbody>
</table>
<ul>
<li>Damaging (probably damaging):Polyphen2.HDIV.score &gt;=0.957</li>
</ul>
</div>
</div>
<div id="constraint-top-51003-gene-set" class="section level2">
<h2>Constraint (top 5%~1003) gene set</h2>
<p>Constraint gene set is from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">this paper</a>.</p>
<div id="cons-simple-burden-analysis" class="section level3">
<h3>Cons: Simple burden analysis</h3>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
The burden of 8 categories is
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
The burden of eight categories
</caption>
<thead>
<tr>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.ca
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
No.co
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
1.0955882
</td>
<td style="text-align:right;">
0.1475154
</td>
<td style="text-align:right;">
298
</td>
<td style="text-align:right;">
272
</td>
</tr>
<tr>
<td style="text-align:right;">
1.0418006
</td>
<td style="text-align:right;">
0.3169813
</td>
<td style="text-align:right;">
324
</td>
<td style="text-align:right;">
311
</td>
</tr>
<tr>
<td style="text-align:right;">
1.0149012
</td>
<td style="text-align:right;">
0.2841436
</td>
<td style="text-align:right;">
3133
</td>
<td style="text-align:right;">
3087
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9455085
</td>
<td style="text-align:right;">
0.9915877
</td>
<td style="text-align:right;">
3505
</td>
<td style="text-align:right;">
3707
</td>
</tr>
<tr>
<td style="text-align:right;">
1.0069470
</td>
<td style="text-align:right;">
0.3619947
</td>
<td style="text-align:right;">
5508
</td>
<td style="text-align:right;">
5470
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9945441
</td>
<td style="text-align:right;">
0.6747191
</td>
<td style="text-align:right;">
13307
</td>
<td style="text-align:right;">
13380
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9783169
</td>
<td style="text-align:right;">
0.9325853
</td>
<td style="text-align:right;">
9114
</td>
<td style="text-align:right;">
9316
</td>
</tr>
<tr>
<td style="text-align:right;">
1.0149839
</td>
<td style="text-align:right;">
0.1684549
</td>
<td style="text-align:right;">
8535
</td>
<td style="text-align:right;">
8409
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<ul>
<li>The first catogry only has one common variant with 298/272.</li>
</ul>
</div>
<div id="cons-estimate-all-parameters-delta-eta" class="section level3">
<h3>Cons: Estimate all parameters, <span class="math inline">\(\delta, \eta\)</span></h3>
<div id="parameter-estimate" class="section level4">
<h4>Parameter estimate</h4>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p><span class="math inline">\(\hat{\delta}=0.18\)</span>.</p>
</div>
<div id="top-genes-overlapping-with-sfari-genes" class="section level4">
<h4>Top genes overlapping with SFARI genes</h4>
<pre><code>##  [1] &quot;CYFIP1&quot;  &quot;EP400&quot;   &quot;FBN1&quot;    &quot;SRCAP&quot;   &quot;DYNC1H1&quot; &quot;ABCA2&quot;   &quot;DOCK4&quot;  
##  [8] &quot;CACNA1D&quot; &quot;CHD8&quot;    &quot;TRIP12&quot;</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  matrix(c(overlap1, overlap2, 10, 1003), nrow = 2)
## p-value = 1.983e-05
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   3.493774 30.539486
## sample estimates:
## odds ratio 
##   10.50575</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  matrix(c(overlap1, overlap2, 10, 1003), nrow = 2)
## p-value = 0.0008249
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   2.78616 47.74520
## sample estimates:
## odds ratio 
##   12.84167</code></pre>
<p>strong ASD gene score 1,2,3 : enrichemnt is 8/10 vs 76/1003 gives OR 10.5058 and p value <span class="math inline">\(1.98\times 10^{-5}\)</span></p>
<p>strong ASD gene score 1,2: enrichemnt is 4/10 vs 31/1003 gives OR 12.8417 and p value <span class="math inline">\(8.24\times 10^{-4}\)</span></p>
</div>
<div id="bayesian-fdr" class="section level4">
<h4>Bayesian FDR</h4>
<p>Below are top risk genes with highest Bayes factor and enrichment with de novo rank.</p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-13-1.png" width="672" /><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-13-2.png" width="672" /></p>
<p>** Enrichment of top MIRAGE genes with Strong ASD gene, plausible ASD gene and DNM candidate genes Strong ASD gene: SFARI score 1,2 plausible ASD gene: SFARI score 3 DNM candidate genes: top 1000 DNM genes</p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-14-1.png" width="672" /><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-14-2.png" width="672" /></p>
<p>This is the how FDR varies as the posterior probability cutoff <span class="math inline">\(\tau\)</span> changes with <span class="math inline">\(\hat{\delta}=0.18\)</span> as prior.</p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<pre><code>## Warning: Column `Gene` joining factors with different levels, coercing to
## character vector</code></pre>
<pre><code>## # A tibble: 15 x 5
##    SKATO.pvalue.adj CMC.pvalue.adj ASUM.pvalue.adj Fisher.pvalue.a~
##               &lt;dbl&gt;          &lt;dbl&gt;           &lt;dbl&gt;            &lt;dbl&gt;
##  1            0.857          0.676               0                1
##  2            0.857          0.676               0                1
##  3            0.857          0.676               0                1
##  4            0.857          0.676               0                1
##  5            0.857          0.676               0                1
##  6            0.857          0.676               0                1
##  7            0.857          0.676               0                1
##  8            0.857          0.676               0                1
##  9            0.857          0.676               0                1
## 10            0.857          0.676               0                1
## 11            0.857          0.676               0                1
## 12            0.857          0.676               0                1
## 13            0.857          0.676               0                1
## 14            0.857          0.676               0                1
## 15            0.857          0.676               0                1
## # ... with 1 more variable: Fisher.adj.pvalue.adj &lt;dbl&gt;</code></pre>
<pre class="r"><code>MIRAGE_gene=const.gene.BF.pvalue.combine %&gt;% filter(post_prob&gt;0.72)%&gt;%select(Gene, BF, post_prob)
MIRAGE_gene%&gt;%arrange(desc(post_prob))  # FDR&lt;0.2</code></pre>
<pre><code>## # A tibble: 3 x 3
##   Gene      BF post_prob
##   &lt;chr&gt;  &lt;dbl&gt;     &lt;dbl&gt;
## 1 CYFIP1  27.4     0.858
## 2 EP400   15.3     0.771
## 3 FBN1    11.9     0.722</code></pre>
<pre class="r"><code>const.gene.BF.pvalue.combine %&gt;% filter(post_prob&gt;0.52)%&gt;%select(Gene, BF, post_prob) %&gt;% arrange(desc(post_prob)) # FDR&lt;0.3</code></pre>
<pre><code>## # A tibble: 9 x 3
##   Gene       BF post_prob
##   &lt;chr&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 CYFIP1  27.4      0.858
## 2 EP400   15.3      0.771
## 3 FBN1    11.9      0.722
## 4 SRCAP    9.83     0.683
## 5 DYNC1H1  9.48     0.675
## 6 ABCA2    8.92     0.662
## 7 DOCK4    8.45     0.650
## 8 CACNA1D  6.28     0.579
## 9 CHD8     5.19     0.532</code></pre>
<pre class="r"><code>const.gene.BF.pvalue.combine %&gt;% filter(post_prob&gt;0.4)%&gt;%select(Gene, BF, post_prob) %&gt;% arrange(desc(post_prob))</code></pre>
<pre><code>## # A tibble: 24 x 3
##    Gene       BF post_prob
##    &lt;chr&gt;   &lt;dbl&gt;     &lt;dbl&gt;
##  1 CYFIP1  27.4      0.858
##  2 EP400   15.3      0.771
##  3 FBN1    11.9      0.722
##  4 SRCAP    9.83     0.683
##  5 DYNC1H1  9.48     0.675
##  6 ABCA2    8.92     0.662
##  7 DOCK4    8.45     0.650
##  8 CACNA1D  6.28     0.579
##  9 CHD8     5.19     0.532
## 10 TRIP12   4.79     0.513
## # ... with 14 more rows</code></pre>
<pre class="r"><code>ASUM_gene=const.gene.BF.pvalue.combine %&gt;% filter(ASUM.pvalue.adj&lt;0.05) %&gt;% select(Gene,BF, post_prob, ASUM.pvalue.adj)
ASUM_gene %&gt;% arrange(ASUM.pvalue.adj) %&gt;% print(n=20)</code></pre>
<pre><code>## # A tibble: 16 x 4
##    Gene       BF post_prob ASUM.pvalue.adj
##    &lt;chr&gt;   &lt;dbl&gt;     &lt;dbl&gt;           &lt;dbl&gt;
##  1 CHST1    1.43     0.239          0     
##  2 TRIM8    1.32     0.225          0     
##  3 RBM14    1.19     0.208          0     
##  4 ZIC2     1.19     0.208          0     
##  5 ZSCAN10  1.42     0.238          0     
##  6 SMARCA4  1.19     0.208          0     
##  7 AP2A1    1.32     0.225          0     
##  8 DHX30    1.38     0.233          0     
##  9 TRA2B    1.29     0.220          0     
## 10 NR2F1    1.23     0.213          0     
## 11 EBF1     1.32     0.225          0     
## 12 NRF1     1.19     0.208          0     
## 13 TUBG1    1.25     0.215          0     
## 14 PCBP1    1.25     0.215          0     
## 15 MAFB     1.16     0.203          0     
## 16 ADAR     3.10     0.405          0.0290</code></pre>
<pre class="r"><code>right_join(as_tibble(const.gene.BF), ASUM_gene, by=&quot;Gene&quot;)%&gt;% select(-post_prob.y, -BF.y)%&gt;%arrange(desc(post_prob.x)) </code></pre>
<pre><code>## Warning: Column `Gene` joining factor and character vector, coercing into
## character vector</code></pre>
<pre><code>## # A tibble: 16 x 6
##    Gene     BF.x LoF.BF nonLoF.BF post_prob.x ASUM.pvalue.adj
##    &lt;chr&gt;   &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;           &lt;dbl&gt;
##  1 ADAR     3.10   1.29      2.41       0.405          0.0290
##  2 CHST1    1.43   1.29      1.11       0.239          0     
##  3 ZSCAN10  1.42   1         1.42       0.238          0     
##  4 DHX30    1.38   1         1.38       0.233          0     
##  5 TRIM8    1.32   1         1.32       0.225          0     
##  6 AP2A1    1.32   1         1.32       0.225          0     
##  7 EBF1     1.32   1         1.32       0.225          0     
##  8 TRA2B    1.29   1         1.29       0.220          0     
##  9 TUBG1    1.25   1         1.25       0.215          0     
## 10 PCBP1    1.25   1         1.25       0.215          0     
## 11 NR2F1    1.23   1         1.23       0.213          0     
## 12 RBM14    1.19   1         1.19       0.208          0     
## 13 ZIC2     1.19   1         1.19       0.208          0     
## 14 SMARCA4  1.19   1         1.19       0.208          0     
## 15 NRF1     1.19   1         1.19       0.208          0     
## 16 MAFB     1.16   1         1.16       0.203          0</code></pre>
<pre class="r"><code>left_join(as_tibble(MIRAGE_gene), ASUM_gene, by=&quot;Gene&quot;)%&gt;% select(-post_prob.y, -BF.y)%&gt;%arrange(desc(post_prob.x))%&gt;% print(n=29) ## ASUM: does not work for genes with one variant, resutling in NA </code></pre>
<pre><code>## # A tibble: 3 x 4
##   Gene    BF.x post_prob.x ASUM.pvalue.adj
##   &lt;chr&gt;  &lt;dbl&gt;       &lt;dbl&gt;           &lt;dbl&gt;
## 1 CYFIP1  27.4       0.858              NA
## 2 EP400   15.3       0.771              NA
## 3 FBN1    11.9       0.722              NA</code></pre>
<p>FDR has the minimum of 0.14</p>
<table>
<caption>Number of genes identified by different methods at different FDR levels.</caption>
<thead>
<tr class="header">
<th>method</th>
<th>FDR&lt;0.2</th>
<th>FDR&lt;0.3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MIRAGE</td>
<td>3</td>
<td>9</td>
</tr>
<tr class="even">
<td>SKATO</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>CMC</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>ASUM</td>
<td>20</td>
<td>20</td>
</tr>
<tr class="odd">
<td>Burden</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Burden.adj</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of genes identified by different methods at different FDR levels.</caption>
<thead>
<tr class="header">
<th>method</th>
<th>FDR&lt;0.2</th>
<th>FDR&lt;0.3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MIRAGE</td>
<td>3</td>
<td>9</td>
</tr>
<tr class="even">
<td>SKATO</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>CMC</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Burden</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Burden.adj</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
</div>
<div id="highlight-genes" class="section level4">
<h4>highlight genes</h4>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
</div>
<div id="cons-estimate-eta_i" class="section level3">
<h3>Cons: Estimate <span class="math inline">\(\eta_i\)</span></h3>
<table style="width:96%;">
<caption>Parameter estimates of 1003 constraint genes by EM with eight partitions.</caption>
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="27%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th>Annotation category</th>
<th>fix <span class="math inline">\(\delta=0.1\)</span> (test.stat)</th>
<th>fix <span class="math inline">\(\delta=0.2\)</span> (test.stat)</th>
<th>estimate <span class="math inline">\(\hat{\delta}=0.18 (9.93)\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td><span class="math inline">\(\eta_1=0\)</span> (0)</td>
<td>0 (0)</td>
<td>0 (0)</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td><span class="math inline">\(\eta_2=0.54\)</span> (0.63)</td>
<td>0.39 (0.76)</td>
<td>0.41 (0.75)</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td><span class="math inline">\(\eta_3=0\)</span> (0)</td>
<td>0 (0)</td>
<td>0 (0)</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td><span class="math inline">\(\eta_4=0\)</span> (0)</td>
<td>0 (0)</td>
<td>0 (0)</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td><span class="math inline">\(\eta_5=0.23\)</span> (5.42)</td>
<td>0.15 (5.71)</td>
<td>0.16 (5.72)</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td><span class="math inline">\(\eta_6=0\)</span> (0)</td>
<td>0 (0)</td>
<td>0 (0)</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td><span class="math inline">\(\eta_7=0\)</span> (0)</td>
<td>0 (0)</td>
<td>0 (0)</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td><span class="math inline">\(\eta_8=0.24\)</span> (3.11)</td>
<td>0.16 (3.49)</td>
<td>0.17 (3.46)</td>
</tr>
</tbody>
</table>
<pre class="r"><code>para.est=c(0, 0.54, 0, 0, 0.23, 0, 0, 0.24)
x=1:8
plot(x, para.est, type=&quot;p&quot;, ylim=c(0,1), cex.lab=0.7, ylab=expression(paste(hat(beta))), pch=16, xlab=&quot;Variant Category&quot;, main=&quot;Parameter estimate of constraint genes with fixed delta=0.1&quot;)</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<pre><code>## 
## Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     combine</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<pre class="r"><code>kable(cbind(rank.BF.const.gene.common.delta[1:20,1:2], no.case, no.contr), row.names=NA, caption =(&quot;Top 20 genes of constraint genes with fixed delta&quot;), &quot;html&quot;)%&gt;%
kable_styling() %&gt;%
scroll_box(height = &quot;200px&quot;)</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 20 genes of constraint genes with fixed delta
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Gene
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
BF
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
no.case
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
no.contr
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
212
</td>
<td style="text-align:left;">
CYFIP1
</td>
<td style="text-align:right;">
47.172299
</td>
<td style="text-align:right;">
151
</td>
<td style="text-align:right;">
156
</td>
</tr>
<tr>
<td style="text-align:left;">
45
</td>
<td style="text-align:left;">
EP400
</td>
<td style="text-align:right;">
27.197227
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
58
</td>
</tr>
<tr>
<td style="text-align:left;">
214
</td>
<td style="text-align:left;">
FBN1
</td>
<td style="text-align:right;">
19.281002
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
247
</td>
</tr>
<tr>
<td style="text-align:left;">
62
</td>
<td style="text-align:left;">
DYNC1H1
</td>
<td style="text-align:right;">
16.723128
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:right;">
69
</td>
</tr>
<tr>
<td style="text-align:left;">
677
</td>
<td style="text-align:left;">
DOCK4
</td>
<td style="text-align:right;">
15.512253
</td>
<td style="text-align:right;">
334
</td>
<td style="text-align:right;">
343
</td>
</tr>
<tr>
<td style="text-align:left;">
217
</td>
<td style="text-align:left;">
SRCAP
</td>
<td style="text-align:right;">
13.958606
</td>
<td style="text-align:right;">
207
</td>
<td style="text-align:right;">
159
</td>
</tr>
<tr>
<td style="text-align:left;">
706
</td>
<td style="text-align:left;">
ABCA2
</td>
<td style="text-align:right;">
12.227103
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
567
</td>
<td style="text-align:left;">
TRIP12
</td>
<td style="text-align:right;">
7.989234
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
24
</td>
</tr>
<tr>
<td style="text-align:left;">
53
</td>
<td style="text-align:left;">
CHD8
</td>
<td style="text-align:right;">
7.779481
</td>
<td style="text-align:right;">
78
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
137
</td>
<td style="text-align:left;">
PLXNB1
</td>
<td style="text-align:right;">
7.507818
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
584
</td>
<td style="text-align:left;">
HIRA
</td>
<td style="text-align:right;">
6.707113
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
609
</td>
<td style="text-align:left;">
CACNA1D
</td>
<td style="text-align:right;">
6.523912
</td>
<td style="text-align:right;">
147
</td>
<td style="text-align:right;">
123
</td>
</tr>
<tr>
<td style="text-align:left;">
141
</td>
<td style="text-align:left;">
KALRN
</td>
<td style="text-align:right;">
5.577831
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
193
</td>
</tr>
<tr>
<td style="text-align:left;">
213
</td>
<td style="text-align:left;">
INO80
</td>
<td style="text-align:right;">
4.981064
</td>
<td style="text-align:right;">
155
</td>
<td style="text-align:right;">
131
</td>
</tr>
<tr>
<td style="text-align:left;">
247
</td>
<td style="text-align:left;">
TP63
</td>
<td style="text-align:right;">
4.972342
</td>
<td style="text-align:right;">
79
</td>
<td style="text-align:right;">
71
</td>
</tr>
<tr>
<td style="text-align:left;">
109
</td>
<td style="text-align:left;">
ANAPC1
</td>
<td style="text-align:right;">
4.615794
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
35
</td>
</tr>
<tr>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
ADAR
</td>
<td style="text-align:right;">
4.613623
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
28
</td>
</tr>
<tr>
<td style="text-align:left;">
450
</td>
<td style="text-align:left;">
MYH10
</td>
<td style="text-align:right;">
4.481565
</td>
<td style="text-align:right;">
161
</td>
<td style="text-align:right;">
135
</td>
</tr>
<tr>
<td style="text-align:left;">
54
</td>
<td style="text-align:left;">
METTL3
</td>
<td style="text-align:right;">
4.426548
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
20
</td>
</tr>
<tr>
<td style="text-align:left;">
394
</td>
<td style="text-align:left;">
BAZ1A
</td>
<td style="text-align:right;">
4.424587
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
52
</td>
</tr>
</tbody>
</table>
</div>
<div id="enrichment-with-de-novo-genes" class="section level4">
<h4>enrichment with de novo genes</h4>
<p>In top 1000 denovo genes, there are 9 in top 20 constarint genes and 87 in 1003 constraint genes. By Fisher exact test (9/20 vs 87/1003), OR=5.17 and p value of <span class="math inline">\(4.10 \times 10^{-4}\)</span>.</p>
</div>
<div id="lof-category" class="section level4">
<h4>LoF category</h4>
<pre class="r"><code>beta.est=c(0.49419653, 0.31677155, 0.22161936, 0.16934539, 0.13697687, 0.11502861, 0.09918004, 0.08717663, 0.07778499, 0.07022805)
delta=seq(0.1, 1, by=0.1)
plot(beta.est, xaxt=&#39;n&#39;, ylab=expression(paste(hat(beta))), main=&quot;Parameter estimate for LoF category&quot;, type=&quot;o&quot;, xlab=expression(paste(delta)))
axis.label=c(&quot;0.1&quot;, &quot;0.2&quot;, &quot;0.3&quot;, &quot;0.4&quot;, &quot;0.5&quot;, &quot;0.6&quot;, &quot;0.7&quot;, &quot;0.8&quot;, &quot;0.9&quot;, &quot;1&quot;)
axis(1,at=1:10,labels=axis.label)</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<pre class="r"><code>test.stat=c(0.6105703, 0.7301446, 0.7621092, 0.7772918, 0.7866515, 0.7931309, 0.7979188, 0.8016140, 0.8045560, 0.8069559, cex.axis=0.8)</code></pre>
</div>
</div>
<div id="cons-fixed-eta_i" class="section level3">
<h3>Cons: Fixed <span class="math inline">\(\eta_i\)</span></h3>
<p>With fixed parameter in every variant group, calculate the BF of every gene which is the product of BF of every variant in this gene. Below are top 20 genes with large BF.</p>
<table>
<caption>Bayes factor partition of top 20 genes from 1003 constraint genes. Rank: Rank by DNM; p-val: empirical p values by perturbating case control labels with null distribution of all constraint genes. ``-&quot; means no variants in the category.</caption>
<thead>
<tr class="header">
<th>Gene</th>
<th>BF</th>
<th>C1</th>
<th>C2</th>
<th>C3</th>
<th>C4</th>
<th>C5</th>
<th>C6</th>
<th>C7</th>
<th>C8</th>
<th>Rank</th>
<th>p-val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CYFIP1</td>
<td>12.44</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>0.81</td>
<td>13.53</td>
<td>-</td>
<td>0.90</td>
<td>1.26</td>
<td>18200</td>
<td>0.0015</td>
</tr>
<tr class="even">
<td>ABCA2</td>
<td>11.96</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>1.18</td>
<td>-</td>
<td>-</td>
<td>10.13</td>
<td>18210</td>
<td>0.0016</td>
</tr>
<tr class="odd">
<td>CACNA1D</td>
<td>11.26</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>9.46</td>
<td>0.97</td>
<td>0.91</td>
<td>1.35</td>
<td>186</td>
<td>0.0018</td>
</tr>
<tr class="even">
<td>FBN1</td>
<td>6.94</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>0.83</td>
<td>5.83</td>
<td>0.95</td>
<td>0.89</td>
<td>1.71</td>
<td>1324</td>
<td>0.0041</td>
</tr>
<tr class="odd">
<td>CIT</td>
<td>6.68</td>
<td>-</td>
<td>1.63</td>
<td>-</td>
<td>5.90</td>
<td>0.45</td>
<td>0.95</td>
<td>0.70</td>
<td>2.31</td>
<td>13225</td>
<td>0.0044</td>
</tr>
<tr class="even">
<td>PLXNB1</td>
<td>5.78</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>3.45</td>
<td>-</td>
<td>-</td>
<td>1.68</td>
<td>364</td>
<td>0.0057</td>
</tr>
<tr class="odd">
<td>CREBBP</td>
<td>5.51</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>2.22</td>
<td>-</td>
<td>1.47</td>
<td>1.70</td>
<td>1232</td>
<td>0.0062</td>
</tr>
<tr class="even">
<td>GTF3C4</td>
<td>5.50</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.85</td>
<td>-</td>
<td>0.92</td>
<td>1.23</td>
<td>15612</td>
<td>0.0063</td>
</tr>
<tr class="odd">
<td>MYH10</td>
<td>5.25</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.45</td>
<td>-</td>
<td>0.79</td>
<td>1.50</td>
<td>48</td>
<td>0.0068</td>
</tr>
<tr class="even">
<td>DNM3</td>
<td>4.91</td>
<td>-</td>
<td>1.63</td>
<td>-</td>
<td>-</td>
<td>2.02</td>
<td>-</td>
<td>1.27</td>
<td>1.18</td>
<td>16092</td>
<td>0.0076</td>
</tr>
<tr class="odd">
<td>CHD8</td>
<td>4.79</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>0.77</td>
<td>6.66</td>
<td>0.97</td>
<td>-</td>
<td>0.96</td>
<td>2</td>
<td>0.008</td>
</tr>
<tr class="even">
<td>TP63</td>
<td>4.05</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>0.81</td>
<td>2.66</td>
<td>-</td>
<td>0.92</td>
<td>2.05</td>
<td>15512</td>
<td>0.0108</td>
</tr>
<tr class="odd">
<td>DOCK4</td>
<td>3.91</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.47</td>
<td>0.90</td>
<td>0.90</td>
<td>1.07</td>
<td>18342</td>
<td>0.0116</td>
</tr>
<tr class="even">
<td>CNOT1</td>
<td>3.73</td>
<td>-</td>
<td>-</td>
<td>0.91</td>
<td>-</td>
<td>3.25</td>
<td>1.06</td>
<td>-</td>
<td>1.19</td>
<td>8577</td>
<td>0.0128</td>
</tr>
<tr class="odd">
<td>HIRA</td>
<td>3.04</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>2.37</td>
<td>1.06</td>
<td>-</td>
<td>1.21</td>
<td>17444</td>
<td>0.0192</td>
</tr>
<tr class="even">
<td>ITSN1</td>
<td>2.94</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>0.91</td>
<td>3.23</td>
<td>-</td>
<td>-</td>
<td>1.00</td>
<td>1479</td>
<td>0.0205</td>
</tr>
<tr class="odd">
<td>HDLBP</td>
<td>2.91</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>1.08</td>
<td>-</td>
<td>4.11</td>
<td>0.65</td>
<td>480</td>
<td>0.0210</td>
</tr>
<tr class="even">
<td>BAZ1A</td>
<td>2.89</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>2.79</td>
<td>1.02</td>
<td>0.85</td>
<td>1.20</td>
<td>1392</td>
<td>0.0212</td>
</tr>
<tr class="odd">
<td>SLC4A8</td>
<td>2.81</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>1.90</td>
<td>-</td>
<td>-</td>
<td>1.48</td>
<td>16562</td>
<td>0.0225</td>
</tr>
<tr class="even">
<td>VPRBP</td>
<td>2.79</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>3.80</td>
<td>-</td>
<td>-</td>
<td>0.73</td>
<td>1393</td>
<td>0.0230</td>
</tr>
</tbody>
</table>
<p>In top 1000 denovo genes, 5 are found in top 20 genes and 81 in all 1003 genes. Fisher exact test (5/20 vs 81/1003) gives OR=3.7857 and p value=0.0211.</p>
<p>ASD gene priors for 1003 constraint genes. (1) Enrichment with top 1000 genes: 11/20 vs 261/1003, OR=3.4696, pvalue=0.008017; (2) enrichment with top 2000 genes: 12/20 vs 419/1003, OR=2.0892, p value=0.1135.</p>
</div>
<div id="cons-estimate-eta_i-gene-specific-delta_i" class="section level3">
<h3>Cons: Estimate <span class="math inline">\(\eta_i\)</span>, gene specific <span class="math inline">\(\delta_i\)</span></h3>
<div id="lof-only" class="section level4">
<h4>(1) LoF only</h4>
<p>The burden is 622/583. <span class="math inline">\(\widehat{\beta}=0.74\)</span> and test statistic is 2.99.</p>
</div>
<div id="eight-categories" class="section level4">
<h4>(2) Eight categories</h4>
<table>
<caption>Parameter estimates of 1003 constraint genes by EM with eight partitions with gene specific <span class="math inline">\(\delta\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\eta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0.81</td>
<td>2.98</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.12</td>
<td>6.88</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.09</td>
<td>0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(rank.BF.const.gene.specific.delta[1:20,1:2], row.names=NA, caption =(&quot;Top 20 genes of constraint genes with gene specific delta&quot;), &quot;html&quot;)%&gt;%
kable_styling() %&gt;%
scroll_box(height = &quot;200px&quot;)</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 20 genes of constraint genes with gene specific delta
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Gene
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
BF
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
212
</td>
<td style="text-align:left;">
CYFIP1
</td>
<td style="text-align:right;">
19.233983
</td>
</tr>
<tr>
<td style="text-align:left;">
62
</td>
<td style="text-align:left;">
DYNC1H1
</td>
<td style="text-align:right;">
7.268263
</td>
</tr>
<tr>
<td style="text-align:left;">
45
</td>
<td style="text-align:left;">
EP400
</td>
<td style="text-align:right;">
7.175237
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
UBR4
</td>
<td style="text-align:right;">
6.546470
</td>
</tr>
<tr>
<td style="text-align:left;">
217
</td>
<td style="text-align:left;">
SRCAP
</td>
<td style="text-align:right;">
6.075509
</td>
</tr>
<tr>
<td style="text-align:left;">
677
</td>
<td style="text-align:left;">
DOCK4
</td>
<td style="text-align:right;">
5.871390
</td>
</tr>
<tr>
<td style="text-align:left;">
214
</td>
<td style="text-align:left;">
FBN1
</td>
<td style="text-align:right;">
5.577938
</td>
</tr>
<tr>
<td style="text-align:left;">
706
</td>
<td style="text-align:left;">
ABCA2
</td>
<td style="text-align:right;">
5.437373
</td>
</tr>
<tr>
<td style="text-align:left;">
609
</td>
<td style="text-align:left;">
CACNA1D
</td>
<td style="text-align:right;">
5.328570
</td>
</tr>
<tr>
<td style="text-align:left;">
53
</td>
<td style="text-align:left;">
CHD8
</td>
<td style="text-align:right;">
5.264254
</td>
</tr>
<tr>
<td style="text-align:left;">
199
</td>
<td style="text-align:left;">
MACF1
</td>
<td style="text-align:right;">
4.142091
</td>
</tr>
<tr>
<td style="text-align:left;">
137
</td>
<td style="text-align:left;">
PLXNB1
</td>
<td style="text-align:right;">
4.110564
</td>
</tr>
<tr>
<td style="text-align:left;">
75
</td>
<td style="text-align:left;">
GLG1
</td>
<td style="text-align:right;">
4.000271
</td>
</tr>
<tr>
<td style="text-align:left;">
141
</td>
<td style="text-align:left;">
KALRN
</td>
<td style="text-align:right;">
3.779369
</td>
</tr>
<tr>
<td style="text-align:left;">
84
</td>
<td style="text-align:left;">
ACACA
</td>
<td style="text-align:right;">
3.316591
</td>
</tr>
<tr>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
ADAR
</td>
<td style="text-align:right;">
3.246357
</td>
</tr>
<tr>
<td style="text-align:left;">
82
</td>
<td style="text-align:left;">
ASIC2
</td>
<td style="text-align:right;">
3.156752
</td>
</tr>
<tr>
<td style="text-align:left;">
584
</td>
<td style="text-align:left;">
HIRA
</td>
<td style="text-align:right;">
3.129200
</td>
</tr>
<tr>
<td style="text-align:left;">
567
</td>
<td style="text-align:left;">
TRIP12
</td>
<td style="text-align:right;">
3.012621
</td>
</tr>
<tr>
<td style="text-align:left;">
126
</td>
<td style="text-align:left;">
ITSN1
</td>
<td style="text-align:right;">
2.949216
</td>
</tr>
</tbody>
</table>
</div>
<p>In top 1000 denovo genes, 6 are found in top 20 genes and 102 in all 1003 genes. Fisher exact test (6/20 vs 102/1003) gives OR=2.95, pvalue=0.0311.</p>
<p>ASD gene priors for 1003 constraint genes. (1) Enrichment with top 1000 genes: 10/20 vs 331/1003, OR=1.5146, pvalue=0.2896.</p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<pre class="r"><code>para.est=c(0, 0.54, 0, 0, 0.23, 0, 0, 0.24)
x=1:8
plot(x, para.est, type=&quot;p&quot;, ylim=c(0,1), cex.lab=0.7, ylab=expression(paste(hat(beta))), pch=15, col=&quot;black&quot;, xlab=&quot;Variant Category&quot;, main=&quot;Parameter estimates of constraint genes&quot;)
points(x, para.summ[,1], type=&quot;p&quot;, ylim=c(0,1), cex.lab=0.7, ylab=expression(paste(hat(beta))), pch=16,col=&quot;red&quot;)
legend(6, 0.8, c(expression(paste(&quot;common &quot;, delta, &quot;=0.1&quot;)), expression(paste(&quot;gene specific &quot;, delta[i]))), col=c(&quot;black&quot;, &quot;red&quot;), pch=c(15,16))
abline(h=0, lty=2)</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<p>Replace “damaging” with “consensus damaging”</p>
<table>
<caption>Parameter estimates of 1003 constraint genes by EM with eight partitions with gene specific <span class="math inline">\(\delta\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\eta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0.7318</td>
<td>2.99</td>
</tr>
<tr class="odd">
<td>Consensus Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Consensus Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Consensus Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.1310</td>
<td>10.06</td>
</tr>
<tr class="even">
<td>Non-Consensus Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Non-Consensus Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Non-Consensus Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.0737</td>
<td>0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="qq-plot-of-p-values-by-different-methods" class="section level3">
<h3>QQ plot of p values by different methods</h3>
<pre><code>## [1] 0.3069552</code></pre>
<pre><code>## [1] 0.6547919</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<p>This is for LoF and damaging variants only.</p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-35-1.png" width="672" /></p>
</div>
<div id="empirical-distribution-of-bf-by-permutations" class="section level3">
<h3>Empirical distribution of BF by permutations</h3>
<p>permutate the variant count in cases and controls for every gene 100 times, calculate Bayes factor by MIRAGE and get their empirical p value by comparing original BF and 100 permutated BF. The null distribution of BF is pooling BF’s of all genes through permutations together.</p>
<pre><code>## # A tibble: 771 x 6
##    Gene       BF LoF.BF nonLoF.BF post_prob empirical.pvalue
##    &lt;fct&gt;   &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;            &lt;dbl&gt;
##  1 CYFIP1  27.4   1         27.4      0.858          0.00115
##  2 EP400   15.3   0.940     16.3      0.771          0.00154
##  3 FBN1    11.9   1         11.9      0.722          0.00200
##  4 SRCAP    9.83  1          9.83     0.683          0.00246
##  5 DYNC1H1  9.48  1.29       7.37     0.675          0.00259
##  6 ABCA2    8.92  1          8.92     0.662          0.00279
##  7 DOCK4    8.45  1          8.45     0.650          0.00293
##  8 CACNA1D  6.28  1          6.28     0.579          0.00428
##  9 CHD8     5.19  1.29       4.03     0.532          0.00559
## 10 KALRN    4.79  1.29       3.72     0.513          0.00658
## # ... with 761 more rows</code></pre>
<p>Histogram of empirical p values by permutations</p>
<pre><code>## Warning: Calling `as_tibble()` on a vector is discouraged, because the behavior is likely to change in the future. Use `tibble::enframe(name = NULL)` instead.
## This warning is displayed once per session.</code></pre>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
</div>
</div>
<div id="constraint-genes" class="section level2">
<h2>Constraint genes</h2>
<p>There are 17,416 genes.</p>
<div id="con-estimate-beta_i-gene-specific-delta_i" class="section level3">
<h3>Con: Estimate <span class="math inline">\(\beta_i\)</span>, gene specific <span class="math inline">\(\delta_i\)</span></h3>
<div id="non-nested-gene-sets" class="section level4">
<h4>Non-nested gene sets</h4>
<p>With their constraint scores, focus on top 10% genes, top 10-20% genes, top 20-30% genes, top 30-40% genes, top 40-50% genes.</p>
<pre class="r"><code>beta=matrix(nrow=5, ncol=8)
rownames(beta)=c(&quot;&lt;10%&quot;, &quot;10-20%&quot;, &quot;20-30%&quot;, &quot;30-40%&quot;, &quot;40-50%&quot;)
beta[1,]=c(0, 0.6195,  0, 0.0025,  0.0427, 0,  0,  0.0855)
beta[2,]=c(0, 0.0073,  0, 0,   0.0316, 0, 0,  0.0588)
beta[3,]=c(0,  0.0179, 0,   0,   0.0284,0,  0.0294,  0.0390)
beta[4,]=c(0,  0.0883,  0, 0,   0.0245,0,  0,   0.0017)
beta[5,]=c(0,  0.2706,  0, 0,   0.0038,0,   0.0187,  0.0006)
plot(beta[,1], ylim=c(0,1),cex=0.1, type=&quot;o&quot;, xaxt=&quot;n&quot;, xlab=&quot;Score quantile &quot;, pch=16, ylab=expression(paste(hat(beta))), cex.lab=0.75, main=&quot;Parameter estimates of top five non-nested constraint gene sets&quot;,)
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.8)
for (i in 2:8)
 lines(beta[,i], type=&quot;o&quot;, col=i, pch=16)
betas=c(expression(paste(hat(beta)[1])),expression(paste(hat(beta)[2])), expression(paste(hat(beta)[3])),expression(paste(hat(beta)[4])), 
expression(paste(hat(beta)[5])),expression(paste(hat(beta)[6])), expression(paste(hat(beta)[7])),expression(paste(hat(beta)[8])))
#legend(2, 1, c(paste(&quot;beta&quot;, seq(1,8), sep=&quot;&quot;)), col=seq(1,8), lty=rep(1,8))
legend(4, 1, betas, col=seq(1,8), lty=rep(1,8), ncol=2)</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<p>Replace “Damaging” with “Consensus damaging”</p>
<pre class="r"><code>beta=matrix(nrow=5, ncol=8)
rownames(beta)=c(&quot;top10%&quot;, &quot;top10-20%&quot;, &quot;top20-30%&quot;, &quot;top30-40%&quot;, &quot;top40-50%&quot;)
beta[1,]=c(0, 0.6275,  0, 0,  0.0454, 0,  0,  0.0848)
beta[2,]=c(0, 0.0326,  0, 0,   0.0667, 0, 0,  0.0071)
beta[3,]=c(0,  0.0379, 0,   0,   0.0365,0,  0.0509,  0.0034)
beta[4,]=c(0,  0.1009,  0, 0,   0.0020,0,  0,   0.0008)
beta[5,]=c(0,  0.2652,  0, 0,   0.0350,0,   0.0080,  0.0004)
plot(beta[,1], ylim=c(0,1), type=&quot;o&quot;, xaxt=&quot;n&quot;, xlab=&quot;&quot;, ylab=expression(paste(hat(beta))), main=&quot;Parameter estimate of five non-nested gene sets&quot;)
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type=&quot;o&quot;, col=i)
legend(4, 1, c(paste(&quot;beta&quot;, seq(1,8), sep=&quot;&quot;)), col=seq(1,8), lty=rep(1,8))</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
<table style="width:81%;">
<caption>Number of variants in C5 and C8 at different constrait level.</caption>
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="27%" />
</colgroup>
<thead>
<tr class="header">
<th>constraint level</th>
<th>C5: Consensus Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></th>
<th>C8: Non-Consensus Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>top 10%</td>
<td>34,028/33,508=1.0155</td>
<td>30,922/30,736=1.0061</td>
</tr>
<tr class="even">
<td>top 10-20%</td>
<td>17,184/16,925=1.0153</td>
<td>15,738/15,894=0.9902</td>
</tr>
<tr class="odd">
<td>top 20-30%</td>
<td>12,371/12,588=0.9828</td>
<td>12,035/12,216=0.9852</td>
</tr>
<tr class="even">
<td>top 30-40%</td>
<td>11,972/12,038=0.9945</td>
<td>10,652/10,605=1.0044</td>
</tr>
<tr class="odd">
<td>top 40-50%</td>
<td>11,055/11,147=0.9917</td>
<td>9,548/9,903=0.9642</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tada-gene-q5" class="section level2">
<h2>TADA gene (q&lt;5%)</h2>
<p>Only 46 genes with TADA q values less than 5%.</p>
<div id="tada-estimate-beta-fixed-delta1" class="section level3">
<h3>TADA: Estimate <span class="math inline">\(\beta\)</span>, fixed <span class="math inline">\(\delta=1\)</span></h3>
<div id="only-lof-variants" class="section level4">
<h4>Only LoF variants</h4>
<p>In the simple burden analysis (see Table above), LoF variants is 55/27. With <span class="math inline">\(\delta=1\)</span>, <span class="math inline">\(\widehat{\beta}=0.5692\)</span>, and test statistic is 12.0211.</p>
</div>
</div>
<div id="tada-estimate-beta_i-gene-specific-delta_i" class="section level3">
<h3>TADA: Estimate <span class="math inline">\(\beta_i\)</span>, gene specific <span class="math inline">\(\delta_i\)</span></h3>
<p>Use 1-FDR of every gene as priors to estimate <span class="math inline">\(\beta_i\)</span>.</p>
<div id="only-lof-category" class="section level4">
<h4>Only LoF category</h4>
<p><span class="math inline">\(\widehat{\beta}=0.63\)</span>, test statistics is 10.07.</p>
</div>
</div>
</div>
<div id="rvis-gene" class="section level2">
<h2>RVIS gene</h2>
<div id="rvisestimate-beta_i-gene-specific-delta_i" class="section level3">
<h3>RVIS:Estimate <span class="math inline">\(\beta_i\)</span>, gene specific <span class="math inline">\(\delta_i\)</span></h3>
<div id="quantile-score-50-8437-genes" class="section level4">
<h4>Quantile score &lt;50 (8437) genes</h4>
<table>
<caption>Parameter estimates of 8437 ( quantile&lt;50 ) RVIS genes by EM with eight partitions with gene specific <span class="math inline">\(\delta_i\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0.17</td>
<td>6.06</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.06</td>
<td>13.55</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.08</td>
<td>11.34</td>
</tr>
</tbody>
</table>
</div>
<div id="with-different-quantile-score" class="section level4">
<h4>with different quantile score</h4>
<p>Replace “Damaging” with “Consensus damaging”</p>
<pre class="r"><code>beta=matrix(nrow=5, ncol=8)
rownames(beta)=c(&quot;Quantile score&lt;10&quot;, &quot;10-20&quot;, &quot;20-30&quot;, &quot;30-40&quot;, &quot;40-50&quot;)
beta[1,]=c(0, 0.5309,  0, 0,  0.0414, 0,  0,  0.0794)
beta[2,]=c(0, 0.1402,  0, 0,   0.1098, 0, 0,  0.0464)
beta[3,]=c(0,  0.0026, 0,   0,   0.0222,0,  0.0197,  0.1779)
beta[4,]=c(0,  0.0016,  0, 0,   0.0770,0,  0,   0.0558)
beta[5,]=c(0,  0.1103,  0, 0,   0.1246,0,   0,  0.0369)
plot(beta[,1], ylim=c(0,0.6), type=&quot;o&quot;, xaxt=&quot;n&quot;, xlab=&quot;&quot;, ylab=expression(paste(hat(beta))), main=&quot;Parameter estimate of five non-nested gene sets&quot;)
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type=&quot;o&quot;, col=i)
legend(4, 0.5, c(paste(&quot;beta&quot;, seq(1,8), sep=&quot;&quot;)), col=seq(1,8), lty=rep(1,8))</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="whole-genome" class="section level2">
<h2>Whole genome</h2>
<p>consider all 15624 genes in ASD data.</p>
<div id="whole-estimate-beta_i-gene-specific-delta_i" class="section level3">
<h3>Whole: Estimate <span class="math inline">\(\beta_i\)</span>, gene specific <span class="math inline">\(\delta_i\)</span></h3>
<table>
<caption>Parameter estimates of 15624 genes by EM with eight partitions with gene specific <span class="math inline">\(\delta_i\)</span>.</caption>
<thead>
<tr class="header">
<th>Annotation category</th>
<th><span class="math inline">\(\widehat{\beta}\)</span></th>
<th>test.stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LoF, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>LoF, <span class="math inline">\(AF &lt; 0.01\)</span></td>
<td>0.08</td>
<td>3.94</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.06</td>
<td>23.85</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(0.01 \leq AF &lt; 0.05\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Non-Damaging, <span class="math inline">\(0.001 \leq AF &lt; 0.01\)</span></td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>Non-Damaging, <span class="math inline">\(AF &lt; 0.001\)</span></td>
<td>0.06</td>
<td>11.42</td>
</tr>
</tbody>
</table>
<pre class="r"><code>para.est=c(0, 0.08, 0, 0, 0.06, 0, 0, 0.06)
x=1:8
plot(para.est, type=&quot;p&quot;, ylim=c(0,0.5), cex.lab=0.7, ylab=expression(paste(hat(beta))), pch=16, xlab=&quot;Variant Category&quot;, main=&quot;Parameter estimate of all genes in the whole genome&quot;)</code></pre>
<p><img src="GeneSetAnalysis_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
