<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-11-07" />

<title>QQ plot of burden and ACAT</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">QQ plot of burden and ACAT</h1>
<h4 class="date">2023-11-07</h4>

</div>


<pre class="r"><code>################## DO NOT RUN ########################
path=&quot;C:\\Shengtong\\research\\rare-var\\&quot;

########### read into Laura&#39;s data all inherited variants aggregated 
mirage_input=as_tibble(read.table(paste(path, &quot;ASC\\Laura_result\\mirage_input_20231013.txt&quot;, sep=&quot;&quot;), header=T))
N_all=14578  # from Laura&#39;s script: NOTES_wrap_mirage_100323
#res=mirage(data.frame(mirage_input),n1=N_all,n2=N_all,gamma=c(6,6,6,3,3,3),delta.init = 0.05,estimate.delta = FALSE)
#mirage_input_for_gnomAD_AF=mirage_input %&gt;% select(Variant, Gene)
#write.csv(mirage_input_for_gnomAD_AF, file=paste(path, &quot;ASC\\Laura_result\\mirage_input_for_gnomAD_AF_20231106.csv&quot;, sep=&quot;&quot;))

########## read into gnomAD allele frequency 
ASC_v17_TDT_by_variant_20200916_gnomad=as_tibble(read.table(paste(path, &quot;ASC\\Laura_result\\ASC_v17_TDT_by_variant_2020-09-16_gnomad.txt&quot;, sep=&quot;&quot;), header=T))
mirage_input_with_gnomad_AF=right_join(ASC_v17_TDT_by_variant_20200916_gnomad, mirage_input, by=&quot;Variant&quot;)</code></pre>
<pre class="r"><code>variant_counts &lt;- mirage_input %&gt;%
  group_by(Gene) %&gt;%
  summarise(variant_count = n()) %&gt;%
  arrange(desc(variant_count))
 summary(variant_counts$variant_count)</code></pre>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##     1.0    38.0    70.0    93.8   118.0  6425.0</code></pre>
<div id="qq-plot-of-staar" class="section level2">
<h2>QQ plot of STAAR</h2>
<pre class="r"><code>################# calculate ACAT pvalue for every gene 
#BF_proband_AllGene=as_tibble(read.csv(&quot;../output/kyleData/BF.proband.AllGene_1105_2022.csv&quot;))
#BF_proband_AllGene_sorted=BF_proband_AllGene%&gt;%arrange(desc(post.prob))
colnames(mirage_input_with_gnomad_AF)[4:6]=c(&quot;Transmitted_proband&quot;, &quot;Untransmitted_proband&quot;, &quot;Group_Index&quot;)
interest_gene=unique(mirage_input_with_gnomad_AF$Gene)  # use all genes 
new_data=mirage_input_with_gnomad_AF %&gt;% filter(Gene %in% interest_gene) %&gt;% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Group_Index) %&gt;% drop_na() ##### remove variants with NA AF 
acat_gene_pvalue=numeric()
a1=1; a2=25
for (i in 1:length(interest_gene))
{
  #i=6819
  cat(i, &quot;th gene is running&quot;, &quot;\n&quot;)
  single_gene_data=new_data %&gt;% filter(Gene==interest_gene[i])
  var_group_index=single_gene_data %&gt;% select(Group_Index) %&gt;%pull
  num_var_group=unique(var_group_index)
  
  acat_var_pvalue=numeric()
  acat_var_weight=numeric()  # give weights when use ACAT to collapse all variants 
  #####################
  if (length(num_var_group)==1) # only one variant group
  {
    if (nrow(single_gene_data)==1) # case 1: only one variant in one group
    {
      binom_test=binom.test(c(single_gene_data$Transmitted_proband, single_gene_data$Untransmitted_proband), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
      acat_var_pvalue[1]=binom_test$p.value
      maf=single_gene_data$Gnomad_non_neuro_AF
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300) ##### use weight, beta density  as in ACAT paper  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
    } # end of   if (nrow(single_gene_data)==1) # only one variant in one group
    
    if (nrow(single_gene_data)&gt;1) # case 2: multiple variants in one group
    {
      minor_alele_count=numeric()
      for (j in 1:nrow(single_gene_data)) # count the total number of alleles for every variant 
        minor_alele_count[j]=single_gene_data$Transmitted_proband[j]+single_gene_data$Untransmitted_proband[j]
      
      minor_alele_count_lessthan10=which(minor_alele_count&lt;10) # find variants with MAC&lt;10
      
      #############
      if (length(minor_alele_count_lessthan10)&lt;=1) # if only one or no variant with MAC&lt;10, no need collapsing 
        for (j in 1:nrow(single_gene_data))
        {
         binom_test=binom.test(c(single_gene_data$Transmitted_proband[j], single_gene_data$Untransmitted_proband[j]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
      acat_var_pvalue[j]=binom_test$p.value  
        maf=single_gene_data$Gnomad_non_neuro_AF[j]
        acat_var_weight[j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
        }
    #############
      if (length(minor_alele_count_lessthan10)&gt;1 &amp; length(minor_alele_count_lessthan10)&lt;nrow(single_gene_data)) # some, not all variants have MAC&lt;10
      {
        binom_test=binom.test(c(sum(single_gene_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT ## collapse all variants with MAC&lt;10
      acat_var_pvalue[1]=binom_test$p.value
      maf=single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10]
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      minor_alele_count_morethan10=which(minor_alele_count&gt;=10) # find variants with MAC&gt;=10
      
      if (length(minor_alele_count_morethan10)==1)
      {
      binom_test=binom.test(c(single_gene_data$Transmitted_proband[minor_alele_count_morethan10], single_gene_data$Untransmitted_proband[minor_alele_count_morethan10]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
      acat_var_pvalue[2]=binom_test$p.value
      maf=single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10]
      acat_var_weight[2]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      }
      if (length(minor_alele_count_morethan10)&gt;1)  
        for (j in 1:length(minor_alele_count_morethan10))
        {
         binom_test=binom.test(c(single_gene_data$Transmitted_proband[minor_alele_count_morethan10[j]], single_gene_data$Untransmitted_proband[minor_alele_count_morethan10[j]]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
        acat_var_pvalue[1+j]=binom_test$p.value  
        maf=single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10[j]]
        acat_var_weight[1+j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
        }
      } # end of if (length(minor_alele_count_lessthan10)&gt;1 &amp; length(minor_alele_count_lessthan10)&lt;nrow(single_gene_data)) # some, not all variants have MAC&lt;10
      
       #############
      if (length(minor_alele_count_lessthan10)&gt;1 &amp; length(minor_alele_count_lessthan10)==nrow(single_gene_data)) #  all variants have MAC&lt;10
      {
        binom_test=binom.test(c(sum(single_gene_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT ## collapse all variants with MAC&lt;10
      acat_var_pvalue[1]=binom_test$p.value
      maf=mean(single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10]) # use average MAF https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300) # avoid weight to be 0 
        
      }
      
    }  # end of if (nrow(single_gene_data)&gt;1) # case 2: multiple variants in one group
  ##################### 
    if (max(acat_var_pvalue)==1 &amp; length(acat_var_pvalue)&gt;1)
      acat_var_pvalue[acat_var_pvalue==1]=1-1/(length(acat_var_pvalue)) # adjust pvalues https://github.com/yaowuliu/ACAT
    if (max(acat_var_pvalue)==1 &amp; length(acat_var_pvalue)==1)
      acat_var_pvalue[acat_var_pvalue==1]=1-1e-300
      
    acat_gene_pvalue[i]=ACAT(acat_var_pvalue, acat_var_weight)
    
  } # end of if (length(num_var_group)==1) # only one variant group
  
############
  if (length(num_var_group)&gt;1) # multiple variant groups
  {
    acat_group_pvalue=numeric()
    for (g in 1: length(num_var_group)) # use ACAT within every variant group 
    {
     #g=1
     single_gene_single_group_data=single_gene_data %&gt;% filter(Group_Index==var_group_index[g])
     
     acat_var_pvalue=numeric()  # within each variant group, use ACAT
     acat_var_weight=numeric()
     
     if (nrow(single_gene_single_group_data)==1) # case 1: only one variant in one group
    {
      binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband, single_gene_single_group_data$Untransmitted_proband), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
      acat_var_pvalue[1]=binom_test$p.value
      maf=single_gene_single_group_data$Gnomad_non_neuro_AF
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)  ##### use weight as in ACAT paper, avoid zero weights  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
    } # end of   if (nrow(single_gene_data)==1) # only one variant in one group
    
    if (nrow(single_gene_single_group_data)&gt;1) # case 2: multiple variants in one group
    {
      minor_alele_count=numeric()
      for (j in 1:nrow(single_gene_single_group_data)) # count the total number of alleles for every variant 
        minor_alele_count[j]=single_gene_single_group_data$Transmitted_proband[j]+single_gene_single_group_data$Untransmitted_proband[j]
      
      minor_alele_count_lessthan10=which(minor_alele_count&lt;10) # find variants with MAC&lt;10
      #############
      if (length(minor_alele_count_lessthan10)&lt;=1) # if only one or no variant with MAC&lt;10, no need collapsing 
        for (j in 1:nrow(single_gene_single_group_data))
        {
         binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband[j], single_gene_single_group_data$Untransmitted_proband[j]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
      acat_var_pvalue[j]=binom_test$p.value  
        maf=single_gene_single_group_data$Gnomad_non_neuro_AF[j]
        acat_var_weight[j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
        }
    #############
      if (length(minor_alele_count_lessthan10)&gt;1 &amp; length(minor_alele_count_lessthan10)&lt;nrow(single_gene_single_group_data)) # some, not all variants have MAC&lt;10
      {
        binom_test=binom.test(c(sum(single_gene_single_group_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_single_group_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT ## collapse all variants with MAC&lt;10
      acat_var_pvalue[1]=binom_test$p.value
      maf=mean(single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10])
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      minor_alele_count_morethan10=which(minor_alele_count&gt;=10) # find variants with MAC&gt;=10
      
      if (length(minor_alele_count_morethan10)==1)
      {
      binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband[minor_alele_count_morethan10], single_gene_single_group_data$Untransmitted_proband[minor_alele_count_morethan10]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
      acat_var_pvalue[2]=binom_test$p.value
      maf=single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10]
      acat_var_weight[2]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      }
      if (length(minor_alele_count_morethan10)&gt;1)  
        for (j in 1:length(minor_alele_count_morethan10))
        {
         binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband[minor_alele_count_morethan10[j]], single_gene_single_group_data$Untransmitted_proband[minor_alele_count_morethan10[j]]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
        acat_var_pvalue[1+j]=binom_test$p.value  
        maf=single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10[j]]
        acat_var_weight[1+j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5,1e-300)
        }
      } # end of if (length(minor_alele_count_lessthan10)&gt;1 &amp; length(minor_alele_count_lessthan10)&lt;nrow(single_gene_data)) # some, not all variants have MAC&lt;10
      
       #############
      if (length(minor_alele_count_lessthan10)&gt;1 &amp; length(minor_alele_count_lessthan10)==nrow(single_gene_single_group_data)) #  all variants have MAC&lt;10
      {
        binom_test=binom.test(c(sum(single_gene_single_group_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_single_group_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT ## collapse all variants with MAC&lt;10
      acat_var_pvalue[1]=binom_test$p.value
      maf=mean(single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10]) # use average MAF https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300) # avoid weight to be 0 
        
      }
      
    }  # end of if (nrow(single_gene_data)&gt;1) # case 2: multiple variants in one group
  ##################### 
     if (max(acat_var_pvalue)==1 &amp; length(acat_var_pvalue)&gt;1)
      acat_var_pvalue[acat_var_pvalue==1]=1-1/(length(acat_var_pvalue))
      if (max(acat_var_pvalue)==1 &amp; length(acat_var_pvalue)==1)  # avoid enforcing singular p values to be 0
        acat_var_pvalue[acat_var_pvalue==1]=1-1e-300
     
   acat_group_pvalue[g]=ACAT(acat_var_pvalue, acat_var_weight)  
      
    } # end of for (g in length(num_var_group))
    
    if (max(acat_group_pvalue)==1 &amp; length(acat_group_pvalue)&gt;1)
      acat_group_pvalue[acat_group_pvalue==1]=1-1/(length(acat_group_pvalue))
    if (max(acat_group_pvalue)==1 &amp; length(acat_group_pvalue)==1)
      acat_group_pvalue[acat_group_pvalue==1]=1-1e-300
    
   acat_gene_pvalue[i]=ACAT(acat_group_pvalue) # use ACAT to combine group level p values with equal weights 
    
    
  } # end of if (length(num_var_group)&gt;1) # multiple variant groups
  
  
} # end of   for (i in 1:length(interest_gene))


 # write.csv(data.frame(Gene=interest_gene[1:16344], ACAT_pvalue=acat_gene_pvalue), file=&quot;../output/LauraData/AllGene_6vargroups_ACAT_pvalue_2023_1108.csv&quot;)
 # write.csv(combine_001, file=&quot;../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv&quot;)</code></pre>
<pre class="r"><code>staar_pvalue=as_tibble(read.csv(&quot;../output/LauraData/AllGene_6vargroups_ACAT_pvalue_2023_1108.csv&quot;))
theo &lt;- (ppoints(length(staar_pvalue$ACAT_pvalue)))

  #qqplot(-log(theo, base=10), -log(Burden.pvalue, base=10), xlab=expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)), ylab=expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)), main=&quot;(A) Burden&quot;, frame=F, cex.lab=0.8, pch=20)
  #abline(0,1,col=&#39;red&#39;, lwd=2)

#################  use ggplot draw qq plot 
sx &lt;- sort(-log(theo, base=10)); sy &lt;- sort(-log(staar_pvalue$ACAT_pvalue, base=10))
lenx &lt;- length(sx)
leny &lt;- length(sy)
if (leny &lt; lenx)sx &lt;- approx(1L:lenx, sx, n = leny)$y
if (leny &gt; lenx)sy &lt;- approx(1L:leny, sy, n = lenx)$y

burden=tibble(sx=sx, sy=sy)
 ggplot(burden) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)))+
   ylab(expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)))+
   theme(plot.title = element_text(hjust = 0.5, size=10,face=&quot;bold&quot;))+  #center the title+
   ggtitle(&quot;STAAR&quot;)+
   geom_abline(intercept = 0, slope = 1, color=&quot;red&quot;, 
                 linetype=&quot;solid&quot;, size=1.5)+
theme(text = element_text(size=10))</code></pre>
<p><img src="qqplot_burden_acat_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>for STAAR analysis</p>
<ul>
<li><p>apply AF filter <span
class="math inline">\(AF&lt;5\%\)</span></p></li>
<li><p>within each of 6 variant groups, use ACAT-V to combine single
variant p values of binomial test using weights in the paper <span
class="math inline">\(w_{i, ACAT-V}=Beta(MAF_i, a1=1,
a2=25)*\sqrt{MAF_i(1-MAF_i)}\)</span>. Variants with MAC&lt;10 are
collapsed for burden test, with weight <span class="math inline">\(w_{0,
ACAT-V}\)</span>.</p></li>
<li><p>use ACAT-O to combine 6 group level ACAT p values in the last
step with equal weight as in the paper, as STAAR p value for each
gene.</p></li>
<li><p>all variants have valid AF and those with <code>NA</code> AF are
removed</p></li>
</ul>
</div>
<div id="qq-plot-of-burden" class="section level2">
<h2>QQ plot of burden</h2>
<pre class="r"><code>################# calculate burden for each gene 
colnames(mirage_input_with_gnomad_AF)[4:6]=c(&quot;Transmitted_proband&quot;, &quot;Untransmitted_proband&quot;, &quot;Group_Index&quot;)
interest_gene=unique(mirage_input_with_gnomad_AF$Gene)  # use all genes 
new_data=mirage_input_with_gnomad_AF %&gt;% filter(Gene %in% interest_gene) %&gt;% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Group_Index)

LoF_burden_005=matrix(nrow=length(interest_gene), ncol=4)
Missense_burden_005=matrix(nrow=length(interest_gene), ncol=4)
LoF_Missense_burden_005=matrix(nrow=length(interest_gene), ncol=4)
p_value_005=matrix(nrow=length(interest_gene), ncol=5) # col1:LoF_pvalue; col2:LoF_missense_pvalue; col3: LoF_missense_fishercombined_pvalue; col4: LoF_ACAT_pvalue; col5: missense_ACAT_pvalue 

LoF_burden_001=matrix(nrow=length(interest_gene), ncol=4)
Missense_burden_001=matrix(nrow=length(interest_gene), ncol=4)
LoF_Missense_burden_001=matrix(nrow=length(interest_gene), ncol=4)
p_value_001=matrix(nrow=length(interest_gene), ncol=5) # col1:LoF_pvalue; col2:LoF_missense_pvalue; col3: LoF_missense_fishercombined_pvalue; col4: LoF_ACAT_pvalue; col5: missense_ACAT_pvalue
threshold=1e-100
num.family=14578

for (i in 1:length(interest_gene))
{
  #i=6819
  cat(i, &quot;th gene is running&quot;, &quot;\n&quot;)
  single_gene_data=new_data %&gt;% filter(Gene==interest_gene[i])
  
  ############################ LoF variants 
  LoF_burden_005[i,1]=sum(single_gene_data %&gt;% filter(Group_Index&lt;4) %&gt;% select(Transmitted_proband)%&gt;% pull) # sum over all T LoF variants   
  LoF_burden_005[i,2]=sum(single_gene_data %&gt;% filter(Group_Index&lt;4) %&gt;% select(Untransmitted_proband)%&gt;% pull) # sum over all UT LoF variants  
  burden_test=fisher.test(matrix(c(LoF_burden_005[i,1],num.family, LoF_burden_005[i,2], num.family), nrow=2)) # two sides burden test 
  binom_test=binom.test(c(LoF_burden_005[i,1], LoF_burden_005[i,2]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
  LoF_burden_005[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  LoF_burden_005[i,4]=binom_test$p.value
  p_value_005[i,1]=LoF_burden_005[i,4]
  ########## use ACAT to combine individual variant p values 
  single_gene_data_variant=data.frame(Trans=single_gene_data %&gt;% filter( Group_Index&lt;4) %&gt;% select(Transmitted_proband)%&gt;% pull, UTrans=single_gene_data %&gt;% filter( Group_Index&lt;4) %&gt;% select(Untransmitted_proband)%&gt;% pull) # use all LoF variants 
  single_gene_data_variant_pvalue=numeric()
  if (nrow(single_gene_data_variant)&gt;0)
    { 
     for ( j in 1:nrow(single_gene_data_variant))
    single_gene_data_variant_pvalue[j]=binom.test(c(single_gene_data_variant$Trans[j], single_gene_data_variant$UTrans[j]), p=0.5, alternative = &quot;greater&quot;)$p.value# calculate p value under alternative: T&gt;UT for each single variant 
  if (max(single_gene_data_variant_pvalue)==1 &amp; length(single_gene_data_variant_pvalue)&gt;1)   
    p_value_005[i,4]=1-1/length(single_gene_data_variant_pvalue) # use adjust ACAT p value 
  if (max(single_gene_data_variant_pvalue)==1 &amp; length(single_gene_data_variant_pvalue)==1)
    p_value_005[i,4]=1
  if (max(single_gene_data_variant_pvalue)&lt;1)
    p_value_005[i,4]=ACAT(single_gene_data_variant_pvalue)  # use equal weight to every single variant 
  }
  
  ##################################### missense (MPC&gt;=2)variants 
  Missense_burden_005[i,1]=sum(single_gene_data %&gt;% filter( Group_Index==4) %&gt;% select(Transmitted_proband)%&gt;% pull)  # only MPC&gt;2 
  Missense_burden_005[i,2]=sum(single_gene_data %&gt;% filter( Group_Index==4) %&gt;% select(Untransmitted_proband)%&gt;% pull) # only MPC&gt;2
  burden_test=fisher.test(matrix(c(Missense_burden_005[i,1],num.family, Missense_burden_005[i,2], num.family), nrow=2))
  binom_test=binom.test(c(Missense_burden_005[i,1], Missense_burden_005[i,2]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT; one side burden test 
  Missense_burden_005[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  Missense_burden_005[i,4]=binom_test$p.value
  ############## avoid p value of 0 or 1 
  if (LoF_burden_005[i,4]==0)
     LoF_burden_005[i,4]=threshold
  if (LoF_burden_005[i,4]&gt;0.9999)
     LoF_burden_005[i,4]=1-threshold
  if (Missense_burden_005[i,4]==0)
     Missense_burden_005[i,4]=threshold
  if (Missense_burden_005[i,4]&gt;0.9999)
     Missense_burden_005[i,4]=1-threshold
    p_value_005[i,3]=fisher(c(LoF_burden_005[i,4], Missense_burden_005[i,4]))$p
    
    ########## use ACAT to combine individual variant p values 
  single_gene_data_variant=data.frame(Trans=single_gene_data %&gt;% filter( Group_Index&gt;=4) %&gt;% select(Transmitted_proband)%&gt;% pull, UTrans=single_gene_data %&gt;% filter( Group_Index&gt;=4) %&gt;% select(Untransmitted_proband)%&gt;% pull) # use all missense variants 
  single_gene_data_variant_pvalue=numeric()
  if (nrow(single_gene_data_variant)&gt;0)
    {
     for ( j in 1:nrow(single_gene_data_variant))
    single_gene_data_variant_pvalue[j]=binom.test(c(single_gene_data_variant$Trans[j], single_gene_data_variant$UTrans[j]), p=0.5, alternative = &quot;greater&quot;)$p.value# calculate p value under alternative: T&gt;UT for each single variant 
  if (max(single_gene_data_variant_pvalue)==1 &amp; length(single_gene_data_variant_pvalue)&gt;1)   
    p_value_005[i,5]=1-1/length(single_gene_data_variant_pvalue) # use adjusted ACAT p value 
  if (max(single_gene_data_variant_pvalue)==1 &amp; length(single_gene_data_variant_pvalue)==1)
    p_value_005[i,5]=1
  if (max(single_gene_data_variant_pvalue)&lt;1)
    p_value_005[i,5]=ACAT(single_gene_data_variant_pvalue)  # use equal weight to every single variant 
  }
  ################################## LoF+missense (MPC&gt;=2) variants 
  LoF_Missense_burden_005[i,1]=sum(single_gene_data %&gt;% filter(Group_Index&lt;=4) %&gt;% select(Transmitted_proband)%&gt;% pull)  
  LoF_Missense_burden_005[i,2]=sum(single_gene_data %&gt;% filter(Group_Index&lt;=4) %&gt;% select(Untransmitted_proband)%&gt;% pull)
  burden_test=fisher.test(matrix(c(LoF_Missense_burden_005[i,1],num.family, LoF_Missense_burden_005[i,2], num.family), nrow=2))
  binom_test=binom.test(c(LoF_Missense_burden_005[i,1], LoF_Missense_burden_005[i,2]), p=0.5, alternative = &quot;greater&quot;) # calculate p value under alternative: T&gt;UT
  LoF_Missense_burden_005[i,3]=burden_test$estimate
  #LoF_Missense_burden_005[i,4]=burden_test$p.value
  LoF_Missense_burden_005[i,4]=binom_test$p.value
  p_value_005[i,2]=LoF_Missense_burden_005[i,4]
  
} # end of for (i in 1:length(interest_gene))

################
  rownames(p_value_005)=interest_gene
  colnames(p_value_005)=c(&quot;LoF_pvalue&quot;, &quot;LoF.Missense_pvalue&quot;, &quot;fisher_combine_pvalue&quot;, &quot;LoF_ACAT_pvalue&quot;, &quot;Missense_ACAT_pvalue&quot;)
  
  combine_005=cbind(LoF_burden_005, Missense_burden_005, LoF_Missense_burden_005)
  rownames(combine_005)=interest_gene
  colnames(combine_005)=paste(rep(c(&quot;LoF&quot;, &quot;Missense&quot;, &quot;LoF.Missense&quot;), each=4), c(&quot;T&quot;, &quot;UT&quot;, &quot;OR&quot;, &quot;P_value&quot;), sep=&quot;_&quot;)
  ############### check the variants counts are correct 
  check_gene=&quot;NAV3&quot;
  sum(mirage_input %&gt;% filter(Gene==check_gene &amp; group.index==4) %&gt;% select(No.case) %&gt;% pull) 
  sum(mirage_input %&gt;% filter(Gene==check_gene &amp; group.index==4) %&gt;% select(No.contr) %&gt;% pull)
  combine_005[which(rownames(combine_005)==check_gene),]
  
  
#  write.csv(p_value_005, file=&quot;../output/LauraData/All_inherited_variants_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_2023_1108.csv&quot;)
#  write.csv(combine_005, file=&quot;../output/LauraData/All_inherited_variants_6vargroups_burden_counts_binom_oneside_pvalues_MPC_larger_than2_2023_1108.csv&quot;)</code></pre>
<pre class="r"><code>plotQQ.unif &lt;- function(p.obs) {
  obs &lt;- (p.obs)
  theo &lt;- (ppoints(length(obs)))
  qqplot(-log(theo, base=10), -log(obs, base=10), xlab=expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)), ylab=expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)), main=&quot;A&quot;)
  abline(0,1,col=&#39;red&#39;)
}</code></pre>
<pre class="r"><code>burden_pvalues_001=as_tibble(read.csv(&quot;../output/LauraData/All_inherited_variants_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_2023_1108.csv&quot;))
combine_001=as_tibble(read.csv(&quot;../output/LauraData/All_inherited_variants_6vargroups_burden_counts_binom_oneside_pvalues_MPC_larger_than2_2023_1108.csv&quot;))

#######################   John storey&#39;s q value 
#### install qvalue package 
#install.packages(&quot;devtools&quot;)
#library(&quot;devtools&quot;)
#install_github(&quot;jdstorey/qvalue&quot;)   https://github.com/StoreyLab/qvalue

#library(qvalue)
#qobj.burden=qvalue(Burden.pvalue)
#localFDR.burden &lt;- qobj.burden$lfdr
#min(localFDR.burden, na.rm=T)

#qobj.SKATO=qvalue(SKATO.pvalue)
#localFDR.SKATO &lt;- qobj.SKATO$lfdr
#min(localFDR.SKATO, na.rm=T)
#######################


#pdf(&quot;C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_othermethod_ASD_topconstgene.pdf&quot;)
#par(mfrow=c(1,2))
theo &lt;- (ppoints(length(burden_pvalues_001$LoF_pvalue)))
  #qqplot(-log(theo, base=10), -log(Burden.pvalue, base=10), xlab=expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)), ylab=expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)), main=&quot;(A) Burden&quot;, frame=F, cex.lab=0.8, pch=20)
  #abline(0,1,col=&#39;red&#39;, lwd=2)

#################  use ggplot draw qq plot 
sx &lt;- sort(-log(theo, base=10)); sy &lt;- sort(-log(burden_pvalues_001$LoF_pvalue, base=10))
lenx &lt;- length(sx)
leny &lt;- length(sy)
if (leny &lt; lenx)sx &lt;- approx(1L:lenx, sx, n = leny)$y
if (leny &gt; lenx)sy &lt;- approx(1L:leny, sy, n = lenx)$y

burden=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF=ggplot(burden) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)))+
   ylab(expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)))+
   theme(plot.title = element_text(hjust = 0.5, size=10,face=&quot;bold&quot;))+  #center the title+
   ggtitle(&quot;LoF&quot;)+
   geom_abline(intercept = 0, slope = 1, color=&quot;red&quot;, 
                 linetype=&quot;solid&quot;, size=1.5)+
theme(text = element_text(size=10))
 #qqplot_burden_LoF

#################
theo &lt;- (ppoints(length(burden_pvalues_001$LoF.Missense_pvalue)))

 sx &lt;- sort(-log(theo, base=10)); sy &lt;- sort(-log(burden_pvalues_001$LoF.Missense_pvalue, base=10))
lenx &lt;- length(sx)
leny &lt;- length(sy)
if (leny &lt; lenx)sx &lt;- approx(1L:lenx, sx, n = leny)$y
if (leny &gt; lenx)sy &lt;- approx(1L:leny, sy, n = lenx)$y

 LoF.Missense=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF.Missense=ggplot(LoF.Missense) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)))+
   ylab(expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)))+
   #ylab(&quot;&quot;)+
   theme(plot.title = element_text(hjust = 0.5, size=10,face=&quot;bold&quot;))+  #center the title+
   ggtitle(&quot;LoF+MPC&gt;2&quot;)+
   geom_abline(intercept = 0, slope = 1, color=&quot;red&quot;, 
                 linetype=&quot;solid&quot;, size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_LoF.Missense
 
 
 ################################
 theo &lt;- (ppoints(length(burden_pvalues_001$fisher_combine_pvalue)))

 sx &lt;- sort(-log(theo, base=10)); sy &lt;- sort(-log(burden_pvalues_001$fisher_combine_pvalue, base=10))
lenx &lt;- length(sx)
leny &lt;- length(sy)
if (leny &lt; lenx)sx &lt;- approx(1L:lenx, sx, n = leny)$y
if (leny &gt; lenx)sy &lt;- approx(1L:leny, sy, n = lenx)$y

 fisher.combine=tibble(sx=sx, sy=sy)
 qqplot_burden_fisher.combine=ggplot(fisher.combine) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste(&quot;Theoretical &quot;,-log[10], &quot;(p-values)&quot;)))+
   ylab(expression(paste(&quot;Observed &quot;, -log[10], &quot;(p-values)&quot;)))+
   #ylab(&quot;&quot;)+
   theme(plot.title = element_text(hjust = 0.5, size=10,face=&quot;bold&quot;))+  #center the title+
   ggtitle(&quot;Fisher combine LoF+MPC&gt;2&quot;)+
   geom_abline(intercept = 0, slope = 1, color=&quot;red&quot;, 
                 linetype=&quot;solid&quot;, size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_fisher.combine

#png(&quot;C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_ASD_topconstgene_enrichemnt.png&quot;) 
figure=ggarrange(qqplot_burden_LoF, qqplot_burden_LoF.Missense,  qqplot_burden_fisher.combine, nrow=1)
annotate_figure(figure,
                top = text_grob(&quot;&quot;, color = &quot;red&quot;, face = &quot;bold&quot;, size = 14),
#                bottom = text_grob(&quot;Datasource: \n ToothGrowth data set&quot;, color = &quot;blue&quot;,
#                                   hjust = 1, x = 1, face = &quot;italic&quot;, size = 10),
#                left = text_grob(&quot;Tooth length&quot;, color = &quot;green&quot;, rot = 90),
#                right = &quot;I&#39;m done, thanks :-)!&quot;,
#                fig.lab = &quot;Figure 1&quot;, fig.lab.face = &quot;bold&quot;
)</code></pre>
<p><img src="qqplot_burden_acat_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
