<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>SCZ analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SCZ analysis</h1>
<h4 class="author">Shengtong Han</h4>
<h4 class="date">YYYY-MM-DD</h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Add your analysis here -->
<!-- Add your analysis here -->
<p>The data is from this <a href="http://www.nature.com/nature/journal/v506/n7487/full/nature12975.html?foxtrotcallback=true">paper</a>.</p>
<div id="quality-control-qc" class="section level2">
<h2>Quality Control (QC)</h2>
<div id="annotated-data-quality." class="section level3">
<h3>Annotated data quality.</h3>
<pre class="r"><code>N1=2536; N0=2543
All.Anno.Data[All.Anno.Data ==&quot;.&quot;] &lt;- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF&lt;0.05 &amp; All.Anno.Data$Annotation!=&quot;synonymous SNV&quot;),] # use AF cutoff and exclude synonumous SNV
var.data=data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr)</code></pre>
<ul>
<li>sample size is N1=2536; N0=2543</li>
<li>set ExacAF of NA to be 0</li>
<li>use AF cutoff and exclude synonymous SNV</li>
</ul>
<table>
<caption>Quality of annotated data</caption>
<thead>
<tr class="header">
<th align="left">item</th>
<th align="right">value.AllVar</th>
<th align="right">value.withoutsyn</th>
<th align="right">value.nonsyn</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">#{Variants}</td>
<td align="right">6.383460e+05</td>
<td align="right">4.424320e+05</td>
<td align="right">3.460550e+05</td>
</tr>
<tr class="even">
<td align="left">#{Gene}</td>
<td align="right">1.632100e+04</td>
<td align="right">1.556500e+04</td>
<td align="right">1.550600e+04</td>
</tr>
<tr class="odd">
<td align="left">rate.ca</td>
<td align="right">1.725218e+04</td>
<td align="right">9.402438e+03</td>
<td align="right">6.316679e+03</td>
</tr>
<tr class="even">
<td align="left">rate.co</td>
<td align="right">1.725746e+04</td>
<td align="right">9.406797e+03</td>
<td align="right">6.320157e+03</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Polyphen2.HDIV.score</td>
<td align="right">5.438320e-01</td>
<td align="right">7.833859e-01</td>
<td align="right">9.921313e-01</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_SIFT.score</td>
<td align="right">5.431239e-01</td>
<td align="right">7.774031e-01</td>
<td align="right">9.571455e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_CADD.raw</td>
<td align="right">5.917543e-01</td>
<td align="right">8.204040e-01</td>
<td align="right">9.955556e-01</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_Exon.Brain.Exp</td>
<td align="right">2.171988e-01</td>
<td align="right">1.906259e-01</td>
<td align="right">2.377686e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_Annotation</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
<tr class="even">
<td align="left">ratio_valid_value_Exon.Exac.Cons</td>
<td align="right">3.612790e-01</td>
<td align="right">3.205329e-01</td>
<td align="right">3.980841e-01</td>
</tr>
<tr class="odd">
<td align="left">ratio_valid_value_ExacAF</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
<td align="right">1.000000e+00</td>
</tr>
</tbody>
</table>
<p>The table summaries the ratio of non-missing values in every annotation feature from different types of variants</p>
<ul>
<li>first column: all variants, i.e. everything; second column: all variants excluding synonymous variants; third column: non-synonymous variants only</li>
<li>“Annotation” row means nonsynonymous SNV, intronic, etc.</li>
</ul>
</div>
<div id="variant-rate-at-different-af-cutoffs" class="section level3">
<h3>Variant rate at different AF cutoffs</h3>
<p>“private” variants is defined as the variants whose ExAC AF is NA or 0.</p>
<table>
<caption>Rate of variants in cases and controls at varying AF cutoffs</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">No.rows</th>
<th align="right">No.var.ca</th>
<th align="right">No.var.co</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td align="right">638346</td>
<td align="right">43751522</td>
<td align="right">43885733</td>
<td align="right">17252.1774</td>
<td align="right">17257.4648</td>
</tr>
<tr class="even">
<td>MAF&lt;0.01</td>
<td align="right">577134</td>
<td align="right">2283549</td>
<td align="right">2280280</td>
<td align="right">900.4531</td>
<td align="right">896.6890</td>
</tr>
<tr class="odd">
<td>MAF&lt;0.001</td>
<td align="right">516737</td>
<td align="right">1246878</td>
<td align="right">1252924</td>
<td align="right">491.6711</td>
<td align="right">492.6952</td>
</tr>
<tr class="even">
<td>MAF&lt;1e-04</td>
<td align="right">383965</td>
<td align="right">877760</td>
<td align="right">885911</td>
<td align="right">346.1199</td>
<td align="right">348.3724</td>
</tr>
<tr class="odd">
<td>MAF&lt;1e-05</td>
<td align="right">128161</td>
<td align="right">563170</td>
<td align="right">563910</td>
<td align="right">222.0702</td>
<td align="right">221.7499</td>
</tr>
<tr class="even">
<td>private</td>
<td align="right">23714</td>
<td align="right">460781</td>
<td align="right">461156</td>
<td align="right">181.6960</td>
<td align="right">181.3433</td>
</tr>
</tbody>
</table>
</div>
<div id="burden-of-synonymous-mutations" class="section level3">
<h3>Burden of synonymous mutations</h3>
<p>Test: one side (“greater”) poisson test to see if the rate in cases is signifcantly greater than that in controls.</p>
<table>
<caption>Burden analysis of synonymous mutations at verying AF cutoffs</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td align="right">0.9998818</td>
<td align="right">0.6455623</td>
<td align="right">7849.73935</td>
<td align="right">7850.66732</td>
</tr>
<tr class="even">
<td>MAF&lt;0.01</td>
<td align="right">1.0056395</td>
<td align="right">0.0004496</td>
<td align="right">275.45032</td>
<td align="right">273.90562</td>
</tr>
<tr class="odd">
<td>MAF&lt;0.001</td>
<td align="right">0.9949281</td>
<td align="right">0.9817900</td>
<td align="right">132.84188</td>
<td align="right">133.51907</td>
</tr>
<tr class="even">
<td>MAF&lt;1e-04</td>
<td align="right">0.9894040</td>
<td align="right">0.9998164</td>
<td align="right">87.54377</td>
<td align="right">88.48132</td>
</tr>
<tr class="odd">
<td>MAF&lt;1e-05</td>
<td align="right">0.9939790</td>
<td align="right">0.9410211</td>
<td align="right">52.49014</td>
<td align="right">52.80810</td>
</tr>
<tr class="even">
<td>private</td>
<td align="right">0.9938003</td>
<td align="right">0.9269896</td>
<td align="right">42.77445</td>
<td align="right">43.04129</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="variant-level-analysis" class="section level2">
<h2>Variant level analysis</h2>
<div id="single-variant" class="section level3">
<h3>single variant</h3>
<pre class="r"><code>kable(summy, caption=&quot;Burden analysis of single variant features&quot;)</code></pre>
<table>
<caption>Burden analysis of single variant features</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MAF&lt;1e-2</td>
<td align="right">1.0009027</td>
<td align="right">0.1399487</td>
<td align="right">1130.79180</td>
<td align="right">1129.77192</td>
</tr>
<tr class="even">
<td>MAF&lt;1e-3</td>
<td align="right">0.9990341</td>
<td align="right">0.7431739</td>
<td align="right">358.82926</td>
<td align="right">359.17617</td>
</tr>
<tr class="odd">
<td>MAF&lt;1e-4</td>
<td align="right">0.9949403</td>
<td align="right">0.9981993</td>
<td align="right">258.57610</td>
<td align="right">259.89107</td>
</tr>
<tr class="even">
<td>Prby Damaging</td>
<td align="right">1.0032179</td>
<td align="right">0.0366178</td>
<td align="right">245.48738</td>
<td align="right">244.69996</td>
</tr>
<tr class="odd">
<td>Psbl Damaging</td>
<td align="right">0.9977596</td>
<td align="right">0.8187783</td>
<td align="right">129.34464</td>
<td align="right">129.63508</td>
</tr>
<tr class="even">
<td>SIFT&lt;0.05</td>
<td align="right">1.0004939</td>
<td align="right">0.3948239</td>
<td align="right">231.52050</td>
<td align="right">231.40621</td>
</tr>
<tr class="odd">
<td>CADD top10%</td>
<td align="right">1.0099608</td>
<td align="right">0.0119795</td>
<td align="right">41.14945</td>
<td align="right">40.74361</td>
</tr>
<tr class="even">
<td>BrainExp top10%</td>
<td align="right">1.0002455</td>
<td align="right">0.4847875</td>
<td align="right">22.11435</td>
<td align="right">22.10893</td>
</tr>
<tr class="odd">
<td>Consensus</td>
<td align="right">1.0016397</td>
<td align="right">0.1371545</td>
<td align="right">351.41364</td>
<td align="right">350.83838</td>
</tr>
<tr class="even">
<td>LoF</td>
<td align="right">1.0000218</td>
<td align="right">0.5000694</td>
<td align="right">17.13565</td>
<td align="right">17.13527</td>
</tr>
<tr class="odd">
<td>Private</td>
<td align="right">1.0044795</td>
<td align="right">0.0304736</td>
<td align="right">138.92153</td>
<td align="right">138.30201</td>
</tr>
</tbody>
</table>
<p>Test: use two sample one side poisson test (“greater”)</p>
<ul>
<li>probably damaging:Polyphen2.HDIV.score &gt;=0.957</li>
<li>possibly damaging: &lt;0.957 &amp; &gt;=0.453;</li>
<li>consensus: “Prby Damaging”, or “SIFT&lt;0.05”, or “CADD top10%” (take union)</li>
<li>LoF:stopgain+ frameshift substitution+splicing+stoploss (take union)</li>
</ul>
</div>
<div id="lof-in-different-gene-set" class="section level3">
<h3>LoF in different gene set</h3>
<pre><code>## Warning: `src_sqlite()` is deprecated as of dplyr 1.0.0.
## Please use `tbl()` directly with a database connection
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
</div>
</div>
<div id="exon-level-analysis" class="section level2">
<h2>Exon level analysis</h2>
<p><strong>All subsequence analsyis foucs on variant with AF &lt; 5% and exluding synonymous mutautions. </strong></p>
<p>Test: use two sample one side poisson test</p>
<ul>
<li>high values means more constraint</li>
<li>critical exon=brian expression exon top 25%+exon constraint score top 25%</li>
</ul>
<pre class="r"><code>kable(exon.summ, caption=&quot;Burden analysis of exons&quot;)</code></pre>
<table>
<caption>Burden analysis of exons</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">OR</th>
<th align="right">p.value</th>
<th align="right">rate.ca</th>
<th align="right">rate.co</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Top%90</td>
<td align="right">1.002738</td>
<td align="right">0.2812777</td>
<td align="right">35.65536</td>
<td align="right">35.55800</td>
</tr>
<tr class="even">
<td>Top%80</td>
<td align="right">1.003102</td>
<td align="right">0.1643592</td>
<td align="right">78.69164</td>
<td align="right">78.44829</td>
</tr>
<tr class="odd">
<td>Top%70</td>
<td align="right">1.002110</td>
<td align="right">0.2018336</td>
<td align="right">124.06703</td>
<td align="right">123.80574</td>
</tr>
<tr class="even">
<td>Top%60</td>
<td align="right">1.004247</td>
<td align="right">0.0231259</td>
<td align="right">174.77563</td>
<td align="right">174.03657</td>
</tr>
<tr class="odd">
<td>Top%50</td>
<td align="right">1.002107</td>
<td align="right">0.1310614</td>
<td align="right">224.25986</td>
<td align="right">223.78844</td>
</tr>
<tr class="even">
<td>critical exon</td>
<td align="right">1.001718</td>
<td align="right">0.2360881</td>
<td align="right">138.72003</td>
<td align="right">138.48211</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-level-analysis" class="section level2">
<h2>Gene level analysis</h2>
<p>See <a href="Geneset-SCZ.html">here</a></p>
</div>
<div id="combined-feature-analysis" class="section level2">
<h2>Combined feature analysis</h2>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 18362)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] knitr_1.23    dplyr_1.0.0   RSQLite_2.1.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.1       magrittr_1.5     tidyselect_1.1.0 bit_1.1-14      
##  [5] R6_2.4.1         rlang_0.4.6      highr_0.8        stringr_1.4.0   
##  [9] blob_1.2.1       tools_3.6.0      xfun_0.7         DBI_1.0.0       
## [13] dbplyr_1.4.4     htmltools_0.3.6  ellipsis_0.3.0   assertthat_0.2.1
## [17] yaml_2.2.0       bit64_0.9-7      digest_0.6.19    tibble_3.0.1    
## [21] lifecycle_0.2.0  crayon_1.3.4     purrr_0.3.4      vctrs_0.3.0     
## [25] memoise_1.1.0    glue_1.4.1       evaluate_0.13    rmarkdown_1.13  
## [29] stringi_1.4.3    pillar_1.4.4     compiler_3.6.0   generics_0.0.2  
## [33] pkgconfig_2.0.3</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
