<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>Simulation study</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Simulation study</h1>
<h4 class="author"><em>Shengtong Han</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-04-11 <!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed --> <strong>Code version:</strong> ea8217c</p>
<!-- Add your analysis here -->
<div id="data-generation" class="section level2">
<h2>Data Generation</h2>
<p>Basically, there are two sequential levels when generating the data</p>
<ul>
<li>gene level: <span class="math inline">\(U_i\)</span> denote the risk status of gene <span class="math inline">\(i\)</span>, <span class="math inline">\(U_i \sim Ber(1, \delta)\)</span>. All genes share the same risk probability <span class="math inline">\(\delta\)</span>.</li>
<li>variant level:
<ul>
<li><span class="math inline">\(U_i=0\)</span>, all variants for gene <span class="math inline">\(i\)</span> are non-causal, which is under null hypothesis. AF <span class="math inline">\(q_{ij} \sim Beta(\alpha_0, \beta_0)\)</span>, <span class="math inline">\(X_{1ij}+X_{0ij} \sim Pois (N1+N0, q_{ij})\)</span>, (alternatively, <span class="math inline">\(X_{1ij} \sim Binom(N1, q_{ij}); X_{0ij} \sim Binom(N0, q_{ij})\)</span>); then filter out variants if <span class="math inline">\(X_{1ij}+X_{0ij}=0\)</span>.<br />
</li>
<li><span class="math inline">\(U_i=1\)</span>, gene <span class="math inline">\(i\)</span> is a causal gene, the variant <span class="math inline">\((i,j)\)</span> whose risk status is denoted by <span class="math inline">\(Z_{ij}\)</span> is generated as <span class="math inline">\(Z_{ij} \sim Ber(1, \pi(\beta))\)</span>, <span class="math inline">\(\pi(\beta)\)</span> is a function linking annotations to the probability of being causal.
<ul>
<li><span class="math inline">\(Z_{ij}=0\)</span> (which is under null), given the generated <span class="math inline">\(X_{1ij}+X_{0ij}\)</span>, split into case and controls by conditional binomial distribution, with <span class="math inline">\(p=\frac{N1}{N1+N0}\)</span></li>
<li><span class="math inline">\(Z_{ij}=1\)</span> (which is under alternative), AF <span class="math inline">\(q_{ij} \sim Beta(\alpha, \beta)\)</span>, <span class="math inline">\(\gamma_{ij} \sim Gamma(\bar{\gamma}*\sigma, \sigma)\)</span>, given generated <span class="math inline">\(X_{1ij}+X_{0ij}\)</span>, split into case and control by conditional binomial distribution with <span class="math inline">\(p=\frac{N1*\gamma_{ij}}{N1*\gamma_{ij}+N0}\)</span>.</li>
</ul></li>
</ul></li>
</ul>
<p>Filtering step: filter variants that have null variant counts in both cases and controls.</p>
<p>Notations:</p>
<ul>
<li>N1, N0 are sample sizes in cases and controls</li>
<li><span class="math inline">\(q_{ij}\)</span> is allele frequency for variant <span class="math inline">\((i,j)\)</span></li>
<li><span class="math inline">\(\bar{\gamma}\)</span> relative risk</li>
</ul>
<p><span class="math inline">\(\pi(\beta)\)</span> is the link function between annotations and prior probability and the interest is in estimating <span class="math inline">\(\beta\)</span>.</p>
</div>
<div id="parameter-estimation" class="section level2">
<h2>Parameter estimation</h2>
<div id="case-1-one-annotation-feature" class="section level3">
<h3>Case 1: One annotation feature</h3>
<div id="paramster-settings" class="section level4">
<h4>Paramster settings:</h4>
<p>This is the simplest case in that every variant has one annotation feature if any. Set <span class="math inline">\(\beta=0.5\)</span>, <span class="math inline">\(\bar{\gamma}=10\)</span> to make signal strong with burden close to 10; 80% elements of of Ajk are 1.</p>
</div>
<div id="likelihood-function" class="section level4">
<h4>likelihood function:</h4>
<p>If the feature has one dimension, the objective likelihood function would be simple to be optimized using common R packages. The log likelihood as a function of <span class="math inline">\(\beta\)</span> (true <span class="math inline">\(\beta=0.5\)</span>)is as <img src="figure/loglikelihood.png" alt="loglikelihood" />.</p>
<p>This plot tells us the likelihood only has one mode in one dimension case. It looks like <span class="math inline">\(\beta\)</span> is overestimated by R package-BFGS. With the fixed randomly generated data, run BFGS 50 times with random initials. The mean of <span class="math inline">\(\hat{\beta}_{MLE}=0.7631\)</span> and SD=2.344796e-05.</p>
</div>
<div id="effect-of-sparsity-of-a_jk-k-is-the-number-of-features-here-k1-on-mle-of-beta" class="section level4">
<h4>Effect of Sparsity of <span class="math inline">\(A_{jk}\)</span> (<span class="math inline">\(k\)</span> is the number of features, here <span class="math inline">\(k=1\)</span>) on MLE of <span class="math inline">\(\beta\)</span></h4>
<p>Set <span class="math inline">\(\beta_{true}=0.5\)</span>, vary the sparsity of <span class="math inline">\(A_{jk}\)</span>, i.e., the ratio of 0’s in <span class="math inline">\(A_{jk}\)</span>. Define sparsity rate as the ratio of 0’s in each feature category.</p>
<table style="width:100%;">
<colgroup>
<col width="15%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th>Sparsity rate</th>
<th>90%</th>
<th>80%</th>
<th>70%</th>
<th>60%</th>
<th>50%</th>
<th>40%</th>
<th>30%</th>
<th>20%</th>
<th>10%</th>
<th>0</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\hat{\beta}_{MLE}\)</span></td>
<td>0.3802</td>
<td>0.4499</td>
<td>0.5845</td>
<td>0.5420</td>
<td>0.5829</td>
<td>0.64444</td>
<td>0.6817</td>
<td>0.7608</td>
<td>0.7941</td>
<td>0.8896</td>
</tr>
</tbody>
</table>
<p>When All entries of <span class="math inline">\(A_{jk}\)</span> are zero, <span class="math inline">\(\tau_j=\frac{1}{2}\)</span>, regardless of whatever values of <span class="math inline">\(\beta\)</span> and the likelihood function is a constant of <span class="math inline">\(\beta\)</span>. Thus <span class="math inline">\(\widehat{\beta}\)</span> can be anywhere.</p>
<p>Set <span class="math inline">\(\beta_{true}=0.2\)</span>.</p>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="8%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th>Sparsity rate</th>
<th>95%</th>
<th>90%</th>
<th>85%</th>
<th>80%</th>
<th>75%</th>
<th>70%</th>
<th>65%</th>
<th>60%</th>
<th>55%</th>
<th>50%</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\hat{\beta}_{MLE}\)</span></td>
<td>0.3075</td>
<td>0.2259</td>
<td>0.3481</td>
<td>0.3944</td>
<td>0.3677</td>
<td>0.4167</td>
<td>0.3956</td>
<td>0.4331</td>
<td>0.4565</td>
<td>0.4091</td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="important-note" class="section level4">
<h4>Important Note:</h4>
<p>There are two variables affecting estimate of <span class="math inline">\(\beta\)</span>, sparsity of <span class="math inline">\(A_{jk}\)</span> and <span class="math inline">\(\beta_{true}\)</span>. (1) <span class="math inline">\(\beta\)</span> cannot be set too large, say &gt;10 because it controls the probability of a variant of being causal. Consider an extreme case where all variants are truly causal risk variants. The likelihood of the data is close to the product of Bayes factor of every variant, free of <span class="math inline">\(\beta\)</span> value. Any large enough <span class="math inline">\(\beta\)</span> will give the same likelihood. (2) Sparsity of <span class="math inline">\(A_{jk}\)</span> will certainly influences the estimate of <span class="math inline">\(\beta\)</span>. The less sparsity, the more information will be used, the more accurate the estimate will be.</p>
</div>
<div id="introducing-ell_2-penalty." class="section level4">
<h4>Introducing <span class="math inline">\(\ell_2\)</span> penalty.</h4>
<p>The same parameter setting. Sparsity rate is 20%. <span class="math inline">\(\beta_{true}=0.5\)</span>. <img src="figure/PenaltyBeta.png" alt="betaandpenalty" /></p>
<p>With this setting, very large tunning parameter <span class="math inline">\(\lambda&gt;1000\)</span> should be used to get the good estimate of <span class="math inline">\(\beta\)</span>.</p>
<pre class="r"><code>Ajk=rep(0, (num.gene*m))
Ajk[sample(length(Ajk), (0.8*length(Ajk)))]=1
############## generate beta 
beta.true=numeric(anno.num)
beta.true[1]=0.5
############# compute tau: risk of every variant being risk variant
tau.true=exp(Ajk*beta.true)/(1+exp(Ajk*beta.true))
Ui=rbinom(num.gene, 1, delta)
  all.data=list(); var.orig.index=numeric(); var.orig.index[1]=NA
  for (i in 1:num.gene)
  {
    data=gene.simu(N0, N1, m, alpha0, beta0, alpha, beta, gamma.mean, sigma, pi=tau.true[((i-1)*m+1):(i*m)], Ui[i], num.group, split.ratio)
    all.data[[i]]=data
    var.orig.index=c(var.orig.index, (i-1)*m+data$var.index)
  }
  var.orig.index=var.orig.index[-1]
  Ajk.effect=Ajk[var.orig.index]
  ################### calculate bayes factor for every variant 
  k=0; BF.var=numeric(); var.count=matrix(nrow=(m*num.gene), ncol=2)
  for (i in 1:length(all.data))
  {
    for (j in 1:ncol(all.data[[i]]$geno))
    {
      k=k+1
      BF.var[k]=BF.var.inte(sum(all.data[[i]]$geno[all.data[[i]]$pheno==1,j]), sum(all.data[[i]]$geno[all.data[[i]]$pheno==0,j]), bar.gamma=6, sig=sigma, N1, N0)
      var.count[k,]=c(sum(all.data[[i]]$geno[all.data[[i]]$pheno==1,j]),sum(all.data[[i]]$geno[all.data[[i]]$pheno==0,j]) )
    }
  } # end of i
 var.count=var.count[-((k+1):nrow(var.count)),]
 ###############################
 ################################
  num.var=length(BF.var)
  lambda.range=rep(0, 50)
  error=numeric()
  beta.fin.est=numeric(length(lambda.range))
###################################
  lambda=0
  beta.range=seq(0, 100, by=10)
  log.lkhd=numeric(length(beta.range))
  for (i in 1:length(beta.range))
  log.lkhd[i]=objec.func(beta.range[i])
  plot(beta.range, log.lkhd, type=&quot;o&quot;, xlab=expression(beta), ylab=&quot;log(likelihood)&quot;)
  abline(v=beta.true, col=2)
###################################  
  
  
  for (k in 1:length(lambda.range))
  {
    cat(k, &quot;is running&quot;, &quot;\n&quot;)  
    lambda=lambda.range[k] 
    beta.est=runif(anno.num, -1,1)
    theta.est=optim(beta.est, objec.func, deriv.objec.func, method=&quot;BFGS&quot;, control=list(fnscale=-1))
    beta.fin.est[k]=theta.est$par
#    theta.est=DEoptim(fn=objec.func.min, lower=rep(-1, anno.num), upper=rep(200, anno.num), control=list(NP=100, itermax=100,trace=FALSE))
#    beta.fin.est[k,]=theta.est$optim$bestmem
    error[k]=sum((beta.fin.est[k]-beta.true)^2)
    
  }</code></pre>
</div>
</div>
<div id="case-2-two-annotation-features" class="section level3">
<h3>Case 2: Two Annotation features</h3>
<p><span class="math inline">\(K=2\)</span>. Set <span class="math inline">\(\beta_1=0.05, \beta_2=0.7\)</span>. Other parameters are the same as in Case 1.</p>
<div id="mle-of-beta." class="section level4">
<h4>MLE of <span class="math inline">\(\beta\)</span>.</h4>
<p>use BFGS method.</p>
</div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.2 (2015-08-14)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] backports_1.0.5 magrittr_1.5    rprojroot_1.2   tools_3.2.2    
 [5] htmltools_0.3.5 yaml_2.1.14     Rcpp_0.12.1     stringi_0.5-5  
 [9] rmarkdown_1.3   knitr_1.15.1    git2r_0.18.0    stringr_1.2.0  
[13] digest_0.6.8    evaluate_0.10  </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
