<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>Simulation study</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rare-var-project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/rare-var-project/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Simulation study</h1>
<h4 class="author"><em>Shengtong Han</em></h4>

</div>


<!-- Add your analysis here -->
<div id="data-generation" class="section level2">
<h2>Data Generation</h2>
<p>Basically, there are two sequential levels when generating the data. First under null hypothesis, generate total number of variants in cases and controls, i.e., AF <span class="math inline">\(q_{ij}^{(0)} \sim Beta(\alpha_0, \beta_0)\)</span>, <span class="math inline">\(X_{1ij}+X_{0ij} \sim Pois (N1+N0, q_{ij}^{(0)})\)</span>, filter out variants will null mutations, then re-distribute all mutations in effective variants into cases and controls depending on the risk status via conditional binomial distribution.</p>
<ul>
<li>gene level: <span class="math inline">\(U_i\)</span> denote the risk status of gene <span class="math inline">\(i\)</span>, <span class="math inline">\(U_i \sim Ber(1, \delta)\)</span>. All genes share the same risk probability <span class="math inline">\(\delta\)</span>.</li>
<li>variant level:
<ul>
<li><span class="math inline">\(U_i=0\)</span>, all variants for gene <span class="math inline">\(i\)</span> are non-causal, which is under null hypothesis. Given the generated <span class="math inline">\(X_{1ij}+X_{0ij}\)</span>, split into case and controls by conditional binomial distribution, with <span class="math inline">\(p=\frac{N1}{N1+N0}\)</span><br />
</li>
<li><span class="math inline">\(U_i=1\)</span>, gene <span class="math inline">\(i\)</span> is a causal gene, the variant <span class="math inline">\((i,j)\)</span> whose risk status is denoted by <span class="math inline">\(Z_{ij}\)</span> is generated as <span class="math inline">\(Z_{ij} \sim Ber(1, \pi(\eta))\)</span>, <span class="math inline">\(\pi(\eta)\)</span> is a function linking annotations to the probability of being causal.
<ul>
<li><span class="math inline">\(Z_{ij}=0\)</span> (which is under null), given the generated <span class="math inline">\(X_{1ij}+X_{0ij}\)</span>, split into case and controls by conditional binomial distribution, with <span class="math inline">\(p=\frac{N1}{N1+N0}\)</span></li>
<li><span class="math inline">\(Z_{ij}=1\)</span> (which is under alternative), AF <span class="math inline">\(q_{ij}^{(1)} \sim Beta(\alpha, \beta)\)</span>, <span class="math inline">\(\gamma_{ij} \sim Gamma(\bar{\gamma}*\sigma, \sigma)\)</span>, <span class="math inline">\(X_{1ij}+X_{0ij} \sim Pois(N1+N0, q_{ij}^{(1)})\)</span>, split into case and control by conditional binomial distribution with <span class="math inline">\(p=\frac{N1*\gamma_{ij}}{N1*\gamma_{ij}+N0}\)</span>.</li>
</ul></li>
</ul></li>
</ul>
<p>Filtering step: filter variants that have null variant counts in both cases and controls.</p>
<p>Notations:</p>
<ul>
<li>N1, N0 are sample sizes in cases and controls</li>
<li><span class="math inline">\(q_{ij}\)</span> is allele frequency for variant <span class="math inline">\((i,j)\)</span></li>
<li><span class="math inline">\(\bar{\gamma}\)</span> relative risk</li>
</ul>
<p><span class="math inline">\(\pi(\eta)\)</span> is the link function between annotations and prior probability and the interest is in estimating <span class="math inline">\(\eta\)</span>.</p>
</div>
<div id="disjoint-group-priors" class="section level2">
<h2>Disjoint group priors</h2>
<div id="all-genes-are-risk-genes-fix-delta1-only-estimate-eta" class="section level3">
<h3>All genes are risk genes, fix <span class="math inline">\(\delta=1\)</span>, only estimate <span class="math inline">\(\eta\)</span></h3>
<ul>
<li>parameter estimate: (I) relative risk effect, <span class="math inline">\(\bar{\gamma}\)</span> on estimating <span class="math inline">\(\eta\)</span></li>
</ul>
<p>Since variant groups are assumed to be independent, it suffices to demonstrate the parameter estimate only for one risk variant group.</p>
<p>Parameter settings: N1=N0=3000, <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>. 100 genes, each having 1000 variants with <span class="math inline">\(\eta_i=0.1\)</span> risk variants, but the actural <span class="math inline">\(\eta_i\)</span> is varying for each generated data set because of the filtering. <span class="math inline">\(\bar{\gamma}=2,3,4,5\)</span> and with each <span class="math inline">\(\bar{\gamma}\)</span>, perform 100 independent runs.</p>
<ul>
<li>Parameter estimate (II): sample size effect on estimating <span class="math inline">\(\eta\)</span></li>
</ul>
<p>Parameter settings: N1=N0=1000, 3000, 5000, <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>. 100 genes, each having 1000 variants with <span class="math inline">\(\eta_i=0.1\)</span> risk variants, but the actural <span class="math inline">\(\eta_i\)</span> is varying for each generated data set because of the filtering. <span class="math inline">\(\bar{\gamma}=5\)</span>. For each fixed sample size, perform 100 independent runs.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<ul>
<li>power comparison with other methods</li>
</ul>
<p>For every simulated risk gene, three different methods are used to test the power of detecting risk genes. Significance level of 0.05 is used as cutoff. Burnden test looks at the variants in cases and controls and performs the enrichment analysis for every gene. SKATO tests the significance of effect size and generate p value without covariates. MIRAGE-LRT first estimates the parameter <span class="math inline">\(\eta\)</span>, then does a likelihood ratio test and returns p value by <span class="math inline">\(\chi^2_k\)</span> (<span class="math inline">\(k\)</span> is the number of unknown parameters <span class="math inline">\(\eta\)</span>).</p>
<p><strong>Case 1: every risk gene only has one risk variant group with the proportion of <span class="math inline">\(\eta\)</span> and fixed <span class="math inline">\(\bar{\gamma}\)</span></strong></p>
<p>Parameter settings: N1=N0=3000 <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>. 100 genes, each having 1000 variants with <span class="math inline">\(\eta=0.1, 0.2, 0.3\)</span> risk variants, but the actural <span class="math inline">\(\eta\)</span> is varying for each generated data set because of the filtering. <span class="math inline">\(\bar{\gamma}=5\)</span>. For every <span class="math inline">\(\eta\)</span>, perform 100 independent runs.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p><strong>Case 2: every risk gene has multiple risk variant groups with varying proportion of <span class="math inline">\(\eta\)</span> and fixed <span class="math inline">\(\bar{\gamma}\)</span></strong></p>
<p>Parameter settings: N1=N0=3000 <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>. 100 genes, each having 1000 variants in three different groups, the first group has 30% variant, the second has 40% and the rest 30% are in the third group. In each group of every gene, the proportion of risk genes <span class="math inline">\(\eta\)</span> is varying at 0.01, 0.05, 0.1 respectively. <span class="math inline">\(\bar{\gamma}=2, 3,4, 5\)</span>. For every <span class="math inline">\(\bar{\gamma}\)</span>, perform 100 independent runs.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p><strong>Case 3: every risk gene has multiple risk variant groups with varying proportion of <span class="math inline">\(\eta\)</span> and varying <span class="math inline">\(\bar{\gamma}\)</span></strong></p>
<p>Parameter settings: N1=N0=3000 <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>. 100 genes, each having 1000 variants in three different groups, the first group has 30% variants, the second has 40% and the rest 30% are in the third group. In each variant group of every gene, the proportion of risk variants <span class="math inline">\(\eta\)</span> is varying at 0.01, 0.05, 0.1 and every risk category has its own <span class="math inline">\(\bar{\gamma}=5, 3,1\)</span> respectively. The basic idea is that the rarer the varinat is, the more effect size it tends to have. Perform 100 independent runs.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="mixture-of-risk-genes-and-non-risk-genes-0delta1-estimate-both-eta-delta" class="section level3">
<h3>Mixture of risk genes and non-risk genes <span class="math inline">\(0&lt;\delta&lt;1\)</span>, estimate both <span class="math inline">\(\eta, \delta\)</span></h3>
<p>Since the simulated data is a mixture of risk genes and non-risk genes, drawing ROC curve is one way to test the power of discriminating risk genes of different methods. For MIRAGE, Bayes factor of every gene was calculated and used to plot ROC, called MIRAGE-BF.</p>
<div id="power-comparison" class="section level4">
<h4>Power comparison</h4>
<p><strong>case 1: one risk gene has only one risk variant category with fixed <span class="math inline">\(\bar{\gamma}\)</span></strong></p>
<p>Parameter settings: N1=N0=3000 <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>. 100 genes, each having 1000 variants with <span class="math inline">\(\eta=0.03, 0.07, 0.1, 0.2\)</span> risk variants, but the actural <span class="math inline">\(\eta\)</span> is varying for each generated data set because of the filtering. <span class="math inline">\(\delta=0.5\)</span>, i.e. half genes are risk genes and the rest half are nonrisk genes. <span class="math inline">\(\bar{\gamma}=5\)</span>.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p><strong>case2: one risk gene has multiple risk variant categories with group specific <span class="math inline">\(\bar{\gamma}\)</span></strong></p>
<p>Parameter settings: N1=N0=3000 <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>, <span class="math inline">\(\delta=0.5\)</span>. 100 genes, each having 1000 variants in three different groups, the first group has 30% variants, the second has 40% and the rest 30% are in the third group. In each variant group of every gene, the proportion of risk variants <span class="math inline">\(\eta\)</span> is varying at 0.01, 0.05, 0.1 and every risk category has its own <span class="math inline">\(\bar{\gamma}=5, 3,1\)</span> respectively. The basic idea is that the rarer the varinat is, the more effect size it tends to have.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Parameter settings: N1=N0=3000 <span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=2000, \beta_0=1000\)</span>, <span class="math inline">\(\delta=0.1, 0.2, 0.3, 0.4\)</span>. 1000 genes, each having 100 variants in three different groups, the first group has 60% variants, the second has 30% and the rest 10% are in the third group. In each variant group of every gene, the proportion of risk variants <span class="math inline">\(\eta\)</span> is varying at 0.05, 0.2, 0.5 and every risk category has its own <span class="math inline">\(\bar{\gamma}\)</span> respectively. The basic idea is that the rarer the varinat is, the more effect size it tends to have.</p>
<p><strong>burden test: for every gene, use all variants in cases and control</strong></p>
<p><span class="math inline">\(\bar{\gamma}=1, 3,5\)</span></p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p><strong>burden-adj: for every gene, compute p values for 3 variant categories, adjust p values by bonferroni correction and take the minimum</strong></p>
<p><span class="math inline">\(\bar{\gamma}=3,3,5\)</span>.</p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p><strong>Parameter estimate</strong></p>
<p><span class="math inline">\(\beta_1=0.05,\beta_2=0.2, \beta_3=0.5, \delta=0.1\)</span></p>
<p><img src="simulation-study_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="logistic-priors" class="section level2">
<h2>Logistic priors</h2>
<div id="case-1-one-annotation-feature" class="section level3">
<h3>Case 1: One annotation feature</h3>
<div id="paramster-settings-1" class="section level4">
<h4>Paramster settings 1 :</h4>
<ul>
<li><span class="math inline">\(\delta=1\)</span></li>
<li><span class="math inline">\(N1=N0=5000\)</span></li>
<li>number of genes <span class="math inline">\(I=1000\)</span>, number of variants per gene <span class="math inline">\(m=200\)</span> before filtering.</li>
<li><span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta_0=1000, \beta=2000\)</span>.</li>
<li><span class="math inline">\(\beta_{true}=0.5\)</span>, <span class="math inline">\(\bar{\gamma}=10\)</span></li>
<li>80% elements of of Ajk are 1</li>
</ul>
<p>This is the simplest case in that every variant has one annotation feature if any.</p>
</div>
<div id="likelihood-function" class="section level4">
<h4>likelihood function:</h4>
<p>If the feature has one dimension, the objective likelihood function would be simple to be optimized using common R packages. The log likelihood as a function of <span class="math inline">\(\beta\)</span> (true <span class="math inline">\(\beta=0.5\)</span>)is as</p>
<p><img src="figure/loglikelihood.png" alt="loglikelihood" />.</p>
<p>This plot tells us the likelihood only has one mode in one dimension case. It looks like <span class="math inline">\(\beta\)</span> is overestimated by R package-BFGS. With the fixed randomly generated data, run BFGS 50 times with random initials. The mean of <span class="math inline">\(\hat{\beta}_{MLE}=0.7631\)</span> and SD=2.344796e-05.</p>
</div>
<div id="effect-of-sparsity-of-a_jk-k-is-the-number-of-features-here-k1-on-mle-of-beta" class="section level4">
<h4>Effect of Sparsity of <span class="math inline">\(A_{jk}\)</span> (<span class="math inline">\(k\)</span> is the number of features, here <span class="math inline">\(k=1\)</span>) on MLE of <span class="math inline">\(\beta\)</span></h4>
<p>Set <span class="math inline">\(\beta_{true}=0.5\)</span>, vary the sparsity of <span class="math inline">\(A_{jk}\)</span>, i.e., the ratio of 0’s in <span class="math inline">\(A_{jk}\)</span>. Define sparsity rate as the ratio of 0’s in each feature category.</p>
<table style="width:100%;">
<caption>MLE at different sparsity rate <span class="math inline">\(\beta_{true}=0.5\)</span></caption>
<colgroup>
<col width="15%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th>Sparsity rate</th>
<th>90%</th>
<th>80%</th>
<th>70%</th>
<th>60%</th>
<th>50%</th>
<th>40%</th>
<th>30%</th>
<th>20%</th>
<th>10%</th>
<th>0</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\hat{\beta}_{MLE}\)</span></td>
<td>0.3802</td>
<td>0.4499</td>
<td>0.5845</td>
<td>0.5420</td>
<td>0.5829</td>
<td>0.64444</td>
<td>0.6817</td>
<td>0.7608</td>
<td>0.7941</td>
<td>0.8896</td>
</tr>
</tbody>
</table>
<p>When All entries of <span class="math inline">\(A_{jk}\)</span> are zero, <span class="math inline">\(\tau_j=\frac{1}{2}\)</span>, regardless of whatever values of <span class="math inline">\(\beta\)</span> and the likelihood function is a constant of <span class="math inline">\(\beta\)</span>. Thus <span class="math inline">\(\widehat{\beta}\)</span> can be anywhere.</p>
<p>Set <span class="math inline">\(\beta_{true}=0.2\)</span>.</p>
<table style="width:100%;">
<caption>MLE at different sparsity rate with <span class="math inline">\(\beta_{true}=0.2\)</span>.</caption>
<colgroup>
<col width="14%" />
<col width="8%" />
<col width="8%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th>Sparsity rate</th>
<th>95%</th>
<th>90%</th>
<th>85%</th>
<th>80%</th>
<th>75%</th>
<th>70%</th>
<th>65%</th>
<th>60%</th>
<th>55%</th>
<th>50%</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\hat{\beta}_{MLE}\)</span></td>
<td>0.3075</td>
<td>0.2259</td>
<td>0.3481</td>
<td>0.3944</td>
<td>0.3677</td>
<td>0.4167</td>
<td>0.3956</td>
<td>0.4331</td>
<td>0.4565</td>
<td>0.4091</td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="important-note" class="section level4">
<h4>Important Note:</h4>
<p>There are two variables affecting estimate of <span class="math inline">\(\beta\)</span>, sparsity of <span class="math inline">\(A_{jk}\)</span> and <span class="math inline">\(\beta_{true}\)</span>. (1) <span class="math inline">\(\beta\)</span> cannot be set too large, say &gt;10 because it controls the probability of a variant of being causal. Consider an extreme case where all variants are truly causal risk variants. The likelihood of the data is close to the product of Bayes factor of every variant, free of <span class="math inline">\(\beta\)</span> value. Any large enough <span class="math inline">\(\beta\)</span> will give the same likelihood. (2) Sparsity of <span class="math inline">\(A_{jk}\)</span> will certainly influences the estimate of <span class="math inline">\(\beta\)</span>. The less sparsity, the more information will be used, the more accurate the estimate will be. (3) AF cannot be too large because for common variants with large odds ratio (large <span class="math inline">\(\bar{\gamma}\)</span>), Bayes factor will become overflow of being Inf.</p>
</div>
<div id="parameter-setting-2-increasing-sample-size" class="section level4">
<h4>Parameter setting 2-increasing sample size</h4>
<ul>
<li><span class="math inline">\(\delta=1\)</span></li>
<li><span class="math inline">\(\beta_{true}=2\)</span>, <span class="math inline">\(\bar{\gamma}=10\)</span></li>
<li>all entries of <span class="math inline">\(A_{jk}\)</span> are 1</li>
<li><span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=1000, \beta_0=2000\)</span></li>
</ul>
<table>
<caption>MLE averaging across 10 replicates with different sample size</caption>
<thead>
<tr class="header">
<th>N1=N0</th>
<th>10,000</th>
<th>50,000</th>
<th>100,000</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\widehat{\beta}_{MLE}\)</span> by Optim</td>
<td>2.7810 (0.0362)</td>
<td>2.4413 (0.0161)</td>
<td>2.3420</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\widehat{\beta}_{MLE}\)</span> by GLM</td>
<td>2.0016 (0.0126)</td>
<td></td>
</tr>
</tbody>
</table>
<p>To use GLM, <span class="math inline">\(Z_{ij}\)</span> is treated as response (but in practice, <span class="math inline">\(Z_{ij}\)</span> is unknown) and each column of <span class="math inline">\(A_{jk}\)</span> is a predictor, but without intercept.</p>
<p>Use function GLM to estimate <span class="math inline">\(\beta\)</span>.</p>
</div>
</div>
<div id="case-2-two-annotation-features" class="section level3">
<h3>Case 2: Two Annotation features</h3>
<p><span class="math inline">\(K=2\)</span>. Set <span class="math inline">\(\beta_1=0.05, \beta_2=2\)</span>. Other parameters are</p>
<ul>
<li><span class="math inline">\(\delta=1\)</span></li>
<li><span class="math inline">\(\bar{\gamma}=10\)</span></li>
<li><span class="math inline">\(\alpha=\alpha_0=0.1\)</span>, <span class="math inline">\(\beta=1000, \beta_0=2000\)</span></li>
</ul>
<p>When there are more than 2 groups, <span class="math inline">\(A_{ij}\)</span> must be designed carefully such that all columns (all annotations) have as little overlap as possible. Otherwise, the method have difficulty in distinguishing between them, leading to poor estimate. In simulations below, these two factures have no overlap.</p>
<table style="width:94%;">
<caption>MLE averaging across 10 replicates with different sample size</caption>
<colgroup>
<col width="12%" />
<col width="11%" />
<col width="12%" />
<col width="58%" />
</colgroup>
<thead>
<tr class="header">
<th>N1=N0</th>
<th>10,000</th>
<th>50,000</th>
<th>80,000</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\widehat{\beta_1}_{MLE}\)</span> by Optim</td>
<td>0.3472 (0.0239)</td>
<td>0.2287 (0.0252)</td>
<td>0.2371 (0.0193)</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\widehat{\beta_2}_{MLE}\)</span> by Optim</td>
<td>2.7807 (0.0919)</td>
<td>2.4239 (0.0211)</td>
<td>2.3636 (0.0243)</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\widehat{\beta_1}_{MLE}\)</span> by GLM</td>
<td>0.0517 (0.0159)</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><span class="math inline">\(\widehat{\beta_2}_{MLE}\)</span> by GLM</td>
<td>1.9952 (0.0177)</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="case-3-many-annotation-features" class="section level3">
<h3>Case 3: Many annotation features</h3>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
