---
title: "ASC new sample gene set analysis"
output: html_doc ment
---


```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
library(DT)
#library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
```

```{r, echo=F, message=F, warning=F}
########## use fixed parameter estimate to calculate BF, this is very useful because it doesn't need to re-run analysis to BF of every gene which takes a long time and large space to store that information 
intergrand=function(aa, var.case, var.contr, bar.gamma, sig, N1, N0)
{
  ff=dbinom(var.case, sum(var.case, var.contr), aa*N1/(aa*N1+N0))*dgamma(aa, bar.gamma*sig, sig)
  return(ff)
}
# calculate the bayes factor of a single variant via integration
BF.var.inte=function(var.case, var.contr, bar.gamma, sig, N1, N0)
{
  marglik0.CC <- dbinom(var.case, sum(var.case, var.contr), N1/(N1+N0))    # Under H0: gamma=1

  marglik1.CC <- integrate(intergrand, var.case, var.contr, bar.gamma, sig, N1, N0, low=0, upper=100, stop.on.error=F)$value # Under H1: gamma~gamma(gamma.mean*sigma, sigma)
  BF.var <- marglik1.CC/marglik0.CC

  return(BF.var)
}

fixed.beta.func=function(new.data, N1, N0, gamma.mean, sigma, beta) # new.data has one column specifying its group index
{
  #######################
  beta.k=beta
  full.info.genevar=list()
  gene.list=new.data$Gene; unique.gene=unique(gene.list) # find the gene list
  num.gene=length(unique.gene)
  BF.gene=numeric()
  ########################
  for (i in 1:num.gene)
  {
#    cat(i, "th gene of ", "\t", num.gene, "\t", "is running", "\n")
    var.index.list=which(gene.list==unique.gene[i])
    indi.gene=new.data[var.index.list,]   # note var.index.list matches new.data
    bb=1; var.BF=numeric()
    if (length(var.index.list)>0) # calculate Bayes factor for variant (i,j)
      for (j in 1:length(var.index.list))
      {
        if (new.data$group.index[var.index.list[j]]<=3)   # note here bar.gamma is group specific, attention what bar.gamma goes to which group 
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=6, sig=sigma, N1, N0)
        if (new.data$group.index[var.index.list[j]]>3)
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=3, sig=sigma, N1, N0)
        bb=bb*((1-beta.k[new.data$group.index[var.index.list[j]]])+beta.k[new.data$group.index[var.index.list[j]]]*var.BF[j])
      }
    full.info.genevar[[i]]=cbind(indi.gene, var.BF)
    BF.gene[i]=bb
  }
  
  return(result=list(BayesFactor=data.frame(Gene=unique.gene, BF=BF.gene), full.info=full.info.genevar))
}
```



```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
path="C:\\Shengtong\\research\\rare-var\\"
ASC_new_sample3=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), sep='\t',  header=T, fill=T)) # this is the correct way that Kyle read the data
ASC_new_sample3=ASC_new_sample3 %>% filter(isSyn=="FALSE") # filter syn variants first 
```


```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
path="C:\\Shengtong\\Research\\rare-var\\"
GeneDB=src_sqlite(path=paste(path, "gene.list.db", sep=""), create=F)
#GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
###### ID gene ######
ID_Gene=as.character(data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))$GeneID)
#### high confidence gene 
High_Confidence_Gene=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
#### moderate confidence gene 
Moderate_Confidence_Gene=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
#### psd gene 
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
PSD_Gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
#### FMRP gene 
FMRP_Gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
#### Autism KB gene 
ASD_KB_Gene=as.character(data.frame(collect(tbl(GeneDB, "AutismKB_gene")))$GeneID)
#### constraint gene 
Constraint_Gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
### RVIS gene 
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS_Gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene
#### Haplo insufficient gene 
Haploinsufficient_Gene=as.character(data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))$GeneID)
#### SCZ data 
SCZ_Gene=purcell.genelist$Gene_symbol
Gene_Set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff", "SCZ gene")
Gene_Set_List=list()
Gene_Set_List[[1]]=ID_Gene
Gene_Set_List[[2]]=High_Confidence_Gene
Gene_Set_List[[3]]=Moderate_Confidence_Gene
Gene_Set_List[[4]]=PSD_Gene
Gene_Set_List[[5]]=FMRP_Gene
Gene_Set_List[[6]]=ASD_KB_Gene
Gene_Set_List[[7]]=Constraint_Gene
Gene_Set_List[[8]]=RVIS_Gene
Gene_Set_List[[9]]=Haploinsufficient_Gene
Gene_Set_List[[10]]=SCZ_Gene
```





```{r, echo=F, warning=F, message=F}

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
######### use ASC_new_sample3 rather than ASC_new_sample2
Transmitted_proband=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) # extract transmitted variants for proband 

Untransmitted_proband=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) # extract untransmitted variants for proband 

Transmitted_sibling=ASC_new_sample3%>% select(starts_with("t_sibling"))%>%  mutate(Transmitted_sibling = select(., t_sibling_male_dad:t_sibling_female_ind) %>% rowSums())%>% select(Transmitted_sibling) # extract transmitted variants for sibling 

Untransmitted_sibling=ASC_new_sample3%>% select(starts_with("u_sibling"))%>%  mutate(Untransmitted_sibling = select(., u_sibling_male_dad:u_sibling_female_ind) %>% rowSums())%>% select(Untransmitted_sibling) # extract untransmitted variants for sibling

##### add new columns of total transmitted and non-transmitted variants 
#ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & VQSLOD<9) # exclude indel & VQSLOD filter 

################################################


ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8)%>%filter(!str_detect(Consequence, 'intron_variant'))# exclude indel & apply other filters for SNP's 

##################
#AF_cutoff=0.001
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Transmitted_proband) %>% pull)   
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Untransmitted_proband) %>% pull)   
########### 1164 vs 1413 not 695 vs 557 in cell paper 


# add one column of variant group index 
ASC_new_sample_GI=ASC_new_sample %>% add_column(Group_Index=NA)

ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.995 & ASC_new_sample_GI$isPTV==T)]=1
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.5 & ASC_new_sample_GI$pLI<0.995 & ASC_new_sample_GI$isPTV==T)]=2
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI<0.5 & ASC_new_sample_GI$isPTV==T)]=3
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=2)]=4
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=1 & ASC_new_sample_GI$MPC<2)]=5
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC<1)]=6

num.family=7291
## test 
```

## Variant count in PTV with `PLI>=0.995` at different AF cutoffs


| AF cutoff      | Trans |   Untrans    | 
| :---        |    :----:   |          ---: |
| <0.000001    | 659    | 655   |
| <0.00001    | 779    | 805   |      
| <0.0001    | 876    | 966   |
| <0.001    | 1164    | 1413   |


When AF <0.1%, there are 1164 transmitted variants and 1413 untransmitted, not consistent with numbers in cell paper 695 vs 557.  

* `pLI>=0.995`
* `isPTV==T`
* `isIndel=="FALSE"`
* `Gnomad_non_neuro_AF<AF_cutoff`
* `num_alleles<5`



| filter      | Trans |   Untrans    | 
| :---        |    :----:   |          ---: |
|  VQSLOD<3.75   | 255    | 546   |
| VQSLOD<5.13    | 300    | 624   |
| VQSLOD<9    | 1151    | 1415  |
| QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8    | 818    | 785   |
| QD >= 6 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8    | 816    | 726   |
| QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8    | 738    | 604   |
| QD >= 10.5 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8    | 714    | 567   |
| QD >= 11 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8    | 678    | 526   |
| QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -2   | 956    | 946   |
| QD >= 1 & AS_SOR <= 5 & AS_ReadPosRankSum >= -0.8   | 821    | 804   |
| QD >= 1 & AS_SOR <= 10 & AS_ReadPosRankSum >= -0.8   | 822    | 861   |

* Kyle suggested: QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8 
* with filter `QD >= 6 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8  `, 0.11 transmitted  (816/7291) vs 0.10 untransmitted  vps
* cutoff for `AS_ReadPosRankSum` could decrease to, for instance -2, or even lower, while maintaining balanced T/UT, but the OR consistently stay around 1.01, a bit away from 1.2 
* OR will increase with increasing cutoff for `QD`. When cutoff =10.5, both OR and variant counts are close to numbers in paper, but when cutoff >10.5, the variant count will become smaller than than in paper, seems not appropriate.  
* With balanced T/UT, increase of cutoff for `AS_SOR` will decrease OR, even below 1, becoming depleted  
* `pLI>=0.995`
* `isPTV==T`
* `isIndel=="FALSE"`
* `Gnomad_non_neuro_AF<0.001`


## plot of QD value 

```{r, echo=F, message=F, warning=F}
############## plot of QD values 
QD=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE"  & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8) %>% select(QD) %>% pull
plot(density(QD))
summary(QD)
```


use filters 

* `isIndel=="FALSE"`
* `AS_SOR <= 3`
* `AS_ReadPosRankSum >= -0.8`



## Run mirage-vs on 6 variant groups in cell paper 



* Use mirage-vs to test if there are signals in six variant groups  
* filter `Gnomad_non_neuro_AF<0.05` was applied 
* `isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`
* `isSyn=="FALSE"`

```{r, echo=F, eval=F}
####### test the toy example 
#data(mirage_vs_toy)
#res_vs = mirage_vs(mirage_vs_toy,n1=1000,n2=1000)
#########
para_est=matrix(nrow=6, ncol=6)
rownames(para_est)=c("PLI>=0.995", "pLI within (0.5,0.995)", "pLI<0.5", "MPC>=2", "MPC within (1,2)", "MPC<1")
colnames(para_est)=c("No_case", "No_contr", "OR", "pval", "eta_est", "eta_pval")
for (i in 1:6)
{ 
#i=1  
 cat(i, "is running", "\n")
 variant_set=ASC_new_sample_GI %>% filter(Group_Index==i & Gnomad_non_neuro_AF<0.05) %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
 para_est[i,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
 pois.test=poisson.test(para_est[i,1:2], c(num.family, num.family), r=1, alternative="greater")
 para_est[i,3:4]=c(pois.test$estimate,pois.test$p.value)
 if (i<=3) # for LoF variants
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=6)
 if (i>3)  # for missense variants 
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=3)
 para_est[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
  
}
as_tibble(para_est)
#write.csv(para_est, file="../output/KyleData/para.est.var.group.proband_0826_2022.csv")
#gene.data=eight.partition(cand.data)
#ds.asd.result=mirage(gene.data, n1=2403, n2=2403)
```


This figure shows the comparison between burden test (one side two sample poisson test) and mirage-vs for proband in 6 variant groups. $\gamma=6$ for 3 PLI variant groups  and $\gamma=3$ for another 3 MPC groups when running mirage_vs.  

```{r, echo=F, warning=F, message=F}
proband_para_est=read.csv("../output/KyleData/para.est.var.group.proband_0826_2022.csv")
as_tibble(proband_para_est)


var_group=c(rep(c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1"), 2))
method=c(rep("Burden", 6), rep("mirage_vs", 6))
pvalue=c(round(-log(proband_para_est$pval, base=10), 2), round(-log(proband_para_est$eta_pval, base=10), 2))
summary=data.frame(var.group=var_group, pvalue=pvalue, method=method)
ggplot(summary, aes(x=var.group, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", 
           position=position_dodge())+
  ylab(expression(-log[10](pvalue)))+
  theme(legend.title=element_blank())+
  xlab("variant groups")+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))
  #ylab(expression(paste(hat(eta))))+ 
  #guides(fill=guide_legend(title=NULL))+
  #ylim(0.05,0.15)+
  # geom_hline(yintercept=0.1,linetype="dashed", color = "red")
```

## Run MIRAGE-VS on 180 variant sets 


```{r, echo=F, eval=F}
####### test the toy example 
#data(mirage_vs_toy)
#res_vs = mirage_vs(mirage_vs_toy,n1=1000,n2=1000)
#########
para_est=matrix(nrow=180, ncol=7)
rownames(para_est)=paste(rep(paste(rep(Gene_Set,each=6), c("PLI>=0.995", "pLI within (0.5,0.995)", "pLI<0.5", "MPC>=2", "MPC within (1,2)", "MPC<1"), sep="_"), each=3), c("0.001<AF<0.01", "0.0001<AF<0.001", "AF<0.0001"), sep="_")
colnames(para_est)=c("No_case", "No_contr", "OR", "pval", "eta_est", "eta_pval", "cate_index")
for (i in 6:6) # for each gene set
  for (j in 6:6) # for each variant group
{ 
#i=1  
 cat(i, "is running", "\n")
    k=((i-1)*6+j-1)*3
    ###################
 variant_set=ASC_new_sample_GI %>% filter( Gene %in% Gene_Set_List[[i]]==T &  Group_Index==j & Gnomad_non_neuro_AF>0.001 & Gnomad_non_neuro_AF<0.01 & isSyn=="FALSE") %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
 if (nrow(variant_set)>0)
 {
 para_est[k+1,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
 pois.test=poisson.test(para_est[k+1,1:2], c(num.family, num.family), r=1, alternative="greater")
 para_est[k+1,3:4]=c(pois.test$estimate,pois.test$p.value)
 if (j<=3) # for LoF variants
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=6)
 if (j>3)  # for missense variants 
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=3)
 para_est[k+1,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
 }
 para_est[k+1,7]=j  # group index need to be recorded regardless of having variants or not 
 ###################
 variant_set=ASC_new_sample_GI %>% filter( Gene %in% Gene_Set_List[[i]]==T &  Group_Index==j & Gnomad_non_neuro_AF>0.0001 & Gnomad_non_neuro_AF<0.001 & isSyn=="FALSE") %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
 if (nrow(variant_set)>0)
 {
 para_est[k+2,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
 pois.test=poisson.test(para_est[k+2,1:2], c(num.family, num.family), r=1, alternative="greater")
 para_est[k+2,3:4]=c(pois.test$estimate,pois.test$p.value)
 if (j<=3) # for LoF variants
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=6)
 if (j>3)  # for missense variants 
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=3)
 para_est[k+2,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
 }  
 para_est[k+2,7]=j  # group index need to be recorded regardless of having variants or not
 ###################
 variant_set=ASC_new_sample_GI %>% filter( Gene %in% Gene_Set_List[[i]]==T &  Group_Index==j &  Gnomad_non_neuro_AF<0.0001 & isSyn=="FALSE") %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
 if (nrow(variant_set)>0)
 {
 para_est[k+3,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
 pois.test=poisson.test(para_est[k+3,1:2], c(num.family, num.family), r=1, alternative="greater")
 para_est[k+3,3:4]=c(pois.test$estimate,pois.test$p.value)
 if (j<=3) # for LoF variants
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=6)
 if (j>3)  # for missense variants 
   vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=3)
 para_est[k+3,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
 }
 para_est[k+3,7]=j  # group index need to be recorded regardless of having variants or not
} # end of  for (j in 1:6) # for each variant group
#as_tibble(para_est)
#write.csv(para_est, file="../output/KyleData/MIRAGE_VS.for.180.variant.group.proband_0829_2022.csv")

para_est[complete.cases(para_est), ]
```


```{r, echo=F, message=F, warning=F}

para_est=read.csv("../output/KyleData/MIRAGE_VS.for.180.variant.group.proband_0829_2022.csv", header=T)
############### plot the results 
#para_est_withoutNA=as_tibble(para_est) %>% drop_na()
para_est_withoutNA=para_est[complete.cases(para_est), ]
############### multiple correction for p values
Burden.pvalue=para_est_withoutNA[,5]
MIRAGE.pvalue=para_est_withoutNA[,7]
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))

Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=para_est_withoutNA[,4], Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=para_est_withoutNA[,6], MIRAGE.pvalue=MIRAGE.pvalue, No.var.case=para_est_withoutNA[,2], No.var.control=para_est_withoutNA[,3], combine.feature=para_est_withoutNA[,1], cate.index=para_est_withoutNA[,8])
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="PTV:pLI>=0.995", "2"="PTV:0.5<pLI<0.995", "3"="PTV: pLI<0.5", "4"="MPC>=2", "5"="1<=MPC<2", "6"="0<MPC<1")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(Burden.MIRAGE.combine.clear, file="../output/CombinedFeature/combined.features.of.gene.set.disjoint.AF.damaging.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=length(MIRAGE.pvalue)
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=1, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max+2))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
  theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=20))+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5))) # symbol size in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max+2))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE_VS")+
  theme_classic()+
  #theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
theme(text = element_text(size=20))


  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 
#dev.off()
```

* exclude syn variants `isSyn=="FALSE"`

* 6 variant groups X 3 AF categories (0.001<AF<0.01, 0.0001<AF<0.001, AF<0.0001) X 10 gene sets=180 variant groups 

* use filter `isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`

* solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/total number of tests.



```{r, echo=F, message=F, warning=F}
ggplot(data.frame(Burden_pval=-log(Burden.MIRAGE.combine.clear$Burden.pvalue, base=10), MIRAGE_VS_pval=-log(Burden.MIRAGE.combine.clear$MIRAGE.pvalue, base=10), OR=Burden.MIRAGE.combine.clear$Burden.OR), aes(x=Burden_pval, y=MIRAGE_VS_pval))+geom_point(aes(size=OR))+
  geom_abline(intercept =0, slope = 1, color="red")+
  xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("MIRAGE_VS vs Burden for 180 variant groups")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=20))  # center the title
  
```

* one outlier with extremely small MIRAGE_Vs p value is variant set `AutismKB_MPC<1_0.0001<AF<0.001` with 6802 (transmitted) vs 6914 (untransmitted) on 1901 loci. Burden p value: 0.9839; MIRAGE_VS p value: 5.195326e-15.

* this variant set has several loci driving the BF 

```
1500 1500      76      262        1 1.185888e+12
1499 1499      56      152        1 1.008106e+05
 
 BF.var.inte(trans=72, nontrans=262, bar.gamma=3, sig=3, num.family, num.family)
[1] 403764608704
```


```{r, echo=F, warning=F, message=F}
gamma_pvalue=data.frame(gamma=seq(1,10),pvalue=c(0.2595, 0.0896, 0.0900, 0.1151, 0.1748, 0.2980, 0.5398, 0.9934, 1,1))
ggplot(data=gamma_pvalue, aes(x=gamma, y=pvalue, group=1)) +
  geom_line(color="red")+
  geom_point()+
  ggtitle("Relative risk vs  MIRAGE_vs pvalues")+
  ylim(c(0,1))+
  ylab("MIRAGE_VS: p value")+
  xlab("Relative risk: gamma")+
  geom_hline(yintercept=0.092, linetype="dashed", 
                color = "blue", size=1)+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=20))  # center the title

```

* variant set `ID gene_PLI>=0.995_0.001<AF<0.01`, two loci, 13 (Trans) vs 15 (Nontrans) and 21 vs 8=> 34 vs 23 in total. 

* varying $\bar{\gamma}$ produces quite different p values for MIRAGE_VS. 

* blue dashed line: burden test p value of 0.09242524



## Run MIRAGE genomewide 

* `Gnomad_non_neuro_AF<0.05` 

* `isSyn=="FALSE"`

* `isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`

* `!str_detect(Consequence, 'intron_variant')` exclude introns 

* `!str_detect(Consequence, 'UTR')` exclude UTRs




### variant types 

```{r, echo=F, message=F, warning=F}
ASC_new_sample%>% filter(!str_detect(Consequence, 'UTR') & Gnomad_non_neuro_AF<0.05) %>%count(Consequence) %>% mutate(prop=round(n/nrow(ASC_new_sample),6)) %>% arrange(desc(prop))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


### AF distribution 

```{r, echo=F, message=F, warning=F}
ASC_new_sample_AF=ASC_new_sample%>% filter(!str_detect(Consequence, 'UTR')& Gnomad_non_neuro_AF<0.05)
  f1=ggplot(ASC_new_sample_AF, aes(x = Gnomad_non_neuro_AF)) + 
  geom_histogram()+
  geom_vline(aes(xintercept=mean(Gnomad_non_neuro_AF)),
            color="red", linetype="dashed", size=1)

a=ASC_new_sample_AF %>% filter(Gnomad_non_neuro_AF<0.001) %>% nrow() #%>%pull() 
b=ASC_new_sample_AF %>% filter(0.001<Gnomad_non_neuro_AF & Gnomad_non_neuro_AF<0.01) %>% nrow() 
c=ASC_new_sample_AF %>% filter(0.01<Gnomad_non_neuro_AF & Gnomad_non_neuro_AF<0.05) %>% nrow() 

var_count_by_AF=tibble(count=c(a,b,c), cate=c("(1) AF<0.1%", "(2) 0.1%<AF<1%", "(3) 1%<AF<5%"))
f2=ggplot(data=var_count_by_AF, aes(x=cate, y=count)) +
  geom_bar(stat="identity", width=0.5, fill="blue")+
  #ggtitle("Variant count in every category by AF")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+ # center the title
  geom_text(aes(label=count), position=position_dodge(width=0.6), vjust=-0.25)
figure=ggarrange(f1, f2, common.legend = TRUE, legend="right") 
annotate_figure(figure,
                top = text_grob("AF distribution", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)

summary(ASC_new_sample_AF$Gnomad_non_neuro_AF)
rm("ASC_new_sample_AF") # remove variables temporarily available in this chunk 
```





### MIRAGE analysis on  6 variant groups 

* Run mirage on all 17850 genes and $\gamma=6$ for 3  PLI variant groups   and $\gamma=3$ for 3  MPC groups. 

* `Gnomad_non_neuro_AF<0.05` 

* `isSyn=="FALSE"`

* `isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`


```{r, echo=F, eval=F}
##!!!!!!!!!!!!!!!!! Do not run !!!!!!!!!!!!!!!!!!!
############  gene set 
num.family=7291
gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family, gamma=c(6,6,6,3,3,3)) # input data must be dataframe 
######### It takes more than one hour to finish running mirage function 

#res=mirage(mirage_toy,n1=4315,n2=4315) # test mirage function 
#write.csv(mirage_result$BF.PP.gene, file="../output/KyleData/BF.proband.AllGene_0830_2022.csv") this is the results with introns 
#write.csv(mirage_result$BF.PP.gene, file="../output/KyleData/BF.proband.AllGene_1105_2022.csv") # this is the result without introns 

```

```{r, echo=F, eval=F}
#### parameter estimate this is the result with introns included

| variant category    | parameter |   estimate   | p value |
| :---        |    :----:   |         :----: |   ---: |
\ |$\hat{\delta}$ | 0.0677 | $1.383766e-42$
(1):PLI>=0.995|$\hat{\eta}_1$ | 0.8341    | $1.887287e-02$
(2):0.5<=pLI<0.995|$\hat{\eta}_2$ | 0.8510    | $1.236803e-02$
(3):pLI<0.5|$\hat{\eta}_3$ | 0.0292    | $6.837139e-01$
(4):MPC>=2|$\hat{\eta}_4$ | 0.4683    | $1.576707e-03$
(5):1<=MPC<2| $\hat{\eta}_5$ | 0.1136    | $1.401804e-12$
(6):0<=MPC<1| $\hat{\eta}_6$ | 0.0120    | $1.471682e-32$

var_group=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1")
estimate=c(0.8341, 0.8510, 0.0292, 0.4683, 0.1136, 0.0120)
summary=data.frame(var.group=var_group, estimate=estimate)
ggplot(summary, aes(x=var.group, y=estimate, fill=var.group)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("variant groups")+ylab(expression(paste(hat(eta))))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))
```



```{r, echo=F, message=F, warning=F}
pvalue_adjust=function(p_value)
{
  p_value_adjust=numeric()
for (i in 1:length(p_value))
{
  if (is.na(p_value[i])==T)
    p_value_adjust[i]=p_value[i]
  if (is.na(p_value[i])==F & p_value[i]<1e-10)
    p_value_adjust[i]="<1e-10"
  if (is.na(p_value[i])==F & p_value[i]>1e-10)
    p_value_adjust[i]=round(p_value[i],10)
}
  return(p_value_adjust)
}

p_value=c(1.340989e-42, 1.915219e-02, 1.249333e-02, 6.840028e-01, 1.466075e-03, 1.324633e-12, 1.629645e-32)


mirage_para_est=data.frame(variant_groups=c("delta:proportion of risk genes","PTV_PLI>=0.995", "PTV_0.5=<PLI<0.995", "PTV_PLI<0.5",  "MPC>=2",  "1=<MPC<2", "MPC<1"), parameter_estimate=round(c(0.06713907, 0.83382753,0.85448876,0.02985442,0.47471948, 0.11521000, 0.01201084),4), p_value=pvalue_adjust(p_value))
mirage_para_est%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

var_group=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1")
estimate=c(0.83382753,0.85448876,0.02985442,0.47471948, 0.11521000, 0.01201084)
summary=data.frame(var.group=var_group, estimate=estimate)
ggplot(summary, aes(x=var.group, y=estimate, fill=var.group)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("variant groups")+ylab(expression(paste(hat(eta))))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))
```



#### top ranked genes



```{r, echo=F}
gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()


BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=BF_proband_AllGene_sorted$Gene[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, BF=round(BF_proband_AllGene_sorted$BF[1:num_gene_to_display], 4), post.prob=round(BF_proband_AllGene_sorted$post.prob[1:num_gene_to_display],4), SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```

* [HLA-DRB1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5955937/)
* [SSC5D](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7697115/)
* [GSTM1](https://pubmed.ncbi.nlm.nih.gov/33546147/)
* [TXNDC2](https://www.nature.com/articles/s41598-021-00123-x)



#### BF in each variant group

```{r, echo=F, message=F, warning=F, eval=F}
################# calculate BF in each group 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
interest_gene=BF_proband_AllGene_sorted$Gene  # use all genes 
beta=c(0.83382753, 0.85448876, 0.02985442,  0.47471948,  0.11521000, 0.01201084)
new.data=ASC_new_sample_GI %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
colnames(new.data)[3:5]=c("No.case", "No.contr", "group.index")
result=fixed.beta.func(new.data, N1=7291, N0=7291, gamma.mean=c(rep(6,3),rep(3,3)),sigma=2, beta=beta) # use fixed beta to calculate Bayes factors

############## verify the calculations 
#gene_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="SSC5D")]]
#group.index=gene_variant_BF$group.index
#gene_BF=prod(1-beta[group.index]+beta[group.index]*gene_variant_BF$var.BF)
#gene_BF
#result$BayesFactor[result$BayesFactor$Gene=="SSC5D",]
##############
variant_BF=matrix(nrow=length(interest_gene), ncol=7)
variant_count=matrix(nrow=length(interest_gene), ncol=7)
SFARI_score=rep("NA", length(interest_gene))
for (i in 1:length(interest_gene))
{
  cat(i, "th gene is running", "\n")
  variant_BF[i,1]=result$BayesFactor$BF[which(result$BayesFactor$Gene==interest_gene[i])]
  var_full=result$full.info[[which(result$BayesFactor$Gene==interest_gene[i])]]
  variant_count[i,1]=paste(paste(sum(var_full$No.case)),"vs", paste(sum(var_full$No.contr)), sep=" ")
   for (j in 1:6)
   {
     if (nrow(var_full %>% filter(group.index==j))==0) # no variants in this group
      { 
        variant_BF[i,j+1]="-"
        variant_count[i,j+1]="-"
     }
     
     if (nrow(var_full %>% filter(group.index==j))>0) # variants exist in this group
      { 
       variant_BF[i,j+1]=prod(1-beta[j]+beta[j]*var_full[which(var_full$group.index==j),]$var.BF) # product of BF of variants within that variant group 
       variant_count[i,j+1]=paste(paste(sum(var_full[which(var_full$group.index==j),]$No.case)),"vs", paste(sum(var_full[which(var_full$group.index==j),]$No.contr)), sep=" ")
     }
     
   }  #### end of 
  if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),]$gene.score
  
  
}
BF_var_count_combine_6groups=data.frame(Gene=interest_gene, BF=round(as.numeric(variant_BF[,1]),4),  SFARI_score=SFARI_score, total_count=variant_count[,1], g1_count=variant_count[,2], g1_BF=round(as.numeric(variant_BF[,2]),2), g2_count=variant_count[,3], g2_BF=round(as.numeric(variant_BF[,3]),2), g3_count=variant_count[,4], g3_BF=round(as.numeric(variant_BF[,4]),2), g4_count=variant_count[,5], g4_BF=round(as.numeric(variant_BF[,5]),2), g5_count=variant_count[,6], g5_BF=round(as.numeric(variant_BF[,6]),2), g6_count=variant_count[,7], g6_BF=round(as.numeric(variant_BF[,7]),2))                                          

#write.csv(BF_var_count_combine_6groups, file="../output/KyleData/BF.proband.AllGene_6vargroups_1105_2022.csv")

```


* [BF's  with 6 variant groups](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/EcspTHTo0IlMh7jWuIvPkYAB1L1B6N2EqQsO9Tmp1CxQpA?e=0mAqUd)


#### Variants driving signal in top genes 


```
> result$full.info[[14663]]
          Variant     Gene No.case No.contr group.index       var.BF
1  6:32580781:C:A HLA-DRB1       0        0           5 1.000000e+00
2  6:32581584:C:G HLA-DRB1      56      152           6 1.008106e+05
3  6:32581672:A:C HLA-DRB1       0        1           5 5.479161e-01
4  6:32581730:C:A HLA-DRB1       1        0           4 1.452084e+00
5  6:32581733:C:G HLA-DRB1       1        0           5 1.452084e+00
6  6:32581836:G:A HLA-DRB1       0        2           3 9.846160e-02
7  6:32584246:G:A HLA-DRB1       0        1           5 5.479161e-01
8  6:32584283:A:C HLA-DRB1       1        1           6 7.666583e-01
9  6:32584294:T:A HLA-DRB1       0        1           5 5.479161e-01
10 6:32584303:T:A HLA-DRB1      76      262           6 1.185888e+12
11 6:32584314:G:T HLA-DRB1       5       32           6 2.010501e+00
12 6:32584344:A:T HLA-DRB1       5        4           6 5.512252e-01
13 6:32584363:T:A HLA-DRB1     278      490           6 9.815683e+03
14 6:32584366:C:T HLA-DRB1     240      373           3 4.213022e-01
```

```
> BF.var.inte(No.case=76, No.contr=262, bar.gamma=3, sig=2, N1=7291, N0=7291)
[1] 1.185888e+12
> 
```

When calculating BF for single variant, under null hypothesis of relative risk of $\gamma=1$, the denominator, $Bin (x_1|x_1+x_0, \frac{N_1}{N_1+N_0})$  could be very small with a large number of total variant counts and too unbalanced numbers in cases and control (could be less counts in case and more in control), driving the BF to be a big number. 



#### Enrichment of top genes 

```{r, echo=F, warning=F, message=F}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))

sort_BF=BF_para_all_est %>% arrange(desc(BF))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 MIRAGE Gene", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.3))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by MIRAGE")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=0.0960", y_position = 0.08  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=0.2059", y_position = 0.08  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.1041", y_position = 0.18  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment
```


* strong ASD: SFARI score 1 or 2

* plausible ASD: SFARI score 3

* DNM: TADA denovo genes  



```{r, echo=F, warning=F, message=F}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))

sort_BF=BF_para_all_est %>% arrange(desc(BF))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 MIRAGE Gene",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.3))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by MIRAGE")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="5.67e-5", y_position = 0.21  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="4.56e-3", y_position = 0.26  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="1.30e-2", y_position = 0.21  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.221", y_position = 0.16  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.069", y_position = 0.06,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.223", y_position = 0.06,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.247", y_position = 0.06,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.513", y_position = 0.06,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="3.51e-3", y_position = 0.26,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.06", y_position = 0.26,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment


```



#### filtering variant groups with negative burden and large BF


```{r, echo=F, message=F, warning=F, eval=F}
################# calculate BF in each group 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
interest_gene=BF_proband_AllGene_sorted$Gene  # use all genes 
beta=c(0.83382753, 0.85448876, 0.02985442,  0.47471948,  0.11521000, 0.01201084)
new.data=ASC_new_sample_GI %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
colnames(new.data)[3:5]=c("No.case", "No.contr", "group.index")
result=fixed.beta.func(new.data, N1=7291, N0=7291, gamma.mean=c(rep(6,3),rep(3,3)),sigma=2, beta=beta) # use fixed beta to calculate Bayes factors

############## verify the calculations 
#gene_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="SSC5D")]]
#group.index=gene_variant_BF$group.index
#gene_BF=prod(1-beta[group.index]+beta[group.index]*gene_variant_BF$var.BF)
#gene_BF
#result$BayesFactor[result$BayesFactor$Gene=="SSC5D",]
##############
variant_BF=matrix(nrow=length(interest_gene), ncol=8)
variant_count=matrix(nrow=length(interest_gene), ncol=7)
SFARI_score=rep("NA", length(interest_gene))
for (i in 1:length(interest_gene))
{
  cat(i, "th gene is running", "\n")
  variant_BF[i,1]=result$BayesFactor$BF[which(result$BayesFactor$Gene==interest_gene[i])]
  var_full=result$full.info[[which(result$BayesFactor$Gene==interest_gene[i])]]
  variant_count[i,1]=paste(paste(sum(var_full$No.case)),"vs", paste(sum(var_full$No.contr)), sep=" ")
  variant_BF_modified=1
   for (j in 1:6)
   {
     if (nrow(var_full %>% filter(group.index==j))==0) # no variants in this group
      { 
        variant_BF[i,j+1]="-"
        variant_count[i,j+1]="-"
     }
     
     if (nrow(var_full %>% filter(group.index==j))>0) # variants exist in this group
      { 
       variant_BF[i,j+1]=prod(1-beta[j]+beta[j]*var_full[which(var_full$group.index==j),]$var.BF) # product of BF of variants within that variant group 
       variant_count[i,j+1]=paste(paste(sum(var_full[which(var_full$group.index==j),]$No.case)),"vs", paste(sum(var_full[which(var_full$group.index==j),]$No.contr)), sep=" ")
       
       ############## filter out variants with negative burden and large BF
       if (sum(var_full[which(var_full$group.index==j),]$No.case) > sum(var_full[which(var_full$group.index==j),]$No.contr) & variant_BF[i,j+1]>1.1) # filter out variant with negative burden and BF>1.1
         variant_BF_modified=variant_BF_modified*as.numeric(variant_BF[i,j+1])
       
       if (sum(var_full[which(var_full$group.index==j),]$No.case) < sum(var_full[which(var_full$group.index==j),]$No.contr) & variant_BF[i,j+1]<1.1) # filter out variant with negative burden and BF>1.1
         variant_BF_modified=variant_BF_modified*as.numeric(variant_BF[i,j+1])
       
     } # end of if (nrow(var_full %>% filter(group.index==j))>0) # variants exist in this group
     
   }  #### end of  for (j in 1:6)
  
  variant_BF[i,8]=variant_BF_modified
  
  
  if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),]$gene.score
  
  
}
BF_var_count_combine_6groups=data.frame(Gene=interest_gene, BF=round(as.numeric(variant_BF[,1]),4), BF_filtered=round(as.numeric(variant_BF[,8]),4), SFARI_score=SFARI_score, total_count=variant_count[,1], g1_count=variant_count[,2], g1_BF=round(as.numeric(variant_BF[,2]),2), g2_count=variant_count[,3], g2_BF=round(as.numeric(variant_BF[,3]),2), g3_count=variant_count[,4], g3_BF=round(as.numeric(variant_BF[,4]),2), g4_count=variant_count[,5], g4_BF=round(as.numeric(variant_BF[,5]),2), g5_count=variant_count[,6], g5_BF=round(as.numeric(variant_BF[,6]),2), g6_count=variant_count[,7], g6_BF=round(as.numeric(variant_BF[,7]),2))                                          

#write.csv(BF_var_count_combine_6groups[order(BF_var_count_combine_6groups$BF_filtered, decreasing = T),], file="../output/KyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv") # filter out variant groups with negative burden and BF>1.1


#write.csv(result$full.info[[3368]], file="../output/KyleData/BF.proband.AllGene_6vargroups_gene_NRXN2_1122_2022.csv") # output gene NRXN2
#write.csv(result$full.info[[6920]], file="../output/KyleData/BF.proband.AllGene_6vargroups_gene_DPH1_1122_2022.csv") # output gene DPH1

#write.csv(result$full.info[[which(result$BayesFactor$Gene=="CTNNA2")]], file="../output/KyleData/BF.proband.AllGene_6vargroups_gene_CTNNA2_1122_2022.csv") # output gene CTNNA2

#write.csv(result$full.info[[which(result$BayesFactor$Gene=="DNM2")]], file="../output/KyleData/BF.proband.AllGene_6vargroups_gene_DNM2_1122_2022.csv") # output gene DNM2
```





```{r, echo=F, message=F, warning=F, eval=T}
gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
################ top genes 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(BF_filtered))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=BF_proband_AllGene_sorted$Gene[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, BF=round(BF_proband_AllGene_sorted$BF[1:num_gene_to_display], 4), 
           BF_filtered=round(BF_proband_AllGene_sorted$BF_filtered[1:num_gene_to_display], 4), 
           SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4))%>%
datatable(extensions = 'Buttons',
          caption = 'Top genes with adjusted BF',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


* `BF_filtered` if the product of BF taking out the groups with negative burden but with BF>1.1, `BF` is original BF with all variant groups 


* [BF's adjusted with 6 variant groups](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/Ec3b10WIx9pOm3cjz-WBipYBYXPM5gHYh_w2yycJts4Nlg?e=xbvpDd) 




```{r, echo=F, message=F, warning=F, eval=T}
#gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
################ top genes 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(BF_filtered))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=20
gene_to_display=BF_proband_AllGene_sorted$Gene[1:num_gene_to_display]
SFARI_score=rep("NA", num_gene_to_display)
Incidence_gene=matrix(nrow=num_gene_to_display, ncol=11)
for (i in 1:length(gene_to_display))
{


if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
for (j in 1:10)
 Incidence_gene[i,1+j]=as.numeric(gene_to_display[i]%in% Gene_Set_List[[j]])

}
rownames(Incidence_gene)=gene_to_display
colnames(Incidence_gene)=c("BF_filtered",Gene_Set)
Incidence_gene[,1]=BF_proband_AllGene_sorted$BF_filtered[1:num_gene_to_display]
Incidence_gene%>%
datatable(extensions = 'Buttons',
          caption = 'Occurrence of top 20 genes in various gene sets',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


* 1 indicates genes in rows belong to gene set in column. 

*  several top genes driven by one single variant 

1.  [NRXN2](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/Ed9BRptUJDNGgmp4VZ_xXykBxtCIBOI8-KDgX-ikD9UR2g?e=eVDeM1) `11:64651517:C:T	NRXN2	12(T)	1(UT)	5	66.4923272(BF)`

2. [CTNNA2](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/ESreMlNVT2FPjNPQ5nCl1dgBOKCg9M0jYPyuDCj-IFamnA?e=AyBYNJ)  `2:80647639:G:A	CTNNA2	15(T)	5(UT)	4	10.55319601(BF)`
3.  [DNM2](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/EX9eZHeVmzNHlbw22wmEAvsBXuAZBGeqIYm6OdfHgwRmAA?e=tpMiD9)   `19:10783059:C:T	DNM2	17(T)	6(UT)	4	11.62929522(BF)`
4.  [DPH1](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/ESrgp6Eeb_pJkfP3VO4iDmIBZzRUkfR0ys9Tw_GF1eU6Wg?e=AIpquJ)  `17:2041487:G:C	DPH1	18(T)	1(UT)	6	1050.851519(BF)`




```{r, echo=F, warning=F, message=F}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))

sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 MIRAGE Gene", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.3))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by MIRAGE")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=0.0960", y_position = 0.08  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=0.2059", y_position = 0.08  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.1041", y_position = 0.18  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment
```



```{r, echo=F, warning=F, message=F}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))

sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 MIRAGE Gene",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.5))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by MIRAGE")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.016", y_position = 0.115  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="3.32e-6", y_position = 0.47  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="6.09e-5", y_position = 0.37  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="8.53e-3", y_position = 0.32  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.069", y_position = 0.07,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.223", y_position = 0.07,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.247", y_position = 0.07,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.041", y_position = 0.17,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="2.02e-6", y_position = 0.47,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="5.06e-4", y_position = 0.47,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment
# pvalue
# [1] 2.232156e-01 6.913626e-02 2.471987e-01 4.121511e-02 6.088691e-05 1.592053e-02 3.323435e-06
# [8] 2.016024e-06 8.532949e-03 5.064991e-04
 
# > Gene_Set
# [1] "ID gene"         "High"            "Mod"             "PSD"             "FMRP"           
# [6] "AutismKB"        "constraint gene" "RVIS"            "Haploinsuff"     "SCZ #gene

```






### MIRAGE analysis on 15 variant groups

```{r, echo=F, message=F, warning=F, eval=T}
# add one column of variant group index 
ASC_new_sample_GI2=ASC_new_sample %>% filter(!str_detect(Consequence, 'UTR')& Gnomad_non_neuro_AF<0.05)%>% add_column(Group_Index=NA)

############# LOF variants 
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$pLI>=0.995 & ASC_new_sample_GI2$isPTV==T & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.01)]=1
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$pLI>=0.995 & ASC_new_sample_GI2$isPTV==T & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.01)]=2

ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$pLI>=0.5 & ASC_new_sample_GI2$pLI<0.995 & ASC_new_sample_GI2$isPTV==T & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.01)]=3
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$pLI>=0.5 & ASC_new_sample_GI2$pLI<0.995 & ASC_new_sample_GI2$isPTV==T & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.01)]=4

ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$pLI<0.5 & ASC_new_sample_GI2$isPTV==T & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.01)]=5
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$pLI<0.5 & ASC_new_sample_GI2$isPTV==T & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.01)]=6
########## missense variants 
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC>=2 & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.001)]=7
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC>=2 & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.001 & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.01)]=8
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC>=2 & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.01 )]=9
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC>=1 & ASC_new_sample_GI2$MPC<2 & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.001)]=10
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC>=1 & ASC_new_sample_GI2$MPC<2 & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.001 & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.01)]=11
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC>=1 & ASC_new_sample_GI2$MPC<2 & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.01)]=12

ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC<1 & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.001)]=13
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC<1 & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.001 & ASC_new_sample_GI2$Gnomad_non_neuro_AF<0.01)]=14
ASC_new_sample_GI2$Group_Index[which(ASC_new_sample_GI2$MPC<1 & ASC_new_sample_GI2$Gnomad_non_neuro_AF>=0.01)]=15
num.family=7291


gene_set=ASC_new_sample_GI2 %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na() # drop all variants neither LoF nor missense variants 
```


```{r, echo=F, message=F, warning=F, eval=F}
################ !!!!!!!!!!! DO NOT RUN #######################################
mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family, gamma=c(rep(6,6), rep(3,9))) # input data must be dataframe 
######### It takes more than one hour to finish running mirage function 

#res=mirage(mirage_toy,n1=4315,n2=4315) # test mirage function 
#write.csv(mirage_result$BF.PP.gene, file="../output/KyleData/BF.proband.AllGene_1015_2022.csv") # this is the result with introns included 
```

```{r, echo=F, message=F, warning=F, eval=F}
################ !!!!!!!!!!! DO NOT RUN ####################################### 
######## use fixed delta=0.1 to estimate eta's 
mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family, gamma=c(rep(6,6), rep(3,9)), sigma = 2, eta.init = 0.1, 
                delta.init = 0.1, estimate.delta = FALSE, max.iter = 10000, 
                tol = 1e-05, verbose = TRUE)  # input data must be dataframe 
######### It takes more than one hour to finish running mirage function 

#res=mirage(mirage_toy,n1=4315,n2=4315) # test mirage function 
#write.csv(mirage_result$BF.PP.gene, file="../output/KyleData/BF.proband.AllGene_1031_2022.csv") 
```

```{r, echo=F, message=F, warning=F, eval=F}
######### estimate both delta and eta, the likelihood function is 4.877431e+122=exp(565/2), 565 is test statistics given p value of 1.407778e-111
######### estimate eta, with fixed delta=0.1, the 3.795194e+103=exp(477/2), 477 is test statistics given p value of 5.20272e-93
```


#### parameter estimate 


```{r, echo=F, message=F, warning=F}
pvalue_adjust=function(p_value)
{
  p_value_adjust=numeric()
for (i in 1:length(p_value))
{
  if (is.na(p_value[i])==T)
    p_value_adjust[i]=p_value[i]
  if (is.na(p_value[i])==F & p_value[i]<1e-10)
    p_value_adjust[i]="<1e-10"
  if (is.na(p_value[i])==F & p_value[i]>1e-10)
    p_value_adjust[i]=round(p_value[i],10)
}
  return(p_value_adjust)
}

p_value=c(1.407778e-111, 1.243930e-03, 1.000000e+00, 1.520364e-02, 1.000000e+00, 3.754382e-01, 1.000000e+00, 1.076301e-08, 1.000000e+00, 1.000000e+00, 1.635885e-21, 1.000000e+00, 7.048646e-10, 2.874112e-72, 1.132749e-16, 1.689324e-01)


mirage_para_est=data.frame(variant_groups=c("delta:proportion of risk genes","PTV_PLI>=0.995_AF<0.01", "PTV_PLI>=0.995_0.01<AF<0.05", "PTV_0.5=<PLI<0.995_AF<0.01", "PTV_0.5=<PLI<0.995_0.01<AF<0.05", "PTV_PLI<0.5_AF<0.01", "PTV_PLI<0.5_0.01<AF<0.05", "MPC>=2_AF<0.001", "MPC>=2_0.001=<AF<0.01", "MPC>=2_0.01=<AF<0.05", "1=<MPC<2_AF<0.001", "1=<MPC<2_0.001=<AF<0.01", "1=<MPC<2_0.01=<AF<0.05", "MPC<1_AF<0.001", "MPC<1_0.001=<AF<0.01", "MPC<1_0.01=<AF<0.05"), parameter_estimate=round(c(0.4984, 2.402658e-01,  0.000000e+00,  1.514065e-01,  0.000000e+00,  1.063114e-02,  0.000000e+00,  1.505522e-01, 1.185758e-322, 1.976263e-323,  8.274313e-02, 7.410985e-323,  1.048833e-02,  4.259534e-02,  2.777784e-04,  5.524852e-04),4), p_value=pvalue_adjust(p_value))
mirage_para_est%>%
datatable(extensions = 'Buttons',
          caption = 'parameter estimates when delta was estimated',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


```{r, echo=F, message=F, warning=F}
pvalue_adjust=function(p_value)
{
  p_value_adjust=numeric()
for (i in 1:length(p_value))
{
  if (is.na(p_value[i])==T)
    p_value_adjust[i]=p_value[i]
  if (is.na(p_value[i])==F & p_value[i]<1e-10)
    p_value_adjust[i]="<1e-10"
  if (is.na(p_value[i])==F & p_value[i]>1e-10)
    p_value_adjust[i]=round(p_value[i],10)
}
  return(p_value_adjust)
}

p_value=c(5.20272e-93, 1.155983e-02, 1.000000e+00, 7.411113e-03, 1.000000e+00,3.861759e-01, 1.000000e+00, 4.294019e-08, 1.000000e+00, 1.000000e+00, 1.054894e-15, 1.000000e+00, 1.351004e-10, 2.996753e-62,1.142594e-16, 5.607992e-02)


mirage_para_est=data.frame(variant_groups=c("delta:proportion of risk genes","PTV_PLI>=0.995_AF<0.01", "PTV_PLI>=0.995_0.01<AF<0.05", "PTV_0.5=<PLI<0.995_AF<0.01", "PTV_0.5=<PLI<0.995_0.01<AF<0.05", "PTV_PLI<0.5_AF<0.01", "PTV_PLI<0.5_0.01<AF<0.05", "MPC>=2_AF<0.001", "MPC>=2_0.001=<AF<0.01", "MPC>=2_0.01=<AF<0.05", "1=<MPC<2_AF<0.001", "1=<MPC<2_0.001=<AF<0.01", "1=<MPC<2_0.01=<AF<0.05", "MPC<1_AF<0.001", "MPC<1_0.001=<AF<0.01", "MPC<1_0.01=<AF<0.05"), parameter_estimate=round(c(0.1, 5.429148e-01,  0.000000e+00,  6.458158e-01,  0.000000e+00,  5.420754e-02,  3.952525e-323,  4.859705e-01, 5.665391e-103, 9.881313e-323,  2.126305e-01, 1.258156e-113,  5.170850e-02,  1.337174e-01,  1.258930e-03,  3.710886e-03),4), p_value=pvalue_adjust(p_value))
mirage_para_est%>%
datatable(extensions = 'Buttons',
          caption = 'parameter estimates when delta was fixed at 0.1',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

#### Top genes 


```{r, echo=F, message=F, warning=F}
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1015_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=BF_proband_AllGene_sorted$Gene[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric()
SFARI_score=rep("NA", num_gene_to_display)
for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, BF=round(BF_proband_AllGene_sorted$BF[1:num_gene_to_display], 4), post.prob=round(BF_proband_AllGene_sorted$post.prob[1:num_gene_to_display],4), SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4))%>%
datatable(extensions = 'Buttons',
          caption = 'Top genes when propotion of risk genes was estimated',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
#BF_proband_AllGene_sorted %>% filter(Gene %in% ASD_102_Gene) %>% print(n=102)
```


```{r, echo=F, message=F, warning=F}
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1031_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=BF_proband_AllGene_sorted$Gene[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric()
SFARI_score=rep("NA", num_gene_to_display)
for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, BF=round(BF_proband_AllGene_sorted$BF[1:num_gene_to_display], 4), post.prob=round(BF_proband_AllGene_sorted$post.prob[1:num_gene_to_display],4), SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4))%>%
datatable(extensions = 'Buttons',
          caption = 'Top genes when propotion of risk genes was fixed at 0.1',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
#BF_proband_AllGene_sorted %>% filter(Gene %in% ASD_102_Gene) %>% print(n=102)
```



#### BF in variant groups 

```{r, echo=F, message=F, warning=F, eval=F}
interest_gene=BF_proband_AllGene_sorted$Gene[1:10]
beta=c(2.402658e-01,  0.000000e+00,  1.514065e-01,  0.000000e+00,  1.063114e-02,  0.000000e+00,  1.505522e-01, 1.185758e-322, 1.976263e-323,  8.274313e-02, 7.410985e-323,  1.048833e-02,  4.259534e-02,  2.777784e-04,  5.524852e-04)
new.data=ASC_new_sample_GI2 %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
colnames(new.data)[3:5]=c("No.case", "No.contr", "group.index")
result=fixed.beta.func(new.data, N1=7291, N0=7291, gamma.mean=c(rep(6,6),rep(3,9)),sigma=2, beta=beta) # use fixed beta to calculate Bayes factors

############## verify the calculations 
#gene_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="SSC5D")]]
#group.index=gene_variant_BF$group.index
#gene_BF=prod(1-beta[group.index]+beta[group.index]*gene_variant_BF$var.BF)
#gene_BF
#result$BayesFactor[result$BayesFactor$Gene=="SSC5D",]
##############
variant_BF=matrix(nrow=length(interest_gene), ncol=16)
variant_count=matrix(nrow=length(interest_gene), ncol=16)
for (i in 1:length(interest_gene))
{
  variant_BF[i,1]=result$BayesFactor$BF[which(result$BayesFactor$Gene==interest_gene[i])]
  var_full=result$full.info[[which(result$BayesFactor$Gene==interest_gene[i])]]
  variant_count[i,1]=paste(paste(sum(var_full$No.case)),"vs", paste(sum(var_full$No.contr)), sep=" ")
   for (j in 1:15)
   {
     if (nrow(var_full %>% filter(group.index==j))==0) # no variants in this group
      { 
        variant_BF[i,j+1]="-"
        variant_count[i,j+1]="-"
     }
     
     if (nrow(var_full %>% filter(group.index==j))>0) # variants exist in this group
      { 
       variant_BF[i,j+1]=prod(1-beta[j]+beta[j]*var_full[which(var_full$group.index==j),]$var.BF) # product of BF of variants within that variant group 
       variant_count[i,j+1]=paste(paste(sum(var_full[which(var_full$group.index==j),]$No.case)),"vs", paste(sum(var_full[which(var_full$group.index==j),]$No.contr)), sep=" ")
     }
     
   }
  
  
}

data.frame(Gene=interest_gene, BF=format(round(as.numeric(variant_BF[,1]), 2),scientific = T), g1_BF=round(as.numeric(variant_BF[,2]), 2), g2_BF=round(as.numeric(variant_BF[,3]), 2), g3_BF=round(as.numeric(variant_BF[,4]), 2), g4_BF=round(as.numeric(variant_BF[,5]), 2), g5_BF=format(round(as.numeric(variant_BF[,6]), 2),scientific = T), g6_BF=round(as.numeric(variant_BF[,7]), 2))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

data.frame(Gene=interest_gene, Trans_vs_Untrans=variant_count[,1], g1_count=variant_count[,2], g2_count=variant_count[,3], g3_count=variant_count[,4], g4_count=variant_count[,5], g5_count=variant_count[,6], g6_count=variant_count[,7])%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
           
data.frame(Gene=interest_gene, BF=format(round(as.numeric(variant_BF[,1]), 2),scientific = T),           g7_BF=format(round(as.numeric(variant_BF[,8]), 2),scientific = T), g8_BF=round(as.numeric(variant_BF[,9]), 2), g9_BF=round(as.numeric(variant_BF[,10]), 2), g10_BF=format(round(as.numeric(variant_BF[,11]), 2),scientific = T), g11_BF=round(as.numeric(variant_BF[,12]), 2), g12_BF=format(round(as.numeric(variant_BF[,13]), 2),scientific = T), g13_BF=format(round(as.numeric(variant_BF[,14]), 2),scientific = T), g14_BF=format(round(as.numeric(variant_BF[,15]), 2),scientific = T),  g15_BF=format(round(as.numeric(variant_BF[,16]), 2),scientific = T))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

data.frame(Gene=interest_gene, Trans_vs_Untrans=variant_count[,1], g7_count=variant_count[,8], g8_count=variant_count[,9], g9_count=variant_count[,10], g10_count=variant_count[,11], g11_count=variant_count[,12], g12_count=variant_count[,13], g13_count=variant_count[,14], g14_count=variant_count[,15], g15_count=variant_count[,16])%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```



```{r, echo=F, message=F, warning=F, eval=F}
################# when delta=0.1, calculate BF in each group 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1031_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
interest_gene=BF_proband_AllGene_sorted$Gene  # use all genes 
beta=c(5.429148e-01,  0.000000e+00,  6.458158e-01,  0.000000e+00,  5.420754e-02,  3.952525e-323,  4.859705e-01, 5.665391e-103, 9.881313e-323,  2.126305e-01, 1.258156e-113,  5.170850e-02,  1.337174e-01,  1.258930e-03,  3.710886e-03)
new.data=ASC_new_sample_GI2 %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
colnames(new.data)[3:5]=c("No.case", "No.contr", "group.index")
result=fixed.beta.func(new.data, N1=7291, N0=7291, gamma.mean=c(rep(6,6),rep(3,9)),sigma=2, beta=beta) # use fixed beta to calculate Bayes factors

############## verify the calculations 
#gene_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="SSC5D")]]
#group.index=gene_variant_BF$group.index
#gene_BF=prod(1-beta[group.index]+beta[group.index]*gene_variant_BF$var.BF)
#gene_BF
#result$BayesFactor[result$BayesFactor$Gene=="SSC5D",]
##############
variant_BF=matrix(nrow=length(interest_gene), ncol=16)
variant_count=matrix(nrow=length(interest_gene), ncol=16)
SFARI_score=rep("NA", length(interest_gene))
for (i in 1:length(interest_gene))
{
  cat(i, "th gene is running", "\n")
  variant_BF[i,1]=result$BayesFactor$BF[which(result$BayesFactor$Gene==interest_gene[i])]
  var_full=result$full.info[[which(result$BayesFactor$Gene==interest_gene[i])]]
  variant_count[i,1]=paste(paste(sum(var_full$No.case)),"vs", paste(sum(var_full$No.contr)), sep=" ")
   for (j in 1:15)
   {
     if (nrow(var_full %>% filter(group.index==j))==0) # no variants in this group
      { 
        variant_BF[i,j+1]="-"
        variant_count[i,j+1]="-"
     }
     
     if (nrow(var_full %>% filter(group.index==j))>0) # variants exist in this group
      { 
       variant_BF[i,j+1]=prod(1-beta[j]+beta[j]*var_full[which(var_full$group.index==j),]$var.BF) # product of BF of variants within that variant group 
       variant_count[i,j+1]=paste(paste(sum(var_full[which(var_full$group.index==j),]$No.case)),"vs", paste(sum(var_full[which(var_full$group.index==j),]$No.contr)), sep=" ")
     }
     
   }  #### end of 
  if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),]$gene.score
  
  
}
BF_var_count_combine_15groups=data.frame(Gene=interest_gene, BF=round(as.numeric(variant_BF[,1]),4),  SFARI_score=SFARI_score, total_count=variant_count[,1], g1_count=variant_count[,2], g1_BF=round(as.numeric(variant_BF[,2]),2), g2_count=variant_count[,3], g2_BF=round(as.numeric(variant_BF[,3]),2), g3_count=variant_count[,4], g3_BF=round(as.numeric(variant_BF[,4]),2), g4_count=variant_count[,5], g4_BF=round(as.numeric(variant_BF[,5]),2), g5_count=variant_count[,6], g5_BF=round(as.numeric(variant_BF[,6]),2), g6_count=variant_count[,7], g6_BF=round(as.numeric(variant_BF[,7]),2), g7_count=variant_count[,8], g7_BF=round(as.numeric(variant_BF[,8]),2), g8_count=variant_count[,9], g8_BF=round(as.numeric(variant_BF[,9]),2), g9_count=variant_count[,10], g9_BF=round(as.numeric(variant_BF[,10]),2), g10_count=variant_count[,11], g10_BF=round(as.numeric(variant_BF[,11]),2), g11_count=variant_count[,12], g11_BF=round(as.numeric(variant_BF[,12]),2), g12_count=variant_count[,13], g12_BF=round(as.numeric(variant_BF[,13]),2), g13_count=variant_count[,14], g13_BF=round(as.numeric(variant_BF[,14]),2), g14_count=variant_count[,15], g4_BF=round(as.numeric(variant_BF[,15]),2), g15_count=variant_count[,16], g15_BF=round(as.numeric(variant_BF[,16]),2))                                          

#write.csv(BF_var_count_combine_15groups, file="../output/KyleData/BF.proband.AllGene_deltafixed0.1_1104_2022.csv")
```


* [BF's when delta was fixed at 0.1](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/EXLTsKW_SZdJkLIMXosQtz8BnRHKvOhh3v5JDAGBjEcepg?e=owqLgw)

#### Enrichment of top genes  with other gene sets 


```{r, echo=F, message=F, warning=F, eval=T}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1031_2022.csv"))

sort_BF=BF_para_all_est %>% arrange(desc(BF))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 MIRAGE Gene", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.3))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by MIRAGE")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=1.0000", y_position = 0.05  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=0.2059", y_position = 0.08  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.2998", y_position = 0.15  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment

```




```{r, echo=F, warning=F, message=F}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1031_2022.csv"))

sort_BF=BF_para_all_est %>% arrange(desc(BF))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 MIRAGE Gene",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.35))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by MIRAGE")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.016", y_position = 0.11  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="0.259", y_position = 0.11  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.210", y_position = 0.11  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.009", y_position = 0.31  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.045,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.071", y_position = 0.16,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.144", y_position = 0.21,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment


```



## 102 ASD genes in cell paper

```{r, echo=F}
ASD_Gene_Cell=as_tibble(read.csv("../../../AutismDatafromASC_Cellpaper/1-s2.0-S0092867419313984-mmc2.csv", header = TRUE))
ASD_102_Gene=ASD_Gene_Cell[1:102,1] %>% pull()
```

```{r, echo=F, eval=F}
#########
para_est=matrix(nrow=6, ncol=6)
rownames(para_est)=c("PLI>=0.995", "pLI>=0.5&pLI<0.995", "pLI<0.5", "MPC>=2", "MPC>=1&MPC<2", "MPC<1")
colnames(para_est)=c("No_case", "No_contr", "OR", "pval", "eta_est", "eta_pval")
for (i in 1:6)
{ 
#i=2  
#cat(i, "is running", "\n")
############  variant set 
variant_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% ASD_102_Gene)%>% filter(Group_Index==i) %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
para_est[i,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
pois.test=poisson.test(para_est[i,1:2], c(num.family, num.family), r=1, alternative="greater")
para_est[i,3:4]=c(pois.test$estimate,pois.test$p.value)
vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=6)
para_est[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
}
as_tibble(para_est)
#write.csv(para_est, file="../output/kyleData/para.est.var.group.proband.102ASDGene_0830_2022.csv")
#gene.data=eight.partition(cand.data)
#ds.asd.result=mirage(gene.data, n1=2403, n2=2403)
```

### variant set (VS) analysis by MIRAGE_VS

This figure shows the comparison between burden test (one side two sample poisson test) and mirage-vs for proband in 6 variant groups. $\gamma=6$ for 3 PLI variant groups and $\gamma=3$ for 3 MPC groups when running mirage_vs.

```{r, echo=F}
proband_para_est=read.csv("../output/kyleData/para.est.var.group.proband.102ASDGene_0830_2022.csv")
as_tibble(proband_para_est)


var_group=c(rep(c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1"), 2))
method=c(rep("Burden", 6), rep("mirage-vs", 6))
pvalue=c(round(-log(proband_para_est$pval, base=10), 2), round(-log(proband_para_est$eta_pval, base=10), 2))
summary=data.frame(var.group=var_group, pvalue=pvalue, method=method)
ggplot(summary, aes(x=var.group, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", 
           position=position_dodge())+
  ylab(expression(-log[10](pvalue)))+
  theme(legend.title=element_blank())+
  xlab(" variant groups")+
  ggtitle(" p value comparison between Burden and MIRAGE-VS")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))
  #ylab(expression(paste(hat(eta))))+ 
  #guides(fill=guide_legend(title=NULL))+
  #ylim(0.05,0.15)+
  # geom_hline(yintercept=0.1,linetype="dashed", color = "red")
```


### gene set analysis by MIRAGE

* $\gamma=6$ for LoF variant and $\gamma=3$ for missense when running mirage.

*  filter `Gnomad_non_neuro_AF<0.05` 

* `isSyn=="FALSE"`

* `isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`


```{r,echo=F, eval=F}
############  gene set 
gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05)%>% filter(Gene %in% ASD_102_Gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
#mirage_result=mirage(data.frame(gene_set),n1=5869, n2=5869, gamma=c(6,6,6,3,3,3))
mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family)# input data must be dataframe
#res=mirage(mirage_toy,n1=4315,n2=4315) # test mirage function 
#write.csv(mirage_result$BF.PP.gene, file="../output/kyleData/BF.proband.102Gene_0830_2022.csv")
```

| variant category    | parameter |   estimate   | p value |
| :---        |    :----:   |         :----: |   ---: |
\ |$\hat{\delta}$ | 0.9998 | $0.0295$
PLI>=0.995| $\hat{\eta}_1$ | 0.4321    | $0.2334$
0.5<=pLI<0.995 | $\hat{\eta}_2$ | 1  | 0.0056
pLI<0.5 | $\hat{\eta}_3$ | 0.5156    | 0.2508
MPC>=2 | $\hat{\eta}_4$ | 0.1176    | 0.1685
1<=MPC<2 | $\hat{\eta}_5$ | 0.0539    | 0.0729
0<=MPC<1 | $\hat{\eta}_6$ | $7.814873e-09$    | 1
Table: Parameter estimate by MIRAGE for 102 ASD genes. 



```{r, echo=F}
var_group=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1")
estimate=c(0.4321, 1, 0.5156, 0.1176, 0.0539, 0)
summary=data.frame(var.group=var_group, estimate=estimate)
ggplot(summary, aes(x=var.group, y=estimate, fill=var.group)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("variant groups")+ylab(expression(paste(hat(eta))))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))
```



## Top 5% constraint genes 


Use below filters and stated otherwise 

*  filter `Gnomad_non_neuro_AF<0.05` 

* `isSyn=="FALSE"`

* `isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`

* `filter(!str_detect(Consequence, 'intron_variant'))`  exclude intron 



### One variant set: constrant genes+ AF<10E-4 

* In old sample, we observe signal in constraint genes with $AF<10^{-4}$. 

```{r, echo=F, warning=F, message=F, cache=T}
Constraint_Gene=as.character(read.csv("../data/GeneSet/Samocha_2014NG_contraintgene.csv", header=T)$gene)
new_sample_trans_data_Constraint_Gene=ASC_new_sample_GI%>% filter(Gene %in% Constraint_Gene & Gnomad_non_neuro_AF<0.0001)%>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband)%>%drop_na()
```

* in new sample, there are 99333 transmitted variants and  94575 untransmitted variants. 

```{r, echo=F}
#poisson.test(c(sum(new_sample_trans_data_Constraint_Gene$Transmitted_proband), sum(new_sample_trans_data_Constraint_Gene$Untransmitted_proband)), c(num.family, num.family), r=1, alternative="greater")

fisher_test=fisher.test(matrix(c(sum(new_sample_trans_data_Constraint_Gene$Transmitted_proband), num.family, sum(new_sample_trans_data_Constraint_Gene$Untransmitted_proband), num.family), nrow=2))

fisher_test_result_all=tibble(OR=fisher_test$estimate, lower=fisher_test$conf.int[1], upper=fisher_test$conf.int[2], var="Constraint_gene_AF<0.0001")

g1=ggplot(fisher_test_result_all, aes(x=var, y=OR, fill=var)) + # draw the OR plot 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_hline(yintercept = 1,linetype="dashed", 
                color = "red", size=1)+
  ggtitle("")+
  theme(legend.position = "none")+  # no legend 
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title
  xlab("")+
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2,
                 position=position_dodge(.9)) 

```


* Put all variants with AF<10E-4 in one category and run MIRAGE_VS. $\widehat{\eta}=0.0882$, p value is $2.585515e-34$.  

```{r, echo=F, eval=F}
################### DO NOT RUN ##################
Constraint_Gene_for_VS=new_sample_trans_data_Constraint_Gene%>% select(Transmitted_proband, Untransmitted_proband)%>%add_column(Group_Index=1)%>% drop_na()
########### here it takes about 5 mins to run mirage-VS 
vs.mirage=mirage_vs(data.frame(Constraint_Gene_for_VS),n1=num.family,n2=num.family)
```



```{r, echo=F, message=F, warning=F}
method=c("MIRAGE_VS", "Burden")
pvalue=c(2.585515e-34, 0.004349)
summary=data.frame(method=method, pvalue=-log(pvalue, base=10))
g2=ggplot(summary, aes(x=method, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("method")+ylab(expression(-log[10](pvalue)))+
  ggtitle("")+
 #  theme(legend.position = "none")+  # no legend
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title 
  theme(axis.text.x = element_text(angle = 0, vjust = 0.5, size=10))


figure=ggarrange(g1, g2, common.legend = F, legend="bottom") 
annotate_figure(figure,
                top = text_grob("Variant set: constraint gene+AF<0.0001", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)
```

### Run MIRAGE_VS on  6 variant groups in cell paper 

```{r, echo=F, eval=F}
#########  DO NOT RUN #########################
Constraint_Gene=as.character(read.csv("../data/GeneSet/Samocha_2014NG_contraintgene.csv", header=T)$gene)
para_est=matrix(nrow=6, ncol=6)
rownames(para_est)=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1")
colnames(para_est)=c("No_case", "No_contr", "OR", "pval", "eta_est", "eta_pval")
for (i in 1:6)
{ 
#i=2  
#cat(i, "is running", "\n")
############  variant set 
variant_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% Constraint_Gene)%>% filter(Group_Index==i) %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

para_est[i,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
pois.test=poisson.test(para_est[i,1:2], c(num.family, num.family), r=1, alternative="greater")
para_est[i,3:4]=c(pois.test$estimate,pois.test$p.value)
vs.mirage=mirage_vs(variant_set[,2:4],n1=num.family,n2=num.family, gamma=6)
para_est[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
}
as_tibble(para_est)
#write.csv(para_est, file="../output/KyleData/para.est.var.group.proband.Constraint.Gene_0830_2022.csv")
#gene.data=eight.partition(cand.data)
#ds.asd.result=mirage(gene.data, n1=2403, n2=2403)
```



* $\gamma=6$ for 3 PLI variant groups and $\gamma=3$ for 3 MPC groups when running mirage_vs.




```{r, echo=F}
proband_para_est=read.csv("../output/KyleData/para.est.var.group.proband.Constraint.Gene_0830_2022.csv")
#as_tibble(proband_para_est) # list the results 
################## plot parameter estimate eta
var_group=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1")
estimate=proband_para_est$eta_est
summary=data.frame(var.group=var_group, estimate=estimate)
ggplot(summary, aes(x=var.group, y=estimate, fill=var.group)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("variant groups")+ylab(expression(paste(hat(eta))))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))

################## burden test for 6 variant groups 
Constraint_Gene=as.character(read.csv("../data/GeneSet/Samocha_2014NG_contraintgene.csv", header=T)$gene)
OR=numeric(); OR_lower=numeric(); OR_upper=numeric()
for (i in 1:6)
{ 
#i=2  
#cat(i, "is running", "\n")
############  variant set 
variant_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% Constraint_Gene)%>% filter(Group_Index==i) %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
fisher_test=fisher.test(matrix(c(sum(variant_set$Transmitted_proband), num.family, sum(variant_set$Untransmitted_proband), num.family), nrow=2))
OR[i]=fisher_test$estimate
OR_lower[i]=fisher_test$conf.int[1]
OR_upper[i]=fisher_test$conf.int[2]
}

fisher_test_result_all=tibble(OR=OR, lower=OR_lower, upper=OR_upper, var=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1"))

g1=ggplot(fisher_test_result_all, aes(x=var, y=OR, fill=var)) + # draw the OR plot 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_hline(yintercept = 1,linetype="dashed", 
                color = "red", size=1)+
  ggtitle("")+
 # theme(legend.position = "none")+  # no legend 
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title
  xlab("")+
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2,
                 position=position_dodge(.9))+
  theme(axis.title.x=element_blank(),   # remove all of x axis labels 
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+
  theme(legend.text=element_text(size=6)) # control text size in legend 
##################
var_group=c(rep(c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1"), 2))
method=c(rep("Burden", 6), rep("mirage-vs", 6))
pvalue=c(round(-log(proband_para_est$pval, base=10), 2), round(-log(proband_para_est$eta_pval, base=10), 2))
summary=data.frame(var.group=var_group, pvalue=pvalue, method=method)
g2=ggplot(summary, aes(x=var.group, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", 
           position=position_dodge())+
  ylab(expression(-log[10](pvalue)))+
  theme(legend.title=element_blank())+
  xlab(" variant groups")+
  ggtitle("p value: Burden vs MIRAGE_VS")+
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title 
  theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=6))
  #ylab(expression(paste(hat(eta))))+ 
  #guides(fill=guide_legend(title=NULL))+
  #ylim(0.05,0.15)+
  # geom_hline(yintercept=0.1,linetype="dashed", color = "red")


figure=ggarrange(g1, g2, common.legend = F, legend="bottom") 
annotate_figure(figure,
                top = text_grob("", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)
```

### gene set analysis by MIRAGE

* $\gamma=6$ for PLI variant groups and $\gamma=3$ for MPC groups when running mirage.

* 836 genes in total 

```{r,echo=F, eval=F}
### !!!!!!!!!!!!!!!!! DO NOT RUN: it takes about ~20 min  to run; not really, heavily depends on the seeds  ####### 
############  gene set 
gene_set=ASC_new_sample_GI %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% Constraint_Gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family, gamma=c(6,6,6,3,3,3))
######## this case takes about 2 hours to finish running 
#set.seed(Sys.time()) # set a unique random seed 
#mirage_result1=mirage(data.frame(gene_set),n1=num.family, n2=num.family)# input data must be dataframe
#write.csv(mirage_result$BF.PP.gene, file="../output/KyleData/BF.proband.ConstraintGene_0830_2022.csv")
#saveRDS(mirage_result1$BF.all, file="../output/KyleData/BF.proband.ConstraintGene.BF.all.rds") #this saves Bayes factor for all variants in each gene, no need to re-run the program 

####### this case takes close to 2 hours to finish running 
#set.seed(Sys.time()) # set a unique random seed 
#mirage_result2=mirage(data.frame(gene_set),n1=num.family, n2=num.family,gamma=c(6,6,6,3,3,3), sigma=2, eta.init=0.01, delta.init=0.01)

#############  this case takes a couple of minutes 
#set.seed(Sys.time()) # set a unique random seed 
#mirage_result3=mirage(data.frame(gene_set),n1=num.family, n2=num.family,gamma=c(6,6,6,3,3,3), sigma=2, eta.init=0.0001, delta.init=0.0001)



############# this case takes about 1.5 hour
#set.seed(Sys.time()) # set a unique random seed 
#mirage_result4=mirage(data.frame(gene_set),n1=num.family, n2=num.family,gamma=c(6,6,6,3,3,3), sigma=2, eta.init=0.5, delta.init=0.5)
```


| variant category    | parameter |   MIRAGE: estimate   | MIRAGE:p value | Trans  | non-Trans  | OR  | MIRAGE_VS: estimate | MIRAGE_VS: p value
| :---        |    :----:   |         :----: |            :----:          |  :----:  |  :----:  | :----:          | :----:          | ---: |
\ |$\hat{\delta}$ | 0.9832 | $2.014043e-06$ 
(1):PLI>=0.995|$\hat{\eta}_1$ | 0.2036    | 0.0150 | 225 |     178 |  1.26 | 0.200  |  0.0150 
(2):0.5<=pLI<0.995|$\hat{\eta}_2$ | $1.118248e-35$    | 1 | 174  |    182 | 0.956 | 0.000342 | 1
(3):pLI<0.5|$\hat{\eta}_3$ | 0.2145    | 0.0011 | 854 |     775 |  1.10 | 0.212 |    0.00105
(4):MPC>=2|$\hat{\eta}_4$ | 0.1115    | 0.0000152777 | 7968 |    7638 | 1.04  | 0.0730  | 0.0000137
(5):1<=MPC<2| $\hat{\eta}_5$ | 0.0175    | 0.0559135877 | 48412 |   47060 |  1.03 |0.00926 | 0.0918
(6):0<=MPC<1| $\hat{\eta}_6$ | 2.259378e-06    | 1 | 146671  | 145348 | 1.01 |0.000871 | 1
Table: Summary of top 5% constraint genes 




```{r, echo=F, message=F, warning=F}

################ plot parameter estimate: eta
var_group=c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1")
estimate=c(0.2036, 0, 0.2145, 0.1115, 0.0175, 0)
summary=data.frame(var.group=var_group, estimate=estimate)
ggplot(summary, aes(x=var.group, y=estimate, fill=var.group)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("variant groups")+ylab(expression(paste(hat(eta))))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))

################  
var_group=c(rep(c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1"), 2))
method=c(rep("Burden", 6), rep("mirage-vs", 6))
pvalue=c(round(-log(proband_para_est$pval, base=10), 2), round(-log(c(0.0150, 1, 0.0011, 0.0000152777, 0.0559135877, 1), base=10),2))
summary=data.frame(var.group=var_group, pvalue=pvalue, method=method)
g3=ggplot(summary, aes(x=var.group, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", 
           position=position_dodge())+
  ylab(expression(-log[10](pvalue)))+
  theme(legend.title=element_blank())+
  xlab(" variant groups")+
  ggtitle("p value: Burden vs MIRAGE")+
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title 
  theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=6))
  #ylab(expression(paste(hat(eta))))+ 
  #guides(fill=guide_legend(title=NULL))+
  #ylim(0.05,0.15)+
  # geom_hline(yintercept=0.1,linetype="dashed", color = "red")


figure=ggarrange(g1, g3, common.legend = F, legend="bottom") 
annotate_figure(figure,
                top = text_grob("", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)


```


#### Top genes with burden 

*  top genes with large BF don't have heavy burden. 


```{r, echo=F, message=F, warning=F}
BF_all=readRDS("../output/KyleData/BF.proband.ConstraintGene.BF.all.rds")
BF_proband_ConstraintGene=as_tibble(read.csv("../output/kyleData/BF.proband.ConstraintGene_0830_2022.csv"))
BF_proband_ConstraintGene_sorted=BF_proband_ConstraintGene%>%arrange(desc(post.prob))
num_gene=100
topgene=BF_proband_ConstraintGene_sorted$Gene[1:num_gene]
Transmitted_proband=numeric(); Untransmitted_proband=numeric(); OR=numeric()
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

SFARI_score=rep("NA", num_gene)
for (i in 1:length(topgene))
{
Transmitted_proband[i]=sum(ASC_new_sample_GI%>%filter(Gene==topgene[i] & Gnomad_non_neuro_AF<=0.5)%>%select(Transmitted_proband)%>%pull())
Untransmitted_proband[i]=sum(ASC_new_sample_GI%>%filter(Gene==topgene[i] & Gnomad_non_neuro_AF<=0.5)%>%select(Untransmitted_proband)%>%pull())
burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==topgene[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==topgene[i]),]$gene.score

}
topgene_summary=tibble(Gene=topgene, BF=round(BF_proband_ConstraintGene_sorted$BF[1:num_gene],4), Transmitted_proband=Transmitted_proband, Untransmitted_proband=Untransmitted_proband, OR=round(OR,4), SFARI_score=SFARI_score)
topgene_summary%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

* [KIAA1109](https://www.nature.com/articles/s41398-021-01578-2.pdf?proof=t)

* [EIF3I](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5785978/pdf/krnb-15-01-1391437.pdf)

* [CTNNA2](https://gene.sfari.org/database/human-gene/CTNNA2)



#### Vairants driving the signal in top gene 

```{r, echo=F, eval=F}
intergrand=function(aa, var.case, var.contr, bar.gamma, sig, N1, N0)
{
  ff=dbinom(var.case, sum(var.case, var.contr), aa*N1/(aa*N1+N0))*dgamma(aa, bar.gamma*sig, sig)
  return(ff)
}

# calculate the bayes factor of a single variant via integration
BF.var.inte=function(var.case, var.contr, bar.gamma, sig, N1, N0)
{
  marglik0.CC <- dbinom(var.case, sum(var.case, var.contr), N1/(N1+N0))    # Under H0: gamma=1
  
  marglik1.CC <- integrate(intergrand, var.case, var.contr, bar.gamma, sig, N1, N0, low=0, upper=100, stop.on.error=F)$value # Under H1: gamma~gamma(gamma.mean*sigma, sigma)
  BF.var <- marglik1.CC/marglik0.CC
  
  return(BF.var)
}

#> BF.var.inte(15, 219, 3, 3, num.family, num.family)
#[1] 6.321461e+36


```




### use fixed $\widehat{\eta}$ to compute BF 


```{r, echo=F, message=F, warning=F, eval=F}

beta=c(2.402658e-01,  0.000000e+00,  1.514065e-01,  0.000000e+00,  1.063114e-02,  0.000000e+00,  1.505522e-01, 1.185758e-322, 1.976263e-323,  8.274313e-02, 7.410985e-323,  1.048833e-02,  4.259534e-02,  2.777784e-04,  5.524852e-04)
new.data=ASC_new_sample_GI2 %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% gene_to_display) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
colnames(new.data)[3:5]=c("No.case", "No.contr", "group.index")
result=fixed.beta.func(new.data, N1=7291, N0=7291, gamma.mean=c(rep(6,6),rep(3,9)),sigma=2, beta=beta) # use fixed beta to calculate Bayes factors

############## verify the calculations 
gene_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="SSC5D")]]
group.index=gene_variant_BF$group.index
gene_BF=prod(1-beta[group.index]+beta[group.index]*gene_variant_BF$var.BF)
gene_BF
result$BayesFactor[result$BayesFactor$Gene=="SSC5D",]
##############

#write.csv(result$BayesFactor, file="../output/KyleData/BF.with.fixed.parameter.Constraint.Gene_1011_2022.csv")
```



```{r, echo=F, message=F, warning=F}
BayesFactor=read.csv("../output/KyleData/BF.with.fixed.parameter.Constraint.Gene_1011_2022.csv", header=T)
constraint_gene_BF=BayesFactor[order(BayesFactor$BF, decreasing = TRUE),]
constraint_gene_BF$BF=round(constraint_gene_BF$BF,4)
constraint_gene_BF[,-1]%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```
 
* $\eta's$ are fixed at values in old results, i.e.  $\eta_1 (PTV, pLI>0.995)=0.41, \eta_4 (MPC>2)=0.16$, all others are set 0, 
  