---
title: "Analysis for 6 variant groups"
output: html_document
date: '2022-11-23'
---

```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
library(DT)
library(poolr) # use fisher method to combine p values 
library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
```

```{r, echo=F, message=F, warning=F}
########## use fixed parameter estimate to calculate BF, this is very useful because it doesn't need to re-run analysis to BF of every gene which takes a long time and large space to store that information 
intergrand=function(aa, var.case, var.contr, bar.gamma, sig, N1, N0)
{
  ff=dbinom(var.case, sum(var.case, var.contr), aa*N1/(aa*N1+N0))*dgamma(aa, bar.gamma*sig, sig)
  return(ff)
}
# calculate the bayes factor of a single variant via integration
BF.var.inte=function(var.case, var.contr, bar.gamma, sig, N1, N0)
{
  marglik0.CC <- dbinom(var.case, sum(var.case, var.contr), N1/(N1+N0))    # Under H0: gamma=1

  marglik1.CC <- integrate(intergrand, var.case, var.contr, bar.gamma, sig, N1, N0, low=0, upper=100, stop.on.error=F)$value # Under H1: gamma~gamma(gamma.mean*sigma, sigma)
  BF.var <- marglik1.CC/marglik0.CC

  return(BF.var)
}

fixed.beta.func=function(new.data, N1, N0, gamma.mean, sigma, beta) # new.data has one column specifying its group index
{
  #######################
  beta.k=beta
  full.info.genevar=list()
  gene.list=new.data$Gene; unique.gene=unique(gene.list) # find the gene list
  num.gene=length(unique.gene)
  BF.gene=numeric()
  ########################
  for (i in 1:num.gene)
  {
#    cat(i, "th gene of ", "\t", num.gene, "\t", "is running", "\n")
    var.index.list=which(gene.list==unique.gene[i])
    indi.gene=new.data[var.index.list,]   # note var.index.list matches new.data
    bb=1; var.BF=numeric()
    if (length(var.index.list)>0) # calculate Bayes factor for variant (i,j)
      for (j in 1:length(var.index.list))
      {
        if (new.data$group.index[var.index.list[j]]<=3)   # note here bar.gamma is group specific, attention what bar.gamma goes to which group 
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=6, sig=sigma, N1, N0)
        if (new.data$group.index[var.index.list[j]]>3)
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=3, sig=sigma, N1, N0)
        bb=bb*((1-beta.k[new.data$group.index[var.index.list[j]]])+beta.k[new.data$group.index[var.index.list[j]]]*var.BF[j])
      }
    full.info.genevar[[i]]=cbind(indi.gene, var.BF)
    BF.gene[i]=bb
  }
  
  return(result=list(BayesFactor=data.frame(Gene=unique.gene, BF=BF.gene), full.info=full.info.genevar))
}
```



```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
path="C:\\Shengtong\\research\\rare-var\\"
ASC_new_sample3=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), sep='\t',  header=T, fill=T)) # this is the correct way that Kyle read the data
ASC_new_sample3=ASC_new_sample3 %>% filter(isSyn=="FALSE") # filter syn variants first 
```


```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
path="C:\\Shengtong\\Research\\rare-var\\"
GeneDB=src_sqlite(path=paste(path, "gene.list.db", sep=""), create=F)
#GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
###### ID gene ######
ID_Gene=as.character(data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))$GeneID)
#### high confidence gene 
High_Confidence_Gene=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
#### moderate confidence gene 
Moderate_Confidence_Gene=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
#### psd gene 
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
PSD_Gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
#### FMRP gene 
FMRP_Gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
#### Autism KB gene 
ASD_KB_Gene=as.character(data.frame(collect(tbl(GeneDB, "AutismKB_gene")))$GeneID)
#### constraint gene 
Constraint_Gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
### RVIS gene 
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS_Gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene
#### Haplo insufficient gene 
Haploinsufficient_Gene=as.character(data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))$GeneID)
#### SCZ data 
SCZ_Gene=purcell.genelist$Gene_symbol
Gene_Set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff", "SCZ gene")
Gene_Set_List=list()
Gene_Set_List[[1]]=ID_Gene
Gene_Set_List[[2]]=High_Confidence_Gene
Gene_Set_List[[3]]=Moderate_Confidence_Gene
Gene_Set_List[[4]]=PSD_Gene
Gene_Set_List[[5]]=FMRP_Gene
Gene_Set_List[[6]]=ASD_KB_Gene
Gene_Set_List[[7]]=Constraint_Gene
Gene_Set_List[[8]]=RVIS_Gene
Gene_Set_List[[9]]=Haploinsufficient_Gene
Gene_Set_List[[10]]=SCZ_Gene
```


```{r, echo=F, warning=F, message=F}

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
######### use ASC_new_sample3 rather than ASC_new_sample2
Transmitted_proband=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) # extract transmitted variants for proband 

Untransmitted_proband=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) # extract untransmitted variants for proband 

Transmitted_sibling=ASC_new_sample3%>% select(starts_with("t_sibling"))%>%  mutate(Transmitted_sibling = select(., t_sibling_male_dad:t_sibling_female_ind) %>% rowSums())%>% select(Transmitted_sibling) # extract transmitted variants for sibling 

Untransmitted_sibling=ASC_new_sample3%>% select(starts_with("u_sibling"))%>%  mutate(Untransmitted_sibling = select(., u_sibling_male_dad:u_sibling_female_ind) %>% rowSums())%>% select(Untransmitted_sibling) # extract untransmitted variants for sibling

##### add new columns of total transmitted and non-transmitted variants 
#ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & VQSLOD<9) # exclude indel & VQSLOD filter 

################################################


ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8)%>%filter(!str_detect(Consequence, 'intron_variant'))# exclude indel & apply other filters for SNP's 

##################
#AF_cutoff=0.001
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Transmitted_proband) %>% pull)   
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Untransmitted_proband) %>% pull)   
########### 1164 vs 1413 not 695 vs 557 in cell paper 


# add one column of variant group index 
ASC_new_sample_GI=ASC_new_sample %>% add_column(Group_Index=NA)

ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.995 & ASC_new_sample_GI$isPTV==T)]=1
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.5 & ASC_new_sample_GI$pLI<0.995 & ASC_new_sample_GI$isPTV==T)]=2
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI<0.5 & ASC_new_sample_GI$isPTV==T)]=3
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=2)]=4
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=1 & ASC_new_sample_GI$MPC<2)]=5
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC<1)]=6

num.family=7291
## test 
```

## two side fisher exact test 


### QQ plot of p values 

```{r, echo=F, message=F, warning=F, eval=F}
################# calculate burden for each gene 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
interest_gene=BF_proband_AllGene_sorted$Gene  # use all genes 
new_data=ASC_new_sample_GI %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

LoF_burden_005=matrix(nrow=length(interest_gene), ncol=4)
Missense_burden_005=matrix(nrow=length(interest_gene), ncol=4)
LoF_Missense_burden_005=matrix(nrow=length(interest_gene), ncol=4)
p_value_005=matrix(nrow=length(interest_gene), ncol=3)

LoF_burden_001=matrix(nrow=length(interest_gene), ncol=4)
Missense_burden_001=matrix(nrow=length(interest_gene), ncol=4)
LoF_Missense_burden_001=matrix(nrow=length(interest_gene), ncol=4)
p_value_001=matrix(nrow=length(interest_gene), ncol=3)
threshold=1e-100

for (i in 1:length(interest_gene))
{
 # i=1143
  cat(i, "th gene is running", "\n")
  single_gene_data=new_data %>% filter(Gene==interest_gene[i])
  
  ########## LoF variants 
  LoF_burden_005[i,1]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.05 & Group_Index<4) %>% select(Transmitted_proband)%>% pull)  
  LoF_burden_005[i,2]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.05 & Group_Index<4) %>% select(Untransmitted_proband)%>% pull)
  burden_test=fisher.test(matrix(c(LoF_burden_005[i,1],num.family, LoF_burden_005[i,2], num.family), nrow=2)) # two sides burden test 
  binom_test=binom.test(c(LoF_burden_005[i,1], LoF_burden_005[i,2]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
  LoF_burden_005[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  LoF_burden_005[i,4]=binom_test$p.value
  p_value_005[i,1]=LoF_burden_005[i,4]
  
  ########## missense variants 
  Missense_burden_005[i,1]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.05 & Group_Index==4) %>% select(Transmitted_proband)%>% pull)  
  Missense_burden_005[i,2]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.05 & Group_Index==4) %>% select(Untransmitted_proband)%>% pull)
  burden_test=fisher.test(matrix(c(Missense_burden_005[i,1],num.family, Missense_burden_005[i,2], num.family), nrow=2))
  binom_test=binom.test(c(Missense_burden_005[i,1], Missense_burden_005[i,2]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT; one side burden test 
  Missense_burden_005[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  Missense_burden_005[i,4]=binom_test$p.value
  ############## avoid p value of 0 or 1 
  if (LoF_burden_005[i,4]==0)
     LoF_burden_005[i,4]=threshold
  if (LoF_burden_005[i,4]>0.9999)
     LoF_burden_005[i,4]=1-threshold
  if (Missense_burden_005[i,4]==0)
     Missense_burden_005[i,4]=threshold
  if (Missense_burden_005[i,4]>0.9999)
     Missense_burden_005[i,4]=1-threshold
    p_value_005[i,3]=fisher(c(LoF_burden_005[i,4], Missense_burden_005[i,4]))$p
  
  ########## LoF+missense variants 
  LoF_Missense_burden_005[i,1]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.05 & Group_Index<=4) %>% select(Transmitted_proband)%>% pull)  
  LoF_Missense_burden_005[i,2]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.05 & Group_Index<=4) %>% select(Untransmitted_proband)%>% pull)
  burden_test=fisher.test(matrix(c(LoF_Missense_burden_005[i,1],num.family, LoF_Missense_burden_005[i,2], num.family), nrow=2))
  binom_test=binom.test(c(LoF_Missense_burden_005[i,1], LoF_Missense_burden_005[i,2]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
  LoF_Missense_burden_005[i,3]=burden_test$estimate
  #LoF_Missense_burden_005[i,4]=burden_test$p.value
  LoF_Missense_burden_005[i,4]=binom_test$p.value
  p_value_005[i,2]=LoF_Missense_burden_005[i,4]
  
  ############################################################
  ########## LoF variants 
  LoF_burden_001[i,1]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.01 & Group_Index<4) %>% select(Transmitted_proband)%>% pull)  
  LoF_burden_001[i,2]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.01 & Group_Index<4) %>% select(Untransmitted_proband)%>% pull)
  burden_test=fisher.test(matrix(c(LoF_burden_001[i,1],num.family, LoF_burden_001[i,2], num.family), nrow=2))
  binom_test=binom.test(c(LoF_burden_001[i,1], LoF_burden_001[i,2]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
  LoF_burden_001[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  LoF_burden_001[i,4]=binom_test$p.value
  p_value_001[i,1]=LoF_burden_001[i,4]
  
  ########## missense variants 
  Missense_burden_001[i,1]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.01 & Group_Index==4) %>% select(Transmitted_proband)%>% pull)  # Group_Index==4: only use the first MPC group
  Missense_burden_001[i,2]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.01 & Group_Index==4) %>% select(Untransmitted_proband)%>% pull)
  burden_test=fisher.test(matrix(c(Missense_burden_001[i,1],num.family, Missense_burden_001[i,2], num.family), nrow=2))
  binom_test=binom.test(c(Missense_burden_001[i,1], Missense_burden_001[i,2]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
  Missense_burden_001[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  Missense_burden_001[i,4]=binom_test$p.value
   ############## avoid p value of 0 or 1 
  if (LoF_burden_001[i,4]==0)
     LoF_burden_001[i,4]=threshold
  if (LoF_burden_001[i,4]>0.9999)
     LoF_burden_001[i,4]=1-threshold
  if (Missense_burden_001[i,4]==0)
     Missense_burden_001[i,4]=threshold
  if (Missense_burden_001[i,4]>0.9999)
     Missense_burden_001[i,4]=1-threshold
    p_value_001[i,3]=fisher(c(LoF_burden_001[i,4], Missense_burden_001[i,4]))$p
  
  ########## LoF+missense variants 
  LoF_Missense_burden_001[i,1]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.01 & Group_Index<=4) %>% select(Transmitted_proband)%>% pull)  
  LoF_Missense_burden_001[i,2]=sum(single_gene_data %>% filter(Gnomad_non_neuro_AF<0.01 & Group_Index<=4) %>% select(Untransmitted_proband)%>% pull)
  burden_test=fisher.test(matrix(c(LoF_Missense_burden_001[i,1],num.family, LoF_Missense_burden_001[i,2], num.family), nrow=2))
  binom_test=binom.test(c(LoF_Missense_burden_001[i,1], LoF_Missense_burden_001[i,2]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
  LoF_Missense_burden_001[i,3]=burden_test$estimate
  #LoF_burden_005[i,4]=burden_test$p.value
  LoF_Missense_burden_001[i,4]=binom_test$p.value
  p_value_001[i,2]=LoF_Missense_burden_001[i,4]
  ################
  
} # end of for (i in 1:length(interest_gene))

################
  rownames(p_value_005)=interest_gene
  colnames(p_value_005)=c("LoF_pvalue", "All_variants_pvalue", "fisher_combine_pvalue")
  
  combine_005=cbind(LoF_burden_005, Missense_burden_005, LoF_Missense_burden_005)
  rownames(combine_005)=interest_gene
  colnames(combine_005)=paste(rep(c("LoF", "Missense", "All_variants"), each=4), c("T", "UT", "OR", "P_value"), sep="_")
 # write.csv(p_value_005, file="../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.05_1208_2022.csv")
#  write.csv(combine_005, file="../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_binom_oneside_pvalues_MPC_larger_than2_AF0.05_1208_2022.csv")

################
  rownames(p_value_001)=interest_gene
  colnames(p_value_001)=c("LoF_pvalue", "All_variants_pvalue", "fisher_combine_pvalue")
  
  combine_001=cbind(LoF_burden_001, Missense_burden_001, LoF_Missense_burden_001)
  rownames(combine_001)=interest_gene
  colnames(combine_001)=paste(rep(c("LoF", "Missense", "All_variants"), each=4), c("T", "UT", "OR", "P_value"), sep="_")
 # write.csv(p_value_001, file="../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv")
 # write.csv(combine_001, file="../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv")


```



```{r, echo=F, message=F, warning=F}
plotQQ.unif <- function(p.obs) {
  obs <- (p.obs)
  theo <- (ppoints(length(obs)))
  qqplot(-log(theo, base=10), -log(obs, base=10), xlab=expression(paste("Theoretical ",-log[10], "(p-values)")), ylab=expression(paste("Observed ", -log[10], "(p-values)")), main="A")
  abline(0,1,col='red')
}

```



```{r, echo=F, message=F, warning=F}
burden_pvalues_005=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_pvalues_AF0.05_1123_2022.csv"))
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_pvalues_AF0.01_1123_2022.csv"))
combine_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_AF0.01_1123_2022.csv"))

#######################   John storey's q value 
#### install qvalue package 
#install.packages("devtools")
#library("devtools")
#install_github("jdstorey/qvalue")   https://github.com/StoreyLab/qvalue

#library(qvalue)
#qobj.burden=qvalue(Burden.pvalue)
#localFDR.burden <- qobj.burden$lfdr
#min(localFDR.burden, na.rm=T)

#qobj.SKATO=qvalue(SKATO.pvalue)
#localFDR.SKATO <- qobj.SKATO$lfdr
#min(localFDR.SKATO, na.rm=T)
#######################


#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_othermethod_ASD_topconstgene.pdf")
#par(mfrow=c(1,2))
theo <- (ppoints(length(burden_pvalues_001$LoF_pvalue)))
  #qqplot(-log(theo, base=10), -log(Burden.pvalue, base=10), xlab=expression(paste("Theoretical ",-log[10], "(p-values)")), ylab=expression(paste("Observed ", -log[10], "(p-values)")), main="(A) Burden", frame=F, cex.lab=0.8, pch=20)
  #abline(0,1,col='red', lwd=2)

#################  use ggplot draw qq plot 
sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$LoF_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

burden=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF=ggplot(burden) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("LoF")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
 #qqplot_burden_LoF

#################
theo <- (ppoints(length(burden_pvalues_001$LoF.Missense_pvalue)))

 sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$LoF.Missense_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

 LoF.Missense=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF.Missense=ggplot(LoF.Missense) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   #ylab("")+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("LoF+Missense")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_LoF.Missense
 
 
 ################################
 theo <- (ppoints(length(burden_pvalues_001$fisher_combine_pvalue)))

 sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$fisher_combine_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

 fisher.combine=tibble(sx=sx, sy=sy)
 qqplot_burden_fisher.combine=ggplot(fisher.combine) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   #ylab("")+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("Fisher combine LoF+Missense")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_fisher.combine

#png("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_ASD_topconstgene_enrichemnt.png") 
figure=ggarrange(qqplot_burden_LoF, qqplot_burden_LoF.Missense,  qqplot_burden_fisher.combine, nrow=1)
annotate_figure(figure,
                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)

#dev.off()
```


* extremely small p values are from variants with overwhelmingly more variants in controls than in cases. 

* patterns for p values of all variants is similar to p values by fisher method that combines LoF p values and missense p values 


### Enrichment of top genes 

```{r, echo=F, message=F, warning=F}
gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()


burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(LoF.Missense_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_test$p.value

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), burden_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```

* `burden_pvalue` is the p value for all variants including LoF and missense.  


```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_pvalues_AF0.01_1123_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(LoF.Missense_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by burden p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.15))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by Burden")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=1.0000", y_position = 0.02  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=0.2059", y_position = 0.06  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.2998", y_position = 0.11  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment
```


* strong ASD: SFARI score 1 or 2

* plausible ASD: SFARI score 3

* DNM: TADA denovo genes 

* top 20 genes are selected by small p values, but may with burden in opposite direction. 


```{r, echo=F, warning=F, message=F}

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_pvalues_AF0.01_1123_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(LoF.Missense_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by burden p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.2))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by burden")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.169", y_position = 0.06  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="0.259", y_position = 0.105  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.565", y_position = 0.055  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="1.000", y_position = 0.08  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.01,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.04,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.595", y_position = 0.06,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.712", y_position = 0.11,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment


```


## one side binom test 


```{r, echo=F, message=F, warning=F}
burden_pvalues_005=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.05_1205_2022.csv"))
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))
combine_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_AF0.01_1123_2022.csv"))

#######################   John storey's q value 
#### install qvalue package 
#install.packages("devtools")
#library("devtools")
#install_github("jdstorey/qvalue")   https://github.com/StoreyLab/qvalue

#library(qvalue)
#qobj.burden=qvalue(Burden.pvalue)
#localFDR.burden <- qobj.burden$lfdr
#min(localFDR.burden, na.rm=T)

#qobj.SKATO=qvalue(SKATO.pvalue)
#localFDR.SKATO <- qobj.SKATO$lfdr
#min(localFDR.SKATO, na.rm=T)
#######################


#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_othermethod_ASD_topconstgene.pdf")
#par(mfrow=c(1,2))
theo <- (ppoints(length(burden_pvalues_001$LoF_pvalue)))
  #qqplot(-log(theo, base=10), -log(Burden.pvalue, base=10), xlab=expression(paste("Theoretical ",-log[10], "(p-values)")), ylab=expression(paste("Observed ", -log[10], "(p-values)")), main="(A) Burden", frame=F, cex.lab=0.8, pch=20)
  #abline(0,1,col='red', lwd=2)

#################  use ggplot draw qq plot 
sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$LoF_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

burden=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF=ggplot(burden) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("LoF")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
 #qqplot_burden_LoF

#################
theo <- (ppoints(length(burden_pvalues_001$All_variants_pvalue)))

 sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$All_variants_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

 LoF.Missense=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF.Missense=ggplot(LoF.Missense) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   #ylab("")+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("All Variants")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_LoF.Missense
 
 
 ################################
 theo <- (ppoints(length(burden_pvalues_001$fisher_combine_pvalue)))

 sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$fisher_combine_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

 fisher.combine=tibble(sx=sx, sy=sy)
 qqplot_burden_fisher.combine=ggplot(fisher.combine) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   #ylab("")+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("Fisher combine LoF+Missense")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_fisher.combine

#png("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_ASD_topconstgene_enrichemnt.png") 
figure=ggarrange(qqplot_burden_LoF, qqplot_burden_LoF.Missense,  qqplot_burden_fisher.combine, nrow=1)
annotate_figure(figure,
                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)

#dev.off()
```

* use one side binomial test 
* `All Variants`: LoF+Missense 

### LoF: enrichment of top genes 


```{r, echo=F, message=F, warning=F}
gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()


burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(LoF_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_pvalues_001_sorted$LoF_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), LoF_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(LoF_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by LoF p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.25))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by LoF p values")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=1.0000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=1.000", y_position = 0.02  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.1041", y_position = 0.16  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment
```

```{r, echo=F, warning=F, message=F}

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(LoF_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by LoF burden p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.2))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by LoF burden")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="1.000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="1.000", y_position = 0.055  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.565", y_position = 0.055  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.221", y_position = 0.155  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.01,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.513", y_position = 0.055,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.237", y_position = 0.105,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.442", y_position = 0.155,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment


```


### All variants: enrichment of top genes 


```{r, echo=F, message=F, warning=F}

gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))
burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(All_variants_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_pvalues_001_sorted$All_variants_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), All_variants_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by all variants p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment1=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.25))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by All variants p values")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=1.0000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=0.024", y_position = 0.11  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.029", y_position = 0.21  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment1
```

```{r, echo=F, warning=F, message=F}

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by all variants burden p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment2=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.2))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by all variants burden")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="1.000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="0.618", y_position = 0.055  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.565", y_position = 0.055  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.221", y_position = 0.155  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.01,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.04,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.071", y_position = 0.155,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.712", y_position = 0.105,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment2

 #figure=ggarrange(topmirage_enrichment1, topmirage_enrichment2, nrow=1)
#annotate_figure(figure,
#                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```


### Fisher combined LoF+Missense: enrichment of top genes 


```{r, echo=F, message=F, warning=F}

gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))
burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(fisher_combine_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_pvalues_001_sorted$fisher_combine_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), fisher_combined_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(fisher_combine_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by fisher combined p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment1=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.25))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by fisher combined p values")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=1.0000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=0.024", y_position = 0.11  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.029", y_position = 0.21  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment1
```

```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.01_1205_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(fisher_combine_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by fisher combined p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment2=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.2))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by fisher combined p values")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.169", y_position = 0.055  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="0.618", y_position = 0.055  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.565", y_position = 0.055  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.221", y_position = 0.155  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.01,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.04,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.237", y_position = 0.105,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.442", y_position = 0.155,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment2

 #figure=ggarrange(topmirage_enrichment1, topmirage_enrichment2, nrow=1)
#annotate_figure(figure,
#                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```



## LoF+MPC>2


```{r, echo=F, message=F, warning=F}
#burden_pvalues_005=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_AF0.05_1205_2022.csv"))
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))
#combine_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_AF0.01_1123_2022.csv"))

#######################   John storey's q value 
#### install qvalue package 
#install.packages("devtools")
#library("devtools")
#install_github("jdstorey/qvalue")   https://github.com/StoreyLab/qvalue

#library(qvalue)
#qobj.burden=qvalue(Burden.pvalue)
#localFDR.burden <- qobj.burden$lfdr
#min(localFDR.burden, na.rm=T)

#qobj.SKATO=qvalue(SKATO.pvalue)
#localFDR.SKATO <- qobj.SKATO$lfdr
#min(localFDR.SKATO, na.rm=T)
#######################


#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_othermethod_ASD_topconstgene.pdf")
#par(mfrow=c(1,2))
theo <- (ppoints(length(burden_pvalues_001$LoF_pvalue)))
  #qqplot(-log(theo, base=10), -log(Burden.pvalue, base=10), xlab=expression(paste("Theoretical ",-log[10], "(p-values)")), ylab=expression(paste("Observed ", -log[10], "(p-values)")), main="(A) Burden", frame=F, cex.lab=0.8, pch=20)
  #abline(0,1,col='red', lwd=2)

#################  use ggplot draw qq plot 
sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$LoF_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

burden=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF=ggplot(burden) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("LoF")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
 #qqplot_burden_LoF

#################
theo <- (ppoints(length(burden_pvalues_001$All_variants_pvalue)))

 sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$All_variants_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

 LoF.Missense=tibble(sx=sx, sy=sy)
 qqplot_burden_LoF.Missense=ggplot(LoF.Missense) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   #ylab("")+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("All Variants")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_LoF.Missense
 
 
 ################################
 theo <- (ppoints(length(burden_pvalues_001$fisher_combine_pvalue)))

 sx <- sort(-log(theo, base=10)); sy <- sort(-log(burden_pvalues_001$fisher_combine_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

 fisher.combine=tibble(sx=sx, sy=sy)
 qqplot_burden_fisher.combine=ggplot(fisher.combine) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   #ylab("")+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("Fisher combine LoF+MPC>2")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
#qqplot_burden_fisher.combine

#png("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-pvalue_QQplot_ASD_topconstgene_enrichemnt.png") 
figure=ggarrange(qqplot_burden_LoF, qqplot_burden_LoF.Missense,  qqplot_burden_fisher.combine, nrow=1)
annotate_figure(figure,
                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
)

#dev.off()
```

* use one side binomial test 
* `All Variants`: LoF+MPC>2 

### LoF: enrichment of top genes 


```{r, echo=F, message=F, warning=F}

gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))
burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(LoF_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_pvalues_001_sorted$LoF_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), LoF_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(LoF_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by LoF p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.25))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by LoF p values")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=1.0000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=1.000", y_position = 0.02  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.1041", y_position = 0.16  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment
```

```{r, echo=F, warning=F, message=F}

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(LoF_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by LoF burden p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.2))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by LoF burden")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="1.000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="1.000", y_position = 0.055  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.565", y_position = 0.055  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.221", y_position = 0.155  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.01,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.513", y_position = 0.055,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.237", y_position = 0.105,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.442", y_position = 0.155,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment


```

* results of LoF are the same as other scenarios 


### All variants (LoF+MPC>2): enrichment of top genes 


```{r, echo=F, message=F, warning=F}

gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))
burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(All_variants_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_pvalues_001_sorted$All_variants_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), All_variants_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


* `All_variants`: LoF+MPC>2 



```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by LoF+MPC>2 p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment1=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.25))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by LoF+MPC>2 p values")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=0.096", y_position = 0.055  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=1.000", y_position = 0.02  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=0.300", y_position = 0.11  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment1
```

```{r, echo=F, warning=F, message=F}

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by LoF+MPC>2 burden p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment2=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.4))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by LoF+MPC>2 burden")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.169", y_position = 0.06  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="0.618", y_position = 0.06  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.210", y_position = 0.11  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.029", y_position = 0.26  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.069", y_position = 0.06,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.02,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.513", y_position = 0.06,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.017", y_position = 0.21,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.057", y_position = 0.26,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment2

 #figure=ggarrange(topmirage_enrichment1, topmirage_enrichment2, nrow=1)
#annotate_figure(figure,
#                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```


### Fisher combined LoF+MPC>2: enrichment of top genes 


```{r, echo=F, message=F, warning=F}

gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.01) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))
burden_pvalues_001_sorted=burden_pvalues_001%>%arrange(fisher_combine_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=burden_pvalues_001_sorted$X[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); burden_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
burden_pvalue[i]=burden_pvalues_001_sorted$fisher_combine_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), fisher_combined_pvalue=round(burden_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(fisher_combine_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
## Top 10 genes 
##top10genes
###################### strong ASD genes 

strong.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol))  # define strong ASD gene with score 1,2
overlap1=length(intersect(topgenes, strong.ASD.gene))
overlap2=length(intersect(all.gene, strong.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))


########################## plausible ASD genes 

plausible.ASD.gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol))  
overlap1=length(intersect(topgenes, plausible.ASD.gene))
overlap2=length(intersect(all.gene, plausible.ASD.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

############### DNM genes 
denovo.gene=read.table("..\\..\\..\\GeneSet\\TADA_SNV_CNV_combined\\TADA_SNV_CNV_combined_Feb7.txt", header=T)
rank.denovo.gene=denovo.gene[order(denovo.gene$qvalue.combined),]
top1000.denovo.gene=rank.denovo.gene$RefSeqName[1:1000]
overlap1=length(intersect(topgenes, top1000.denovo.gene))
overlap2=length(intersect(all.gene, top1000.denovo.gene))
#fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))

######################## draw the enrichment 
topmirage.percent=c(length(intersect(topgenes, strong.ASD.gene)), length(intersect(topgenes, plausible.ASD.gene)), length(intersect(topgenes, top1000.denovo.gene)))/topgene_number
#top20mirage.percent=c(length(intersect(top20genes, strong.ASD.gene)), length(intersect(top20genes, plausible.ASD.gene)), length(intersect(top20genes, top1000.denovo.gene)))/20
all.gene.percent=c(length(intersect(all.gene, strong.ASD.gene)), length(intersect(all.gene, plausible.ASD.gene)), length(intersect(all.gene, top1000.denovo.gene)))/length(all.gene)
topmirage.enrichment=tibble(class=c("Top20 Gene by fisher combined p value", "All Genes"), Strong.ASD=c(topmirage.percent[1], all.gene.percent[1]), Plausible.ASD=c(topmirage.percent[2], all.gene.percent[2]), DNM=c(topmirage.percent[3], all.gene.percent[3]))
topmirage.enrichment.tidy=topmirage.enrichment %>% gather( group, percent,2:4)

topmirage.enrichment.tidy$group =factor(topmirage.enrichment.tidy$group, levels=c("Strong.ASD", "Plausible.ASD", "DNM")) # re order lables in x axis 


topmirage_enrichment1=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
  geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
  theme_classic()+
  xlab("")+ylab("Proportion of putative ASD genes")+
  ylim(c(0, 0.15))+
  theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
  ggtitle("Enrichment of top 20 ASD genes by fisher combined p values")+
  #theme(plot.title = element_text(size=12))
   theme(legend.title = element_blank())+
  geom_signif(annotations ="p=0.096", y_position = 0.055  ,xmin=0.7, xmax=1.2, size=2, vjust=-0.5, textsize=5)+ # textsize is for text over the bar 
  geom_signif(annotations ="p=1.000", y_position = 0.015  ,xmin=1.7, xmax=2.2, size=2, vjust=-0.5, textsize=5)+
  geom_signif(annotations ="p=1.000", y_position = 0.06  ,xmin=2.7, xmax=3.2, size=2, vjust=-0.5, textsize=5)+
  #geom_signif(annotations =0.000825, y_position = 0.45  ,xmin=0.7, xmax=1.2)+
   theme(legend.position="bottom")+
theme(text = element_text(size=15))+ # size is for labels on x axis 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
topmirage_enrichment1
```

```{r, echo=F, warning=F, message=F}
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv"))

sort_pvalue=burden_pvalues_001 %>% arrange(fisher_combine_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$X[1:topgene_number])
all.gene=as.character(sort_pvalue$X)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by fisher combined p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment2=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.35))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by fisher combined p values")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.169", y_position = 0.06  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="0.082", y_position = 0.16  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.058", y_position = 0.16  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.029", y_position = 0.26  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.069", y_position = 0.06,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.169", y_position = 0.11,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.017", y_position = 0.21,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.020", y_position = 0.31,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment2

 #figure=ggarrange(topmirage_enrichment1, topmirage_enrichment2, nrow=1)
#annotate_figure(figure,
#                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```



## Summary 


### Enrichment of top 20 genes 


```{r, echo=F, warning=F, message=F}
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv")) # call top genes by burden 
sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue) # rank genes by p values 
topgenes_burden=as.character(sort_pvalue$X[1:topgene_number])
all.gene_burden=as.character(sort_pvalue$X)
topmirage.percent_burden=numeric()

################
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
###################### strong ASD genes 

SFARI_gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol)) 

################

sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
topgene_number=20
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  ####################
  overlap_burden=length(intersect(topgenes_burden, Gene_Set_List[[i]]))
  topmirage.percent_burden[i]=overlap_burden/topgene_number
}
############ include SFARI gene in gene set 
overlap1=length(intersect(topgenes, SFARI_gene))
  topmirage.percent[i+1]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, SFARI_gene))
  all.gene.percent[i+1]=overlap2/length(all.gene)
  
  overlap_burden=length(intersect(topgenes_burden, SFARI_gene))
  topmirage.percent_burden[i+1]=overlap_burden/topgene_number
  

topmirage.enrichment.tidy=data.frame(class=c(rep("MIRAGE",length(Gene_Set)+1), rep("All Genes", length(Gene_Set)+1), rep("LoF+MPC>2", length(Gene_Set)+1)), group=rep(c(Gene_Set, "SFARI"),3), percent=c(topmirage.percent, all.gene.percent, topmirage.percent_burden))
######################



topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ # this is for numbers on y axis 
   ylim(c(0, 0.5))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by burden and MIRAGE")+
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.016", y_position = 0.115  ,xmin=0.7, xmax=1.3, size=1, vjust=-0.5, textsize=3.0)+ # textsize is for text over the bar; size is for width of lines 
  geom_signif(annotations ="0.169", y_position = 0.065  ,xmin=0.7, xmax=1.0, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="3.32e-6", y_position = 0.465  ,xmin=1.7, xmax=2.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.618", y_position = 0.065  ,xmin=1.7, xmax=2.0, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="6.09e-5", y_position = 0.365  ,xmin=2.7, xmax=3.3, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="0.210", y_position = 0.115  ,xmin=2.7, xmax=3.0, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="8.53e-3", y_position = 0.315  ,xmin=3.7, xmax=4.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.029", y_position = 0.265  ,xmin=3.7, xmax=4.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.069", y_position = 0.065,xmin=4.7, xmax=5.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="", y_position = 0.065,xmin=4.7, xmax=5.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.223", y_position = 0.065,xmin=5.7, xmax=6.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=5.7, xmax=6.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.247", y_position = 0.065,xmin=6.7, xmax=7.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=6.7, xmax=7.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.041", y_position = 0.165,xmin=7.7, xmax=8.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.513", y_position = 0.065,xmin=7.7, xmax=8.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="2.02e-6", y_position = 0.465,xmin=8.7, xmax=9.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.017", y_position = 0.215,xmin=8.7, xmax=9.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="5.06e-4", y_position = 0.465,xmin=9.7, xmax=10.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.057", y_position = 0.265,xmin=9.7, xmax=10.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.046", y_position = 0.113,xmin=10.7, xmax=11.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.280", y_position = 0.065,xmin=10.7, xmax=11.0, size=1, vjust=-0.5, textsize=3.0)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=13))+ # size is for labels on x axis 
  theme(axis.title = element_text(size = 13))+ # Font Size of Axis Titles
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment

```

* Top 20 MIRAGE genes selection: based on 6 variant groups, use MIRAGE to estimate parameters and calculate BF for each variant group, multiply group-wise BF when it has burden or negative burden but with BF<1.1 and ignore it otherwise, rank genes according to adjusted BF and choose top 20 genes. 


* Top 20 genes by burden (LoF+MPC>2): use one side Binomial test to calculate the p value of each gene containing 3 LoF variant groups and one missense variant group MPC>2, rank genes by p value and choose top 20 genes. 


### Top genes with evidence 


```{r, echo=F, message=F, warning=F}
################ top genes 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(BF_filtered))
delta_estimate=0.0671 # the estimate of delta, proportion of risk genes 
top_gene_number=10

post_prob=BF_proband_AllGene_sorted$BF_filtered*delta_estimate/(BF_proband_AllGene_sorted$BF_filtered*delta_estimate+1-delta_estimate) # calculate post probability by definition
Evidence=c("-", "High; FMRP; AutismKB; SCZ", "constraint gene; RVIS", "Haploinsuff", "ID gene; FMRP; SCZ", "-", "-", "-", "FMRP; AutismKB; constraint gene; RVIS; SCZ", "constraint gene; RVIS; Haploinsuff")
Fu_gene_list=as_tibble(read.csv("../../../GeneSet/Fu 2022 Suppl-41588_2022_1104_MOESM3_ESM.csv"))
ASC_gene=Fu_gene_list %>% filter(ASD185=="TRUE") %>% select(gene_gencodeV33) %>% pull
ASC_gene_status=BF_proband_AllGene_sorted$Gene[1:top_gene_number] %in% ASC_gene 
#ASC_gene_status[ASC_gene_status=="FALSE"]="X"; ASC_gene_status[ASC_gene_status=="TRUE"]="V"
Function=c("zinc ion binding", "regulate synaptic function", "putative chromatin structure regulator", "-", "zinc ion binding", "metal ion binding", "protein binding", "protein binding", "metal ion binding; neuroligin family protein binding", "protein binding" )# these information from Genecard 

top_gene_summary=data.frame(Gene=BF_proband_AllGene_sorted$Gene[1:top_gene_number], BF=BF_proband_AllGene_sorted$BF_filtered[1:top_gene_number], posterior=round(post_prob[1:top_gene_number],4),Evidence=Evidence, ASC_gene=ASC_gene_status, Function=Function)
top_gene_summary%>% datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```

* BF: this is filtered BF removing variant groups with negative burden and BF>1.1, for those top 10 genes, their filter BF is their actual BF without filtering 

* ASC_gene: 185 ASC genes with FDR<0.05, published in [NG 2022](https://www.nature.com/articles/s41588-022-01104-0#MOESM3)


### gene Manhattan plot

```{r, echo=F, message=F, warning=F, }
############## run once to get the data for plot 
#BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
#BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(BF_filtered))
#delta_estimate=0.0671 # the estimate of delta, proportion of risk genes 
#post_prob=BF_proband_AllGene_sorted$BF_filtered*delta_estimate/(BF_proband_AllGene_sorted$BF_filtered*delta_estimate+1-delta_estimate) # calculate post probability by definition
#data_for_plot=data.frame(Gene=BF_proband_AllGene_sorted$Gene, BF=BF_proband_AllGene_sorted$BF, BF_filtered=BF_proband_AllGene_sorted$BF_filtered, post_prob=post_prob)
##### extrat chr and position for each gene 
#chr=numeric(); pos=numeric()
#for (i in 1:nrow(data_for_plot))
#for (i in 1:2)  
#{
#  cat(i, "is running", "\n")
#  variant=ASC_new_sample %>% filter(Gene==data_for_plot$Gene[i]) %>% select(Variant)%>% pull() # find all variants for a gene 
#  colon_pos=unlist(gregexpr(":", variant[1])) # location positions of ":"
#  chr[i]=substr(variant[1], 1,colon_pos[1]-1)
#  pos[i]=substr(variant[1], colon_pos[1]+1, colon_pos[2]-1)
#}
#data_manhattan_plot=data_for_plot%>%mutate(chr=as.numeric(chr), pos=as.numeric(pos))
#write.csv(data_manhattan_plot, file="../output/KyleData/data_manhattan_plot_0105_2023.csv")
data_manhattan_plot=read.csv("../output/KyleData/data_manhattan_plot_0105_2023.csv", header=T)
#install.packages("remotes")
#remotes::install_github("kevinlkx/mapgen")  # download kevin's package 
#library(mapgen)


#' @title Make gene manhattan plot
#'
#' @param gene.pip.res Gene level finemapping result
#' @param chr Name of the chr column in the gene finemapping summary statistics data
#' @param pos Name of the pos column in the gene finemapping summary statistics data
#' @param gene_name Name of the gene name column in the gene finemapping summary statistics data
#' @param gene_pip Name of the gene PIP column in the gene finemapping summary statistics data
#' @param sig.pip Signficant gene PIP (default: 0.8)
#' @param highlight Highlight genes with gene PIP > sig.pip
#' @param ylim Truncate gene PIP to ylim value in the plot (default: 1.25).
#' @param point.size Size of the points.
#' @param label.size Size of the labels.
#' @param font.size Font size of the text.
#' @param title Title of the plot
#' @param max.overlaps Exclude text labels that overlap too many things (default: 10).
#' @import ggplot2
#' @importFrom magrittr %>%
#' @export
#'
gene_manhattan_plot <- function(gene.pip.res,
                                chr='chr',
                                pos='pos',
                                gene_name='gene_name',
                                gene_pip='gene_pip',
                                sig.pip = 0.8,
                                highlight = TRUE,
                                ylim = 1.25,
                                point.size = 2,
                                label.size = point.size*2,
                                font.size = 15,
                                max.overlaps = 10,
                                title = '') {

  gene.pip.res <- gene.pip.res %>% dplyr::rename(chr = all_of(chr),
                                                 pos = all_of(pos),
                                                 gene_name = all_of(gene_name),
                                                 gene_pip = all_of(gene_pip))

  # Highlight genes
  gene.pip.res <- gene.pip.res %>%
    dplyr::mutate( is_highlight=(gene_pip >= sig.pip))

  df <- gene.pip.res %>%

    # Compute chromosome size
    dplyr::group_by(chr) %>%
    dplyr::summarise(chr_len = max(as.numeric(pos))) %>%

    # Calculate cumulative position of each chromosome
    dplyr::mutate(tot=cumsum(chr_len)-chr_len) %>%
    dplyr::select(-chr_len) %>%

    # Add this info to the initial dataset
    dplyr::left_join(gene.pip.res, ., by=c("chr"="chr")) %>%

    # Add a cumulative position of each SNP
    dplyr::arrange(chr, pos) %>%
    dplyr::mutate( pos_cum=pos+tot)

  axis.df <- df %>%
    dplyr::group_by(chr) %>%
    dplyr::summarize(center=( max(pos_cum) + min(pos_cum) ) / 2 )

  p <- ggplot(df,
              aes(x = pos_cum, y = gene_pip)) +

    # Show all points
    ggrastr::geom_point_rast( aes(color=as.factor(chr)), size=point.size) +
    scale_color_manual(values = rep(c("grey", "skyblue"), 22 )) +

    # custom X axis:
    scale_x_continuous(label = gsub("chr","", axis.df$chr, ignore.case = TRUE), breaks= axis.df$center) +
    scale_y_continuous(expand = c(0, 0), limits = c(0,ylim)) +     # remove space between plot area and x axis

    # Add highlighted points
    ggrastr::geom_point_rast(data=subset(df, is_highlight==TRUE), color="orange", size=point.size) +

    # Add label using ggrepel to avoid overlapping
    ggrepel::geom_label_repel( data=subset(df, is_highlight==TRUE),
                               aes(label=gene_name),
                               size=label.size,
                               min.segment.length = 0,
                               label.size = NA,
                               fill = alpha(c("white"),0),
                               max.overlaps = max.overlaps) +

    # Custom the theme:
    theme_bw() +
    theme(
      text = element_text(size = font.size),
      legend.position="none",
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank(),
      plot.title = element_text(hjust = 0.5)
    ) +
    ggtitle(title) +
    theme() +
    xlab("Chromosome") +
    #ylab("Gene PIP")
    ylab("Posterior probability")

  return(p)

}


data_manhattan_plot=data_manhattan_plot[complete.cases(data_manhattan_plot), ] # remove rows with NA

#gene_manhattan_plot(data_manhattan_plot, chr=data_manhattan_plot$chr, pos=data_manhattan_plot$pos, gene_name=data_manhattan_plot$Gene, gene_pip = data_manhattan_plot$post_prob)
gene_manhattan_plot(
  data_manhattan_plot,
  chr = "chr",
  pos = "pos",
  gene_name = "Gene",
  gene_pip = "post_prob", 
  sig.pip = 0.5,
  highlight = TRUE,
  ylim = 0.8,
  point.size = 2,
  label.size = 4,
  font.size = 10,
  max.overlaps = 10,
  title = "Manhattan plot of posterior probability")
```