---
title: "syn var in ASC new sample"
output: html_document
---

```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
#library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
```


```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
source("Read.into.Gene.set.R")
```



```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
ASC_new_sample1=as_tibble(read.table("C:\\Shengtong\\Research\\rare-var\\ASC\\ASC_B15_B16_raw_TDT_by_parent_2021-04-19.txt", header=T, fill=T))
ASC_new_sample2=as_tibble(read.table("C:\\Shengtong\\Research\\rare-var\\ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", header=T, fill=T))
```


```{r, echo=F, eval=T}
## identify transmitted and non-transmitted variants 
#transmitted_variant=ASC_new_sample2%>%select(starts_with(c("t", "u")), "Gnomad_non_neuro_AF")
#kable(transmitted_variant[1:10,])%>%
#kable_styling() %>%
#scroll_box(height = "500px")
#write.csv(transmitted_variant, file="C:\\Shengtong\\Research\\rare-var\\ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10_transmitted_variant.csv")

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
Transmitted_proband=ASC_new_sample2%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) 
Untransmitted_proband=ASC_new_sample2%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) 

##### count transmitted variants and untransmitted in proband from parents  by summing over male, female, dad, mom  
Transmitted_proband_parents=ASC_new_sample2%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband_parents = select(., t_proband_male_dad, t_proband_male_mom, t_proband_female_dad, t_proband_female_mom) %>% rowSums())%>% select(Transmitted_proband_parents) 
Untransmitted_proband_parents=ASC_new_sample2%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband_parents= select(., u_proband_male_dad, u_proband_male_mom, u_proband_female_dad, u_proband_female_mom) %>% rowSums())%>% select(Untransmitted_proband_parents) 


##### add new columns of total transmitted and non-transmitted variants  
ASC_new_sample=ASC_new_sample2%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_proband_parents=Transmitted_proband_parents$Transmitted_proband_parents, Untransmitted_proband_parents=Untransmitted_proband_parents$Untransmitted_proband_parents)

num_proband=6026+1265 # male proband+female proband 
```

```{r, echo=F, warning=F, warning=F}
# input must have column Gnomad_non_neuro_AF; Transmitted_proband;  Untransmitted_proband
fisher_test=function(input.data, sample_size, AF_lower, AF_upper)
{
input.data.filter=input.data %>%filter(Gnomad_non_neuro_AF>=AF_lower & Gnomad_non_neuro_AF<AF_upper)  
Trans_variant=input.data.filter%>%select(Transmitted_proband)%>%pull
Untrans_variant=input.data.filter%>%select(Untransmitted_proband)%>%pull
burden_test=fisher.test(matrix(c(sum(Trans_variant), num_proband, sum(Untrans_variant), num_proband), nrow=2))
result=list(OR=burden_test$estimate, CI_lower=burden_test$conf.int[1], CI_upper=burden_test$conf.int[2], pvalue=burden_test$p.value, Trans_variant=sum(Trans_variant), Untrans_variant=sum(Untrans_variant))
return(result)  
}
```



## Burden of syn var partitioned by AF 

* filter `Consequence=="synonymous_variant"`
* AF from `Gnomad_non_neuro_AF` is used to define other categories. 
* When use AF cutoff, be aware of 15614 variants with zero AF going to which category 

```{r, echo=F, message=F, warning=F}
syn_variants_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant" ) %>% select(Variant, Gene, Gnomad_non_neuro_AF,  Transmitted_proband, Untransmitted_proband, Transmitted_proband_parents, Untransmitted_proband_parents)

AF_cutoff=c(0, 0.05, 1) # define AF cutoffs 
OR=numeric(); pvalue=numeric(); CI_lower=numeric(); CI_upper=numeric()
for (i in 1:(length(AF_cutoff)-1))
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[i], AF_cutoff[i+1])
  OR[i]=burden_test$OR; pvalue[i]=burden_test$pvalue; CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}
burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[1], AF_cutoff[3])
  OR[i+1]=burden_test$OR; pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 3 categories; (1) AF<5% (2) AF>5% (3) all 
  
syn_varint_burden=tibble(OR=OR, var=c( "(1): AF<5%", "(2): AF>5%", "(3): all"), lower=CI_lower, upper=CI_upper)

ggplot(syn_varint_burden, aes(x=var, y=OR, fill=var)) + # draw the OR plot 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  ylim(c(0,1.1))+
  ggtitle("Burden of syn variants")+
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2,
                 position=position_dodge(.9))+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
  theme(plot.title = element_text(hjust = 0.5))+ # center the title
  theme(legend.position="none")


```


*  filter `Consequence=="synonymous_variant"`

* Private variants are defined as variants with total allele  count equal to  1 in parents calculated as `Transmitted_proband_parents+Untransmitted_proband_parents=1`

* AF from `Gnomad_non_neuro_AF` is used to define other categories. 



```{r, echo=F}
AF_cutoff=c(0, 0.001, 0.01, 0.05, 1)

OR=numeric(); pvalue=numeric(); CI_lower=numeric(); CI_upper=numeric()

for (i in 1:(length(AF_cutoff)-1))
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[i], AF_cutoff[i+1])
  OR[i]=burden_test$OR; pvalue[i]=burden_test$pvalue; CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}
burden_test=fisher_test(syn_variants_new_sample %>% filter((Transmitted_proband_parents+Untransmitted_proband_parents)==1), num_proband, AF_cutoff[1], AF_cutoff[5]) # private var  
  OR[i+1]=burden_test$OR; pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 5 categories; (1) private (2) AF<0.1% (3) 0.1%=<AF<1% (4) 1%=<AF<5% (5) 5%=<AF
  
syn_varint_burden=tibble(OR=c(OR[5], OR[1:4]), var=c( "(1): private", "(2): AF<0.1%", "(3): 0.1%<AF<1%", "(4): 1%<AF<5%", "(5): AF>5%"), lower=c(CI_lower[5], CI_lower[1:4]), upper=c(CI_upper[5], CI_upper[1:4]))

ggplot(syn_varint_burden, aes(x=var, y=OR, fill=var)) + # draw the OR plot 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  ylim(c(0,1.1))+
  ggtitle("Burden of syn variants")+
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2,
                 position=position_dodge(.9))+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
  theme(plot.title = element_text(hjust = 0.5))+ # center the title
  theme(legend.position="none")
  
  
```
  





## Burden of syn var with  `VQSLOD` filter

* [VQSLOD](https://gatk.broadinstitute.org/hc/en-us/articles/360035531612-Variant-Quality-Score-Recalibration-VQSR-) is short for variant quality score log-odds.  

* at each VQSLOD cutoff, variants with VQSLOD score less than cutoff is chosen. 

```
> summary(ASC_new_sample %>% filter(Consequence=="synonymous_variant") %>% select(VQSLOD)%>%pull())
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
-39450.00     -5.49     -2.65    -40.90     -1.93      1.91
```

```
Kyle looked at using a VQSLOD cutoff of 3.75, but then brought it up to 5.13 after trying a similar calibration procedure with non-coding indels.
```


*  filter `Consequence=="synonymous_variant"`
* Private variants are defined as variants with total allele  count equal to  1 in parents calculated as `Transmitted_proband_parents+Untransmitted_proband_parents=1`

* AF from `Gnomad_non_neuro_AF` is used to define other categories. 

```{r, echo=F, message=F, warning=F}
num.cutoff=100
AF_cutoff=c(0, 0.001, 0.01, 0.05, 1)
#summary(ASC_new_sample %>% filter(Consequence=="synonymous_variant" & Gnomad_non_neuro_AF<0.05)%>%select(VQSLOD)%>%pull)
vqslod.cutoff=seq(-40,1.91,length.out=num.cutoff)
odds.ratio=matrix(nrow=num.cutoff, ncol=5)
for (i in 1:num.cutoff)
{
syn_variants_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant"  & VQSLOD<vqslod.cutoff[i]) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Transmitted_proband_parents, Untransmitted_proband_parents)

for (j in 1:(length(AF_cutoff)-1)) # at every VQSLOD cutoff, consider 5 variant categories 
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[j], AF_cutoff[j+1])
  odds.ratio[i,j]=burden_test$OR;
  #pvalue[i]=burden_test$pvalue; #CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}  # end of for (j in 1:(length(AF_cutoff)-1))
burden_test=fisher_test(syn_variants_new_sample %>% filter((Transmitted_proband_parents+Untransmitted_proband_parents)==1), num_proband, AF_cutoff[1], AF_cutoff[5]) # private var  
  odds.ratio[i,j+1]=burden_test$OR; 
  #pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 5 categories; (1) private (2) AF<0.1% (3) 0.1%=<AF<1% (4) 1%=<AF<5% (5) 5%=<AF
  
  
} # end of for (i in 1:num.cutoff)

odds.ratio.summary=tibble(vqslod=rep(vqslod.cutoff, 5), OR=c( odds.ratio[,5], odds.ratio[,1], odds.ratio[,2], odds.ratio[,3], odds.ratio[,4]), var=c(rep("(1): private", num.cutoff), rep("(2): AF<0.1%", num.cutoff), rep("(3): 0.1%<AF<1%", num.cutoff), rep("(4): 1%<AF<5%", num.cutoff), rep("(5): 5%<AF", num.cutoff)))

ggplot(odds.ratio.summary, aes(x=vqslod, y=OR, group=var)) +
  geom_line(aes(color=var))+
  ylim(c(0,1))+
  geom_point(aes(color=var))+
  ggtitle("odds ratio of syn variants with varying LQSLOD cutoff")+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title; size controls font size in title 


```




## Burden of syn var with  `QD/ReadPosRankSum/SOR` filter


Consider [`QD, ReadPosRankSum, SOR`](https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants) combined filters.

### filter used for indels by Kyle: 

* Kyle suggested `QD >= 3 & AS_SOR <= 3 & AS_ReadPosRankSum >= -8` 


* fix `QD >= 3 & AS_SOR <= 3` and vary `AS_ReadPosRankSum` and variants with score larger than threshold are chosen 


* vertical line is ` AS_ReadPosRankSum = -8`

```{r, echo=F, message=F, warning=F}
num.cutoff=20
#summary(ASC_new_sample %>% filter(Consequence=="synonymous_variant")%>%select(AS_ReadPosRankSum)%>%pull)
ReadPosRankSum.cutoff=seq(-18,11,length.out=num.cutoff) # AS_ReadPosRankSum >= -8 by Kyle 
odds.ratio=matrix(nrow=num.cutoff, ncol=5)
for (i in 1:num.cutoff)
{
syn_variants_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant"  & QD >= 3 & AS_SOR <= 3  & AS_ReadPosRankSum>ReadPosRankSum.cutoff[i]) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Transmitted_proband_parents, Untransmitted_proband_parents)

for (j in 1:(length(AF_cutoff)-1)) # at every VQSLOD cutoff, consider 5 variant categories 
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[j], AF_cutoff[j+1])
  odds.ratio[i,j]=burden_test$OR;
  #pvalue[i]=burden_test$pvalue; #CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}  # end of for (j in 1:(length(AF_cutoff)-1))
burden_test=fisher_test(syn_variants_new_sample %>% filter((Transmitted_proband_parents+Untransmitted_proband_parents)==1), num_proband, AF_cutoff[1], AF_cutoff[5]) # private var  
  odds.ratio[i,j+1]=burden_test$OR; 
  #pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 5 categories; (1) private (2) AF<0.1% (3) 0.1%=<AF<1% (4) 1%=<AF<5% (5) 5%=<AF
  
  
} # end of for (i in 1:num.cutoff)

odds.ratio.summary=tibble(ReadPosRankSum=rep(ReadPosRankSum.cutoff, 5), OR=c(odds.ratio[,5], odds.ratio[,1], odds.ratio[,2], odds.ratio[,3], odds.ratio[,4]), var=c(rep("(1): private", num.cutoff), rep("(2): AF<0.1%", num.cutoff), rep("(3): 0.1%<AF<1%", num.cutoff), rep("(4): 1%<AF<5%", num.cutoff), rep("(5): 5%<AF", num.cutoff)))

ggplot(odds.ratio.summary, aes(x=ReadPosRankSum, y=OR, group=var)) +
  geom_line(aes(color=var))+
  ylim(c(0,1))+
  geom_vline(xintercept = -8, linetype="solid", 
                color = "blue", size=1.5)+
  geom_point(aes(color=var))+
  ggtitle("odds ratio of syn variants with varying AS_ReadPosRankSum cutoff")+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title; size controls font size in title 


```

### filter used for SNPs by Kyle: 


`QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8` for SNPs suggested by Kyle. Fix two of them and vary one. 


#### Vary `AS_ReadPosRankSum`

* fix `QD >= 1 & AS_SOR <= 3` and vary `AS_ReadPosRankSum` and variants with score larger than threshold are chosen 


* vertical line is ` AS_ReadPosRankSum = -0.8`


```{r, echo=F, message=F, warning=F}
num.cutoff=20
#summary(ASC_new_sample %>% filter(Consequence=="synonymous_variant")%>%select(AS_ReadPosRankSum)%>%pull)
ReadPosRankSum.cutoff=seq(-18,11,length.out=num.cutoff) # AS_ReadPosRankSum >= -8 by Kyle 
odds.ratio=matrix(nrow=num.cutoff, ncol=5)
for (i in 1:num.cutoff)
{
syn_variants_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant"  & QD >= 1 & AS_SOR <= 3  & AS_ReadPosRankSum>ReadPosRankSum.cutoff[i]) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Transmitted_proband_parents, Untransmitted_proband_parents)

for (j in 1:(length(AF_cutoff)-1)) # at every VQSLOD cutoff, consider 5 variant categories 
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[j], AF_cutoff[j+1])
  odds.ratio[i,j]=burden_test$OR;
  #pvalue[i]=burden_test$pvalue; #CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}  # end of for (j in 1:(length(AF_cutoff)-1))
burden_test=fisher_test(syn_variants_new_sample %>% filter((Transmitted_proband_parents+Untransmitted_proband_parents)==1), num_proband, AF_cutoff[1], AF_cutoff[5]) # private var  
  odds.ratio[i,j+1]=burden_test$OR; 
  #pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 5 categories; (1) private (2) AF<0.1% (3) 0.1%=<AF<1% (4) 1%=<AF<5% (5) 5%=<AF
  
  
} # end of for (i in 1:num.cutoff)

odds.ratio.summary=tibble(ReadPosRankSum=rep(ReadPosRankSum.cutoff, 5), OR=c(odds.ratio[,5], odds.ratio[,1], odds.ratio[,2], odds.ratio[,3], odds.ratio[,4]), var=c(rep("(1): private", num.cutoff), rep("(2): AF<0.1%", num.cutoff), rep("(3): 0.1%<AF<1%", num.cutoff), rep("(4): 1%<AF<5%", num.cutoff), rep("(5): 5%<AF", num.cutoff)))

ggplot(odds.ratio.summary, aes(x=ReadPosRankSum, y=OR, group=var)) +
  geom_line(aes(color=var))+
  ylim(c(0,1))+
  geom_vline(xintercept = -0.8, linetype="solid", 
                color = "blue", size=1.5)+
  geom_point(aes(color=var))+
  ggtitle("odds ratio of syn variants with varying AS_ReadPosRankSum cutoff")+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title; size controls font size in title 


```


#### vary `QD`

* fix `AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8` and vary `QD` and variants with score larger than threshold are chosen 


* vertical line is ` QD=1`

* when the range of OR is enforced to be (0,1), variant groups (1): private and "(2): AF<0.1%" look broken because their maximum of OR exceeds 1. 


```{r, echo=F, message=F, warning=F}
num.cutoff=20
#summary(ASC_new_sample %>% filter(Consequence=="synonymous_variant")%>%select(QD)%>%pull)
QD.cutoff=seq(0,70,length.out=num.cutoff) # AS_ReadPosRankSum >= -8 by Kyle 
odds.ratio=matrix(nrow=num.cutoff, ncol=5)
for (i in 1:num.cutoff)
{
syn_variants_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant"  & QD >= QD.cutoff[i] & AS_SOR <= 3  & AS_ReadPosRankSum>-0.8) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Transmitted_proband_parents, Untransmitted_proband_parents)

for (j in 1:(length(AF_cutoff)-1)) # at every VQSLOD cutoff, consider 5 variant categories 
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[j], AF_cutoff[j+1])
  odds.ratio[i,j]=burden_test$OR;
  #pvalue[i]=burden_test$pvalue; #CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}  # end of for (j in 1:(length(AF_cutoff)-1))
burden_test=fisher_test(syn_variants_new_sample %>% filter((Transmitted_proband_parents+Untransmitted_proband_parents)==1), num_proband, AF_cutoff[1], AF_cutoff[5]) # private var  
  odds.ratio[i,j+1]=burden_test$OR; 
  #pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 5 categories; (1) private (2) AF<0.1% (3) 0.1%=<AF<1% (4) 1%=<AF<5% (5) 5%=<AF
  
  
} # end of for (i in 1:num.cutoff)

odds.ratio.summary=tibble(QD=rep(QD.cutoff, 5), OR=c(odds.ratio[,5], odds.ratio[,1], odds.ratio[,2], odds.ratio[,3], odds.ratio[,4]), var=c(rep("(1): private", num.cutoff), rep("(2): AF<0.1%", num.cutoff), rep("(3): 0.1%<AF<1%", num.cutoff), rep("(4): 1%<AF<5%", num.cutoff), rep("(5): 5%<AF", num.cutoff)))

y.max=max(odds.ratio.summary$OR)+0.1
ggplot(odds.ratio.summary, aes(x=QD, y=OR, group=var)) +
  geom_line(aes(color=var))+
  ylim(c(0,y.max))+
  geom_vline(xintercept =1, linetype="solid", 
                color = "blue", size=1.5)+
  geom_point(aes(color=var))+
  ggtitle("odds ratio of syn variants with varying QD cutoff")+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title; size controls font size in title 

```


```{r, echo=F, eval=F}
# look at two variant groups that look broken
odds.ratio.summary %>% filter(var=="(1): private")
odds.ratio.summary %>% filter(var=="(2): AF<0.1%")
```




#### vary `As_SOR`

* fix `QD >= 1 & AS_ReadPosRankSum >= -0.8` and vary `AS_SOR` and variants with score less than threshold are chosen 


* vertical line is `AS_SOR=3 `

* there are two Inf `  0    Inf     (4): 1%<AF<5% `, ` 0    Inf     (2): AF<0.1%`


```{r, echo=F, message=F, warning=F}
num.cutoff=20
#summary(ASC_new_sample %>% filter(Consequence=="synonymous_variant")%>%select(AS_SOR)%>%pull)
SOR.cutoff=seq(0,22,length.out=num.cutoff) # AS_SOR <= 3 by Kyle 
odds.ratio=matrix(nrow=num.cutoff, ncol=5)
for (i in 1:num.cutoff)
{
syn_variants_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant"  & AS_SOR <= SOR.cutoff[i] & QD>=1  & AS_ReadPosRankSum>-0.8) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Transmitted_proband_parents, Untransmitted_proband_parents)

for (j in 1:(length(AF_cutoff)-1)) # at every VQSLOD cutoff, consider 5 variant categories 
{
  burden_test=fisher_test(syn_variants_new_sample, num_proband, AF_cutoff[j], AF_cutoff[j+1])
  odds.ratio[i,j]=burden_test$OR;
  #pvalue[i]=burden_test$pvalue; #CI_lower[i]=burden_test$CI_lower; CI_upper[i]=burden_test$CI_upper
}  # end of for (j in 1:(length(AF_cutoff)-1))
burden_test=fisher_test(syn_variants_new_sample %>% filter((Transmitted_proband_parents+Untransmitted_proband_parents)==1), num_proband, AF_cutoff[1], AF_cutoff[5]) # private var  
  odds.ratio[i,j+1]=burden_test$OR; 
  #pvalue[i+1]=burden_test$pvalue; CI_lower[i+1]=burden_test$CI_lower; CI_upper[i+1]=burden_test$CI_upper
# 5 categories; (1) private (2) AF<0.1% (3) 0.1%=<AF<1% (4) 1%=<AF<5% (5) 5%=<AF
  
  
} # end of for (i in 1:num.cutoff)

odds.ratio.summary=tibble(SOR=rep(SOR.cutoff, 5), OR=c(odds.ratio[,5], odds.ratio[,1], odds.ratio[,2], odds.ratio[,3], odds.ratio[,4]), var=c(rep("(1): private", num.cutoff), rep("(2): AF<0.1%", num.cutoff), rep("(3): 0.1%<AF<1%", num.cutoff), rep("(4): 1%<AF<5%", num.cutoff), rep("(5): 5%<AF", num.cutoff)))

y.max=min(max(odds.ratio.summary$OR)+0.1, 1.3) # because there could be Inf 
ggplot(odds.ratio.summary, aes(x=SOR, y=OR, group=var)) +
  geom_line(aes(color=var))+
  ylim(c(0,y.max))+
  geom_vline(xintercept =3, linetype="solid", 
                color = "blue", size=1.5)+
  geom_point(aes(color=var))+
  ggtitle("odds ratio of syn variants with varying SOR cutoff")+
  geom_hline(yintercept=1, linetype="dashed", 
                color = "red", size=1)+
theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title; size controls font size in title 


```



 


```{r, echo=F, warning=F, message=F, eval=F}

## Burden of syn var in different gene sets 


### filter `VQSLOD` 

* `Gnomad_non_neuro_AF<0.05`
* `Consequence=="synonymous_variant"`
* after applying the above two filters, maximum of `VQSLOD` is 1.91, thus no need to use `cutoff of 3.75, but then brought it up to 5.13 ` suggested by Kyle. 
######### obtain syn missense variants in new sample data

rare_syn_variant_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant" & Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband)  # find syn variants   in new sample data with Gnomad AF threshold 

transmitted_new_sample=numeric()
nontransmitted_new_sample=numeric()
for ( i in 1:length(Gene_Set_List))
 {
  Feature_Data=rare_syn_variant_new_sample%>%filter(Gene %in% Gene_Set_List[[i]])%>% drop_na()
  if (nrow(Feature_Data)>0)
  { 
    transmitted_new_sample[i]=Feature_Data%>%select(Transmitted_proband)%>%sum()
    nontransmitted_new_sample[i]=Feature_Data%>%select(Untransmitted_proband)%>%sum()
    
     ##############  run mirage-vs on every variant set, takes several minutes to run
#  Feature_Data_VS=cell_variant_set%>% filter(GENE_NAME %in% Gene_Set_List[[i]])%>% select(Transmitted_proband,Untransmitted_proband) %>% add_column(Group_Index=1)%>% drop_na()
#  Burden_MIRAGE_VS[i,1:2]=c(transmitted_cell_sample[i], nontransmitted_cell_sample[i])
#  pois.test=poisson.test(Burden_MIRAGE_VS[i,1:2], c(5869, 5869), r=1, alternative="greater")
# Burden_MIRAGE_VS[i,3:4]=c(pois.test$estimate,pois.test$p.value)
 
#  vs.mirage=mirage_vs(data.frame(Feature_Data_VS),n1=5869,n2=5869)
#  Burden_MIRAGE_VS[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
  
  ##############
    
  }
  }  ### end of i

#write.csv(Burden_MIRAGE_VS, file="../output/CellDataASC/Variant_Set_missense_10Geneset.csv")

syn_new_sample=tibble(Gene=rep(Gene_Set,2), status=rep(c("transmitted", "nontransmitted"), each=10), variant_count=c(transmitted_new_sample, nontransmitted_new_sample))

g_new_sample=ggplot(syn_new_sample, aes(x = Gene, y = variant_count, fill = status)) +
  geom_col(position = position_dodge())+
  #geom_bar(stat="identity")+
  theme_classic()+
  ylim(c(0,40000))+
  ylab("Variant count")+xlab("Gene set")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))+
theme(text = element_text(size=10))+
ggtitle("syn variants in new sample")+
  theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title
g_new_sample
############ combine two figures into one panel 
#figure=ggarrange( g_old_sample, g_new_sample, common.legend = TRUE, legend="right") 
#annotate_figure(figure,
#                top = text_grob("missense variants; AF<0.0001", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)
```

 

```{r, echo=F, message=F, warning=F, eval=F}
* fisher exact test is used 
* lower, upper indicate 95% confidence interval
#################### print out summary statistics in a  table 
syn_new_sample_summary=tibble()
p.value=numeric(); lower_bound=numeric(); upper_bound=numeric(); OR=numeric()
num_proband=6026+1265 # male proband+female proband 
for (i in 1:length(transmitted_new_sample))
{
  fisher_test=fisher.test(matrix(c(transmitted_new_sample[i], num_proband, nontransmitted_new_sample[i], num_proband), nrow=2))
  p.value[i]=fisher_test$p.val
  lower_bound[i]=fisher_test$conf.int[1]
  upper_bound[i]=fisher_test$conf.int[2]
  OR[i]=fisher_test$estimate
}
syn_new_sample_summary=tibble(gene=Gene_Set,transmitted=transmitted_new_sample, nontransmitted=nontransmitted_new_sample, OR=round(OR,2),  pvalue=round(p.value,4), lower=round(lower_bound,4), upper=round(upper_bound,4))
kable(syn_new_sample_summary, caption = "")%>%
kable_styling() %>%
scroll_box(height = "200px")
```

 

```{r, echo=F, warning=F, message=F, eval=F}

### filter QD and AS_SOR

* `Gnomad_non_neuro_AF<0.05`
* `Consequence=="synonymous_variant"`
* `QD>=3  & AS_SOR <=3 & AS_ReadPosRankSum>-8`, filter used for indel by Kyle 


######### obtain syn missense variants in new sample data

rare_syn_variant_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant" & Gnomad_non_neuro_AF<0.05 & QD>=3  & AS_SOR <=3 & AS_ReadPosRankSum>-8) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband)  # find syn variants   in new sample data with Gnomad AF threshold 

transmitted_new_sample=numeric()
nontransmitted_new_sample=numeric()
for ( i in 1:length(Gene_Set_List))
 {
  Feature_Data=rare_syn_variant_new_sample%>%filter(Gene %in% Gene_Set_List[[i]])%>% drop_na()
  if (nrow(Feature_Data)>0)
  { 
    transmitted_new_sample[i]=Feature_Data%>%select(Transmitted_proband)%>%sum()
    nontransmitted_new_sample[i]=Feature_Data%>%select(Untransmitted_proband)%>%sum()
    
     ##############  run mirage-vs on every variant set, takes several minutes to run
#  Feature_Data_VS=cell_variant_set%>% filter(GENE_NAME %in% Gene_Set_List[[i]])%>% select(Transmitted_proband,Untransmitted_proband) %>% add_column(Group_Index=1)%>% drop_na()
#  Burden_MIRAGE_VS[i,1:2]=c(transmitted_cell_sample[i], nontransmitted_cell_sample[i])
#  pois.test=poisson.test(Burden_MIRAGE_VS[i,1:2], c(5869, 5869), r=1, alternative="greater")
# Burden_MIRAGE_VS[i,3:4]=c(pois.test$estimate,pois.test$p.value)
 
#  vs.mirage=mirage_vs(data.frame(Feature_Data_VS),n1=5869,n2=5869)
#  Burden_MIRAGE_VS[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
  
  ##############
    
  }
  }  ### end of i

#write.csv(Burden_MIRAGE_VS, file="../output/CellDataASC/Variant_Set_missense_10Geneset.csv")

syn_new_sample=tibble(Gene=rep(Gene_Set,2), status=rep(c("transmitted", "nontransmitted"), each=10), variant_count=c(transmitted_new_sample, nontransmitted_new_sample))

g_new_sample=ggplot(syn_new_sample, aes(x = Gene, y = variant_count, fill = status)) +
  geom_col(position = position_dodge())+
  #geom_bar(stat="identity")+
  theme_classic()+
  ylim(c(0,40000))+
  ylab("Variant count")+xlab("Gene set")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))+
theme(text = element_text(size=10))+
ggtitle("syn variants in new sample")+
  theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title
g_new_sample
############ combine two figures into one panel 
#figure=ggarrange( g_old_sample, g_new_sample, common.legend = TRUE, legend="right") 
#annotate_figure(figure,
#                top = text_grob("missense variants; AF<0.0001", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```


 

```{r, echo=F, message=F, warning=F, eval=F}

* fisher exact test is used 
* lower, upper indicate 95% confidence interval

#################### print out summary statistics in a  table 
syn_new_sample_summary=tibble()
p.value=numeric(); lower_bound=numeric(); upper_bound=numeric(); OR=numeric()
num_proband=6026+1265 # male proband+female proband 
for (i in 1:length(transmitted_new_sample))
{
  fisher_test=fisher.test(matrix(c(transmitted_new_sample[i], num_proband, nontransmitted_new_sample[i], num_proband), nrow=2))
  p.value[i]=fisher_test$p.val
  lower_bound[i]=fisher_test$conf.int[1]
  upper_bound[i]=fisher_test$conf.int[2]
  OR[i]=fisher_test$estimate
}
syn_new_sample_summary=tibble(gene=Gene_Set,transmitted=transmitted_new_sample, nontransmitted=nontransmitted_new_sample, OR=round(OR,2), pvalue=round(p.value,4), lower=round(lower_bound,2), upper=round(upper_bound,2))
kable(syn_new_sample_summary, caption = "")%>%
kable_styling() %>%
scroll_box(height = "200px")
```

```{r, echo=F, message=F, warning=F, eval=F}

### filter QD and AS_SOR 


* `Gnomad_non_neuro_AF<0.05`
* `Consequence=="synonymous_variant"`
* `QD>=1  & AS_SOR <=3 & AS_ReadPosRankSum>-0.8`, filter used for SNP by Kyle

```

```{r, echo=F, message=F, warning=F, eval=F}
rare_syn_variant_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant" & Gnomad_non_neuro_AF<0.05 & QD>=1  & AS_SOR <=3 & AS_ReadPosRankSum>-0.8) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband)  # find syn variants   in new sample data with Gnomad AF threshold 


fisher_test=fisher.test(matrix(c(sum(rare_syn_variant_new_sample$Transmitted_proband), num_proband, sum(rare_syn_variant_new_sample$Untransmitted_proband), num_proband), nrow=2))

fisher_test_result_syn=tibble(OR=fisher_test$estimate, lower=fisher_test$conf.int[1], upper=fisher_test$conf.int[2], var="syn")
ggplot(fisher_test_result_syn, aes(x=var, y=OR, fill=var)) + # draw the OR plot 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=lower, ymax=upper), width=.2,
                 position=position_dodge(.9)) 
```


```{r, echo=F, warning=F, message=F, eval=F}
######### obtain syn variants in new sample data

rare_syn_variant_new_sample=ASC_new_sample %>% filter(Consequence=="synonymous_variant" & Gnomad_non_neuro_AF<0.05 & QD>=1  & AS_SOR <=3 & AS_ReadPosRankSum>-0.8) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband)  # find syn variants   in new sample data with Gnomad AF threshold 

transmitted_new_sample=numeric()
nontransmitted_new_sample=numeric()
for ( i in 1:length(Gene_Set_List))
 {
  Feature_Data=rare_syn_variant_new_sample%>%filter(Gene %in% Gene_Set_List[[i]])%>% drop_na()
  if (nrow(Feature_Data)>0)
  { 
    transmitted_new_sample[i]=Feature_Data%>%select(Transmitted_proband)%>%sum()
    nontransmitted_new_sample[i]=Feature_Data%>%select(Untransmitted_proband)%>%sum()
    
     ##############  run mirage-vs on every variant set, takes several minutes to run
#  Feature_Data_VS=cell_variant_set%>% filter(GENE_NAME %in% Gene_Set_List[[i]])%>% select(Transmitted_proband,Untransmitted_proband) %>% add_column(Group_Index=1)%>% drop_na()
#  Burden_MIRAGE_VS[i,1:2]=c(transmitted_cell_sample[i], nontransmitted_cell_sample[i])
#  pois.test=poisson.test(Burden_MIRAGE_VS[i,1:2], c(5869, 5869), r=1, alternative="greater")
# Burden_MIRAGE_VS[i,3:4]=c(pois.test$estimate,pois.test$p.value)
 
#  vs.mirage=mirage_vs(data.frame(Feature_Data_VS),n1=5869,n2=5869)
#  Burden_MIRAGE_VS[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
  
  ##############
    
  }
  }  ### end of i

#write.csv(Burden_MIRAGE_VS, file="../output/CellDataASC/Variant_Set_missense_10Geneset.csv")

syn_new_sample=tibble(Gene=rep(Gene_Set,2), status=rep(c("transmitted", "nontransmitted"), each=10), variant_count=c(transmitted_new_sample, nontransmitted_new_sample))

g_new_sample=ggplot(syn_new_sample, aes(x = Gene, y = variant_count, fill = status)) +
  geom_col(position = position_dodge())+
  #geom_bar(stat="identity")+
  theme_classic()+
  ylim(c(0,30000))+
  ylab("Variant count")+xlab("Gene set")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))+
theme(text = element_text(size=10))+
ggtitle("syn variants in new sample")+
  theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title
g_new_sample
############ combine two figures into one panel 
#figure=ggarrange( g_old_sample, g_new_sample, common.legend = TRUE, legend="right") 
#annotate_figure(figure,
#                top = text_grob("missense variants; AF<0.0001", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```



```{r, echo=F, message=F, warning=F, eval=F}

* fisher exact test is used 
* lower, upper indicate 95% confidence interval 

#################### print out summary statistics in a  table 
syn_new_sample_summary=tibble()
p.value=numeric(); lower_bound=numeric(); upper_bound=numeric()
num_proband=6026+1265 # male proband+female proband 
OR=numeric()
for (i in 1:length(transmitted_new_sample))
{
  fisher_test=fisher.test(matrix(c(transmitted_new_sample[i], num_proband, nontransmitted_new_sample[i], num_proband), nrow=2))
  p.value[i]=fisher_test$p.val
  lower_bound[i]=fisher_test$conf.int[1]
  upper_bound[i]=fisher_test$conf.int[2]
  OR[i]=fisher_test$estimate
}
syn_new_sample_summary=tibble(gene=Gene_Set,transmitted=transmitted_new_sample, nontransmitted=nontransmitted_new_sample, OR=round(OR,2), pvalue=round(p.value,4), lower=round(lower_bound,2), upper=round(upper_bound,2))
kable(syn_new_sample_summary, caption = "")%>%
kable_styling() %>%
scroll_box(height = "200px")
```

