---
title: "Cell data"
output: html_document
---


```{r, echo=F}
rm(list=ls())
set.seed(123)
library(tidyverse)
library(knitr)
#library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
```

This is the transmitted data from [cell paper](https://www.sciencedirect.com/science/article/abs/pii/S0092867419313984)

```{r, echo=F, cache=T}
part1=as_tibble(read.table("../../AutismDatafromASC_Cellpaper/ASC_v15_TDT_GQPL25.hAB30.results.2018-04-01.part1.tsv.gz",sep = '\t', header = TRUE, fill = TRUE))
part2=as_tibble(read.table("../../AutismDatafromASC_Cellpaper/ASC_v15_TDT_GQPL25.hAB30.results.2018-04-01.part2.tsv.gz",sep = '\t', header = TRUE, fill = TRUE))
cell_trans_data=full_join(part1, part2)
```

```{r, echo=T}
cell_trans_data
colnames(cell_trans_data)
```

## Annotations in the data

* PTV: protein-truncating variants including nonsense, frameshift, and essential splice sites variants
* Missense 
* pLI: probability of loss of function intolerance 
* MPC: missense badness, PolyPhen-2, constraint score 

## Replicate Burden analysis for transmitted variants in Figure 1B 

```{r, echo=F}
trans_count=matrix(nrow=2, ncol=3)
rownames(trans_count)=c("transmitted", "untransmitted")
colnames(trans_count)=c("pLI>=0.995", "0.5<pLI<0.995", "MPC>=2")
###### pLI>=0.995
trans_count[1,1]=#cell_trans_data %>% filter(pLI>=0.995) %>% summarise(Transmitted_sibling) %>% sum(na.rm=T)+
cell_trans_data %>% filter(pLI>=0.995) %>% filter(PTV==T) %>% summarise(Transmitted_proband)%>%sum(na.rm=T) ### sum silbing and proband
trans_count[2,1]=#cell_trans_data %>% filter(pLI>=0.995) %>% summarise(Untransmitted_sibling) %>% sum(na.rm=T)+
cell_trans_data %>% filter(pLI>=0.995) %>% filter(PTV==T) %>% summarise(Untransmitted_proband)%>%sum(na.rm=T)
##### 0.5<pLI<0.95
trans_count[1,2]=#cell_trans_data %>% filter(pLI<0.995 & pLI>0.5) %>% summarise(Transmitted_sibling) %>% sum(na.rm=T)+
  cell_trans_data %>% filter(pLI<0.995 & pLI>0.5) %>% filter(PTV==T) %>% summarise(Transmitted_proband)%>%sum(na.rm=T)
trans_count[2,2]=#cell_trans_data %>% filter(pLI<0.995 & pLI>0.5) %>% summarise(Untransmitted_sibling) %>% sum(na.rm=T)+
  cell_trans_data %>% filter(pLI<0.995 & pLI>0.5) %>% filter(PTV==T) %>% summarise(Untransmitted_proband)%>%sum(na.rm=T)
###### MPC>=2
trans_count[1,3]=#cell_trans_data %>% filter(MPC>=2) %>% summarise(Transmitted_sibling) %>% sum(na.rm=T)+
  cell_trans_data %>% filter(MPC>=2) %>% summarise(Transmitted_proband)%>%sum(na.rm=T)
trans_count[2,3]=#cell_trans_data %>% filter(MPC>=2) %>% summarise(Untransmitted_sibling) %>% sum(na.rm=T)+
  cell_trans_data %>% filter(MPC>=2) %>% summarise(Untransmitted_proband)%>%sum(na.rm=T)
no.prob=5869; no.silb=1993

########## burden test
burden=matrix(nrow=3, ncol=3)
rownames(burden)=c("OR", "RR", "p.value")
colnames(burden)=c("pLI>=0.995", "0.5<pLI<0.995", "MPC>=2")
for (i in 1:ncol(trans_count))
  { 
  test=binom.test(trans_count[,i],alternative="greater")
  burden[1,i]=trans_count[1,i]/(no.prob-trans_count[1,i])/(trans_count[2,i]/(no.prob-trans_count[2,i]))
  burden[2,i]=trans_count[1,i]/trans_count[2,i]
  burden[3,i]=test$p.value
}
kable(trans_count, caption="variant count of transmitted PTV and MPC")
kable(burden, caption="Enrichment of transmited PTV and MPC in proband using binomial exact test")
```

In the paper, when $pLI>=0.995$ for PTV, there are 695 transmitted vs 557 untransmitted from 5869 parents, leading to RR 1.2 and p value 0.07?, and  these numbers are close to that in the above table. 

```{r, echo=F, eval=F}
# in Figure1 B. for PTV with pLI>=0.995, there are 695 transmitted vs 557 untransmitted in 5869 parents, thus 
RR=round(695/5869/(557/5869),1)
# how come the p value is 0.07? 
```

## Annotate the data by ANNOVAR

Since the data doesn't have allele frequency, use ANNOVAR to annotate it 

```{r, echo=T}
head(cell_trans_data)
```

### prepare the data in the appropriate format  for annovar 

To run annovar, the simplist format has 5 columns chr, star, end, ref, alt and each row is one variant. 

1.  use R to extract 5 required columns 

```{r, echo=T, eval=F}

cell_data=read.table("cell_trans_data.txt", header=T, fill=T) # note "fill" some lines with less than 38 elements
locus=matrix(cell_data$Variant)
split_locus=strsplit(locus, ":") # ":"  separates chr, start and end 
avinput=matrix(nrow=length(split_locus), ncol=5)  # format of avinput; chr, start, end, ref, alt
for (i in 1:length(split_locus))
  { 
   avinput[i,1:2]=split_locus[[i]][1:2]
   avinput[i,3]=split_locus[[i]][2]
   avinput[i,4:5]=split_locus[[i]][3:4]
}

#write.table(avinput, file="avinput.txt",row.names=F, col.names=F, quote=F)

```

2.  then transfer into tab-delimited columns

```{r, echo=T, eval=F}
head avinput.txt
1 861283 861283 G C
1 861287 861287 C T
1 861289 861289 G A
1 861341 861341 A G
1 861389 861389 C T
1 861398 861398 G A
1 865635 865635 G A
1 865645 865645 T G
1 865655 865655 T G
1 865686 865686 A G
```


```{r, echo=T, eval=F}
awk '{ for(i=1;i<=NF;i++){if(i==NF){printf("%s\n",$NF);}else {printf("%s\t",$i)}}}'
input.data > output.bed
```

3.  download relevant database 

```{r, echo=T, eval=F}
perl annotate_variation.pl -buildver hg19 -downdb -webfrom annovar exac03 humandb/

```


4.  run annovar function 

```{r, echo=T, eval=F}
perl annotate_variation.pl -filter -build hg19 -dbtype exac03 avinput humandb/
  # the output file with AF annotated is avinput.hg19_exac03_dropped
```


## Refine annotations 

Stratify six variant groups PTV (PLI>=0.995, 0.5-0.995, <0.5) and missense (MPC>=2, 1-2, 0-1) by AF (0.01-0.05, 0.001-0.01, <0.001), resulting in 18 categories. 

```{r, echo=F, cache=T}
## add Exac Allele frequency to the data
avinput.hg19_exac03_dropped=as_tibble(read.table("../../AutismDatafromASC_Cellpaper/avinput.hg19_exac03_dropped"))
ASC_ExacAF=tibble(Variant=paste(avinput.hg19_exac03_dropped$V3, avinput.hg19_exac03_dropped$V4, avinput.hg19_exac03_dropped$V6, avinput.hg19_exac03_dropped$V7, sep=":"), ExacAF=avinput.hg19_exac03_dropped$V2)
cell_trans_data_AF=left_join(cell_trans_data, ASC_ExacAF)
cell_trans_data_AF$ExacAF[is.na(cell_trans_data_AF$ExacAF)==T]=0 # fill NA AF with 0
hist(cell_trans_data_AF$ExacAF)
summary(cell_trans_data_AF$ExacAF)
```

```{r, echo=F}
## partition vriants into disjoint groups 

variant_groups=list()
variant_groups[[1]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.01 & ExacAF<=0.05)
variant_groups[[2]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.001 & ExacAF<=0.01)
variant_groups[[3]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter( ExacAF<=0.001)

variant_groups[[4]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.01 & ExacAF<=0.05)
variant_groups[[5]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.001 & ExacAF<=0.01)
variant_groups[[6]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter( ExacAF<=0.001)

variant_groups[[7]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter(ExacAF>0.01 & ExacAF<=0.05)
variant_groups[[8]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter(ExacAF>0.001 & ExacAF<=0.01)
variant_groups[[9]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter( ExacAF<=0.001)

variant_groups[[10]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter(ExacAF>0.01 & ExacAF<=0.05)
variant_groups[[11]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter(ExacAF>0.001 & ExacAF<=0.01)
variant_groups[[12]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter( ExacAF<=0.001)

variant_groups[[13]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(ExacAF>0.01 & ExacAF<=0.05)
variant_groups[[14]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(ExacAF>0.001 & ExacAF<=0.01)
variant_groups[[15]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter( ExacAF<=0.001)

variant_groups[[16]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter(ExacAF>0.01 & ExacAF<=0.05)
variant_groups[[17]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter(ExacAF>0.001 & ExacAF<=0.01)
variant_groups[[18]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter( ExacAF<=0.001)

count_variant_groups=matrix(nrow=18, ncol=2)
colnames(count_variant_groups)=c("transmitted", "untransmitted")
rownames(count_variant_groups)=paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=3), c("0.01<AF<=0.05", "0.001<AF<=0.01", "AF<=0.001"), sep="&")
for (i in 1:length(variant_groups))
  if (nrow(variant_groups[[i]])>0)
    count_variant_groups[i,]=c(variant_groups[[i]]%>% summarise(Transmitted_proband)%>%sum(na.rm=T), variant_groups[[i]]%>% summarise(Untransmitted_proband)%>%sum(na.rm=T))
count_variant_groups=count_variant_groups[complete.cases(count_variant_groups), ]
count_variant_groups
```
It looks like all variants in the data are filtered with cutoff, like $0.1%$. 

## Genome-wide analysis

### Run mirage-vs on 6 variant groups in cell paper 



```{r, echo=F}
# add one column of variant group index 
cell_trans_data_AF_GI=cell_trans_data_AF %>% add_column(Group_Index=NA)
cell_trans_data_AF_GI$Group_Index[which(cell_trans_data_AF_GI$pLI>=0.995 & cell_trans_data_AF_GI$PTV==T)]=1
cell_trans_data_AF_GI$Group_Index[which(cell_trans_data_AF_GI$pLI>=0.5 & cell_trans_data_AF_GI$pLI<0.995 &  cell_trans_data_AF_GI$PTV==T)]=2
cell_trans_data_AF_GI$Group_Index[which(cell_trans_data_AF_GI$pLI<0.5 &  cell_trans_data_AF_GI$PTV==T)]=3
cell_trans_data_AF_GI$Group_Index[which(cell_trans_data_AF_GI$MPC>=2)]=4
cell_trans_data_AF_GI$Group_Index[which(cell_trans_data_AF_GI$MPC>=1 & cell_trans_data_AF_GI$MPC<2)]=5
cell_trans_data_AF_GI$Group_Index[which(cell_trans_data_AF_GI$MPC<1)]=6
#cell_trans_data_AF_GI=cell_trans_data_AF_GI %>% drop_na()
```

 

Use mirage-vs to test if there are signals in six variant groups  

```{r, echo=F, eval=F}
####### test the toy example 
#data(mirage_vs_toy)
#res_vs = mirage_vs(mirage_vs_toy,n1=1000,n2=1000)
#########
para_est=matrix(nrow=6, ncol=6)
rownames(para_est)=c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1")
colnames(para_est)=c("No_case", "No_contr", "OR", "pval", "eta_est", "eta_pval")
for (i in 1:6)
{ 
#i=1  
 cat(i, "is running", "\n")
 variant_set=cell_trans_data_AF_GI %>% filter(Group_Index==i) %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
 para_est[i,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
 pois.test=poisson.test(para_est[i,1:2], c(5869, 5869), r=1, alternative="greater")
 para_est[i,3:4]=c(pois.test$estimate,pois.test$p.value)
 if (i<=3) # for LoF variants
   vs.mirage=mirage_vs(variant_set[,2:4],n1=5869,n2=5869, gamma=6)
 if (i>3)  # for missense variants 
   vs.mirage=mirage_vs(variant_set[,2:4],n1=5869,n2=5869, gamma=3)
 para_est[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
  
}
as_tibble(para_est)
#write.csv(para_est, file="../output/CellDataASC/para.est.var.group.proband.csv")
#gene.data=eight.partition(cand.data)
#ds.asd.result=mirage(gene.data, n1=2403, n2=2403)
```

This figure shows the comparison between burden test (one side two sample poisson test) and mirage-vs for proband in 6 variant groups. $\gamma=6$ for LoF variant and $\gamma=3$ for missense when running mirage.  

```{r, echo=F}
proband_para_est=read.csv("../output/CellDataASC/para.est.var.group.proband.csv")
as_tibble(proband_para_est)


var_group=c(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"), 2))
method=c(rep("Burden", 6), rep("mirage-vs", 6))
pvalue=c(round(-log(proband_para_est$pval, base=10), 2), round(-log(proband_para_est$eta_pval, base=10), 2))
summary=data.frame(var.group=var_group, pvalue=pvalue, method=method)
ggplot(summary, aes(x=var.group, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", 
           position=position_dodge())+
  ylab(expression(-log[10](pvalue)))+
  theme(legend.title=element_blank())+
  xlab(" variant groups")+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))  #center the title 
  #ylab(expression(paste(hat(eta))))+ 
  #guides(fill=guide_legend(title=NULL))+
  #ylim(0.05,0.15)+
  # geom_hline(yintercept=0.1,linetype="dashed", color = "red")
```

### gene set analysis 

Run mirage on all 17653 genes and $\gamma=6$ for LoF variants and $\gamma=3$ for missense. 


```{r, echo=F, eval=F}
############  gene set 
gene_set=cell_trans_data_AF_GI %>% select(Variant, GENE_NAME, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
mirage_result=mirage(data.frame(gene_set),n1=5869, n2=5869, gamma=c(6,6,6,3,3,3)) # input data must be dataframe 
######### It takes about one day to finish running mirage function 

#res=mirage(mirage_toy,n1=4315,n2=4315) # test mirage function 
#write.csv(mirage_result$BF.PP.gene, file="../output/CellDataASC/BF.proband.AllGene.csv")
```

parameter | estimate |  p value |
-----|-----|------|--------
$\hat{\delta}$ | 0.15 | $1.64\times 10^{-8}$
$\hat{\eta}_1$ | 1    | $3.52\times 10^{-6}$
$\hat{\eta}_2$ | 0.48    | 0.03
$\hat{\eta}_3$ | 0.12    | 0.01
$\hat{\eta}_4$ | 0.20    | 0.05
$\hat{\eta}_5$ | 0.02    | 0.46
$\hat{\eta}_6$ | 0.0014    | 1


```{r, echo=F}
BF_proband_AllGene=as_tibble(read.csv("../output/CellDataASC/BF.proband.AllGene.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
BF_proband_AllGene_sorted
#BF_proband_AllGene_sorted %>% filter(Gene %in% ASD_102_Gene) %>% print(n=102)
```

## Run mirage on 102 ASD genes in cell paper

```{r, echo=F}
ASD_Gene_Cell=as_tibble(read.csv("../../AutismDatafromASC_Cellpaper/1-s2.0-S0092867419313984-mmc2.csv", header = TRUE))
ASD_102_Gene=ASD_Gene_Cell[1:102,] %>% pull(ï..gene)
```

```{r, echo=F, eval=F}
#########
para_est=matrix(nrow=6, ncol=6)
rownames(para_est)=c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1")
colnames(para_est)=c("No_case", "No_contr", "OR", "pval", "eta_est", "eta_pval")
for (i in 1:6)
{ 
#i=2  
#cat(i, "is running", "\n")
############  variant set 
variant_set=cell_trans_data_AF_GI %>% filter(GENE_NAME %in% ASD_102_Gene)%>% filter(Group_Index==i) %>% select(Variant, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
para_est[i,1:2]=c(sum(variant_set$Transmitted_proband),sum(variant_set$Untransmitted_proband))
pois.test=poisson.test(para_est[i,1:2], c(5869, 5869), r=1, alternative="greater")
para_est[i,3:4]=c(pois.test$estimate,pois.test$p.value)
vs.mirage=mirage_vs(variant_set[,2:4],n1=5869,n2=5869, gamma=6)
para_est[i,5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
}
as_tibble(para_est)
#write.csv(para_est, file="../output/CellDataASC/para.est.var.group.proband.102ASDGene.csv")
#gene.data=eight.partition(cand.data)
#ds.asd.result=mirage(gene.data, n1=2403, n2=2403)
```

### VS analysis

This figure shows the comparison between burden test (one side two sample poisson test) and mirage-vs for proband in 6 variant groups. $\gamma=6$ for LoF variant and $\gamma=3$ for missense when running mirage.

```{r, echo=F}
proband_para_est=read.csv("../output/CellDataASC/para.est.var.group.proband.102ASDGene.csv")
as_tibble(proband_para_est)


var_group=c(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"), 2))
method=c(rep("Burden", 6), rep("mirage-vs", 6))
pvalue=c(round(-log(proband_para_est$pval, base=10), 2), round(-log(proband_para_est$eta_pval, base=10), 2))
summary=data.frame(var.group=var_group, pvalue=pvalue, method=method)
ggplot(summary, aes(x=var.group, y=pvalue, fill=method)) + 
   geom_bar(stat="identity", 
           position=position_dodge())+
  ylab(expression(-log[10](pvalue)))+
  theme(legend.title=element_blank())+
  xlab(" variant groups")+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))  #center the title 
  #ylab(expression(paste(hat(eta))))+ 
  #guides(fill=guide_legend(title=NULL))+
  #ylim(0.05,0.15)+
  # geom_hline(yintercept=0.1,linetype="dashed", color = "red")
```

### gene set analysis

```{r,echo=F, eval=F}
############  gene set 
gene_set=cell_trans_data_AF_GI %>% filter(GENE_NAME %in% ASD_102_Gene) %>% select(Variant, GENE_NAME, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
mirage_result=mirage(data.frame(gene_set),n1=5869, n2=5869) # input data must be dataframe
#res=mirage(mirage_toy,n1=4315,n2=4315) # test mirage function 

```


