---
title: "ASD_dbGaP"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(rlang)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(ggrepel)
library(forcats)
library(scales)
```

Four cohort studies 


Group | No.subject | No.var called 
-------|-----------|----------
DS-ASD | 10008 | 3452056 (locus with observed alleles)
GRU    | 2509
DS-AOND-MDS| 162
HMB-MDS | 93 
Total | 12772 



## DS-ASD sample  

[Sample data](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000298.v4.p3) has 10008 individuals 




## old sample vs DS-ASD sample


```{r, echo=F, cache=T}
path1="C:\\Shengtong\\Research\\rare-var\\"
path2="C:\\Shengtong\\Research\\rare-var\\73416-DS-ASD\\"
All.Anno.Data=as_tibble(read.table(paste(path1, "AnnotatedTrans.txt", sep=""), header=T))
DS.ASD.sample=as_tibble(read.csv(paste(path2, "73416-DS-ASD.sample.csv", sep=""), header=T)[,-1])

```

### sample decomposition  by AF 


```{r, echo=F}

All.Anno.count=numeric(); DS.ASD.count=numeric()

All.Anno.count[1]=nrow(All.Anno.Data); DS.ASD.count[1]=nrow(DS.ASD.sample)
All.Anno.count[2]=nrow(All.Anno.Data %>% filter(ExacAF<0.001)); DS.ASD.count[2]=nrow(DS.ASD.sample %>% filter(ExacAF<0.001))
All.Anno.count[3]=nrow(All.Anno.Data %>% filter(ExacAF>0.001 & ExacAF<0.01)); DS.ASD.count[3]=nrow(DS.ASD.sample %>% filter(ExacAF>0.001 & ExacAF<0.01))
All.Anno.count[4]=nrow(All.Anno.Data %>% filter(ExacAF>0.01 & ExacAF<0.05)); DS.ASD.count[4]=nrow(DS.ASD.sample %>% filter(ExacAF>0.01 & ExacAF<0.05))
All.Anno.count[5]=nrow(All.Anno.Data %>% filter(ExacAF>0.05))
DS.ASD.count[5]=nrow(DS.ASD.sample %>% filter(ExacAF>0.05))

var.count=tibble(cutoff=rep(c("All", "AF<0.1%", "0.1%<AF<1%", "1%<AF<5%", "AF>5%"),2), group=rep(c("old", "DS-ASD"),each=5), counts=c(All.Anno.count, DS.ASD.count) )

ggplot(data=var.count, aes(x=cutoff, y=counts, fill=group)) +
geom_bar(stat="identity", position=position_dodge())
var.count
```



```{r, echo=F}

old.sample.label=var.count%>% filter(group=="old")%>% .$cutoff
label=old.sample.label[-1]
old.sample.value=var.count%>% filter(group=="old")%>% .$counts

old.sample <- structure(list(Group = structure(c(1L, 2L, 3L, 4L), .Label = label, class = "factor"), value = old.sample.value[-1]), .Names = c("Group", "value"), class = "data.frame", row.names = c("1", "2", "3", "4"))

old.sample=old.sample %>% arrange(desc(value)) %>% mutate(prop = paste(round(value/sum(value),4)*100, "%", sep="")) 

pie <- ggplot(old.sample, aes(x = "", y = value, fill = fct_inorder(Group))) +
       geom_bar(width = 1, stat = "identity") +
       coord_polar("y", start = 0) +
       geom_label_repel(aes(label = prop), size=5, show.legend = F, nudge_x = 1) +
       guides(fill = guide_legend(title = "Group"))+
       ggtitle("Old sample")+
       theme(plot.title = element_text(hjust = 0.5)) # centering ggplot title 

pie



DS_ASD.sample.value=var.count%>% filter(group=="DS-ASD")%>% .$counts
DS_ASD.sample <- structure(list(Group = structure(c(1L, 2L, 3L, 4L), .Label = label, class = "factor"), value = DS_ASD.sample.value[-1]), .Names = c("Group", "value"), class = "data.frame", row.names = c("1", "2", "3", "4"))

DS_ASD.sample=DS_ASD.sample %>% arrange(desc(value)) %>% mutate(prop = paste(round(value/sum(value),4)*100, "%", sep="")) 

pie <- ggplot(DS_ASD.sample, aes(x = "", y = value, fill = fct_inorder(Group))) +
       geom_bar(width = 1, stat = "identity") +
       coord_polar("y", start = 0) +
       geom_label_repel(aes(label = prop), size=5, show.legend = F, nudge_x = 1) +
       guides(fill = guide_legend(title = "Group"))+
  ggtitle("DS-ASD sample")+
  theme(plot.title = element_text(hjust = 0.5)) # centering ggplot title 

pie

```


### several top genes 

```{r, echo=F}
DS.ASD.rare.var=DS.ASD.sample%>% filter(ExacAF<0.05 & Annotation!="synonymous SNV")
All.Anno.rare.var=All.Anno.Data%>% filter(ExacAF<0.05 & Annotation!="synonymous SNV")

gene.set=c("CYFIP1", "EP400", "FBN1", "SRCAP", "DYNC1H1", "ABCA2", "DOCK4", "CACNA1D", "CHD8", "TRIP12")
DS.ASD.gene.set=matrix(nrow=length(gene.set), ncol=2)
All.Anno.gene.set=matrix(nrow=length(gene.set), ncol=2)
for (i in 1:length(gene.set))
{
DS.ASD.gene.set[i,1]=DS.ASD.rare.var%>% filter(Gene==gene.set[i])%>% summarise(count=sum(No.case))%>%as.numeric()
DS.ASD.gene.set[i,2]=DS.ASD.rare.var%>% filter(Gene==gene.set[i])%>% summarise(count=sum(No.contr))%>%as.numeric()  
  
All.Anno.gene.set[i,1]=All.Anno.rare.var%>% filter(Gene==gene.set[i])%>% summarise(count=sum(No.case))%>%as.numeric()
All.Anno.gene.set[i,2]=All.Anno.rare.var%>% filter(Gene==gene.set[i])%>% summarise(count=sum(No.contr))%>%as.numeric() 
}
combine=rbind(DS.ASD.gene.set, All.Anno.gene.set)
colnames(combine)=c("No.case", "No.control")
rownames(combine)=c(paste("DS.ASD", gene.set), paste("old.sample", gene.set))
combine

combine.gene.set=tibble(Gene=rep(gene.set,4), status=c(rep(c("case", "control"), each=10), rep(c("case", "control"), each=10)), sample=rep(c("DS.ASD", "old.sample"), each=20), no.var=c(combine[1:10,1], combine[1:10,2], combine[11:20,1], combine[11:20,2]))
ggplot(combine.gene.set, aes(x = Gene, y = no.var, fill = status)) + 
  geom_bar(stat="identity") + 
  facet_grid(.~sample) +
  theme(axis.text.x = element_text(angle = 90, size = 8))

```


### Gene CYFIP1


```{r, echo=T}
All.Anno.rare.var%>%filter(Gene=="CYFIP1", ID=="15:22963816-22963816_A_G")
DS.ASD.rare.var%>%filter(Gene=="CYFIP1", ID=="chr15:22963816_22963817_A_G")
```

\ | No.case| No.control |
---|----------|-------------|-
old | 8 | 0  
DS-ASD | 6 | 2  
Table: chr15:22963816-22963816_A~G



### QC for annotations 

```{r, echo=F}
# Data is a tibble having columns"Annotation"; 
# N1: sample size in case: N0: sample size in control 
missing.ratio=function(Data, N1, N0)  # check the ratio of missingness in the data
{  
  nosyn.data=Data %>% filter(Annotation!="synonymous SNV")
  nonsyn.data=Data %>% filter(Annotation=="nonsynonymous SNV")
Anno.fea=paste(rep("ratio_valid_value_", ncol(Data)), colnames(Data), sep=""); Anno.fea[1:4]=c("#{Variants}", "#{Gene}", "rate.ca", "rate.co")
ratio.data=numeric(); ratio.nosyn=numeric(); ratio.nonsyn=numeric()
for (i in 5:length(Anno.fea))
{
  ratio.data[i]=1-sum(is.na(Data[,i]))/nrow(Data)
  ratio.nosyn[i]=1-sum(is.na(nosyn.data[,i]))/nrow(nosyn.data)
  ratio.nonsyn[i]=1-sum(is.na(nonsyn.data[,i]))/nrow(nonsyn.data)
}  
ratio.data[1]=nrow(Data); ratio.data[2]=length(unique(Data$Gene))
ratio.data[3]=sum(Data$No.case)/N1; ratio.data[4]=sum(Data$No.contr)/N0

ratio.nosyn[1]=nrow(nosyn.data); ratio.nosyn[2]=length(unique(nosyn.data$Gene))
ratio.nosyn[3]=sum(nosyn.data$No.case)/N1; ratio.nosyn[4]=sum(nosyn.data$No.contr)/N0

ratio.nonsyn[1]=nrow(nonsyn.data); ratio.nonsyn[2]=length(unique(nonsyn.data$Gene))
ratio.nonsyn[3]=sum(nonsyn.data$No.case)/N1; ratio.nonsyn[4]=sum(nonsyn.data$No.contr)/N0

data.summ=data.frame(item=Anno.fea, value.AllVar=ratio.data, value.withoutsyn=ratio.nosyn, value.nonsyn=ratio.nonsyn)

return(data.summ)
}

all.anno=missing.ratio(All.Anno.Data, N1=4315, N0=4315)
ds.asd=missing.ratio(DS.ASD.sample, N1=2403, N0=2403)
library(knitr)
kable(all.anno, caption="Old sample")
kable(ds.asd, caption="DS-ASD sample")
```


## Gene set analysis 

```{r defined function, echo=F, results='hide'}
test.func=function(evid, Data, N1, N0) # given evid, and sample size, perform the burden analysis
{
evid.data=Data[Data$ID %in% evid,]
count=c(sum(evid.data$No.case), sum(evid.data$No.contr))
Time=c(N1, N0)
pois.test=poisson.test(count, Time, r=1, alternative="greater")
return(result=list(odds.ratio=pois.test$estimate, p.value=pois.test$p.value, rate.case=count[1]/N1, rate.contr=count[2]/N0))
}
```


```{r, echo=F}
eight.partition=function(cand.data) # given gene data and annotations, do variant partitions
{
  par.evid=list()
  LoF.def=c("stopgain", "frameshift substitution", "splicing", "stoploss")
  par.evid[[1]]=which(cand.data$Annotation %in% LoF.def==T & cand.data$ExacAF<0.05 & cand.data$ExacAF>=0.01 )
  par.evid[[2]]=which(cand.data$Annotation %in% LoF.def==T &  cand.data$ExacAF<0.01)

  par.evid[[3]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
  par.evid[[4]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
  par.evid[[5]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 &  cand.data$ExacAF<0.001)

  par.evid[[6]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
  par.evid[[7]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
  par.evid[[8]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF<0.001)

  group.index=rep(NA, nrow(cand.data))
  for (i in 1:length(par.evid))
    group.index[par.evid[[i]]]=i
  gene.data=data.frame(ID=cand.data$ID, Gene=cand.data$Gene, No.case=cand.data$No.case, No.contr=cand.data$No.contr, group.index=group.index)
  gene.data=gene.data[complete.cases(gene.data),]
  return(gene.data)
}
```

```{r set paramters and modifying the data, echo=T, results='hide'}
N1=2403; N0=2403
DS.ASD.sample[DS.ASD.sample =="."] <- NA
DS.ASD.sample$ExacAF[is.na(DS.ASD.sample$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=DS.ASD.sample[which(DS.ASD.sample$ExacAF<0.05 & DS.ASD.sample$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonumous SNV
var.data=as_tibble(data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr))
LoF.def=c("stopgain", "frameshift substitution", "splicing", "stoploss")
LoF.var=as.character(Anno.Data$ID[which(Anno.Data$Annotation %in% LoF.def==T)]) 
```



### Simple burden test 

```{r burdern analysis-gene level, results='hide', echo=F}
GeneDB=src_sqlite("C:\\Shengtong\\Research\\rare-var\\gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
IDGene=data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
ASD.gene=data.frame(collect(tbl(GeneDB, "AutismKB_gene")))
constraint.gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS.gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene 
haploinsuff.gene=data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))
gene.set=c("ID gene","High conf", "Mod conf", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff gene", "SCZ gene", "Olfac.gene")
gene.fea=c("cate1", "cate2", "cate3", "cate4", "cate5", "cate6", "cateS", "TADAq<5%", "TADAq<20%", "TADAq<30%", "TADAq<40%", "TADAq<50%", "TADAq>90%", gene.set); max.gene=length(gene.fea)
gene.summy=matrix(nrow=max.gene+1, ncol=4)
gene.evid=list(); var.index=1
gene.evid[[1]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate1$GeneID )])
gene.evid[[2]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate2$GeneID )])
gene.evid[[3]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate3$GeneID )])
gene.evid[[4]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate4$GeneID )])
gene.evid[[5]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate5$GeneID )])
gene.evid[[6]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate6$GeneID )])
gene.evid[[7]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cateS$GeneID )])
gene.evid[[8]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan5percentgene )])
gene.evid[[9]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan20percentgene )])
gene.evid[[10]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan30percentgene )])
gene.evid[[11]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan40percentgene)])
gene.evid[[12]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan50percentgene )])
gene.evid[[13]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlargerthan90percentgene )])
gene.evid[[14]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% IDGene$GeneID )])
high.conf=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
mod.conf=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
gene.evid[[15]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% high.conf)])
gene.evid[[16]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% mod.conf )])
psd.gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
FMRP.gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
gene.evid[[17]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% psd.gene )])
gene.evid[[18]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% FMRP.gene )])
gene.evid[[19]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% ASD.gene$GeneID )])
gene.evid[[20]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% constraint.gene )])
gene.evid[[21]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% RVIS.gene )])
gene.evid[[22]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% haploinsuff.gene$GeneID )])
gene.evid[[23]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% purcell.genelist$Gene_symbol )])
olfac.gene=data.frame(collect(tbl(GeneDB, "Olfac_gene")))
gene.evid[[24]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% olfac.gene$GeneID )])
#sczgene=as.character(read.table("D:\\ResearchWork\\StatisticalGenetics\\NumericAnalysis\\RealData\\SCZData\\GeneList\\SCZ.67gene.q0.3.txt", header=T)[[1]])
#gene.evid[[25]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% sczgene )])

colnames(gene.summy)=c("OR", "p.value", "rate.ca", "rate.co")
rownames(gene.summy)=c(gene.fea, "67SCZriskgene")
LoF.summy=gene.summy
colnames(LoF.summy)=c("OR", "p.value", "No.ca", "No.co")
for (gene in 1:(max.gene))
{
  cat(gene, "is running", "\n")
  pois.test=test.func(gene.evid[[gene]], var.data, N1, N0)
  gene.summy[gene,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
  
  LoF.test=test.func(intersect(gene.evid[[gene]], LoF.var), var.data, N1, N0)
  LoF.summy[gene,]=c(LoF.test$odds.ratio, LoF.test$p.value, LoF.test$rate.case*N1, LoF.test$rate.contr*N0)
}

```


```{r gene level, echo=F}
library(kableExtra)
kable(gene.summy, caption="Burden analysis of gene set", "html")%>%
kable_styling() %>%
scroll_box(height = "200px")

kable(LoF.summy, caption="Burden analysis of LoF in different gene sets", "html")%>%
kable_styling() %>%
scroll_box(height = "200px")
```


```{r, echo=F}
library(viridis)
library(ggsignif)
library(ggpubr)
Gene=row.names(gene.summy)[1:22]
OR=gene.summy[1:22,1]; pval=gene.summy[1:22,2]
result.summary=tibble(Gene=Gene, OR=OR, pval=pval)
# pdf("../../Figure/burden_geneset.pdf")
  ggplot(result.summary, aes(x=Gene, y=OR, fill=Gene))+
  geom_bar(stat="identity")+
  ylab("")+xlab("Gene")+ ggtitle("Burden analysis for gene sets")+
   geom_point(mapping=aes(x=Gene, y=-log(pval, base=10), size=-log(pval, base=10), fill=Gene), alpha=0.8)+ 
    coord_flip()+
    scale_color_viridis(discrete=T)+
    geom_hline(yintercept=1,linetype="dashed")+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
    #theme(legend.position = "none")+  # no legend at all 
    guides(fill = FALSE)+
  theme(plot.title = element_text(hjust = 0.5, size=10)) #center the title 
  
#dev.off()
```


## Constraint genes 

### Cons: Simple burden analysis

```{r, echo=F, results='hide'}
gene.set=as.character(read.csv("C:/Shengtong/Research/rare-var/GeneSet/Samocha_2014NG/Samocha_2014NG_contraintgene.csv", header=T)$gene)
cand.data=as_tibble(Anno.Data[which(Anno.Data$Gene %in% gene.set),])
cand.data=cand.data %>% rename(Polyphen2.HDIV.score=Polyphen2.HDIV)
par.evid=list()
par.evid[[1]]=cand.data %>% filter(Annotation%in%LoF.def)%>% filter(between(ExacAF, 0.01, 0.05))
par.evid[[2]]=cand.data %>% filter(Annotation%in%LoF.def) %>% filter(between(ExacAF, 0, 0.01))
par.evid[[3]]=cand.data %>% filter(Annotation %in% LoF.def==F) %>% filter(between(as.numeric(as.character(Polyphen2.HDIV.score)), 0.957, 1)) %>% filter(between(ExacAF, 0.01, 0.05))
par.evid[[4]]=cand.data %>% filter(Annotation %in% LoF.def==F) %>% filter(between(as.numeric(as.character(Polyphen2.HDIV.score)), 0.957, 1)) %>% filter(between(ExacAF, 0.001, 0.01))
par.evid[[5]]=cand.data %>% filter(Annotation %in% LoF.def==F) %>% filter(between(as.numeric(as.character(Polyphen2.HDIV.score)), 0.957, 1)) %>% filter(between(ExacAF, 0, 0.001))
par.evid[[6]]=cand.data %>% filter(Annotation %in% LoF.def==F) %>% filter(between(as.numeric(as.character(Polyphen2.HDIV.score)), 0, 0.957)) %>% filter(between(ExacAF, 0.01, 0.05))
par.evid[[7]]=cand.data %>% filter(Annotation %in% LoF.def==F) %>% filter(between(as.numeric(as.character(Polyphen2.HDIV.score)), 0, 0.957)) %>% filter(between(ExacAF, 0.001, 0.01))
par.evid[[8]]=cand.data %>% filter(Annotation %in% LoF.def==F) %>% filter(between(as.numeric(as.character(Polyphen2.HDIV.score)), 0, 0.957)) %>% filter(between(ExacAF, 0, 0.001))

burden.summy=matrix(nrow=length(par.evid), ncol=4)
for (i in 1:length(par.evid))
{
  cat(i, "is running", "\n")
  pois.test=test.func(par.evid[[i]]$ID, var.data, N1, N0)
  burden.summy[i,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case*N1, pois.test$rate.contr*N0)
  }

```


```{r, echo=F}
unique.gene=unique(cand.data$Gene)
var.count=numeric()
for (i in 1:length(unique.gene))
{
  single.gene=cand.data[which(cand.data$Gene==unique.gene[i]),]
  var.count[i]=sum(single.gene$No.case)+sum(single.gene$No.contr)
}
```

```{r, echo=F, warning=F}
gene.var.count=data.frame(Gene=unique.gene, var.count=var.count)
ggplot(gene.var.count, aes(x=var.count))+ 
  geom_histogram()+
  geom_vline(aes(xintercept=mean(var.count)),
            color="blue", linetype="dashed", size=1)+
  xlab("#.variant")+
  ylab("#.gene")+
  ggtitle("Histogram of variants in top constraint genes")+
  theme(plot.title = element_text(hjust = 0.5, size=10)) #center the title 

```

The burden of 8 categories is 
```{r, echo=F}
colnames(burden.summy)=c("OR", "p.value", "No.ca", "No.co")
kable(burden.summy, caption="The burden of eight categories", "html")%>%
kable_styling() %>%
scroll_box(height = "200px")
```


```{r, echo=F}
Cate=c("1", "2", "3", "4", "5", "6", "7", "8")
OR=burden.summy[,1]; pval=burden.summy[,2]
result.summary=tibble(Cate=Cate, OR=OR, pval=pval)
result.summary$Cate=result.summary$Cate %>% recode("1"="LoF; 1%<AF<5%", "2"="LoF; AF<1%", "3" = "Damaging; 1%<AF<5%","4"= "Damaging; 0.1%<FA<1%", "5" ="Damaging; AF<0.1%", "6" ="Non-damaging; 1%<AF<5%", "7" ="Non-damaging; 0.1%<AF<1%", "8" ="Non-damaging; AF<0.1%")

 #pdf("C:\\Users\\han24\\OneDrive - UWM\\rare-var\\Figure\\Supp\\Fig-burden_variant_group_of_constgene.pdf")
  ggplot(result.summary, aes(x=Cate, y=OR, fill=Cate))+
  geom_bar(stat="identity")+
  ylab("")+xlab("")+ ggtitle("")+
   geom_point(aes(x=Cate, y=-log(pval, base=10), size=-log(pval, base=10)))+ 
    coord_flip()+
    geom_hline(yintercept=1,linetype="dashed")+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
    #scale_color_viridis(discrete=T)+
    guides(fill = FALSE)+
  theme(plot.title = element_text(hjust = 0.5, size=10))+ #center the title
    theme(axis.text.y = element_text(angle = 0, hjust = 1,size=7))+
    labs(size=expression(paste("-log"[10], "(p value)")))+ # the argument in labs could be size, color, fill depends on which character the legend demonstrate!!!!!!!!
    theme(legend.text = element_text(size = 7))+ # change legend text size 
    theme(legend.title=element_text(size=10))#+
    #theme(legend.position="bottom")
#dev.off()
```



```{r, echo=F, eval=F}
library("devtools")
install.packages("Rtools")
install_github('xinhe-lab/mirage')
library(mirage)
gene.data=eight.partition(cand.data)
ds.asd.result=mirage(gene.data, n1=2403, n2=2403)
gene.data.nogroup1=gene.data[-which(gene.data$group.index=="1"),]
ds.asd.res=mirage(gene.data.nogroup1, n1=2403, n2=2403)
####### for test data 
res= mirage(mirage_toy,n1=1000,n2=1000)
```


```{r, echo=F, eval=F}
 bcftools stats pseudocc.tucc.vcf
# This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
# The command line was:	bcftools stats  pseudocc.tucc.vcf
#
# Definition of sets:
# ID	[2]id	[3]tab-separated file names
ID	0	pseudocc.tucc.vcf
# SN, Summary numbers:
#   number of records   .. number of data rows in the VCF
#   number of no-ALTs   .. reference-only sites, ALT is either "." or identical to REF
#   number of SNPs      .. number of rows with a SNP
#   number of MNPs      .. number of rows with a MNP, such as CC>TT
#   number of indels    .. number of rows with an indel
#   number of others    .. number of rows with other type, for example a symbolic allele or
#                          a complex substitution, such as ACT>TCGA
#   number of multiallelic sites     .. number of rows with multiple alternate alleles
#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
#
#   Note that rows containing multiple types will be counted multiple times, in each
#   counter. For example, a row with a SNP and an indel increments both the SNP and
#   the indel counter.
#
# SN	[2]id	[3]key	[4]value
SN	0	number of samples:	5512
SN	0	number of records:	3452066
SN	0	number of no-ALTs:	35521
SN	0	number of SNPs:	3193922
SN	0	number of MNPs:	0
SN	0	number of indels:	212812
SN	0	number of others:	9811
SN	0	number of multiallelic sites:	0
SN	0	number of multiallelic SNP sites:	0
# TSTV, transitions/transversions:
# TSTV	[2]id	[3]ts	[4]tv	[5]ts/tv	[6]ts (1st ALT)	[7]tv (1st ALT)	[8]ts/tv (1st ALT)
TSTV	0	2250696	942762	2.39	2250696	942762	2.39
# SiS, Singleton stats:
# SiS	[2]id	[3]allele count	[4]number of SNPs	[5]number of transitions	[6]number of transversions	[7]number of indels	[8]repeat-consistent	[9]repeat-inconsistent	[10]not applicable
SiS	0	1	3193458	2250696	942762	212812	0	0	212812
# AF, Stats by non-reference allele frequency:
# AF	[2]id	[3]allele frequency	[4]number of SNPs	[5]number of transitions	[6]number of transversions	[7]number of indels	[8]repeat-consistent	[9]repeat-inconsistent	[10]not applicable
AF	0	0.000000	3193458	2250696	942762	212812	0	0	212812
# QUAL, Stats by quality:
# QUAL	[2]id	[3]Quality	[4]number of SNPs	[5]number of transitions (1st ALT)	[6]number of transversions (1st ALT)	[7]number of indels
QUAL	0	998	3193458	2250696	942762	212812
# IDD, InDel distribution:
# IDD	[2]id	[3]length (deletions negative)	[4]count
IDD	0	-60	838
IDD	0	-59	10
IDD	0	-58	12
IDD	0	-57	32
IDD	0	-56	14
IDD	0	-55	10
IDD	0	-54	35
IDD	0	-53	12
IDD	0	-52	16
IDD	0	-51	28
IDD	0	-50	24
IDD	0	-49	21
IDD	0	-48	94
IDD	0	-47	19
IDD	0	-46	35
IDD	0	-45	57
IDD	0	-44	35
IDD	0	-43	21
IDD	0	-42	82
IDD	0	-41	40
IDD	0	-40	42
IDD	0	-39	79
IDD	0	-38	48
IDD	0	-37	41
IDD	0	-36	106
IDD	0	-35	58
IDD	0	-34	74
IDD	0	-33	106
IDD	0	-32	69
IDD	0	-31	79
IDD	0	-30	232
IDD	0	-29	112
IDD	0	-28	104
IDD	0	-27	285
IDD	0	-26	159
IDD	0	-25	167
IDD	0	-24	432
IDD	0	-23	219
IDD	0	-22	245
IDD	0	-21	589
IDD	0	-20	341
IDD	0	-19	335
IDD	0	-18	663
IDD	0	-17	398
IDD	0	-16	489
IDD	0	-15	908
IDD	0	-14	652
IDD	0	-13	722
IDD	0	-12	1611
IDD	0	-11	932
IDD	0	-10	1144
IDD	0	-9	1895
IDD	0	-8	1347
IDD	0	-7	1649
IDD	0	-6	3548
IDD	0	-5	4679
IDD	0	-4	12231
IDD	0	-3	23718
IDD	0	-2	23919
IDD	0	-1	54129
IDD	0	1	41404
IDD	0	2	6596
IDD	0	3	5461
IDD	0	4	4154
IDD	0	5	1968
IDD	0	6	2319
IDD	0	7	1039
IDD	0	8	979
IDD	0	9	1093
IDD	0	10	609
IDD	0	11	489
IDD	0	12	664
IDD	0	13	345
IDD	0	14	343
IDD	0	15	489
IDD	0	16	318
IDD	0	17	285
IDD	0	18	523
IDD	0	19	234
IDD	0	20	254
IDD	0	21	370
IDD	0	22	210
IDD	0	23	157
IDD	0	24	236
IDD	0	25	123
IDD	0	26	109
IDD	0	27	150
IDD	0	28	101
IDD	0	29	89
IDD	0	30	128
IDD	0	31	66
IDD	0	32	75
IDD	0	33	85
IDD	0	34	51
IDD	0	35	51
IDD	0	36	68
IDD	0	37	55
IDD	0	38	48
IDD	0	39	58
IDD	0	40	55
IDD	0	41	39
IDD	0	42	71
IDD	0	43	50
IDD	0	44	33
IDD	0	45	36
IDD	0	46	49
IDD	0	47	29
IDD	0	48	63
IDD	0	49	37
IDD	0	50	28
IDD	0	51	35
IDD	0	52	18
IDD	0	53	16
IDD	0	54	15
IDD	0	55	13
IDD	0	56	9
IDD	0	57	28
IDD	0	58	8
IDD	0	59	8
IDD	0	60	385
# ST, Substitution types:
# ST	[2]id	[3]type	[4]count
ST	0	A>C	93692
ST	0	A>G	338794
ST	0	A>T	74903
ST	0	C>A	142266
ST	0	C>G	162908
ST	0	C>T	787530
ST	0	G>A	784925
ST	0	G>C	162209
ST	0	G>T	142017
ST	0	T>A	72872
ST	0	T>C	339447
ST	0	T>G	91895
# DP, Depth distribution
# DP	[2]id	[3]bin	[4]number of genotypes	[5]fraction of genotypes (%)	[6]number of sites	[7]fraction of sites (%)
```


