---
title: "ASD_dbGaP"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(ggrepel)
library(forcats)
library(scales)
```

Four cohort studies 


Group | No.subject | No.var called 
-------|-----------|----------
DS-ASD | 10008 | 3452056 (locus with observed alleles)
GRU    | 2509
DS-AOND-MDS| 162
HMB-MDS | 93 
Total | 12772 



## DS-ASD sample  

[Sample data](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000298.v4.p3) has 10008 individuals 




## old sample vs DS-ASD sample


```{r, echo=T, cache=T}
path1="C:\\Shengtong\\Research\\rare-var\\"
path2="C:\\Shengtong\\Research\\rare-var\\73416-DS-ASD\\"
All.Anno.Data=as_tibble(read.table(paste(path1, "AnnotatedTrans.txt", sep=""), header=T))
DS.ASD.sample=as_tibble(read.csv(paste(path2, "73416-DS-ASD.sample.csv", sep=""), header=T)[,-1])


```

### sample decomposition  by AF 


```{r, echo=F}
valid.DS.ASD=DS.ASD.sample %>% filter(is.na(No.case)==F & is.na(No.contr)==F) # only keep samples with observed alleles in cases and controls 
DS.ASD=valid.DS.ASD

All.Anno.count=numeric(); DS.ASD.count=numeric()

All.Anno.count[1]=nrow(All.Anno.Data); DS.ASD.count[1]=nrow(DS.ASD)
All.Anno.count[2]=nrow(All.Anno.Data %>% filter(ExacAF<0.001)); DS.ASD.count[2]=nrow(DS.ASD %>% filter(ExacAF<0.001))
All.Anno.count[3]=nrow(All.Anno.Data %>% filter(ExacAF>0.001 & ExacAF<0.01)); DS.ASD.count[3]=nrow(DS.ASD %>% filter(ExacAF>0.001 & ExacAF<0.01))
All.Anno.count[4]=nrow(All.Anno.Data %>% filter(ExacAF>0.01 & ExacAF<0.05)); DS.ASD.count[4]=nrow(DS.ASD %>% filter(ExacAF>0.01 & ExacAF<0.05))
All.Anno.count[5]=nrow(All.Anno.Data %>% filter(ExacAF>0.05))
DS.ASD.count[5]=nrow(DS.ASD %>% filter(ExacAF>0.05))

var.count=tibble(cutoff=rep(c("All", "AF<0.1%", "0.1%<AF<1%", "1%<AF<5%", "AF>5%"),2), group=rep(c("old", "DS-ASD"),each=5), counts=c(All.Anno.count, DS.ASD.count) )

ggplot(data=var.count, aes(x=cutoff, y=counts, fill=group)) +
geom_bar(stat="identity", position=position_dodge())
var.count
```



```{r, echo=F}


old.sample <- structure(list(Group = structure(c(3L, 1L, 2L, 4L), .Label = c("AF<0.1%", "0.1%<AF<1%", "1%<AF<5%", "AF>5%"), class = "factor"), value = c(464077, 52143, 11145, 15279)), .Names = c("Group", "value"), class = "data.frame", row.names = c("1", "2", "3", "4"))

old.sample=old.sample %>% arrange(desc(value)) %>% mutate(prop = paste(round(value/sum(value),4)*100, "%", sep="")) 

pie <- ggplot(old.sample, aes(x = "", y = value, fill = fct_inorder(Group))) +
       geom_bar(width = 1, stat = "identity") +
       coord_polar("y", start = 0) +
       geom_label_repel(aes(label = prop), size=5, show.legend = F, nudge_x = 1) +
       guides(fill = guide_legend(title = "Group"))+
       ggtitle("Old sample")+
       theme(plot.title = element_text(hjust = 0.5)) # centering ggplot title 

pie



DS_ASD.sample <- structure(list(Group = structure(c(3L, 1L, 2L, 4L), .Label = c("AF<0.1%", "0.1%<AF<1%", "1%<AF<5%", "AF>5%"), class = "factor"), value = c(1706881, 222404, 59491, 75427)), .Names = c("Group", "value"), class = "data.frame", row.names = c("1", "2", "3", "4"))

DS_ASD.sample=DS_ASD.sample %>% arrange(desc(value)) %>% mutate(prop = paste(round(value/sum(value),4)*100, "%", sep="")) 

pie <- ggplot(DS_ASD.sample, aes(x = "", y = value, fill = fct_inorder(Group))) +
       geom_bar(width = 1, stat = "identity") +
       coord_polar("y", start = 0) +
       geom_label_repel(aes(label = prop), size=5, show.legend = F, nudge_x = 1) +
       guides(fill = guide_legend(title = "Group"))+
  ggtitle("DS-ASD sample")+
  theme(plot.title = element_text(hjust = 0.5)) # centering ggplot title 

pie

```



### 

```{r, echo=F, eval=F}
 bcftools stats pseudocc.tucc.vcf
# This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
# The command line was:	bcftools stats  pseudocc.tucc.vcf
#
# Definition of sets:
# ID	[2]id	[3]tab-separated file names
ID	0	pseudocc.tucc.vcf
# SN, Summary numbers:
#   number of records   .. number of data rows in the VCF
#   number of no-ALTs   .. reference-only sites, ALT is either "." or identical to REF
#   number of SNPs      .. number of rows with a SNP
#   number of MNPs      .. number of rows with a MNP, such as CC>TT
#   number of indels    .. number of rows with an indel
#   number of others    .. number of rows with other type, for example a symbolic allele or
#                          a complex substitution, such as ACT>TCGA
#   number of multiallelic sites     .. number of rows with multiple alternate alleles
#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
#
#   Note that rows containing multiple types will be counted multiple times, in each
#   counter. For example, a row with a SNP and an indel increments both the SNP and
#   the indel counter.
#
# SN	[2]id	[3]key	[4]value
SN	0	number of samples:	5512
SN	0	number of records:	3452066
SN	0	number of no-ALTs:	35521
SN	0	number of SNPs:	3193922
SN	0	number of MNPs:	0
SN	0	number of indels:	212812
SN	0	number of others:	9811
SN	0	number of multiallelic sites:	0
SN	0	number of multiallelic SNP sites:	0
# TSTV, transitions/transversions:
# TSTV	[2]id	[3]ts	[4]tv	[5]ts/tv	[6]ts (1st ALT)	[7]tv (1st ALT)	[8]ts/tv (1st ALT)
TSTV	0	2250696	942762	2.39	2250696	942762	2.39
# SiS, Singleton stats:
# SiS	[2]id	[3]allele count	[4]number of SNPs	[5]number of transitions	[6]number of transversions	[7]number of indels	[8]repeat-consistent	[9]repeat-inconsistent	[10]not applicable
SiS	0	1	3193458	2250696	942762	212812	0	0	212812
# AF, Stats by non-reference allele frequency:
# AF	[2]id	[3]allele frequency	[4]number of SNPs	[5]number of transitions	[6]number of transversions	[7]number of indels	[8]repeat-consistent	[9]repeat-inconsistent	[10]not applicable
AF	0	0.000000	3193458	2250696	942762	212812	0	0	212812
# QUAL, Stats by quality:
# QUAL	[2]id	[3]Quality	[4]number of SNPs	[5]number of transitions (1st ALT)	[6]number of transversions (1st ALT)	[7]number of indels
QUAL	0	998	3193458	2250696	942762	212812
# IDD, InDel distribution:
# IDD	[2]id	[3]length (deletions negative)	[4]count
IDD	0	-60	838
IDD	0	-59	10
IDD	0	-58	12
IDD	0	-57	32
IDD	0	-56	14
IDD	0	-55	10
IDD	0	-54	35
IDD	0	-53	12
IDD	0	-52	16
IDD	0	-51	28
IDD	0	-50	24
IDD	0	-49	21
IDD	0	-48	94
IDD	0	-47	19
IDD	0	-46	35
IDD	0	-45	57
IDD	0	-44	35
IDD	0	-43	21
IDD	0	-42	82
IDD	0	-41	40
IDD	0	-40	42
IDD	0	-39	79
IDD	0	-38	48
IDD	0	-37	41
IDD	0	-36	106
IDD	0	-35	58
IDD	0	-34	74
IDD	0	-33	106
IDD	0	-32	69
IDD	0	-31	79
IDD	0	-30	232
IDD	0	-29	112
IDD	0	-28	104
IDD	0	-27	285
IDD	0	-26	159
IDD	0	-25	167
IDD	0	-24	432
IDD	0	-23	219
IDD	0	-22	245
IDD	0	-21	589
IDD	0	-20	341
IDD	0	-19	335
IDD	0	-18	663
IDD	0	-17	398
IDD	0	-16	489
IDD	0	-15	908
IDD	0	-14	652
IDD	0	-13	722
IDD	0	-12	1611
IDD	0	-11	932
IDD	0	-10	1144
IDD	0	-9	1895
IDD	0	-8	1347
IDD	0	-7	1649
IDD	0	-6	3548
IDD	0	-5	4679
IDD	0	-4	12231
IDD	0	-3	23718
IDD	0	-2	23919
IDD	0	-1	54129
IDD	0	1	41404
IDD	0	2	6596
IDD	0	3	5461
IDD	0	4	4154
IDD	0	5	1968
IDD	0	6	2319
IDD	0	7	1039
IDD	0	8	979
IDD	0	9	1093
IDD	0	10	609
IDD	0	11	489
IDD	0	12	664
IDD	0	13	345
IDD	0	14	343
IDD	0	15	489
IDD	0	16	318
IDD	0	17	285
IDD	0	18	523
IDD	0	19	234
IDD	0	20	254
IDD	0	21	370
IDD	0	22	210
IDD	0	23	157
IDD	0	24	236
IDD	0	25	123
IDD	0	26	109
IDD	0	27	150
IDD	0	28	101
IDD	0	29	89
IDD	0	30	128
IDD	0	31	66
IDD	0	32	75
IDD	0	33	85
IDD	0	34	51
IDD	0	35	51
IDD	0	36	68
IDD	0	37	55
IDD	0	38	48
IDD	0	39	58
IDD	0	40	55
IDD	0	41	39
IDD	0	42	71
IDD	0	43	50
IDD	0	44	33
IDD	0	45	36
IDD	0	46	49
IDD	0	47	29
IDD	0	48	63
IDD	0	49	37
IDD	0	50	28
IDD	0	51	35
IDD	0	52	18
IDD	0	53	16
IDD	0	54	15
IDD	0	55	13
IDD	0	56	9
IDD	0	57	28
IDD	0	58	8
IDD	0	59	8
IDD	0	60	385
# ST, Substitution types:
# ST	[2]id	[3]type	[4]count
ST	0	A>C	93692
ST	0	A>G	338794
ST	0	A>T	74903
ST	0	C>A	142266
ST	0	C>G	162908
ST	0	C>T	787530
ST	0	G>A	784925
ST	0	G>C	162209
ST	0	G>T	142017
ST	0	T>A	72872
ST	0	T>C	339447
ST	0	T>G	91895
# DP, Depth distribution
# DP	[2]id	[3]bin	[4]number of genotypes	[5]fraction of genotypes (%)	[6]number of sites	[7]fraction of sites (%)
```


## [HMB-MDS](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000298.v4.p3):  93 individuals 










###  Phenotype: .txt file 

```{r, echo=F, eval=F}
# Study accession: phs000298.v4.p3
# Table accession: pht002283.v3.p3.c4
# Consent group: Health/Medical/Biomedical (MDS)
# Citation instructions: The study accession (phs000298.v4.p3) is used to cite the study and its data tables and documents. The data in this file should be cited using the accession pht002283.v3.p3.c4.
# To cite columns of data within this file, please use the variable (phv#) accessions below:
#
# 1) the table name and the variable (phv#) accessions below; or
# 2) you may cite a variable as phv#.v3.p3.c4.

##      phv00160418.v3.p3.c4    phv00167339.v3.p3.c4    phv00167341.v3.p3.c4    phv00167342.v3.p3.c4    phv00369219.v1.p3.c4
dbGaP_Subject_ID        SUBJID  SEX     AGE     RACE    DIAGNOSIS
2533851 19-952879-11    M                       CONTROL
```


### Genotype: vcf file 


```{r, echo=T, eval=F}
##fileformat=VCFv4.1
##ALT=<ID=NON_REF,Description="Represents any possible alternative allele at this location">
##FILTER=<ID=InbreedingCoeff,Description="InbreedingCoeff < -0.3">
##FILTER=<ID=LowQual,Description="Low quality">
##FILTER=<ID=VQSRTrancheINDEL95.00to96.00,Description="Truth sensitivity tranche level for INDEL model at VQS Lod: 0.6951 <= x < 0.8703">
##FILTER=<ID=VQSRTrancheINDEL96.00to97.00,Description="Truth sensitivity tranche level for INDEL model at VQS Lod: 0.5424 <= x < 0.6951">
##FILTER=<ID=VQSRTrancheINDEL97.00to99.00,Description="Truth sensitivity tranche level for INDEL model at VQS Lod: -0.2106 <= x < 0.5424">
##FILTER=<ID=VQSRTrancheINDEL99.00to99.50,Description="Truth sensitivity tranche level for INDEL model at VQS Lod: -0.9744 <= x < -0.2106">
##FILTER=<ID=VQSRTrancheINDEL99.50to99.90,Description="Truth sensitivity tranche level for INDEL model at VQS Lod: -9.6841 <= x < -0.9744">
##FILTER=<ID=VQSRTrancheINDEL99.90to99.95,Description="Truth sensitivity tranche level for INDEL model at VQS Lod: -26.8213 <= x < -9.6841">
```
