---
title: "Gene set analysis for SCZ data"
author: "Shengtong Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r load packages, results='hide', include=FALSE}
library(RSQLite)
library(dplyr)
library(knitr)
```

```{r defined function, echo=F, results='hide'}
test.func=function(evid, Data, N1, N0) # given evid, and sample size, perform the burden analysis
{
evid.data=Data[Data$ID %in% evid,]
count=c(sum(evid.data$No.case), sum(evid.data$No.contr))
Time=c(N1, N0)
pois.test=poisson.test(count, Time, r=1, alternative="greater")
return(result=list(odds.ratio=pois.test$estimate, p.value=pois.test$p.value, rate.case=count[1]/N1, rate.contr=count[2]/N0))
}
```

```{r read into data, echo=F, results='hide', cache=T}
All.Anno.Data=read.table("D:\\ResearchWork\\StatisticalGenetics\\Rare-variant-project\\AnnotatedSCZ.txt", header=T)
colnames(All.Anno.Data)[10]="Exon.Exac.Cons"
```


```{r set paramters and modifying the data, echo=T, results='hide'}
N1=2536; N0=2543
All.Anno.Data[All.Anno.Data =="."] <- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF<0.05 & All.Anno.Data$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonumous SNV
var.data=data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr)
LoF.def=c("stopgain", "frameshift substitution", "splicing", "stoploss")
LoF.var=as.character(Anno.Data$ID[which(Anno.Data$Annotation %in% LoF.def==T)]) 
```

## Simple burden test 

```{r burdern analysis-gene level, results='hide', echo=F}
GeneDB=src_sqlite(path="D:\\ResearchWork\\StatisticalGenetics\\Rare-variant-project\\gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
IDGene=data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
ASD.gene=data.frame(collect(tbl(GeneDB, "AutismKB_gene")))
constraint.gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS.gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene 
haploinsuff.gene=data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))
gene.set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff gene", "SCZ gene", "Olfac.gene")
gene.fea=c("cate1", "cate2", "cate3", "cate4", "cate5", "cate6", "cateS", "TADAq<5%", "TADAq<20%", "TADAq<30%", "TADAq<40%", "TADAq<50%", "TADAq>90%", gene.set); max.gene=length(gene.fea)
gene.summy=matrix(nrow=max.gene+1, ncol=4)
gene.evid=list(); var.index=1
gene.evid[[1]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate1$GeneID )])
gene.evid[[2]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate2$GeneID )])
gene.evid[[3]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate3$GeneID )])
gene.evid[[4]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate4$GeneID )])
gene.evid[[5]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate5$GeneID )])
gene.evid[[6]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate6$GeneID )])
gene.evid[[7]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cateS$GeneID )])
gene.evid[[8]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan5percentgene )])
gene.evid[[9]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan20percentgene )])
gene.evid[[10]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan30percentgene )])
gene.evid[[11]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan40percentgene)])
gene.evid[[12]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan50percentgene )])
gene.evid[[13]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlargerthan90percentgene )])
gene.evid[[14]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% IDGene$GeneID )])
high.conf=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
mod.conf=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
gene.evid[[15]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% high.conf)])
gene.evid[[16]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% mod.conf )])
psd.gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
FMRP.gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
gene.evid[[17]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% psd.gene )])
gene.evid[[18]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% FMRP.gene )])
gene.evid[[19]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% ASD.gene$GeneID )])
gene.evid[[20]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% constraint.gene )])
gene.evid[[21]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% RVIS.gene )])
gene.evid[[22]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% haploinsuff.gene$GeneID )])
gene.evid[[23]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% purcell.genelist$Gene_symbol )])
olfac.gene=data.frame(collect(tbl(GeneDB, "Olfac_gene")))
gene.evid[[24]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% olfac.gene$GeneID )])
sczgene=as.character(read.table("D:\\ResearchWork\\StatisticalGenetics\\NumericAnalysis\\RealData\\SCZData\\GeneList\\SCZ.67gene.q0.3.txt", header=T)[[1]])
gene.evid[[25]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% sczgene )])

colnames(gene.summy)=c("OR", "p.value", "rate.ca", "rate.co")
rownames(gene.summy)=c(gene.fea, "67SCZriskgene")
LoF.summy=gene.summy
colnames(LoF.summy)=c("OR", "p.value", "No.ca", "No.co")
for (gene in 1:(max.gene+1))
{
  cat(gene, "is running", "\n")
  pois.test=test.func(gene.evid[[gene]], var.data, N1, N0)
  gene.summy[gene,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
  
  LoF.test=test.func(intersect(gene.evid[[gene]], LoF.var), var.data, N1, N0)
  LoF.summy[gene,]=c(LoF.test$odds.ratio, LoF.test$p.value, LoF.test$rate.case*N1, LoF.test$rate.contr*N0)
}

```
* Two sample one side ("greater") test
* high confidence gene: TADA q<0.05, [SFARI][sfari] cate1&2
* Moder confi: 0.3>TADA q>0.05 SFARI cat3 & S
* ID gene: 252 genes from AJHG [paper][paper] 
* 842 [FMRP][fmrp] gene
* 171 [AutismKB][autismkb] genes 
* 1003 [constraint][constraint] genes 
* [RVIS][rvis] gene
* [Haploinsufficient][haploinsufficient] genes 
* [Schizophrenia][schizophrenia], SCZ gene
* 861 [Olfactory][olfactory] gene 
* 67SCZ genes are from SCZ genes with cutoff 0.3. 
* de novo gene: TADA_SNV_CNV_combined_Feb7, look at column qvalue.combined, the smaller q value, the more likely to be risk gene 
* Constraint gene set: use Exac Constraint data ConstraintMat.RDS, look at the last column lof.Z, the smaller value the less constraint
* ASD prior list: ASD_Priors, look at the column-post, the larger, the more likely to be risk. 



[paper]:https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4067558/
[sfari]:https://sfari.org/
[fmrp]:https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232425/
[autismkb]:http://autismkb.cbi.pku.edu.cn/
[constraint]:http://www.nature.com/ng/journal/v46/n9/abs/ng.3050.html?foxtrotcallback=true
[rvis]:http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709
[haploinsufficient]:http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709
[schizophrenia]:https://www.ncbi.nlm.nih.gov/pubmed/24463508
[olfactory]:http://www.genenames.org/genefamilies/OR






```{r gene level, echo=F}
kable(gene.summy, caption="Burden analysis of gene set")
kable(LoF.summy, caption="Burden analysis of LoF in different gene sets")
```



## Variant partitions
A gene usually has many variants such as missense, LoF, etc which contribute to the disease at different degree of deleteriousness. So we partition variants from all genes into different variant categories based on their annotation priors. 

### Eight variant groups 



Category | AF 
---------|--------------------------|-------------------------------
C1:LoF|   $0.01 \leq AF < 0.05$  
C2:LoF|  $AF < 0.01$         
C3:Damaging| $0.01 \leq AF < 0.05$  
C4:Damaging|  $0.001 \leq  AF < 0.01$  
C5:Damaging|  $AF < 0.001$  
C6:Non-Damaging| $0.01 \leq  AF < 0.05$  
C7:Non-Damaging| $0.001 \leq AF < 0.01$   
C8:Non-Damaging|  $AF < 0.001$  
Table: Fixed parameters for eight variant categories. 



* Damaging (probably damaging):Polyphen2.HDIV.score >=0.957


## Constraint (top 5%-1003) gene set 

Constraint gene set is from [this paper].

[this paper]: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/



### Cons: Simple burden analysis

```{r, echo=F, results='hide'}
#gene.set=as.character(read.csv("D:\\ResearchWork\\StatisticalGenetics\\Rare-variant-project\\rare-var-project\\data\\GeneSet\\Samocha_2014NG_contraintgene.csv", header=T)$gene)
gene.set=as.character(read.csv("D:\\ResearchWork\\StatisticalGenetics\\NumericAnalysis\\RealData\\GeneSet\\Samocha_2014NG\\Samocha_2014NG_contraintgene.csv", header=T)$gene)
cand.data=Anno.Data[which(Anno.Data$Gene %in% gene.set),]
par.evid=list()
par.evid[[1]]=which(cand.data$Annotation %in% LoF.def==T & cand.data$ExacAF<0.05 & cand.data$ExacAF>=0.01)
par.evid[[2]]=which(cand.data$Annotation %in% LoF.def==T &  cand.data$ExacAF<0.01)
par.evid[[3]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
par.evid[[4]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
par.evid[[5]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF<0.001)
par.evid[[6]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
par.evid[[7]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
par.evid[[8]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF<0.001)
burden.summy=matrix(nrow=length(par.evid), ncol=4)
for (i in 1:length(par.evid))
{
  cat(i, "is running", "\n")
  pois.test=test.func(as.character(cand.data$ID[par.evid[[i]]]), var.data, N1, N0)
  burden.summy[i,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case*N1, pois.test$rate.contr*N0)
  }

```


The burden of 8 categories is 
```{r, echo=T}
colnames(burden.summy)=c("OR", "p.value", "No.ca", "No.co")
kable(burden.summy, caption="The burden of eight categories")
```


### Cons: Estimate $\beta_i$, ASD gene specific $\delta_i$


#### (2) Eight categories 


Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     NA | NA 
LoF,  $AF < 0.01$ |       0 | 0 
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0292 | 2.3766
Damaging,  $AF < 0.001$ | 0.0391| 1.1027
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0 | 0 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0 | 0 
Non-Damaging,  $AF < 0.001$ | 0.0862 | 1.6528
Table: Parameter estimates of 1003 constraint genes by EM with eight partitions with gene specific  $\delta$. NA means no variants. 


## RVIS gene 

### RVIS:Estimate $\beta_i$, ASD gene specific $\delta_i$ 

```{r, echo=T}
beta=matrix(nrow=5, ncol=8)
rownames(beta)=c("Quantile score<10", "10-20", "20-30", "30-40", "40-50")
beta[1,]=c(0, 0.0977,  0, 0.0179,  0.0132, 0,  0.0203,  0.0415)
beta[2,]=c(0, 0.1240,  0, 0.0206,   0, 0, 0,  0.0004)
beta[3,]=c(0,  0, 0,   0,   0.0940,0.0615,  0,  0.0639)
beta[4,]=c(0,  0,  0, 0.0285,   0.0813,0,  0.0889,   0.0005)
beta[5,]=c(0, 0.0500,   0, 0,  0,  0.0370, 0.0315,    0.0005)
plot(beta[,1], ylim=c(0,0.5), type="o", xaxt="n", xlab="", ylab=expression(paste(hat(beta))), main="Parameter estimate of five non-nested gene sets")
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type="o", col=i)
legend(4, 0.5, c(paste("beta", seq(1,8), sep="")), col=seq(1,8), lty=rep(1,8))
```




## Constraint genes 

There are 17,416 genes. 

### Con: Estimate $\beta_i$, ASD gene specific $\delta_i$

#### Non-nested gene sets 

With their constraint scores, focus on top 10% genes, top 10-20% genes, top 20-30% genes, top 30-40% genes, top 40-50% genes. 


```{r, echo=T}
beta=matrix(nrow=5, ncol=8)
rownames(beta)=c("top10%", "10-20%", "20-30%", "30-40%", "40-50%")
beta[1,]=c(0.9999, 0,  0, 0.0065,  0.0002, 0,  0.0250,  0.0029)
beta[2,]=c(0, 0.1189,  0, 0.0087,   0, 0, 0,  0.00005)
beta[3,]=c(0,  0, 0,   0.0097,   0,0,  0,  0.0046)
beta[4,]=c(0,  0,  0, 0,   0.0046,0,  0.0003,   0.0601)
beta[5,]=c(0, 0,  0,  0.0017, 0.0323,   0,0.0074,   0.0207)
plot(beta[,1], ylim=c(0,1), type="o", xaxt="n", xlab="", ylab=expression(paste(hat(beta))), main="Parameter estimate of five non-nested gene sets")
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type="o", col=i)
legend(4, 0.8, c(paste("beta", seq(1,8), sep="")), col=seq(1,8), lty=rep(1,8))

```


## Whole genome 

consider all 15624 genes in ASD data. 

### Whole: Estimate $\beta_i$, ASD gene specific $\delta_i$

Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     0 | 0 
LoF,  $AF < 0.01$ |       0 | 0  
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0063 | 2.2903
Damaging,  $AF < 0.001$ | 0.0238| 10.8729
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0.0029 | 0.6222 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0.0166 | 4.1023 
Non-Damaging,  $AF < 0.001$ | 0.0043 | 0
Table: Parameter estimates of 15624 genes by EM with eight partitions with gene specific  $\delta_i$.



### Whole: Estimate $\beta_i$, common  $\delta$

$\widehat{\delta}=0.2$ with test statistics of 24.2446. 


Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     0 | 0 
LoF,  $AF < 0.01$ |       0 | 0  
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0024 | 0.5010
Damaging,  $AF < 0.001$ | 0.0228| 17.4871
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0.0010 | 0.3323 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0.0134 | 5.4018 
Non-Damaging,  $AF < 0.001$ | 0.0103 | 0.5224
Table: Parameter estimates of all  genes by EM with eight partitions with commom  $\delta$.



<!-- Add your analysis here -->

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
