---
title: "Gene set analysis for SCZ data"
author: "Shengtong Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```


```{r load packages, results='hide', include=FALSE}
rm(list=ls())
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(ggplot2)
library(tidyverse)
library(viridis)
library(ggsignif)
library(ggpubr)
set.seed(123)
```

```{r defined function, echo=F, results='hide'}
test.func=function(evid, Data, N1, N0) # given evid, and sample size, perform the burden analysis
{
evid.data=Data[Data$ID %in% evid,]
count=c(sum(evid.data$No.case), sum(evid.data$No.contr))
Time=c(N1, N0)
pois.test=poisson.test(count, Time, r=1, alternative="greater")
return(result=list(odds.ratio=pois.test$estimate, p.value=pois.test$p.value, rate.case=count[1]/N1, rate.contr=count[2]/N0))
}
```

```{r read into data, echo=F, results='hide', cache=T}
All.Anno.Data=read.table("../../AnnotatedSCZ.txt", header=T)
colnames(All.Anno.Data)[10]="Exon.Exac.Cons"
```


```{r set paramters and modifying the data, echo=T, results='hide'}
N1=2536; N0=2543
All.Anno.Data[All.Anno.Data =="."] <- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF<0.05 & All.Anno.Data$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonumous SNV
var.data=data.frame(ID=Anno.Data$ID, No.case=Anno.Data$No.case, No.contr=Anno.Data$No.contr)
LoF.def=c("stopgain", "frameshift substitution", "splicing", "stoploss")
LoF.var=as.character(Anno.Data$ID[which(Anno.Data$Annotation %in% LoF.def==T)]) 
```

## Simple burden test 

```{r burdern analysis-gene level, results='hide', echo=F}
GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
IDGene=data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
ASD.gene=data.frame(collect(tbl(GeneDB, "AutismKB_gene")))
constraint.gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS.gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene 
haploinsuff.gene=data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))
gene.set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff gene", "SCZ gene", "Olfac.gene")
gene.fea=c("cate1", "cate2", "cate3", "cate4", "cate5", "cate6", "cateS", "TADAq<5%", "TADAq<20%", "TADAq<30%", "TADAq<40%", "TADAq<50%", "TADAq>90%", gene.set); max.gene=length(gene.fea)
gene.summy=matrix(nrow=max.gene+1, ncol=4)
gene.evid=list(); var.index=1
gene.evid[[1]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate1$GeneID )])
gene.evid[[2]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate2$GeneID )])
gene.evid[[3]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate3$GeneID )])
gene.evid[[4]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate4$GeneID )])
gene.evid[[5]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate5$GeneID )])
gene.evid[[6]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cate6$GeneID )])
gene.evid[[7]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% gene_cateS$GeneID )])
gene.evid[[8]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan5percentgene )])
gene.evid[[9]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan20percentgene )])
gene.evid[[10]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan30percentgene )])
gene.evid[[11]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan40percentgene)])
gene.evid[[12]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlessthan50percentgene )])
gene.evid[[13]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% Qlargerthan90percentgene )])
gene.evid[[14]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% IDGene$GeneID )])
high.conf=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
mod.conf=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
gene.evid[[15]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% high.conf)])
gene.evid[[16]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% mod.conf )])
psd.gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
FMRP.gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
gene.evid[[17]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% psd.gene )])
gene.evid[[18]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% FMRP.gene )])
gene.evid[[19]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% ASD.gene$GeneID )])
gene.evid[[20]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% constraint.gene )])
gene.evid[[21]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% RVIS.gene )])
gene.evid[[22]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% haploinsuff.gene$GeneID )])
gene.evid[[23]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% purcell.genelist$Gene_symbol )])
olfac.gene=data.frame(collect(tbl(GeneDB, "Olfac_gene")))
gene.evid[[24]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% olfac.gene$GeneID )])
sczgene=as.character(read.table("../../SCZ.67gene.q0.3.txt", header=T)[[1]])
gene.evid[[25]]=as.character(Anno.Data$ID[which(Anno.Data$Gene %in% sczgene )])

colnames(gene.summy)=c("OR", "p.value", "rate.ca", "rate.co")
rownames(gene.summy)=c(gene.fea, "67SCZriskgene")
LoF.summy=gene.summy
colnames(LoF.summy)=c("OR", "p.value", "No.ca", "No.co")
for (gene in 1:(max.gene+1))
{
  cat(gene, "is running", "\n")
  pois.test=test.func(gene.evid[[gene]], var.data, N1, N0)
  gene.summy[gene,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
  
  LoF.test=test.func(intersect(gene.evid[[gene]], LoF.var), var.data, N1, N0)
  LoF.summy[gene,]=c(LoF.test$odds.ratio, LoF.test$p.value, LoF.test$rate.case*N1, LoF.test$rate.contr*N0)
}

```
* Two sample one side ("greater") test
* high confidence gene: TADA q<0.05, [SFARI][sfari] cate1&2
* Moder confi: 0.3>TADA q>0.05 SFARI cat3 & S
* ID gene: 252 genes from AJHG [paper][paper] 
* 842 [FMRP][fmrp] gene
* 171 [AutismKB][autismkb] genes 
* 1003 [constraint][constraint] genes 
* [RVIS][rvis] gene
* [Haploinsufficient][haploinsufficient] genes 
* [Schizophrenia][schizophrenia], SCZ gene
* 861 [Olfactory][olfactory] gene 
* 67SCZ genes are from SCZ genes with cutoff 0.3. 
* de novo gene: TADA_SNV_CNV_combined_Feb7, look at column qvalue.combined, the smaller q value, the more likely to be risk gene 
* Constraint gene set: use Exac Constraint data ConstraintMat.RDS, look at the last column lof.Z, the smaller value the less constraint
* ASD prior list: ASD_Priors, look at the column-post, the larger, the more likely to be risk. 



[paper]:https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4067558/
[sfari]:https://sfari.org/
[fmrp]:https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232425/
[autismkb]:http://autismkb.cbi.pku.edu.cn/
[constraint]:http://www.nature.com/ng/journal/v46/n9/abs/ng.3050.html?foxtrotcallback=true
[rvis]:http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709
[haploinsufficient]:http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709
[schizophrenia]:https://www.ncbi.nlm.nih.gov/pubmed/24463508
[olfactory]:http://www.genenames.org/genefamilies/OR






```{r gene level, echo=F}

kable(gene.summy, caption="Burden analysis of gene set", "html")%>%
kable_styling() %>%
scroll_box(height = "200px")

kable(LoF.summy, caption="Burden analysis of LoF in different gene sets", "html")%>%
kable_styling() %>%
scroll_box(height = "200px")
```





```{r, echo=F}
Gene=row.names(gene.summy)[1:22]
OR=gene.summy[1:22,1]; pval=gene.summy[1:22,2]
result.summary=tibble(Gene=Gene, OR=OR, pval=pval)
# pdf("../../Figure/burden_geneset.pdf")
  ggplot(result.summary, aes(x=Gene, y=OR, fill=Gene))+
  geom_bar(stat="identity")+
  ylab("")+xlab("Gene")+ ggtitle("Burden analysis for gene sets")+
   geom_point(mapping=aes(x=Gene, y=-log(pval, base=10), size=-log(pval, base=10), fill=Gene), alpha=0.8)+ 
    coord_flip()+
    scale_color_viridis(discrete=T)+
    geom_hline(yintercept=1,linetype="dashed")+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
    #theme(legend.position = "none")+  # no legend at all 
    guides(fill = FALSE)+
  theme(plot.title = element_text(hjust = 0.5, size=10)) #center the title 
  
#dev.off()
```


```{r, echo=F}
Gene=row.names(LoF.summy)[1:22]
OR=LoF.summy[1:22,1]; pval=LoF.summy[1:22,2]
result.summary=tibble(Gene=Gene, OR=OR, pval=pval)
# pdf("../../Figure/burden_geneset.pdf")
  ggplot(result.summary, aes(x=Gene, y=OR, fill=Gene))+
  geom_bar(stat="identity")+
  ylab("")+xlab("Gene")+ ggtitle("Burden analysis for LoF in different gene sets")+
   geom_point(mapping=aes(x=Gene, y=-log(pval, base=10), size=-log(pval, base=10), fill=Gene), alpha=0.8)+ 
    coord_flip()+
    scale_color_viridis(discrete=T)+
    geom_hline(yintercept=1,linetype="dashed")+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
    #theme(legend.position = "none")+  # no legend at all 
    guides(fill = FALSE)+
  theme(plot.title = element_text(hjust = 0.5, size=10)) #center the title 
  
#dev.off()
```



## Variant partitions
A gene usually has many variants such as missense, LoF, etc which contribute to the disease at different degree of deleteriousness. So we partition variants from all genes into different variant categories based on their annotation priors. 

### Eight variant groups 



Category | AF 
---------|--------------------------|-------------------------------
C1:LoF|   $0.01 \leq AF < 0.05$  
C2:LoF|  $AF < 0.01$         
C3:Damaging| $0.01 \leq AF < 0.05$  
C4:Damaging|  $0.001 \leq  AF < 0.01$  
C5:Damaging|  $AF < 0.001$  
C6:Non-Damaging| $0.01 \leq  AF < 0.05$  
C7:Non-Damaging| $0.001 \leq AF < 0.01$   
C8:Non-Damaging|  $AF < 0.001$  
Table: Fixed parameters for eight variant categories. 



* Damaging (probably damaging):Polyphen2.HDIV.score >=0.957


## Constraint (top 5%-1003) gene set 

Constraint gene set is from [this paper].

[this paper]: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/



### Cons: Simple burden analysis

```{r, echo=F, results='hide'}
#gene.set=as.character(read.csv("D:\\ResearchWork\\StatisticalGenetics\\Rare-variant-project\\rare-var-project\\data\\GeneSet\\Samocha_2014NG_contraintgene.csv", header=T)$gene)
gene.set=as.character(read.csv("../../GeneSet/Samocha_2014NG/Samocha_2014NG_contraintgene.csv", header=T)$gene)
cand.data=as_tibble(Anno.Data[which(Anno.Data$Gene %in% gene.set),])
par.evid=list()
par.evid[[1]]=which(cand.data$Annotation %in% LoF.def==T & cand.data$ExacAF<0.05 & cand.data$ExacAF>=0.01)
par.evid[[2]]=which(cand.data$Annotation %in% LoF.def==T &  cand.data$ExacAF<0.01)
par.evid[[3]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
par.evid[[4]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
par.evid[[5]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF<0.001)
par.evid[[6]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
par.evid[[7]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
par.evid[[8]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF<0.001)
burden.summy=matrix(nrow=length(par.evid), ncol=4)
for (i in 1:length(par.evid))
{
  cat(i, "is running", "\n")
  pois.test=test.func(as.character(cand.data$ID[par.evid[[i]]]), var.data, N1, N0)
  burden.summy[i,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case*N1, pois.test$rate.contr*N0)
  }

```




Category | AF                       | OR | pval | No.case | No.control
---------|--------------------------|----|-------|--------|------------
C1:LoF|   $0.01 \leq AF < 0.05$     | NaN | 1.00000000    | 0 |    0
C2:LoF|  $AF < 0.01$                |0.8736921| 0.84119952|    88 |  101
C3:Damaging| $0.01 \leq AF < 0.05$  | 1.0269332 | 0.07253966|  6160 | 6015
C4:Damaging|  $0.001 \leq  AF < 0.01$|1.0381518 | 0.03470484 | 4840 | 4675  
C5:Damaging|  $AF < 0.001$           | 0.9724785| 0.87835050 | 3372  |3477
C6:Non-Damaging| $0.01 \leq  AF < 0.05$| 1.0110971 | 0.13713984| 19890| 19726  
C7:Non-Damaging| $0.001 \leq AF < 0.01$|0.9916250| 0.74314422| 11755 |11887   
C8:Non-Damaging|  $AF < 0.001$ | 0.9978114| 0.56347097 | 9678 | 9726 
Table: Burden of 8 variant categories  


### Cons: run mirage

#### parameter estimate  when AF < 5\%

$\widehat{\delta}=0.09$ with p value of 0.56. 

```{r, echo=F}
eight.partition=function(cand.data) # given gene data and annotations, do variant partitions
{
  par.evid=list()
  LoF.def=c("stopgain", "frameshift substitution", "splicing", "stoploss")
  par.evid[[1]]=which(cand.data$Annotation %in% LoF.def==T & cand.data$ExacAF<0.05 & cand.data$ExacAF>=0.01 )
  par.evid[[2]]=which(cand.data$Annotation %in% LoF.def==T &  cand.data$ExacAF<0.01)

  par.evid[[3]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
  par.evid[[4]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
  par.evid[[5]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))>=0.957 &  cand.data$ExacAF<0.001)

  par.evid[[6]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.01 & cand.data$ExacAF<0.05)
  par.evid[[7]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF>=0.001 & cand.data$ExacAF<0.01)
  par.evid[[8]]=which(cand.data$Annotation %in% LoF.def==F & as.numeric(as.character(cand.data$Polyphen2.HDIV.score))<0.957 & cand.data$ExacAF<0.001)

  group.index=rep(NA, nrow(cand.data))
  for (i in 1:length(par.evid))
    group.index[par.evid[[i]]]=i
  gene.data=data.frame(ID=cand.data$ID, Gene=cand.data$Gene, No.case=cand.data$No.case, No.contr=cand.data$No.contr, group.index=group.index)
  gene.data=gene.data[complete.cases(gene.data),]
  return(gene.data)
}
```


```{r, echo=F, eval=F}
#library("devtools")
#install_github('xinhe-lab/mirage')
library(mirage)
gene.data=eight.partition(cand.data)
#N1=2536; N0=2543
#scz.result=mirage(gene.data, n1=2536, n2=2543)

no.row=1+length(scz.result$eta.est); no.col=2
parameter.estimate=matrix(nrow=no.row, ncol=no.col) # it may not have all 8 variant categories 
parameter.estimate[,1]=c(scz.result$delta.est, scz.result$eta.est)
parameter.estimate[,2]=c(scz.result$delta.pvalue, scz.result$eta.pvalue)
rownames(parameter.estimate)=c("delta", paste(names(scz.result$eta.est), sep=""))
colnames(parameter.estimate)=c("estimate", "pval")
BF.PP.gene=scz.result$BF.PP.gene
#save(parameter.estimate, BF.PP.gene, file="../output/para.est.constraint.gene.estimate.delta.SCZ.AFcutoff5%.RData")
```

```{r, echo=F, warning=F}
library(dplyr)
library(tidyverse)
parameter.estimate=tibble(cate=c("LoF; 1%<AF<5%", "LoF; AF<1%", "Damaging; 1%<AF<5%", "Damaging; 0.1%<AF<1%", "Damaging; AF<0.1%", "Non-damaging; 1%<AF<5%", "Non-damaging; 0.1%<AF<1%", "Non-damaging; AF<0.1%"), estimate=c(0, 0, 0, 0.16, 0, 0, 0, 0.26))

ggplot(data=parameter.estimate, aes(x=cate, y=estimate)) +
geom_bar(stat="identity", position=position_dodge(), fill="blue")+
   xlab("")+
   ylab(expression(paste(hat(eta))))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))+
theme(text = element_text(size=15))
```


Category | AF                       | eta.est             | p value
---------|--------------------------|---------------------|----------
C1:LoF|   $0.01 \leq AF < 0.05$     | NA |NA
C2:LoF|  $AF < 0.01$                | 0  | 1
C3:Damaging| $0.01 \leq AF < 0.05$  |0 | 1
C4:Damaging|  $0.001 \leq  AF < 0.01$ | 0.16 | 0.14  
C5:Damaging|  $AF < 0.001$          | 0 | 1
C6:Non-Damaging| $0.01 \leq  AF < 0.05$ | 0 | 1   
C7:Non-Damaging| $0.001 \leq AF < 0.01$ | 0 | 1
C8:Non-Damaging|  $AF < 0.001$          | 0.26 | 0.12 
Table: parameter estimates and p values in variant groups. 0's estimates are not exactly estimated as 0 but  they are extremely close to 0, like $e^{-10}$. 


#### Top genes 

```{r, echo=F, warning=F}
rm("parameter.estimate", "BF.PP.gene")
#const.gene.BF.DS.ASD=loadRData("../output/para.est.constraint.gene.estimate.delta.DS.ASD.RData") loadRData only load one item in RData
load("../output/para.est.constraint.gene.estimate.delta.SCZ.AFcutoff5%.RData") # load all items stored in RData
BF.PP.gene=as_tibble(BF.PP.gene)
BF.PP.gene.sort=arrange(BF.PP.gene, desc(post.prob))
BF.PP.gene.sort

```

#### highlight gene DNMT3A 


```{r, echo=T, eval=F}
Gene: DNMT3A briven by damaging variants              
             ID   Gene No.case No.contr category      var.BF original.index
949 chr2:25523096-25523096_T_G DNMT3A      39       13        3 566.1746060              4
943 chr2:25457243-25457243_G_A DNMT3A       1        0        4   1.4530292              5
944 chr2:25467078-25467078_C_A DNMT3A       0        1        7   0.5482178              8
945 chr2:25468174-25468174_T_C DNMT3A       1        0        7   1.4530292              8
946 chr2:25469073-25469073_G_A DNMT3A      10        8        7   0.3861151              8
947 chr2:25469142-25469142_A_G DNMT3A       0        1        7   0.5482178              8
948 chr2:25469614-25469614_G_A DNMT3A       1        0        7   1.4530292              8
```


```{r, echo=T, eval=F}

Gene: MACF1, signal driven by rare damaing and non-damaging variants 

                            ID  Gene No.case No.contr category      var.BF original.index
379 chr1:39847730-39847730_G_A MACF1      45       47        2  0.04355588              3
353 chr1:39775303-39775303_A_G MACF1      19       13        3  0.58277783              4
359 chr1:39783013-39783013_G_T MACF1      90       74        3  0.15193265              4
362 chr1:39788295-39788295_C_A MACF1       2        4        3  0.23939249              4
367 chr1:39816547-39816547_C_G MACF1      19        7        3 13.10494586              4
386 chr1:39853640-39853640_A_C MACF1      45       25        3  6.09805554              4
387 chr1:39853641-39853641_G_T MACF1      90       74        3  0.15193265              4
394 chr1:39853959-39853959_G_A MACF1      19       12        3  0.87925645              4
403 chr1:39900230-39900230_C_T MACF1       2        3        3  0.36622226              4
420 chr1:39951304-39951304_T_G MACF1      24       23        3  0.10043482              4
345 chr1:39549925-39549925_G_A MACF1       1        1        4  0.76755484              5
346 chr1:39549942-39549942_C_T MACF1       0        2        4  0.32948455              5
348 chr1:39752993-39752993_G_A MACF1       1        0        4  1.45302921              5
349 chr1:39753145-39753145_C_T MACF1       0        1        4  0.54821782              5
356 chr1:39776562-39776562_C_T MACF1       1        2        4  0.44240738              5
357 chr1:39781210-39781210_T_G MACF1       1        1        4  0.76755484              5
358 chr1:39781321-39781321_A_G MACF1       1        2        4  0.44240738              5
360 chr1:39784189-39784189_A_T MACF1       1        0        4  1.45302921              5
361 chr1:39785384-39785384_C_G MACF1       1        2        4  0.44240738              5
363 chr1:39788346-39788346_C_T MACF1       0        1        4  0.54821782              5
364 chr1:39788635-39788635_C_G MACF1       1        0        4  1.45302921              5
375 chr1:39827287-39827287_C_T MACF1       0        1        4  0.54821782              5
376 chr1:39835745-39835745_C_G MACF1       1        1        4  0.76755484              5
377 chr1:39835746-39835746_G_A MACF1       0        1        4  0.54821782              5
380 chr1:39851427-39851427_G_A MACF1       2        1        4  1.09359980              5
381 chr1:39851451-39851451_G_C MACF1       1        0        4  1.45302921              5
382 chr1:39851518-39851518_T_C MACF1       0        1        4  0.54821782              5
383 chr1:39851554-39851554_G_A MACF1       1        0        4  1.45302921              5
384 chr1:39851563-39851563_C_T MACF1       0        2        4  0.32948455              5
389 chr1:39853740-39853740_C_G MACF1       0        5        4  0.12191241              5
391 chr1:39853818-39853818_G_A MACF1       0        1        4  0.54821782              5
392 chr1:39853893-39853893_C_T MACF1       0        1        4  0.54821782              5
393 chr1:39853900-39853900_G_A MACF1       1        0        4  1.45302921              5
396 chr1:39854170-39854170_G_A MACF1       0        1        4  0.54821782              5
397 chr1:39889792-39889792_T_G MACF1       0        3        4  0.21687257              5
398 chr1:39893177-39893177_C_T MACF1       0        1        4  0.54821782              5
399 chr1:39893669-39893669_G_A MACF1       0        2        4  0.32948455              5
401 chr1:39895535-39895535_G_A MACF1       0        1        4  0.54821782              5
402 chr1:39896485-39896485_C_T MACF1       0        1        4  0.54821782              5
404 chr1:39903464-39903464_G_A MACF1       3        9        4  0.08613987              5
405 chr1:39904981-39904981_T_A MACF1       0        1        4  0.54821782              5
406 chr1:39908490-39908490_A_C MACF1       3        3        4  0.49340212              5
408 chr1:39910413-39910413_G_A MACF1       1        2        4  0.44240738              5
409 chr1:39910418-39910418_G_A MACF1       5        5        4  0.34806276              5
410 chr1:39913252-39913252_A_T MACF1       1        0        4  1.45302921              5
412 chr1:39916854-39916854_C_A MACF1       0        1        4  0.54821782              5
413 chr1:39917864-39917864_G_A MACF1       1        0        4  1.45302921              5
414 chr1:39917931-39917931_A_C MACF1       0        1        4  0.54821782              5
418 chr1:39951230-39951230_C_T MACF1       1        0        4  1.45302921              5
419 chr1:39951243-39951243_G_A MACF1       5        5        4  0.34806276              5
350 chr1:39757595-39757595_G_A MACF1      14       16        6  0.09148606              7
351 chr1:39763365-39763365_G_T MACF1      54       44        6  0.16436446              7
378 chr1:39838200-39838200_A_G MACF1       6       10        6  0.09719669              7
347 chr1:39715702-39715702_A_G MACF1       2        0        7  2.14039567              8
352 chr1:39765991-39765991_A_G MACF1       1        1        7  0.76755484              8
354 chr1:39775999-39775999_G_A MACF1       1        0        7  1.45302921              8
355 chr1:39776527-39776527_A_G MACF1       4        0        7  4.80333231              8
365 chr1:39790337-39790337_C_T MACF1       0        1        7  0.54821782              8
366 chr1:39790348-39790348_G_C MACF1       0        1        7  0.54821782              8
368 chr1:39823159-39823159_T_C MACF1      12        3        7 13.43065361              8
369 chr1:39823306-39823306_G_A MACF1       0        1        7  0.54821782              8
370 chr1:39823349-39823349_C_A MACF1       1        0        7  1.45302921              8
371 chr1:39823363-39823363_C_T MACF1       0        1        7  0.54821782              8
372 chr1:39823365-39823365_T_C MACF1       1        3        7  0.27799691              8
373 chr1:39824468-39824468_G_A MACF1       0        3        7  0.21687257              8
374 chr1:39826587-39826587_A_G MACF1       2        2        7  0.60727166              8
385 chr1:39853356-39853356_A_G MACF1       1        1        7  0.76755484              8
388 chr1:39853687-39853687_T_A MACF1       4        9        7  0.09258787              8
390 chr1:39853797-39853797_G_T MACF1       1        0        7  1.45302921              8
395 chr1:39854002-39854002_G_A MACF1       7        2        7  3.71945900              8
400 chr1:39893676-39893676_A_G MACF1       6        5        7  0.46235530              8
407 chr1:39908511-39908511_G_T MACF1       2        1        7  1.09359980              8
411 chr1:39914356-39914356_G_A MACF1       1        1        7  0.76755484              8
415 chr1:39919407-39919407_G_A MACF1       4        1        7  2.31557551              8
416 chr1:39920758-39920758_G_A MACF1       2        1        7  1.09359980              8
417 chr1:39924788-39924788_C_T MACF1       1        0        7  1.45302921              8

```


### Cons: Estimate $\beta_i$, ASD gene specific $\delta_i$


#### (2) Eight categories 


Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     NA | NA 
LoF,  $AF < 0.01$ |       0 | 0 
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0292 | 2.3766
Damaging,  $AF < 0.001$ | 0.0391| 1.1027
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0 | 0 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0 | 0 
Non-Damaging,  $AF < 0.001$ | 0.0862 | 1.6528
Table: Parameter estimates of 1003 constraint genes by EM with eight partitions with gene specific  $\delta$. NA means no variants. 


## RVIS gene 

### RVIS:Estimate $\beta_i$, ASD gene specific $\delta_i$ 

```{r, echo=T}
beta=matrix(nrow=5, ncol=8)
rownames(beta)=c("Quantile score<10", "10-20", "20-30", "30-40", "40-50")
beta[1,]=c(0, 0.0977,  0, 0.0179,  0.0132, 0,  0.0203,  0.0415)
beta[2,]=c(0, 0.1240,  0, 0.0206,   0, 0, 0,  0.0004)
beta[3,]=c(0,  0, 0,   0,   0.0940,0.0615,  0,  0.0639)
beta[4,]=c(0,  0,  0, 0.0285,   0.0813,0,  0.0889,   0.0005)
beta[5,]=c(0, 0.0500,   0, 0,  0,  0.0370, 0.0315,    0.0005)
plot(beta[,1], ylim=c(0,0.5), type="o", xaxt="n", xlab="", ylab=expression(paste(hat(beta))), main="Parameter estimate of five non-nested gene sets")
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type="o", col=i)
legend(4, 0.5, c(paste("beta", seq(1,8), sep="")), col=seq(1,8), lty=rep(1,8))
```




## Constraint genes 

There are 17,416 genes. 

### Con: Estimate $\beta_i$, ASD gene specific $\delta_i$

#### Non-nested gene sets 

With their constraint scores, focus on top 10% genes, top 10-20% genes, top 20-30% genes, top 30-40% genes, top 40-50% genes. 


```{r, echo=T}
beta=matrix(nrow=5, ncol=8)
rownames(beta)=c("top10%", "10-20%", "20-30%", "30-40%", "40-50%")
beta[1,]=c(0.9999, 0,  0, 0.0065,  0.0002, 0,  0.0250,  0.0029)
beta[2,]=c(0, 0.1189,  0, 0.0087,   0, 0, 0,  0.00005)
beta[3,]=c(0,  0, 0,   0.0097,   0,0,  0,  0.0046)
beta[4,]=c(0,  0,  0, 0,   0.0046,0,  0.0003,   0.0601)
beta[5,]=c(0, 0,  0,  0.0017, 0.0323,   0,0.0074,   0.0207)
plot(beta[,1], ylim=c(0,1), type="o", xaxt="n", xlab="", ylab=expression(paste(hat(beta))), main="Parameter estimate of five non-nested gene sets")
axis(side=1, at=1:5, labels=rownames(beta),cex.axis=0.7)
for (i in 2:8)
 lines(beta[,i], type="o", col=i)
legend(4, 0.8, c(paste("beta", seq(1,8), sep="")), col=seq(1,8), lty=rep(1,8))

```


## Whole genome 

consider all 15624 genes in ASD data. 

### Whole: Estimate $\beta_i$, ASD gene specific $\delta_i$

Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     0 | 0 
LoF,  $AF < 0.01$ |       0 | 0  
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0063 | 2.2903
Damaging,  $AF < 0.001$ | 0.0238| 10.8729
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0.0029 | 0.6222 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0.0166 | 4.1023 
Non-Damaging,  $AF < 0.001$ | 0.0043 | 0
Table: Parameter estimates of 15624 genes by EM with eight partitions with gene specific  $\delta_i$.



### Whole: Estimate $\beta_i$, common  $\delta$

$\widehat{\delta}=0.2$ with test statistics of 24.2446. 


Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     0 | 0 
LoF,  $AF < 0.01$ |       0 | 0  
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0024 | 0.5010
Damaging,  $AF < 0.001$ | 0.0228| 17.4871
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0.0010 | 0.3323 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0.0134 | 5.4018 
Non-Damaging,  $AF < 0.001$ | 0.0103 | 0.5224
Table: Parameter estimates of all  genes by EM with eight partitions with commom  $\delta$.



### Whole: Estimate $\beta_i$, SCZ de novo  prior $\delta_i$

Use posterior probability PP as priors 

* prior by  2 categoris 



Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     0 | 0 
LoF,  $AF < 0.01$ |       0 | 0  
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0.0012 | 0.03
Damaging,  $AF < 0.001$ | 0.0707| 11.67
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0.0090 | 0.57 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0.0689 | 5.27 
Non-Damaging,  $AF < 0.001$ | 0.0029 | 0.05
Table: Parameter estimates  by EM with eight partitions with gene specific  $\delta_i$.




* prior by  3 categoris 



Annotation category  | $\widehat{\beta}$ | test.stat
---------------------|---------------|-------------------
LoF,   $0.01 \leq AF < 0.05$  |     0 | 0 
LoF,  $AF < 0.01$ |       0 | 0  
Damaging, $0.01 \leq AF < 0.05$ |   0| 0 
Damaging,  $0.001 \leq  AF < 0.01$ |  0 | 0
Damaging,  $AF < 0.001$ | 0.0774| 9.95
Non-Damaging, $0.01 \leq  AF < 0.05$ |  0.0132 | 0.65 
Non-Damaging, $0.001 \leq AF < 0.01$ |  0.0871 | 5.11 
Non-Damaging,  $AF < 0.001$ | 0.0016 | 0.02
Table: Parameter estimates  by EM with eight partitions with gene specific  $\delta_i$.



<!-- Add your analysis here -->

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
