---
title: "Laura analysis"
output: html_document
date: "2023-05-24"
---


```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
library(DT)
library(poolr) # use fisher method to combine p values 
library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)

#############
#devtools::install_github("yaowuliu/ACAT")  # install package ACAT once 
library(ACAT)
```



```{r, echo=F, message=F, warning=F}
########## use fixed parameter estimate to calculate BF, this is very useful because it doesn't need to re-run analysis to BF of every gene which takes a long time and large space to store that information 
intergrand=function(aa, var.case, var.contr, bar.gamma, sig, N1, N0)
{
  ff=dbinom(var.case, sum(var.case, var.contr), aa*N1/(aa*N1+N0))*dgamma(aa, bar.gamma*sig, sig)
  return(ff)
}
# calculate the bayes factor of a single variant via integration
BF.var.inte=function(var.case, var.contr, bar.gamma, sig, N1, N0)
{
  marglik0.CC <- dbinom(var.case, sum(var.case, var.contr), N1/(N1+N0))    # Under H0: gamma=1

  marglik1.CC <- integrate(intergrand, var.case, var.contr, bar.gamma, sig, N1, N0, low=0, upper=100, stop.on.error=F)$value # Under H1: gamma~gamma(gamma.mean*sigma, sigma)
  BF.var <- marglik1.CC/marglik0.CC

  return(BF.var)
}

fixed.beta.func=function(new.data, N1, N0, gamma.mean, sigma, beta) # new.data has one column specifying its group index
{
  #######################
  beta.k=beta
  full.info.genevar=list()
  gene.list=new.data$Gene; unique.gene=unique(gene.list) # find the gene list
  num.gene=length(unique.gene)
  BF.gene=numeric()
  ########################
  for (i in 1:num.gene)
  {
#    cat(i, "th gene of ", "\t", num.gene, "\t", "is running", "\n")
    var.index.list=which(gene.list==unique.gene[i])
    indi.gene=new.data[var.index.list,]   # note var.index.list matches new.data
    bb=1; var.BF=numeric()
    if (length(var.index.list)>0) # calculate Bayes factor for variant (i,j)
      for (j in 1:length(var.index.list))
      {
        if (new.data$group.index[var.index.list[j]]<=3)   # note here bar.gamma is group specific, attention what bar.gamma goes to which group 
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=6, sig=sigma, N1, N0)
        if (new.data$group.index[var.index.list[j]]>3)
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=3, sig=sigma, N1, N0)
        bb=bb*((1-beta.k[new.data$group.index[var.index.list[j]]])+beta.k[new.data$group.index[var.index.list[j]]]*var.BF[j])
      }
    full.info.genevar[[i]]=cbind(indi.gene, var.BF)
    BF.gene[i]=bb
  }
  
  return(result=list(BayesFactor=data.frame(Gene=unique.gene, BF=BF.gene), full.info=full.info.genevar))
}
```



```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
path="C:\\Shengtong\\research\\rare-var\\"
ASC_new_sample3=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), sep='\t',  header=T, fill=T)) # this is the correct way that Kyle read the data
ASC_new_sample3=ASC_new_sample3 %>% filter(isSyn=="FALSE") # filter syn variants first 
```





```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
path="C:\\Shengtong\\Research\\rare-var\\"
GeneDB=src_sqlite(path=paste(path, "gene.list.db", sep=""), create=F)
#GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
###### ID gene ######
ID_Gene=as.character(data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))$GeneID)
#### high confidence gene 
High_Confidence_Gene=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
#### moderate confidence gene 
Moderate_Confidence_Gene=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
#### psd gene 
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
PSD_Gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
#### FMRP gene 
FMRP_Gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
#### Autism KB gene 
ASD_KB_Gene=as.character(data.frame(collect(tbl(GeneDB, "AutismKB_gene")))$GeneID)
#### constraint gene 
Constraint_Gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
### RVIS gene 
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS_Gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene
#### Haplo insufficient gene 
Haploinsufficient_Gene=as.character(data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))$GeneID)
#### SCZ data 
SCZ_Gene=purcell.genelist$Gene_symbol
Gene_Set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff", "SCZ gene")
Gene_Set_List=list()
Gene_Set_List[[1]]=ID_Gene
Gene_Set_List[[2]]=High_Confidence_Gene
Gene_Set_List[[3]]=Moderate_Confidence_Gene
Gene_Set_List[[4]]=PSD_Gene
Gene_Set_List[[5]]=FMRP_Gene
Gene_Set_List[[6]]=ASD_KB_Gene
Gene_Set_List[[7]]=Constraint_Gene
Gene_Set_List[[8]]=RVIS_Gene
Gene_Set_List[[9]]=Haploinsufficient_Gene
Gene_Set_List[[10]]=SCZ_Gene
```



```{r, echo=F, warning=F, message=F}

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
######### use ASC_new_sample3 rather than ASC_new_sample2
Transmitted_proband=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) # extract transmitted variants for proband 

Untransmitted_proband=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) # extract untransmitted variants for proband 

Transmitted_sibling=ASC_new_sample3%>% select(starts_with("t_sibling"))%>%  mutate(Transmitted_sibling = select(., t_sibling_male_dad:t_sibling_female_ind) %>% rowSums())%>% select(Transmitted_sibling) # extract transmitted variants for sibling 

Untransmitted_sibling=ASC_new_sample3%>% select(starts_with("u_sibling"))%>%  mutate(Untransmitted_sibling = select(., u_sibling_male_dad:u_sibling_female_ind) %>% rowSums())%>% select(Untransmitted_sibling) # extract untransmitted variants for sibling

##### add new columns of total transmitted and non-transmitted variants 
#ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & VQSLOD<9) # exclude indel & VQSLOD filter 

################################################


ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8)%>%filter(!str_detect(Consequence, 'intron_variant'))# exclude indel & apply other filters for SNP's 

##################
#AF_cutoff=0.001
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Transmitted_proband) %>% pull)   
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Untransmitted_proband) %>% pull)   
########### 1164 vs 1413 not 695 vs 557 in cell paper 


# add one column of variant group index 
ASC_new_sample_GI=ASC_new_sample %>% add_column(Group_Index=NA)

ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.995 & ASC_new_sample_GI$isPTV==T)]=1
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.5 & ASC_new_sample_GI$pLI<0.995 & ASC_new_sample_GI$isPTV==T)]=2
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI<0.5 & ASC_new_sample_GI$isPTV==T)]=3
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=2)]=4
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=1 & ASC_new_sample_GI$MPC<2)]=5
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC<1)]=6

num.family=7291
## test 
```


```{r, echo=F, message=F, warning=F, cache=T}
Laura_input_v17=as_tibble(read.table("C:\\Shengtong\\Research\\rare-var\\ASC\\Laura_result\\mirage_input_v17.txt", header=T))

num.family=7291
Han_input_v17=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
#write.csv(gene_set, file="C:/Shengtong/Research/rare-var/ASC/mirage_input_20230519.csv")  # input data fpr MIRAGE 
#mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family, gamma=c(6,6,6,3,3,3)) # input data must be dataframe 

Laura_Han_combine_v17=Laura_input_v17 %>% full_join(Han_input_v17)

############ transmitted variants 

g1=ggplot(Laura_Han_combine_v17, aes(x=No.case, y=Transmitted_proband))+geom_point()+
   xlim(c(0,30))+ylim(c(0,30))+
  xlab("Laura_v17")+ylab("Han_v17")+
  geom_abline(intercept =0, slope = 1, color="red", size=1)+
   ggtitle("overlapping transmitted alleles")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title

g2=ggplot(Laura_Han_combine_v17, aes(x=No.contr, y=Untransmitted_proband))+geom_point()+
   xlim(c(0,30))+ylim(c(0,30))+
  xlab("Laura_v17")+ylab("Han_v17")+
  geom_abline(intercept =0, slope = 1, color="red", size=1)+
   ggtitle("overlapping untransmitted alleles")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title

ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom")
```

* for overlapping variants  (89.1% of Laura's data, 78.9% of Han's data), they share same number of transmitted variants and untransmitted ones 





## gene set overlapping 


* `han_v17` ASCv17 from Kyle 

* `laura_v17`, additional filters applied to `han_v17`, with 85% overlapping, Laura ran MIRAGE to get parameter estimate that are close to parameter estimate in `han_v17`. 

* `laura_all` combined all dataset, including two SPARK data sets, `laura_v17`, and ASC_B15_B16 


```{r echo=F, message=F, warning=F}
topgene_number=20

BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
topgenes=as.character(sort_BF$Gene[1:topgene_number])

laura_result=as_tibble(read.csv(paste(path, "ASC\\Laura_result\\Comparing Shengtong and Laura's inputs_2023_0524.csv", sep="")))
laura_result=laura_result %>% select(Gene, BF.ALL_combined, BF.ASC_v17_TDT_by_variant_2020.09.16) %>% mutate(BF.ASC_v17=as.numeric(BF.ASC_v17_TDT_by_variant_2020.09.16)) %>% drop_na()

laura_result_sort_by_ASCv17=laura_result%>% arrange(desc(BF.ASC_v17))
laura_result_sort_by_ASCv17_top_genes=laura_result_sort_by_ASCv17$Gene[1:topgene_number]

laura_result_sort_by_AllData=laura_result%>% arrange(desc(BF.ALL_combined))
laura_result_sort_by_AllData_top_genes=laura_result_sort_by_AllData$Gene[1:topgene_number]

sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
mirage_topgenes=as.character(sort_BF$Gene[1:topgene_number])
#intersect(mirage_topgenes, laura_result_sort_by_ASCv17_top_genes)

##############
library("ggvenn")
x=list(
  laura_v17=laura_result_sort_by_ASCv17_top_genes,
  han_v17=mirage_topgenes
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 


x=list(
  laura_all=laura_result_sort_by_AllData_top_genes,
  han_v17=mirage_topgenes
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 

x=list(
  laura_all=laura_result_sort_by_AllData_top_genes,
  laura_v17=laura_result_sort_by_ASCv17_top_genes
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 


#############
han_BF=laura_ASCv17_BF=laura_allcombined_BF=numeric()
for (i in 1:(length(mirage_topgenes)-1))
{
  han_BF[i]=BF_para_all_est %>% filter(Gene==mirage_topgenes[i]) %>% select(BF_filtered) %>% pull()
  laura_ASCv17_BF[i]=laura_result%>% filter(Gene==mirage_topgenes[i]) %>% select(BF.ASC_v17) %>% pull()
  laura_allcombined_BF[i]=laura_result%>% filter(Gene==mirage_topgenes[i]) %>% select(BF.ALL_combined) %>% pull()
}


ggplot(data.frame(Gene=rep(mirage_topgenes[1:19],3), logBF=c(log(han_BF), log(laura_ASCv17_BF), log(laura_allcombined_BF)), sample=rep(c("han_v17", "laura_v17", "laura_all"), each=19)), aes(x=Gene, y=logBF, fill=sample)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge())+ 
  theme(legend.title=element_blank())+
  ylab("log(BF)")+
  ggtitle("BF of top 19 genes in han_v17")+
  theme(plot.title = element_text(hjust = 0.5, size=15, color = 2))+ #center the title
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))#+
 #theme(text = element_text(size=10))+
  #geom_text(aes(label=round(logBF,2)), position=position_dodge(width=0.8), vjust=-0.5, color="blue")

```

```{r, echo=F, message=F, warning=F}
han_BF=laura_ASCv17_BF=laura_allcombined_BF=numeric()
for (i in 1:(length(laura_result_sort_by_ASCv17_top_genes)))
{
  han_BF[i]=BF_para_all_est %>% filter(Gene==laura_result_sort_by_ASCv17_top_genes[i]) %>% select(BF_filtered) %>% pull()
  laura_ASCv17_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_ASCv17_top_genes[i]) %>% select(BF.ASC_v17) %>% pull()
  laura_allcombined_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_ASCv17_top_genes[i]) %>% select(BF.ALL_combined) %>% pull()
}


ggplot(data.frame(Gene=rep(laura_result_sort_by_ASCv17_top_genes,3), logBF=c(log(han_BF), log(laura_ASCv17_BF), log(laura_allcombined_BF)), sample=rep(c("han_v17", "laura_v17", "laura_all"), each=topgene_number)), aes(x=Gene, y=logBF, fill=sample)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge())+ 
  theme(legend.title=element_blank())+
  ylab("log(BF)")+
  ggtitle("BF of top 20 genes in laura_v17")+
  theme(plot.title = element_text(hjust = 0.5, size=15, color = 2))+ #center the title
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))#+
 #theme(text = element_text(size=10))+
  #geom_text(aes(label=round(logBF,2)), position=position_dodge(width=0.8), vjust=-0.5, color="blue")

```


```{r, echo=F, message=F, warning=F}
han_BF=laura_ASCv17_BF=laura_allcombined_BF=numeric()
for (i in 1:(length(laura_result_sort_by_AllData_top_genes)))
{
  han_BF[i]=BF_para_all_est %>% filter(Gene==laura_result_sort_by_AllData_top_genes[i]) %>% select(BF_filtered) %>% pull()
  laura_ASCv17_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_AllData_top_genes[i]) %>% select(BF.ASC_v17) %>% pull()
  laura_allcombined_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_AllData_top_genes[i]) %>% select(BF.ALL_combined) %>% pull()
}


ggplot(data.frame(Gene=rep(laura_result_sort_by_AllData_top_genes,3), logBF=c(log(han_BF), log(laura_ASCv17_BF), log(laura_allcombined_BF)), sample=rep(c("han_v17", "laura_v17", "laura_all"), each=topgene_number)), aes(x=Gene, y=logBF, fill=sample)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge())+ 
  theme(legend.title=element_blank())+
  ylab("log(BF)")+
  ggtitle("BF of top 20 genes in laura_all")+
  theme(plot.title = element_text(hjust = 0.5, size=15, color = 2))+ #center the title
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))#+
 #theme(text = element_text(size=10))+
  #geom_text(aes(label=round(logBF,2)), position=position_dodge(width=0.8), vjust=-0.5, color="blue")

```
