---
title: "Laura analysis"
output: html_document
date: "2023-05-24"
---


```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
library(DT)
library(poolr) # use fisher method to combine p values 
library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)

#############
#devtools::install_github("yaowuliu/ACAT")  # install package ACAT once 
library(ACAT)
```



```{r, echo=F, message=F, warning=F}
########## use fixed parameter estimate to calculate BF, this is very useful because it doesn't need to re-run analysis to BF of every gene which takes a long time and large space to store that information 
intergrand=function(aa, var.case, var.contr, bar.gamma, sig, N1, N0)
{
  ff=dbinom(var.case, sum(var.case, var.contr), aa*N1/(aa*N1+N0))*dgamma(aa, bar.gamma*sig, sig)
  return(ff)
}
# calculate the bayes factor of a single variant via integration
BF.var.inte=function(var.case, var.contr, bar.gamma, sig, N1, N0)
{
  marglik0.CC <- dbinom(var.case, sum(var.case, var.contr), N1/(N1+N0))    # Under H0: gamma=1

  marglik1.CC <- integrate(intergrand, var.case, var.contr, bar.gamma, sig, N1, N0, low=0, upper=100, stop.on.error=F)$value # Under H1: gamma~gamma(gamma.mean*sigma, sigma)
  BF.var <- marglik1.CC/marglik0.CC

  return(BF.var)
}

fixed.beta.func=function(new.data, N1, N0, gamma.mean, sigma, beta) # new.data has one column specifying its group index
{
  #######################
  beta.k=beta
  full.info.genevar=list()
  gene.list=new.data$Gene; unique.gene=unique(gene.list) # find the gene list
  num.gene=length(unique.gene)
  BF.gene=numeric()
  ########################
  for (i in 1:num.gene)
  {
#    cat(i, "th gene of ", "\t", num.gene, "\t", "is running", "\n")
    var.index.list=which(gene.list==unique.gene[i])
    indi.gene=new.data[var.index.list,]   # note var.index.list matches new.data
    bb=1; var.BF=numeric()
    if (length(var.index.list)>0) # calculate Bayes factor for variant (i,j)
      for (j in 1:length(var.index.list))
      {
        if (new.data$group.index[var.index.list[j]]<=3)   # note here bar.gamma is group specific, attention what bar.gamma goes to which group 
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=6, sig=sigma, N1, N0)
        if (new.data$group.index[var.index.list[j]]>3)
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=3, sig=sigma, N1, N0)
        bb=bb*((1-beta.k[new.data$group.index[var.index.list[j]]])+beta.k[new.data$group.index[var.index.list[j]]]*var.BF[j])
      }
    full.info.genevar[[i]]=cbind(indi.gene, var.BF)
    BF.gene[i]=bb
  }
  
  return(result=list(BayesFactor=data.frame(Gene=unique.gene, BF=BF.gene), full.info=full.info.genevar))
}
```



```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
path="C:\\Shengtong\\research\\rare-var\\"
ASC_new_sample33=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), sep='\t',  header=T, fill=T)) # this is the correct way that Kyle read the data # 2,732,526 variants, 
ASC_new_sample3=ASC_new_sample33 %>% filter(isSyn=="FALSE") # filter syn variants first 
```



```{r, echo=F, message=F, warning=F, eval=F}
###### from my v17 to laura's v17

############## below is Kyle's filters 
# Determine rarity threshold
#rarity_threshold = 0.001

# Apply rarity threshold... in gnomad
#tdt = subset(tdt, is.na(Gnomad_non_neuro_AF) | (Gnomad_non_neuro_AF <= rarity_threshold)) # 2,543,183
# ...and in this data:
#tdt = subset(tdt, AF <= rarity_threshold) # 2,521,044

# Apply VQSLOD threshold I used
#tdt = subset(tdt, VQSLOD >= 5.13) # 2,094,224

# Also apply additional filters used for this dataset
#tdt = subset(tdt, (isIndel & QD >= 3 & AS_SOR <= 3 & AS_ReadPosRankSum >= -8) |
#               (!isIndel & QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8))
# 1,777,785
################# to replicate laura's v17
ASC_new_sample33 %>% filter(Gnomad_non_neuro_AF <=0.001) # 2,543,183

ASC_new_sample33 %>% filter(Gnomad_non_neuro_AF <=0.001 & AF<=0.001) # 2,521,044

ASC_new_sample33 %>% filter(Gnomad_non_neuro_AF <=0.001 & AF<=0.001 & VQSLOD >= 5.13) # 2,094,224
ASC_new_sample33 %>% filter(Gnomad_non_neuro_AF <=0.001 & AF<=0.001 & VQSLOD >= 5.13) %>% filter((isIndel & QD >= 3 & AS_SOR <= 3 & AS_ReadPosRankSum >= -8)| (!isIndel & QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8)) # 1,777,785

```



```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
path="C:\\Shengtong\\Research\\rare-var\\"
GeneDB=src_sqlite(path=paste(path, "gene.list.db", sep=""), create=F)
#GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
###### ID gene ######
ID_Gene=as.character(data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))$GeneID)
#### high confidence gene 
High_Confidence_Gene=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
#### moderate confidence gene 
Moderate_Confidence_Gene=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
#### psd gene 
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
PSD_Gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
#### FMRP gene 
FMRP_Gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
#### Autism KB gene 
ASD_KB_Gene=as.character(data.frame(collect(tbl(GeneDB, "AutismKB_gene")))$GeneID)
#### constraint gene 
Constraint_Gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
### RVIS gene 
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS_Gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene
#### Haplo insufficient gene 
Haploinsufficient_Gene=as.character(data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))$GeneID)
#### SCZ data 
SCZ_Gene=purcell.genelist$Gene_symbol
Gene_Set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff", "SCZ gene")
Gene_Set_List=list()
Gene_Set_List[[1]]=ID_Gene
Gene_Set_List[[2]]=High_Confidence_Gene
Gene_Set_List[[3]]=Moderate_Confidence_Gene
Gene_Set_List[[4]]=PSD_Gene
Gene_Set_List[[5]]=FMRP_Gene
Gene_Set_List[[6]]=ASD_KB_Gene
Gene_Set_List[[7]]=Constraint_Gene
Gene_Set_List[[8]]=RVIS_Gene
Gene_Set_List[[9]]=Haploinsufficient_Gene
Gene_Set_List[[10]]=SCZ_Gene
```



```{r, echo=F, warning=F, message=F}

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
######### use ASC_new_sample3 rather than ASC_new_sample2
Transmitted_proband=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) # extract transmitted variants for proband 

Untransmitted_proband=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) # extract untransmitted variants for proband 

Transmitted_sibling=ASC_new_sample3%>% select(starts_with("t_sibling"))%>%  mutate(Transmitted_sibling = select(., t_sibling_male_dad:t_sibling_female_ind) %>% rowSums())%>% select(Transmitted_sibling) # extract transmitted variants for sibling 

Untransmitted_sibling=ASC_new_sample3%>% select(starts_with("u_sibling"))%>%  mutate(Untransmitted_sibling = select(., u_sibling_male_dad:u_sibling_female_ind) %>% rowSums())%>% select(Untransmitted_sibling) # extract untransmitted variants for sibling

##### add new columns of total transmitted and non-transmitted variants 
#ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & VQSLOD<9) # exclude indel & VQSLOD filter 

################################################


ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8)%>%filter(!str_detect(Consequence, 'intron_variant'))# exclude indel & apply other filters for SNP's 

##################
#AF_cutoff=0.001
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Transmitted_proband) %>% pull)   
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Untransmitted_proband) %>% pull)   
########### 1164 vs 1413 not 695 vs 557 in cell paper 


# add one column of variant group index 
ASC_new_sample_GI=ASC_new_sample %>% add_column(Group_Index=NA)

ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.995 & ASC_new_sample_GI$isPTV==T)]=1
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.5 & ASC_new_sample_GI$pLI<0.995 & ASC_new_sample_GI$isPTV==T)]=2
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI<0.5 & ASC_new_sample_GI$isPTV==T)]=3
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=2)]=4
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=1 & ASC_new_sample_GI$MPC<2)]=5
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC<1)]=6

num.family=7291
## test 
```







```{r, echo=F, message=F, warning=F, cache=T}
Laura_input_v17=as_tibble(read.table("C:\\Shengtong\\Research\\rare-var\\ASC\\Laura_result\\mirage_input_v17.txt", header=T))

num.family=7291
Han_input_v17=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
#write.csv(gene_set, file="C:/Shengtong/Research/rare-var/ASC/mirage_input_20230519.csv")  # input data fpr MIRAGE 
#mirage_result=mirage(data.frame(gene_set),n1=num.family, n2=num.family, gamma=c(6,6,6,3,3,3)) # input data must be dataframe 

Laura_Han_combine_v17=Laura_input_v17 %>% full_join(Han_input_v17)

############ transmitted variants 

g1=ggplot(Laura_Han_combine_v17, aes(x=No.case, y=Transmitted_proband))+geom_point()+
   xlim(c(0,30))+ylim(c(0,30))+
  xlab("Laura_v17")+ylab("Han_v17")+
  geom_abline(intercept =0, slope = 1, color="red", size=1)+
   ggtitle("overlapping transmitted alleles")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title

g2=ggplot(Laura_Han_combine_v17, aes(x=No.contr, y=Untransmitted_proband))+geom_point()+
   xlim(c(0,30))+ylim(c(0,30))+
  xlab("Laura_v17")+ylab("Han_v17")+
  geom_abline(intercept =0, slope = 1, color="red", size=1)+
   ggtitle("overlapping untransmitted alleles")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title

ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom")
```

* for overlapping variants  (89.1% of Laura's data, 78.9% of Han's data), they share same number of transmitted variants and untransmitted ones 


```{r, echo=T, message=F, warning=F}
Laura_Han_combine_v17 %>% filter(is.na(No.case)==T & is.na(Transmitted_proband)==F) # variants not in  Laura's data, but in mine 

Laura_Han_combine_v17 %>% filter(is.na(No.case)==F & is.na(Transmitted_proband)==T) # variants  in Laura's data, but not in mine

```


## gene set overlapping 


* `han_v17` ASCv17 from Kyle 

* `laura_v17`, additional filters applied to `han_v17`, with 85% overlapping, Laura ran MIRAGE to get parameter estimate that are close to parameter estimate in `han_v17`. 

* `laura_all` combined all dataset, including two SPARK data sets, `laura_v17`, and `ASC_B15_B16`. For every variant, aggregate all variant counts across 4 data sets, two SPARK data sets, ASCv17 and ASC_B15_B16. Then run MIRAGE to estimate parameters, and calculate BF.   




```{r echo=F, message=F, warning=F}

path="C:\\Shengtong\\research\\rare-var\\"

topgene_number=20

BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
topgenes=as.character(sort_BF$Gene[1:topgene_number])

laura_result=as_tibble(read.csv(paste(path, "ASC\\Laura_result\\Comparing Shengtong and Laura's inputs_2023_0524.csv", sep="")))
laura_result=laura_result %>% select(Gene, BF.ALL_combined, BF.ASC_v17_TDT_by_variant_2020.09.16) %>% mutate(BF.ASC_v17=as.numeric(BF.ASC_v17_TDT_by_variant_2020.09.16)) %>% drop_na()

###########   this is the result with parameteres estimated 
laura_result2=as_tibble(read.table(paste(path, "ASC\\Laura_result\\ALL_combined.combinedBF_2023_0729.txt", sep=""), header=T))

########## this is the result with parameteres fixed at 5% 
laura_result3=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))


laura_result_sort_by_ASCv17=laura_result%>% arrange(desc(BF.ASC_v17))
laura_result_sort_by_ASCv17_top_genes=laura_result_sort_by_ASCv17$Gene[1:topgene_number]

laura_result_sort_by_AllData=laura_result%>% arrange(desc(BF.ALL_combined))
laura_result_sort_by_AllData_top_genes=laura_result_sort_by_AllData$Gene[1:topgene_number]

laura_result_sort_by_AllData2=laura_result2%>% arrange(desc(BF_mirage))
laura_result_sort_by_AllData_top_genes2=laura_result_sort_by_AllData2$gene[1:topgene_number]

laura_result_sort_by_AllData3=laura_result3%>% arrange(desc(BF))
laura_result_sort_by_AllData_top_genes3=laura_result_sort_by_AllData3$Gene[1:topgene_number]

sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
mirage_topgenes=as.character(sort_BF$Gene[1:topgene_number])
#intersect(mirage_topgenes, laura_result_sort_by_ASCv17_top_genes)

##############
library("ggvenn")
x=list(
  laura_v17=laura_result_sort_by_ASCv17_top_genes,
  han_v17=mirage_topgenes
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 


x=list(
  laura_all=laura_result_sort_by_AllData_top_genes,
  han_v17=mirage_topgenes
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 

x=list(
  laura_all=laura_result_sort_by_AllData_top_genes,
  laura_v17=laura_result_sort_by_ASCv17_top_genes
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 

x=list(
  laura_all=laura_result_sort_by_AllData_top_genes,
  laura_all2=laura_result_sort_by_AllData_top_genes2
)
ggvenn(x)+
  ggtitle("overlap of top 20 genes")+
  theme(plot.title = element_text(hjust = 0.5, size=20, color = 2)) #center the title 




#############
han_BF=laura_ASCv17_BF=laura_allcombined_BF=laura_allcombined_BF2=numeric()
for (i in 1:(length(mirage_topgenes)-1))
{
  han_BF[i]=BF_para_all_est %>% filter(Gene==mirage_topgenes[i]) %>% select(BF_filtered) %>% pull()
  laura_ASCv17_BF[i]=laura_result%>% filter(Gene==mirage_topgenes[i]) %>% select(BF.ASC_v17) %>% pull()
  laura_allcombined_BF[i]=laura_result%>% filter(Gene==mirage_topgenes[i]) %>% select(BF.ALL_combined) %>% pull()
  laura_allcombined_BF2[i]=laura_result2%>% filter(gene==mirage_topgenes[i]) %>% select(BF_mirage) %>% pull()
}


ggplot(data.frame(Gene=rep(mirage_topgenes[1:19],4), logBF=c(log(han_BF), log(laura_ASCv17_BF), log(laura_allcombined_BF), log(laura_allcombined_BF2)), sample=rep(c("han_v17", "laura_v17", "laura_all", "laura_all2"), each=19)), aes(x=Gene, y=logBF, fill=sample)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge())+ 
  theme(legend.title=element_blank())+
  ylab("log(BF)")+
  ggtitle("BF of top 19 genes in han_v17")+
  theme(plot.title = element_text(hjust = 0.5, size=15, color = 2))+ #center the title
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))#+
 #theme(text = element_text(size=10))+
  #geom_text(aes(label=round(logBF,2)), position=position_dodge(width=0.8), vjust=-0.5, color="blue")

```


```{r, echo=F, message=F, warning=F}
han_BF=laura_ASCv17_BF=laura_allcombined_BF=numeric()
for (i in 1:(length(laura_result_sort_by_ASCv17_top_genes)))
{
  han_BF[i]=BF_para_all_est %>% filter(Gene==laura_result_sort_by_ASCv17_top_genes[i]) %>% select(BF_filtered) %>% pull()
  laura_ASCv17_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_ASCv17_top_genes[i]) %>% select(BF.ASC_v17) %>% pull()
  laura_allcombined_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_ASCv17_top_genes[i]) %>% select(BF.ALL_combined) %>% pull()
}


ggplot(data.frame(Gene=rep(laura_result_sort_by_ASCv17_top_genes,3), logBF=c(log(han_BF), log(laura_ASCv17_BF), log(laura_allcombined_BF)), sample=rep(c("han_v17", "laura_v17", "laura_all"), each=topgene_number)), aes(x=Gene, y=logBF, fill=sample)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge())+ 
  theme(legend.title=element_blank())+
  ylab("log(BF)")+
  ggtitle("BF of top 20 genes in laura_v17")+
  theme(plot.title = element_text(hjust = 0.5, size=15, color = 2))+ #center the title
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))#+
 #theme(text = element_text(size=10))+
  #geom_text(aes(label=round(logBF,2)), position=position_dodge(width=0.8), vjust=-0.5, color="blue")

```


```{r, echo=F, message=F, warning=F}
han_BF=laura_ASCv17_BF=laura_allcombined_BF=numeric()
for (i in 1:(length(laura_result_sort_by_AllData_top_genes)))
{
  han_BF[i]=BF_para_all_est %>% filter(Gene==laura_result_sort_by_AllData_top_genes[i]) %>% select(BF_filtered) %>% pull()
  laura_ASCv17_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_AllData_top_genes[i]) %>% select(BF.ASC_v17) %>% pull()
  laura_allcombined_BF[i]=laura_result%>% filter(Gene==laura_result_sort_by_AllData_top_genes[i]) %>% select(BF.ALL_combined) %>% pull()
}


ggplot(data.frame(Gene=rep(laura_result_sort_by_AllData_top_genes,3), logBF=c(log(han_BF), log(laura_ASCv17_BF), log(laura_allcombined_BF)), sample=rep(c("han_v17", "laura_v17", "laura_all"), each=topgene_number)), aes(x=Gene, y=logBF, fill=sample)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge())+ 
  theme(legend.title=element_blank())+
  ylab("log(BF)")+
  ggtitle("BF of top 20 genes in laura_all")+
  theme(plot.title = element_text(hjust = 0.5, size=15, color = 2))+ #center the title
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))#+
 #theme(text = element_text(size=10))+
  #geom_text(aes(label=round(logBF,2)), position=position_dodge(width=0.8), vjust=-0.5, color="blue")

```



```{r, echo=F, message=F, warning=F}
path="C:\\Shengtong\\research\\rare-var\\"
topgene_number=20
BF_para_all_est=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_6vargroups_BFFiltered_1118_2022.csv"))
burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv")) # call top genes by burden 
sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue) # rank genes by p values 
topgenes_burden=as.character(sort_pvalue$X[1:topgene_number])
all.gene_burden=as.character(sort_pvalue$X)
topmirage.percent_burden=numeric()

all_combined=as_tibble(read.table(paste(path, "ASC\\Laura_result\\ALL_combined.combinedBF.txt", sep=""), header=T))
all_combined_sort=all_combined %>% arrange(desc(BF_with_dn))
all_combined_top_gene=all_combined_sort$gene[1:topgene_number]



################
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
###################### strong ASD genes 

SFARI_gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol)) 

################

sort_BF=BF_para_all_est %>% arrange(desc(BF_filtered))
topgenes_han_v17=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
han_v17_percent=numeric(); all_gene_percent=numeric()
laura_v17_percent=numeric(); laura_all_percent=numeric(); laura_all_percent2=numeric(); laura_all_percent3=numeric()
all_combined_percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes_han_v17, Gene_Set_List[[i]]))
  han_v17_percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all_gene_percent[i]=overlap2/length(all.gene)
  
  overlap3=length(intersect(laura_result_sort_by_ASCv17_top_genes, Gene_Set_List[[i]]))
  laura_v17_percent[i]=overlap3/topgene_number
  
  overlap4=length(intersect(laura_result_sort_by_AllData_top_genes, Gene_Set_List[[i]]))
  laura_all_percent[i]=overlap4/topgene_number
  
  overlap6=length(intersect(laura_result_sort_by_AllData_top_genes2, Gene_Set_List[[i]]))
  laura_all_percent2[i]=overlap6/topgene_number
  
  overlap7=length(intersect(laura_result_sort_by_AllData_top_genes3, Gene_Set_List[[i]]))
  laura_all_percent3[i]=overlap7/topgene_number
  
  overlap5=length(intersect(all_combined_top_gene, Gene_Set_List[[i]]))
  all_combined_percent[i]=overlap5/topgene_number
  
  #pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  ####################
  overlap_burden=length(intersect(topgenes_burden, Gene_Set_List[[i]]))
  topmirage.percent_burden[i]=overlap_burden/topgene_number
}
############ include SFARI gene in gene set 
overlap1=length(intersect(topgenes_han_v17, SFARI_gene))
  han_v17_percent[i+1]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, SFARI_gene))
  all_gene_percent[i+1]=overlap2/length(all.gene)
  
  overlap3=length(intersect(laura_result_sort_by_ASCv17_top_genes, SFARI_gene))
  laura_v17_percent[i+1]=overlap3/topgene_number
  
  overlap4=length(intersect(laura_result_sort_by_AllData_top_genes, SFARI_gene))
  laura_all_percent[i+1]=overlap4/topgene_number
  
  overlap5=length(intersect(all_combined_top_gene, SFARI_gene))
  all_combined_percent[i+1]=overlap5/topgene_number
  
  overlap6=length(intersect(laura_result_sort_by_AllData_top_genes2, SFARI_gene))
  laura_all_percent2[i+1]=overlap6/topgene_number
  
  overlap7=length(intersect(laura_result_sort_by_AllData_top_genes3, SFARI_gene))
  laura_all_percent3[i+1]=overlap7/topgene_number

#topmirage.enrichment.tidy=data.frame(class=c(rep("han_v17",length(Gene_Set)+1), rep("All Genes", length(Gene_Set)+1), rep("laura_v17", length(Gene_Set)+1), rep("laura_all", length(Gene_Set)+1), rep("all_combined", length(Gene_Set)+1), rep("laura_all2", length(Gene_Set)+1), rep("laura_all3", length(Gene_Set)+1)), group=rep(c(Gene_Set, "SFARI"),7), percent=c(han_v17_percent, all_gene_percent, laura_v17_percent, laura_all_percent, all_combined_percent, laura_all_percent2, laura_all_percent3))
  
  topmirage.enrichment.tidy=data.frame(class=c(rep("han_v17",length(Gene_Set)+1), rep("All Genes", length(Gene_Set)+1),  rep("laura_all", length(Gene_Set)+1), rep("laura_all2", length(Gene_Set)+1), rep("laura_all3", length(Gene_Set)+1)), group=rep(c(Gene_Set, "SFARI"),5), percent=c(han_v17_percent, all_gene_percent,  laura_all_percent,  laura_all_percent2, laura_all_percent3))
######################



topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of genes in the gene set")+
  theme(axis.text.y = element_text(size = 10))+ # this is for numbers on y axis 
   ylim(c(0, 0.75))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by MIRAGE across different samples")+
    theme(legend.title = element_blank(), legend.text=element_text(size=9))+
   #geom_signif(annotations ="0.016", y_position = 0.115  ,xmin=0.7, xmax=1.3, size=1, vjust=-0.5, textsize=3.0)+ # textsize is for text over the bar; size is for width of lines 
  #geom_signif(annotations ="0.169", y_position = 0.065  ,xmin=0.7, xmax=1.0, size=1, vjust=-0.5, textsize=3.0)+
  # geom_signif(annotations ="3.32e-6", y_position = 0.465  ,xmin=1.7, xmax=2.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.618", y_position = 0.065  ,xmin=1.7, xmax=2.0, size=1, vjust=-0.5, textsize=3.0)+
  # geom_signif(annotations ="6.09e-5", y_position = 0.365  ,xmin=2.7, xmax=3.3, size=1, vjust=-0.5, textsize=3.0)+
  # geom_signif(annotations ="0.210", y_position = 0.115  ,xmin=2.7, xmax=3.0, size=1, vjust=-0.5, textsize=3.0)+
  # geom_signif(annotations ="8.53e-3", y_position = 0.315  ,xmin=3.7, xmax=4.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.029", y_position = 0.265  ,xmin=3.7, xmax=4.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.069", y_position = 0.065,xmin=4.7, xmax=5.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="", y_position = 0.065,xmin=4.7, xmax=5.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.223", y_position = 0.065,xmin=5.7, xmax=6.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="1.000", y_position = 0.025,xmin=5.7, xmax=6.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.247", y_position = 0.065,xmin=6.7, xmax=7.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="1.000", y_position = 0.025,xmin=6.7, xmax=7.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.041", y_position = 0.165,xmin=7.7, xmax=8.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.513", y_position = 0.065,xmin=7.7, xmax=8.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="2.02e-6", y_position = 0.465,xmin=8.7, xmax=9.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.017", y_position = 0.215,xmin=8.7, xmax=9.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="5.06e-4", y_position = 0.465,xmin=9.7, xmax=10.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.057", y_position = 0.265,xmin=9.7, xmax=10.0, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.046", y_position = 0.113,xmin=10.7, xmax=11.3, size=1, vjust=-0.5, textsize=3.0)+
  #geom_signif(annotations ="0.280", y_position = 0.065,xmin=10.7, xmax=11.0, size=1, vjust=-0.5, textsize=3.0)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=13))+ # size is for labels on x axis 
  theme(axis.title = element_text(size = 13))+ # Font Size of Axis Titles
 theme(legend.spacing.x = unit(0.5, 'cm')) # add space between symbols in legend 


 
 #png("C:/Shengtong/Research/rare-var/RareVariant/2023_0413_Figure_for_paper/Fig-enrichment_of_top_genes.png")
 topmirage_enrichment
 #dev.off()
```

* `all_combined`: combine de novo mutation (TADA BF), and transmitted, case control (MIRAGE BF). 

* only difference of `laura_all2` to `laura_all` is that parameters are estimated in `laura_all2`.  

* `laura_all3` fixed delta=0.05. 

* `laura_all` fixed delta=0.067. 



## Analysis of all transmitted variants aggregated


```{r, echo=F, message=F, warning=F, eval=F}
################## DO NOT RUN ########################
path="C:\\Shengtong\\research\\rare-var\\"

mirage_input=as_tibble(read.table(paste(path, "ASC\\Laura_result\\mirage_input_20231013.txt", sep=""), header=T))
N_all=14578  # from Laura's script: NOTES_wrap_mirage_100323
#res=mirage(data.frame(mirage_input),n1=N_all,n2=N_all,gamma=c(6,6,6,3,3,3),delta.init = 0.05,estimate.delta = FALSE)
mirage_input_for_gnomAD_AF=mirage_input %>% select(Variant, Gene)
#write.csv(mirage_input_for_gnomAD_AF, file=paste(path, "ASC\\Laura_result\\mirage_input_for_gnomAD_AF_20231106.csv", sep=""))
```



### Parameter estimates 


```{r, echo=F, message=F, warning=F}

#mirage_para_est=data.frame(variant_groups=c("delta:proportion of risk genes","PTV_PLI>=0.995", "PTV_0.5=<PLI<0.995", "PTV_PLI<0.5",  "MPC>=2",  "1=<MPC<2", "MPC<1"), parameter_estimate=round(c(0.06713907, 0.83382753,0.85448876,0.02985442,0.47471948, 0.11521000, 0.01201084),4), p_value=pvalue_adjust(p_value))


var_group=rep(c("(1):PLI>=0.995", "(2):pLI>=0.5&pLI<0.995", "(3):pLI<0.5", "(4):MPC>=2", "(5):MPC>=1&MPC<2", "(6):MPC<1"),3)
estimate=c(0.83382753,0.85448876,0.02985442,0.47471948, 0.11521000, 0.01201084,0.353828740524254,
0.157066593507005,
0.0215284458160296,
0.0912380102963087,
0.033530954375487,
0.00664397682712002, 
0.738664291977854,
0.701330924907413,
0.0931459290392402,
0.366135847737628,
0.0731974976958973,
0.0386167362363095
)
sample=rep(c("ASC_v17_estimate_delta", "all_inherited_estimate_delta", "all_inherited_fixed_delta"), each=6)
summary=data.frame(var.group=var_group, estimate=estimate, sample=sample)
ggplot(summary, aes(x=var.group, y=estimate, fill=sample)) + 
   geom_bar(stat="identity", position=position_dodge())+
  theme(legend.title=element_blank())+
  xlab("variant groups")+ylab(expression(paste(hat(eta))))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  #center the title 
  theme(axis.text.x = element_text(angle = 30, vjust = 0.5, size=10))
  
```  


* [parameter estimates by MIRAGE for all inherited variants](https://www.dropbox.com/scl/fi/42yulompgjmiaf7jxg3vi/MIRAGE-parameters-072023.docx?rlkey=dnt8cb5e5y0jdkvobrjnaadgd&dl=0) 

* [parameter estimates by MIRAGE for all inherited variants with fixed delta=5%](https://www.dropbox.com/scl/fo/33polh7c3tx2lu7mi8qzh/h/ALL_combined?dl=0&preview=eta.est.txt&rlkey=geomwls9yq1vrw87rpm0tnrxk&subfolder_nav_tracking=1)

* $\widehat{\delta}=0.307$ for all inherited variants and  $\widehat{\delta}=0.067$ for ASC_v17.  



### Occurrence of top 20 ASD genes in other gene sets 


```{r, echo=F, message=F, warning=F, eval=T}
#gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
################ top genes 
BF_proband_AllGene=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))

BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=20
gene_to_display=BF_proband_AllGene_sorted$Gene[1:num_gene_to_display]
SFARI_score=rep("NA", num_gene_to_display)
Incidence_gene=matrix(nrow=num_gene_to_display, ncol=11)
for (i in 1:length(gene_to_display))
{


if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
for (j in 1:10)
 Incidence_gene[i,1+j]=as.numeric(gene_to_display[i]%in% Gene_Set_List[[j]])

}
rownames(Incidence_gene)=gene_to_display
colnames(Incidence_gene)=c("BF",Gene_Set)
Incidence_gene[,1]=round(BF_proband_AllGene_sorted$BF[1:num_gene_to_display],2)
Incidence_gene%>%
datatable(extensions = 'Buttons',
          caption = 'Occurrence of top 20 genes in various gene sets',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```

* top 20 ASD genes by running MIRAGE on all inherited variants with $\delta=5\%$. 


### BF in each variant group with fixed $\delta=5\%$

```{r, echo=F, message=F, warning=F, eval=F}
################# calculate BF in each group 
#BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))
########## this is the result with parameters fixed at 5% 
BF_proband_AllGene=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))

BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
interest_gene=BF_proband_AllGene_sorted$Gene  # use all genes 
beta=c(0.738664291977854,
0.701330924907413,
0.0931459290392402,
0.366135847737628,
0.0731974976958973,
0.0386167362363095)
#####################
### read into input data
#new.data=ASC_new_sample_GI %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()
new.data=read.table(paste(path, "ASC\\Laura_result\\mirage_input_20231013.txt", sep=""), header=T)
N_all=14578  # from Laura's script: NOTES_wrap_mirage_100323
#res=mirage(data.frame(mirage_input),n1=N_all,n2=N_all,gamma=c(6,6,6,3,3,3),delta.init = 0.05,estimate.delta = FALSE)
#colnames(new.data)[3:5]=c("No.case", "No.contr", "group.index")
result=fixed.beta.func(new.data, N1=N_all, N0=N_all, gamma.mean=c(rep(6,3),rep(3,3)), sigma=2,beta=beta) # use fixed beta to calculate Bayes factors

############## verify the calculations 
#gene_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="SSC5D")]]
#group.index=gene_variant_BF$group.index
#gene_BF=prod(1-beta[group.index]+beta[group.index]*gene_variant_BF$var.BF)
#gene_BF
#result$BayesFactor[result$BayesFactor$Gene=="SSC5D",]
##############
variant_BF=matrix(nrow=length(interest_gene), ncol=7)
variant_count=matrix(nrow=length(interest_gene), ncol=7)
SFARI_score=rep("NA", length(interest_gene))
for (i in 1:length(interest_gene))
{
  cat(i, "th gene is running", "\n")
  variant_BF[i,1]=result$BayesFactor$BF[which(result$BayesFactor$Gene==interest_gene[i])]
  var_full=result$full.info[[which(result$BayesFactor$Gene==interest_gene[i])]]
  variant_count[i,1]=paste(paste(sum(var_full$No.case)),"vs", paste(sum(var_full$No.contr)), sep=" ")
   for (j in 1:6)
   {
     if (nrow(var_full %>% filter(group.index==j))==0) # no variants in this group
      { 
        variant_BF[i,j+1]="-"
        variant_count[i,j+1]="-"
     }
     
     if (nrow(var_full %>% filter(group.index==j))>0) # variants exist in this group
      { 
       variant_BF[i,j+1]=prod(1-beta[j]+beta[j]*var_full[which(var_full$group.index==j),]$var.BF) # product of BF of variants within that variant group 
       variant_count[i,j+1]=paste(paste(sum(var_full[which(var_full$group.index==j),]$No.case)),"vs", paste(sum(var_full[which(var_full$group.index==j),]$No.contr)), sep=" ")
     }
     
   }  #### end of 
  if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==interest_gene[i]),]$gene.score
  
  
}
BF_var_count_combine_6groups=data.frame(Gene=interest_gene, BF=round(as.numeric(variant_BF[,1]),4),  SFARI_score=SFARI_score, total_count=variant_count[,1], g1_count=variant_count[,2], g1_BF=round(as.numeric(variant_BF[,2]),2), g2_count=variant_count[,3], g2_BF=round(as.numeric(variant_BF[,3]),2), g3_count=variant_count[,4], g3_BF=round(as.numeric(variant_BF[,4]),2), g4_count=variant_count[,5], g4_BF=round(as.numeric(variant_BF[,5]),2), g5_count=variant_count[,6], g5_BF=round(as.numeric(variant_BF[,6]),2), g6_count=variant_count[,7], g6_BF=round(as.numeric(variant_BF[,7]),2))                                          

#write.csv(BF_var_count_combine_6groups, file="../output/LauraData/MIRAGE_all_inherited_variant_fixed_delta_2023_1023.csv")


#################### given a gene, output its variant level BF
#Gene_MYH10_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="MYH10")]]
#write.csv(Gene_MYH10, file=paste(path, "ASC\\Laura_result\\Gene_MYH10_T&UT_variant_BF_20231102.csv", sep=""))

#Gene_PTPRF_variant_BF=result$full.info[[which(result$BayesFactor$Gene=="PTPRF")]]
#write.csv(Gene_PTPRF, file=paste(path, "ASC\\Laura_result\\Gene_PTPRF_T&UT_variant_BF_20231102.csv", sep=""))

```

[variant count and BF in every group](https://marq-my.sharepoint.com/:x:/g/personal/shengtong_han_marquette_edu/Ee0vmWgnInhDh5_SNQBmRFwB9b09gPto2JhX-Hk7tQFpxg?e=CLVnF7)



### Enrichment of top 20 genes with fixed $\delta=5\%$ 


```{r, echo=F, warning=F, message=F}
path="C:\\Shengtong\\research\\rare-var\\"
topgene_number=20
BF_para_all_est=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))
#burden_pvalues_001=as_tibble(read.csv("../output/KyleData/BF.proband.AllGene_6vargroups_burden_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv")) # call top genes by burden 
#sort_pvalue=burden_pvalues_001 %>% arrange(All_variants_pvalue) # rank genes by p values 
#topgenes_burden=as.character(sort_pvalue$X[1:topgene_number])
#all.gene_burden=as.character(sort_pvalue$X)
#topmirage.percent_burden=numeric()

################
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)
###################### strong ASD genes 

SFARI_gene=c(as.character(SFARI.gene[which(SFARI.gene$gene.score==1), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==2), ]$gene.symbol), as.character(SFARI.gene[which(SFARI.gene$gene.score==3), ]$gene.symbol)) 

################

sort_BF=BF_para_all_est %>% arrange(desc(BF))
topgenes=as.character(sort_BF$Gene[1:topgene_number])
all.gene=as.character(sort_BF$Gene)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  ####################
 # overlap_burden=length(intersect(topgenes_burden, Gene_Set_List[[i]]))
#  topmirage.percent_burden[i]=overlap_burden/topgene_number
}
############ include SFARI gene in gene set 
overlap1=length(intersect(topgenes, SFARI_gene))
  topmirage.percent[i+1]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, SFARI_gene))
  all.gene.percent[i+1]=overlap2/length(all.gene)
  
#  overlap_burden=length(intersect(topgenes_burden, SFARI_gene))
#  topmirage.percent_burden[i+1]=overlap_burden/topgene_number
  

#topmirage.enrichment.tidy=data.frame(class=c(rep("MIRAGE",length(Gene_Set)+1), rep("All Genes", length(Gene_Set)+1), rep("LoF+MPC>2", length(Gene_Set)+1)), group=rep(c(Gene_Set, "SFARI"),3), percent=c(topmirage.percent, all.gene.percent, topmirage.percent_burden))
  
  ######## without burden test 
  topmirage.enrichment.tidy=data.frame(class=c(rep("MIRAGE",length(Gene_Set)+1), rep("All Genes", length(Gene_Set)+1)), group=rep(c(Gene_Set, "SFARI"),2), percent=c(topmirage.percent, all.gene.percent))
######################



topmirage_enrichment=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of genes in the gene set")+
  theme(axis.text.y = element_text(size = 10))+ # this is for numbers on y axis 
   ylim(c(0, 0.5))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by burden and MIRAGE")+
    theme(legend.title = element_blank())+
   geom_signif(annotations ="0.016", y_position = 0.115  ,xmin=0.7, xmax=1.3, size=1, vjust=-0.5, textsize=3.0)+ # textsize is for text over the bar; size is for width of lines 
  geom_signif(annotations ="0.169", y_position = 0.065  ,xmin=0.7, xmax=1.0, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="3.32e-6", y_position = 0.465  ,xmin=1.7, xmax=2.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.618", y_position = 0.065  ,xmin=1.7, xmax=2.0, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="6.09e-5", y_position = 0.365  ,xmin=2.7, xmax=3.3, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="0.210", y_position = 0.115  ,xmin=2.7, xmax=3.0, size=1, vjust=-0.5, textsize=3.0)+
   geom_signif(annotations ="8.53e-3", y_position = 0.315  ,xmin=3.7, xmax=4.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.029", y_position = 0.265  ,xmin=3.7, xmax=4.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.069", y_position = 0.065,xmin=4.7, xmax=5.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="", y_position = 0.065,xmin=4.7, xmax=5.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.223", y_position = 0.065,xmin=5.7, xmax=6.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=5.7, xmax=6.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.247", y_position = 0.065,xmin=6.7, xmax=7.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="1.000", y_position = 0.025,xmin=6.7, xmax=7.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.041", y_position = 0.165,xmin=7.7, xmax=8.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.513", y_position = 0.065,xmin=7.7, xmax=8.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="2.02e-6", y_position = 0.465,xmin=8.7, xmax=9.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.017", y_position = 0.215,xmin=8.7, xmax=9.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="5.06e-4", y_position = 0.465,xmin=9.7, xmax=10.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.057", y_position = 0.265,xmin=9.7, xmax=10.0, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.046", y_position = 0.113,xmin=10.7, xmax=11.3, size=1, vjust=-0.5, textsize=3.0)+
  geom_signif(annotations ="0.280", y_position = 0.065,xmin=10.7, xmax=11.0, size=1, vjust=-0.5, textsize=3.0)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=13))+ # size is for labels on x axis 
  theme(axis.title = element_text(size = 13))+ # Font Size of Axis Titles
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 


 
 #png("C:/Shengtong/Research/rare-var/RareVariant/2023_0413_Figure_for_paper/Fig-enrichment_of_top_genes.png")
 topmirage_enrichment
 #dev.off()
```


### Manhattan plot


```{r, echo=F, message=F, warning=F, eval=F}
############## run once to get the data for plot 
path="C:\\Shengtong\\research\\rare-var\\"

###### this is the input aggregating all inherited variants  
new.data=as_tibble(read.table(paste(path, "ASC\\Laura_result\\mirage_input_20231013.txt", sep=""), header=T))


########## this is the result with parameters fixed at 5% 
BF_proband_AllGene=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(BF))
#delta_estimate=0.0671 # the estimate of delta, proportion of risk genes 
#post_prob=BF_proband_AllGene_sorted$BF_filtered*delta_estimate/(BF_proband_AllGene_sorted$BF_filtered*delta_estimate+1-delta_estimate) # calculate post probability by definition

data_for_plot=data.frame(Gene=BF_proband_AllGene_sorted$Gene, BF=BF_proband_AllGene_sorted$BF, post_prob=BF_proband_AllGene_sorted$post.prob)
##### extrat chr and position for each gene 
chr=numeric(); pos=numeric()
for (i in 1:nrow(data_for_plot))
#for (i in 1:2)  
{
  cat(i, "is running", "\n")
  variant=new.data %>% filter(Gene==data_for_plot$Gene[i]) %>% select(Variant)%>% pull() # find all variants for a gene 
  colon_pos=unlist(gregexpr(":", variant[1])) # location positions of ":" # use the "first" variant in the gene 
  chr[i]=substr(variant[1], 1,colon_pos[1]-1)
  pos[i]=substr(variant[1], colon_pos[1]+1, colon_pos[2]-1)
} 
data_manhattan_plot=data_for_plot%>%mutate(chr=as.numeric(chr), pos=as.numeric(pos))
#write.csv(data_manhattan_plot, file=paste(path, "ASC\\Laura_result\\data_manhattan_plot_AllInheritedVariant_deltafixed_20231102.csv", sep=""))

Gene_MYH10=new.data %>% filter(Gene=="MYH10") 
#write.csv(Gene_MYH10, file=paste(path, "ASC\\Laura_result\\Gene_MYH10_T&UT_20231102.csv", sep=""))

Gene_PTPRF=new.data %>% filter(Gene=="PTPRF") 
#write.csv(Gene_PTPRF, file=paste(path, "ASC\\Laura_result\\Gene_PTPRF_T&UT_20231102.csv", sep=""))

```




###  false positive control 

* Run MIRAGE to synonymous variants only, and $\widehat{\delta}=0.017$ with p value 0.086, $\widehat{\eta}=0.079$ with p value 0.027. 


```{r, echo=F, message=F, warning=F, results=F}
path="C:\\Shengtong\\research\\rare-var\\"
multiplesheets <- function(fname) {
   
  # getting info about all excel sheets
  sheets <- readxl::excel_sheets(fname)
  tibble <- lapply(sheets, function(x) readxl::read_excel(fname, sheet = x))
  data_frame <- lapply(tibble, as.data.frame)
    
  # assigning names to data frames
  names(data_frame) <- sheets
    
  # print data frame
  print(data_frame)
}
syn_genes=multiplesheets(paste(path, "ASC\\Laura_result\\Examining highest and lowest BFs 100723.xlsx", sep=""))
```


```{r, echo=F, message=F, warning=F}

path="C:\\Shengtong\\research\\rare-var\\"

all_combined_syn=as_tibble(read.table(paste(path, "ASC\\Laura_result\\mirage_input_20231012.txt", sep=""), header=T))

syn_top_genes=syn_genes[[1]][2:16,1]
OR=numeric(); Trans=numeric(); Untrans=numeric()
for (i in 1:length(syn_top_genes))
{
  Trans[i]=sum(all_combined_syn %>% filter(Gene==syn_top_genes[i]) %>% select(No.case) %>% pull)
  Untrans[i]=sum(all_combined_syn %>% filter(Gene==syn_top_genes[i]) %>% select(No.contr) %>% pull)
  OR[i]=round(Trans[i]/Untrans[i],4)
  
}
data.frame(Gene=syn_top_genes, Trans=Trans, Untrans=Untrans, OR=OR, BF=round(syn_genes[[1]][2:16,2],4), Post_prob=round(syn_genes[[1]][2:16,3],4))%>%
datatable(extensions = 'Buttons',
          caption = "top genes when run MIRAGE on syn mutations", 
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))



ggplot(data.frame(Gene=syn_top_genes, Trans=Trans, Untrans=Untrans), aes(x=Trans, y=Untrans))+
  geom_point()+
 #  xlim(c(0,30))+ylim(c(0,30))+
  xlab("T")+ylab("UT")+
  geom_abline(intercept =0, slope = 1, color="red", size=1)+
   ggtitle("T vs UT for syns")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title
```


### BF breakdown per gene 


```{r, echo=F, message=F, warning=F, results=F}
path="C:\\Shengtong\\research\\rare-var\\"
multiplesheets <- function(fname) {
   
  # getting info about all excel sheets
  sheets <- readxl::excel_sheets(fname)
  tibble <- lapply(sheets, function(x) readxl::read_excel(fname, sheet = x))
  data_frame <- lapply(tibble, as.data.frame)
    
  # assigning names to data frames
  names(data_frame) <- sheets
    
  # print data frame
  print(data_frame)
}
BF_breakdown_gene=multiplesheets(paste(path, "ASC\\Laura_result\\Rough approximation variants for each gene for all datasets combined_20231010.xlsx", sep=""))
```

```{r, echo=F, message=F, warning=F, eval=F}
########## this is the result with parameteres fixed at 5% 
laura_result3=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))
var_group_BF=matrix(nrow=nrow(laura_result3), ncol=6)
#for (g in 1:nrow(laura_result3))
{
  g=1
  single_gene=BF_breakdown_gene[[1]] %>% filter(Gene...1==laura_result3$Gene[g])
  
  ### one variant only 
  if (nrow(single_gene)==1)
  var_group_BF[g, single_gene$group.index]=laura_result3$BF[g]*single_gene$norm
  
  ### more than one variant
  if (nrow(single_gene)>1)
   for (v in 1:nrow(single_gene))
    {
     var_group_BF[g, single_gene$group.index[v]]=laura_result3$BF[g]*single_gene$norm[v]
    
  }
  
} # end of for (g in 1:nrow(laura_result3))

BF_breakdown=data.frame(Gene=laura_result3$Gene, highPLI_BF=var_group_BF[,1], medPLI_BF=var_group_BF[,2], lowPLI_BF=var_group_BF[,3], misB_BF=var_group_BF[,4], misA_BF=var_group_BF[,5], mis_BF=var_group_BF[,6], BF=laura_result3$BF, post_prob=laura_result3$post.prob)

```


### Bayesian FDR


```{r, echo=F, message=F, warning=F, eval=T}
#data_manhattan_plot=read.csv("../output/KyleData/data_manhattan_plot_0105_2023.csv", header=T)

########
#laura_result=as_tibble(read.csv(paste(path, "ASC\\Laura_result\\Comparing Shengtong and Laura's inputs_2023_0524.csv", sep="")))
#laura_result=laura_result %>% select(Gene, BF.ALL_combined) %>% drop_na()

#laura_result_sort_by_ASCv17=laura_result%>% arrange(desc(BF.ASC_v17))
#laura_result_sort_by_ASCv17_top_genes=laura_result_sort_by_ASCv17$Gene[1:topgene_number]

##############
########## this is the result with parameteres fixed at 5% 
laura_result3=as_tibble(read.table(paste(path, "ASC\\Laura_result\\BF.PP.gene_2023_1009.txt", sep=""), header=T))


fdr_function=function(fdr_threshold, data)
{
tau=seq(0, 0.999, by=0.001)
num_pred=NULL
false_disc=NULL
FDR=NULL
for (i in 1:length(tau))
{
num_pred[i]=sum(ifelse(data$post_prob>tau[i], 1, 0))
false_disc[i]=sum((1-data$post_prob)*ifelse(data$post_prob>tau[i], 1, 0))
FDR[i]=false_disc[i]/num_pred[i]
}
tau_fdr=tibble(tau=tau, fdr=FDR)%>%drop_na()  # drop rows with NA
tau_at_threshold=tau_fdr%>%filter(fdr<=fdr_threshold)
if (nrow(tau_at_threshold)==0)
  return("No gene is selected")
if (nrow(tau_at_threshold)>0)
{
  tau_threshold=tau_at_threshold$tau[1]
  return(data%>% filter(post_prob>tau_threshold))
}

}
```

```{r, echo=T, message=F, warning=F, eval=T}
input_data=laura_result3
colnames(input_data)[3]="post_prob"
fdr_function(fdr_threshold = 0.2, input_data)
#fdr_function(fdr_threshold = 0.45, data_manhattan_plot)
#fdr_function(fdr_threshold = 0.5, data_manhattan_plot)
```
