---
title: "liftover samples"
output: html_document
date: ''
---

```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
#install.packages("devtools") # run once 
library(devtools)
library(DT)
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
```



```{r, echo=F, cache=T, warning=F, message=F}
path="C:\\Shengtong\\Research\\rare-var\\"
### old sample
########## read into data
All.Anno.Data=as_tibble(read.table(paste(path, "AnnotatedTrans.txt", sep=""), header=T))
N1=4315; N0=4315
All.Anno.Data[All.Anno.Data =="."] <- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF<0.05 & All.Anno.Data$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonymous SNV
#Anno.Data=All.Anno.Data[which(All.Anno.Data$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonymous SNV

##### old sample after liftover 
All.Anno.Data_liftover=as_tibble(readRDS("C:\\Shengtong\\Research\\rare-var\\ASC\\hg19tohg38.rds"))
All.Anno.Data_liftover[All.Anno.Data_liftover =="."] <- NA
All.Anno.Data_liftover$ExacAF[is.na(All.Anno.Data_liftover$ExacAF)]=0 # set AF of NA to zero 
Anno.Data_liftover=All.Anno.Data_liftover[which(All.Anno.Data_liftover$ExacAF<0.05 & All.Anno.Data_liftover$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonymous SNV
```


```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
#ASC_new_sample1=as_tibble(read.table(paste(path, "ASC\\ASC_B15_B16_raw_TDT_by_parent_2021-04-19.txt", sep=""), header=T, fill=T))
#ASC_new_sample2=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), header=T, fill=T))  ##### this way of reading data is not correct

ASC_new_sample3=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), sep='\t',  header=T, fill=T)) # this is the correct way Kyle read the data

################ new sample after liftover 
ASC_new_sample_liftover1=as_tibble(readRDS("C:\\Shengtong\\Research\\rare-var\\ASC\\hg38tohg19.rds"))
ASC_new_sample_liftover2=ASC_new_sample_liftover1  # without any filters 
ASC_new_sample_liftover1=ASC_new_sample_liftover1 %>% filter(!str_detect(Consequence, 'intron_variant'))%>%filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8) %>% filter(isSyn=="FALSE") %>% filter(Gnomad_non_neuro_AF<0.05)# exclude indel & apply other filters for SNP's
```


```{r, echo=F, message=F, warning=F, eval=T}
## identify transmitted and non-transmitted variants 
#transmitted_variant=ASC_new_sample2%>%select(starts_with(c("t", "u")), "Gnomad_non_neuro_AF")
#kable(transmitted_variant[1:10,])%>%
#kable_styling() %>%
#scroll_box(height = "500px")
#write.csv(transmitted_variant, file="C:\\Shengtong\\Research\\rare-var\\ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10_transmitted_variant.csv")

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
######### use ASC_new_sample3 rather than ASC_new_sample2
Transmitted_proband=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) 
Untransmitted_proband=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) 



##### count transmitted variants and untransmitted in proband from parents  by summing over male, female, dad, mom  
Transmitted_proband_parents=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband_parents = select(., t_proband_male_dad, t_proband_male_mom, t_proband_female_dad, t_proband_female_mom) %>% rowSums())%>% select(Transmitted_proband_parents) 
Untransmitted_proband_parents=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband_parents= select(., u_proband_male_dad, u_proband_male_mom, u_proband_female_dad, u_proband_female_mom) %>% rowSums())%>% select(Untransmitted_proband_parents) 


##### add new columns of total transmitted and non-transmitted variants 
ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_proband_parents=Transmitted_proband_parents$Transmitted_proband_parents, Untransmitted_proband_parents=Untransmitted_proband_parents$Untransmitted_proband_parents)

#################################
ASC_new_sample_nofilter=ASC_new_sample
ASC_new_sample=ASC_new_sample%>%  filter(!str_detect(Consequence, 'intron_variant')) %>% filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8) %>% filter(isSyn=="FALSE") %>% filter(Gnomad_non_neuro_AF<0.05) # exclude indel & apply other filters for SNP's

```





## Allele frequency: old sample vs new sample 

```{r, echo=F, message=F, warning=F, cache=T}
old_data=Anno.Data %>% select(ID, ExacAF)
old_data_liftover=Anno.Data_liftover %>% select(seqnames, start, ExacAF, allele1, allele2) 
new_data=ASC_new_sample %>% select(Variant, Gnomad_non_neuro_AF)
### match IDs in both data sets 
old_pos=unlist(gregexpr("-", old_data$ID))  # locate the position of "-"
old_ID=substr(old_data$ID, 1,old_pos-1)
###########
#old_data_new_ID=str_replace_all(old_data$ID, "_", ":")
old_pos_liftover=nchar(as.character(old_data_liftover$seqnames)) # find the length of string, the last position
old_ID_liftover=paste(substr(old_data_liftover$seqnames, 4, old_pos_liftover), old_data_liftover$start, old_data_liftover$allele1, old_data_liftover$allele2, sep=":") # extract loci position without "chr", starting from 4th position to the end, to be consistent with format with other data 

new_pos=unlist(gregexpr(":", new_data$Variant))[2]
new_ID=substr(new_data$Variant, 1,new_pos-1)



old_data1=tibble(ID=old_ID, AF_old=old_data$ExacAF)
old_data1_liftover=tibble(ID=old_ID_liftover, AF_old_liftover=old_data_liftover$ExacAF)
new_data1=tibble(ID=new_ID, AF_new=new_data$Gnomad_non_neuro_AF)
new_data2=new_data %>% rename(ID=Variant, AF_new=Gnomad_non_neuro_AF)

old_new_combine=old_data1 %>% full_join(new_data1) # combine two data sets: old & new 
old_liftover_new_combine=old_data1_liftover %>% full_join(new_data2) # combine two data sets: old_liftover & new
  
old_new_combine_withoutNA=old_new_combine %>% drop_na()  # remove rows with NA 
old_liftover_new_combine_withoutNA=old_liftover_new_combine %>% drop_na() # removes rows with NA

g1=ggplot(old_new_combine_withoutNA, aes(x=AF_new, y=AF_old))+geom_point()+
  geom_abline(intercept =0, slope = 1, color="red")+
  #xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  #ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("Overlapping variants")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title
#g1

old_new_combine2=old_new_combine %>% replace_na(list(AF_old=2, AF_new=2))  # replace NA with 2 
g2=ggplot(old_new_combine2, aes(x=AF_new, y=AF_old))+geom_point()+
  geom_abline(intercept =0, slope = 1, color="red")+
  #xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  #ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("Union of variants")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title
#g2

ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 


g3=ggplot(old_liftover_new_combine_withoutNA, aes(x=AF_new, y=AF_old_liftover))+geom_point()+
  geom_abline(intercept =0, slope = 1, color="red", size=2)+
  #xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  #ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("Overlapping variants")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title


old_new_combine2=old_liftover_new_combine %>% replace_na(list(AF_old_liftover=2, AF_new=2))  # replace NA with 2 
g4=ggplot(old_new_combine2, aes(x=AF_new, y=AF_old_liftover))+geom_point()+
  geom_abline(intercept =0, slope = 1, color="red")+
  #xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  #ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("Union of variants")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title

ggarrange(g3, g4,nrow=1, common.legend = TRUE, legend="bottom")
```




```{r, echo=F, message=F, warning=F, cache=T}
########### compare AF in new sample after liftover and old sample
new_pos_liftover=nchar(as.character(ASC_new_sample_liftover1$seqnames)) # find the length of string, the last position
new_ID_liftover=paste(substr(ASC_new_sample_liftover1$seqnames, 4, new_pos_liftover), ":", ASC_new_sample_liftover1$start, "-", ASC_new_sample_liftover1$end, "_", ASC_new_sample_liftover1$allele1,"_", ASC_new_sample_liftover1$allele2, sep="") # extract loci position without "chr", starting from 4th position to the end, to be consistent with format with other data

############ combine old & new data after liftover 
old_data1=tibble(ID=old_data$ID, AF_old=old_data$ExacAF)
new_data1=tibble(ID=new_ID_liftover, AF_new_liftover=ASC_new_sample_liftover1$Gnomad_non_neuro_AF)
old_new_liftover_combine=old_data1 %>% full_join(new_data1) # combine two data sets: old & new 
old_new_liftover_combine_withoutNA=old_new_liftover_combine %>% drop_na()  # remove rows with NA


g5=ggplot(old_new_liftover_combine_withoutNA, aes(x=AF_new_liftover, y=AF_old))+geom_point()+
  geom_abline(intercept =0, slope = 1, color="red", size=2)+
  #xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  #ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("Overlapping variants")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title
#g1

old_new_liftover_combine2=old_new_liftover_combine %>% replace_na(list(AF_old=2, AF_new_liftover=2))  # replace NA with 2 
g6=ggplot(old_new_liftover_combine2, aes(x=AF_new_liftover, y=AF_old))+geom_point()+
  geom_abline(intercept =0, slope = 1, color="red")+
  #xlab(expression(paste("Burden: -log" [10], "(p value)")))+
  #ylab(expression(paste("MIRAGE_VS: -log" [10], "(p value)")))+
   ggtitle("Union of variants")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=15))  # center the title
#g2
ggarrange(g5, g6,nrow=1, common.legend = TRUE, legend="bottom")
```



* for both samples, focus on variants with AF<5% and exclude syn mutations. 
* each point is one single variant  
* left panel: 344 (0.0188%) overlapping loci with numeric AF from both old and new sample; one example is `7:100683050 0.042 (AF_old)      0.0000816(AF_new)`, in Gnomad database, V2 say AF is `4.05e-2`, ExAC AF is `4.22e-2`, and v3 (hg38) AF is `3.28e-5`. This example suggests `Gnomad_non_neuro_AF` is close to genomewide AF under hg38. (**old sample is under hg 19 and new sample under hg38=> they are incompatible**)

```
> ASC_new_sample %>% filter(Variant=="7:100683050:G:A") %>% select(Variant, Gene, pLI, Consequence, Gnomad_non_neuro_AF)
# A tibble: 1 x 5
  Variant         Gene     pLI Consequence      Gnomad_non_neuro_AF
  <chr>           <chr>  <dbl> <chr>                          <dbl>
1 7:100683050:G:A GIGYF1 0.979 missense_variant           0.0000816
```


```
> Anno.Data %>%
+     filter(str_detect(ID, '7:100683050')) %>% select(ID, Gene, ExacAF)
# A tibble: 1 x 3
  ID                        Gene  ExacAF
  <chr>                     <chr>  <dbl>
1 7:100683050-100683050_T_G MUC17  0.042
```


* right panel: 1826590 loci with `NA` AF replaced with 2 either from new or old sample 


* in the two figures in the middle, old samples are liftover, a larger number of loci (272,141, 19.31%, 272,141/1,409,319) are overlapping between new sample and liftover sample of old data.  


* a small number of loci with variants not in Gnomad, their new AF will be zero, thus a small number of loci stacking at `AF_new=0`  for example, 

```
ID              AF_old_liftover AF_new
  <chr>                     <dbl>  <dbl>
1 1:1527777:G:A           0.00540      0
2 2:172491230:G:T         0.01         0
3 4:9444548:T:G           0.00616      0
4 5:137870800:A:C         0.00772      0
5 9:18928678:G:T          0.00800      0
```

* new sample both with and without liftover, apply the filter `isIndel=="FALSE" & QD >= 1 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8`,  `!str_detect(Consequence, 'intron_variant')`, `isSyn=="FALSE"`, and `Gnomad_non_neuro_AF<0.05` to exclude intron variants 


## Gene-CYFIP1


```{r, echo=F, message=F, warning=F}
CYFIP1_old=Anno.Data %>% filter(Gene=="CYFIP1") %>% select(ID, Gene, ExacAF)
old_pos=unlist(gregexpr("-", CYFIP1_old$ID))  # locate the position of "-"
old_ID=substr(CYFIP1_old$ID, 1,old_pos-1)
CYFIP1_old=CYFIP1_old %>% mutate(ID2=old_ID)

CYFIP1_new_liftover_no_filter=ASC_new_sample_liftover2 %>% filter(Gene=="CYFIP1") %>% select(group_name, start, allele1, allele2, Gene, Gnomad_non_neuro_AF) %>% mutate(ID=paste("15", ":", start, "-", start, "_", allele1, "_", allele2, sep="")) %>% mutate(ID2=paste("15",":", start, sep=""))
CYFIP1_new_liftover_with_filter=ASC_new_sample_liftover1 %>% filter(Gene=="CYFIP1") %>% select(group_name, start, allele1, allele2, Gene, Gnomad_non_neuro_AF)%>% mutate(ID=paste("15",":", start, "-", start, "_", allele1, "_", allele2, sep="")) %>% mutate(ID2=paste("15",":", start, sep=""))
############# there is multi-allelic issue here 
ID_overlap=intersect(CYFIP1_old$ID2, CYFIP1_new_liftover_with_filter$ID2)

CYFIP1_old %>% filter(ID2 %in% ID_overlap==T)%>%
datatable(extensions = 'Buttons',
          caption = "33 overlapping variants in old sample",
            options = list(dom = 'Blfrtip',
                           #buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

#%>%
#  kbl() %>%
#  kable_styling()%>%
#scroll_box(height = "300px") # display overlapping variants in old sample 


CYFIP1_new_liftover_with_filter %>% filter(ID2 %in% ID_overlap==T) %>% select(group_name, ID, ID2, Gene,Gnomad_non_neuro_AF )%>%
datatable(extensions = 'Buttons',
          caption = "33 overlapping variants in new sample",
            options = list(dom = 'Blfrtip',
                           #buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

CYFIP1_new_liftover_with_filter %>% filter(ID2 %in% ID_overlap==F) %>% select(group_name, ID, ID2, Gene,Gnomad_non_neuro_AF )%>%
datatable(extensions = 'Buttons',
          caption = "non-overlapping variants in new sample",
            options = list(dom = 'Blfrtip',
                           #buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

CYFIP1_old_new_liftover_combine=CYFIP1_old %>% full_join(CYFIP1_new_liftover_with_filter) %>% select(ID, Gene, ExacAF, Gnomad_non_neuro_AF, group_name)
```



* With the samle filter, gene `CYFIP1` has 46 variants, new sample after liftover has 112 variants, and 33 variants are overlapping between themm, shown in above two tables. The third table list non-overlapping variants in new sample, all of which have 0 AF. 

* the variants are complementary to each other, e.g. in old sample, `15:22947019-22947019_G_A` and new sample after liftover `15:22947019-22947019_C_T`. 

* variants in old sample `15:22963816` with variants `8 vs 0` doesn't survive the filter `AS_ReadPosRankSum >= -0.8`, and in new sample with variants `12 vs 0`. 

```{r, echo=F, message=F, warning=F}
#ASC_new_sample_liftover2 %>% filter(start=="22963816") %>% select(group_name)
ASC_new_sample_nofilter %>% filter(Variant=="15:22909252:T:C") %>% select(Variant, pLI, MPC, Consequence, Gnomad_non_neuro_AF, AS_ReadPosRankSum, Transmitted_proband, Untransmitted_proband)
```