---
title: "staar analysis"
output: html_document
date: "2023-02-20"
---


```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
library(DT)
library(poolr) # use fisher method to combine p values 
library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)

#############
#devtools::install_github("yaowuliu/ACAT")  # install package ACAT once 
library(ACAT)
```


```{r, echo=F, message=F, warning=F}
########## use fixed parameter estimate to calculate BF, this is very useful because it doesn't need to re-run analysis to BF of every gene which takes a long time and large space to store that information 
intergrand=function(aa, var.case, var.contr, bar.gamma, sig, N1, N0)
{
  ff=dbinom(var.case, sum(var.case, var.contr), aa*N1/(aa*N1+N0))*dgamma(aa, bar.gamma*sig, sig)
  return(ff)
}
# calculate the bayes factor of a single variant via integration
BF.var.inte=function(var.case, var.contr, bar.gamma, sig, N1, N0)
{
  marglik0.CC <- dbinom(var.case, sum(var.case, var.contr), N1/(N1+N0))    # Under H0: gamma=1

  marglik1.CC <- integrate(intergrand, var.case, var.contr, bar.gamma, sig, N1, N0, low=0, upper=100, stop.on.error=F)$value # Under H1: gamma~gamma(gamma.mean*sigma, sigma)
  BF.var <- marglik1.CC/marglik0.CC

  return(BF.var)
}

fixed.beta.func=function(new.data, N1, N0, gamma.mean, sigma, beta) # new.data has one column specifying its group index
{
  #######################
  beta.k=beta
  full.info.genevar=list()
  gene.list=new.data$Gene; unique.gene=unique(gene.list) # find the gene list
  num.gene=length(unique.gene)
  BF.gene=numeric()
  ########################
  for (i in 1:num.gene)
  {
#    cat(i, "th gene of ", "\t", num.gene, "\t", "is running", "\n")
    var.index.list=which(gene.list==unique.gene[i])
    indi.gene=new.data[var.index.list,]   # note var.index.list matches new.data
    bb=1; var.BF=numeric()
    if (length(var.index.list)>0) # calculate Bayes factor for variant (i,j)
      for (j in 1:length(var.index.list))
      {
        if (new.data$group.index[var.index.list[j]]<=3)   # note here bar.gamma is group specific, attention what bar.gamma goes to which group 
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=6, sig=sigma, N1, N0)
        if (new.data$group.index[var.index.list[j]]>3)
          var.BF[j]=BF.var.inte(new.data$No.case[var.index.list[j]], new.data$No.contr[var.index.list[j]], bar.gamma=3, sig=sigma, N1, N0)
        bb=bb*((1-beta.k[new.data$group.index[var.index.list[j]]])+beta.k[new.data$group.index[var.index.list[j]]]*var.BF[j])
      }
    full.info.genevar[[i]]=cbind(indi.gene, var.BF)
    BF.gene[i]=bb
  }
  
  return(result=list(BayesFactor=data.frame(Gene=unique.gene, BF=BF.gene), full.info=full.info.genevar))
}
```



```{r, echo=F, cache=T, warning=F, message=F}
### read into new sample 
path="C:\\Shengtong\\research\\rare-var\\"
ASC_new_sample3=as_tibble(read.table(paste(path, "ASC\\ASC_v17_raw_TDT_by_parent_2021-05-10.txt", sep=""), sep='\t',  header=T, fill=T)) # this is the correct way that Kyle read the data
ASC_new_sample3=ASC_new_sample3 %>% filter(isSyn=="FALSE") # filter syn variants first 
```




```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
path="C:\\Shengtong\\Research\\rare-var\\"
GeneDB=src_sqlite(path=paste(path, "gene.list.db", sep=""), create=F)
#GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
###### ID gene ######
ID_Gene=as.character(data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))$GeneID)
#### high confidence gene 
High_Confidence_Gene=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
#### moderate confidence gene 
Moderate_Confidence_Gene=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
#### psd gene 
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
PSD_Gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
#### FMRP gene 
FMRP_Gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
#### Autism KB gene 
ASD_KB_Gene=as.character(data.frame(collect(tbl(GeneDB, "AutismKB_gene")))$GeneID)
#### constraint gene 
Constraint_Gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
### RVIS gene 
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS_Gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene
#### Haplo insufficient gene 
Haploinsufficient_Gene=as.character(data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))$GeneID)
#### SCZ data 
SCZ_Gene=purcell.genelist$Gene_symbol
Gene_Set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff", "SCZ gene")
Gene_Set_List=list()
Gene_Set_List[[1]]=ID_Gene
Gene_Set_List[[2]]=High_Confidence_Gene
Gene_Set_List[[3]]=Moderate_Confidence_Gene
Gene_Set_List[[4]]=PSD_Gene
Gene_Set_List[[5]]=FMRP_Gene
Gene_Set_List[[6]]=ASD_KB_Gene
Gene_Set_List[[7]]=Constraint_Gene
Gene_Set_List[[8]]=RVIS_Gene
Gene_Set_List[[9]]=Haploinsufficient_Gene
Gene_Set_List[[10]]=SCZ_Gene
```


```{r, echo=F, warning=F, message=F}

##### count transmitted variants and untransmitted in proband by summing over male, female, dad, mom and ind 
######### use ASC_new_sample3 rather than ASC_new_sample2
Transmitted_proband=ASC_new_sample3%>% select(starts_with("t_proband"))%>%  mutate(Transmitted_proband = select(., t_proband_male_dad:t_proband_female_ind) %>% rowSums())%>% select(Transmitted_proband) # extract transmitted variants for proband 

Untransmitted_proband=ASC_new_sample3%>% select(starts_with("u_proband"))%>%  mutate(Untransmitted_proband = select(., u_proband_male_dad:u_proband_female_ind) %>% rowSums())%>% select(Untransmitted_proband) # extract untransmitted variants for proband 

Transmitted_sibling=ASC_new_sample3%>% select(starts_with("t_sibling"))%>%  mutate(Transmitted_sibling = select(., t_sibling_male_dad:t_sibling_female_ind) %>% rowSums())%>% select(Transmitted_sibling) # extract transmitted variants for sibling 

Untransmitted_sibling=ASC_new_sample3%>% select(starts_with("u_sibling"))%>%  mutate(Untransmitted_sibling = select(., u_sibling_male_dad:u_sibling_female_ind) %>% rowSums())%>% select(Untransmitted_sibling) # extract untransmitted variants for sibling

##### add new columns of total transmitted and non-transmitted variants 
#ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & VQSLOD<9) # exclude indel & VQSLOD filter 

################################################


ASC_new_sample=ASC_new_sample3%>%mutate(Transmitted_proband=Transmitted_proband$Transmitted_proband, Untransmitted_proband=Untransmitted_proband$Untransmitted_proband, Transmitted_sibling=Transmitted_sibling$Transmitted_sibling, Untransmitted_sibling=Untransmitted_sibling$Untransmitted_sibling) %>% filter(isIndel=="FALSE" & QD >= 10 & AS_SOR <= 3 & AS_ReadPosRankSum >= -0.8)%>%filter(!str_detect(Consequence, 'intron_variant'))# exclude indel & apply other filters for SNP's 

##################
#AF_cutoff=0.001
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Transmitted_proband) %>% pull)   
#sum(ASC_new_sample %>% filter(pLI>=0.995 & isPTV==T & Gnomad_non_neuro_AF<AF_cutoff) %>% select(Untransmitted_proband) %>% pull)   
########### 1164 vs 1413 not 695 vs 557 in cell paper 


# add one column of variant group index 
ASC_new_sample_GI=ASC_new_sample %>% add_column(Group_Index=NA)

ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.995 & ASC_new_sample_GI$isPTV==T)]=1
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI>=0.5 & ASC_new_sample_GI$pLI<0.995 & ASC_new_sample_GI$isPTV==T)]=2
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$pLI<0.5 & ASC_new_sample_GI$isPTV==T)]=3
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=2)]=4
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC>=1 & ASC_new_sample_GI$MPC<2)]=5
ASC_new_sample_GI$Group_Index[which(ASC_new_sample_GI$MPC<1)]=6

num.family=7291
## test 

#write.table(ASC_new_sample_GI, file="C:\\Shengtong\\Research\\rare-var\\ASC\\Laura_result\\ASC_new_sample_GI_2023_0901.txt")
```




```{r, echo=F, message=F, warning=F, eval=F}
################# calculate burden for each gene 
BF_proband_AllGene=as_tibble(read.csv("../output/kyleData/BF.proband.AllGene_1105_2022.csv"))
BF_proband_AllGene_sorted=BF_proband_AllGene%>%arrange(desc(post.prob))
interest_gene=BF_proband_AllGene_sorted$Gene  # use all genes 
new_data=ASC_new_sample_GI %>%  filter(Gnomad_non_neuro_AF<0.05) %>% filter(Gene %in% interest_gene) %>% select(Variant, Gene, Gnomad_non_neuro_AF, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()  #apply AF filter AF<5% 
acat_gene_pvalue=numeric()
a1=1; a2=25
for (i in 1:length(interest_gene))
{
  #i=6819
  cat(i, "th gene is running", "\n")
  single_gene_data=new_data %>% filter(Gene==interest_gene[i])
  var_group_index=single_gene_data %>% select(Group_Index) %>%pull
  num_var_group=unique(var_group_index)
  
  acat_var_pvalue=numeric()
  acat_var_weight=numeric()  # give weights when use ACAT to collapse all variants 
  #####################
  if (length(num_var_group)==1) # only one variant group
  {
    if (nrow(single_gene_data)==1) # case 1: only one variant in one group
    {
      binom_test=binom.test(c(single_gene_data$Transmitted_proband, single_gene_data$Untransmitted_proband), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
      acat_var_pvalue[1]=binom_test$p.value
      maf=single_gene_data$Gnomad_non_neuro_AF
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300) ##### use weight, beta density  as in ACAT paper  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
    } # end of   if (nrow(single_gene_data)==1) # only one variant in one group
    
    if (nrow(single_gene_data)>1) # case 2: multiple variants in one group
    {
      minor_alele_count=numeric()
      for (j in 1:nrow(single_gene_data)) # count the total number of alleles for every variant 
        minor_alele_count[j]=single_gene_data$Transmitted_proband[j]+single_gene_data$Untransmitted_proband[j]
      
      minor_alele_count_lessthan10=which(minor_alele_count<10) # find variants with MAC<10
      
      #############
      if (length(minor_alele_count_lessthan10)<=1) # if only one or no variant with MAC<10, no need collapsing 
        for (j in 1:nrow(single_gene_data))
        {
         binom_test=binom.test(c(single_gene_data$Transmitted_proband[j], single_gene_data$Untransmitted_proband[j]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
      acat_var_pvalue[j]=binom_test$p.value  
        maf=single_gene_data$Gnomad_non_neuro_AF[j]
        acat_var_weight[j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
        }
    #############
      if (length(minor_alele_count_lessthan10)>1 & length(minor_alele_count_lessthan10)<nrow(single_gene_data)) # some, not all variants have MAC<10
      {
        binom_test=binom.test(c(sum(single_gene_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT ## collapse all variants with MAC<10
      acat_var_pvalue[1]=binom_test$p.value
      maf=single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10]
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      minor_alele_count_morethan10=which(minor_alele_count>=10) # find variants with MAC>=10
      
      if (length(minor_alele_count_morethan10)==1)
      {
      binom_test=binom.test(c(single_gene_data$Transmitted_proband[minor_alele_count_morethan10], single_gene_data$Untransmitted_proband[minor_alele_count_morethan10]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
      acat_var_pvalue[2]=binom_test$p.value
      maf=single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10]
      acat_var_weight[2]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      }
      if (length(minor_alele_count_morethan10)>1)  
        for (j in 1:length(minor_alele_count_morethan10))
        {
         binom_test=binom.test(c(single_gene_data$Transmitted_proband[minor_alele_count_morethan10[j]], single_gene_data$Untransmitted_proband[minor_alele_count_morethan10[j]]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
        acat_var_pvalue[1+j]=binom_test$p.value  
        maf=single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10[j]]
        acat_var_weight[1+j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
        }
      } # end of if (length(minor_alele_count_lessthan10)>1 & length(minor_alele_count_lessthan10)<nrow(single_gene_data)) # some, not all variants have MAC<10
      
       #############
      if (length(minor_alele_count_lessthan10)>1 & length(minor_alele_count_lessthan10)==nrow(single_gene_data)) #  all variants have MAC<10
      {
        binom_test=binom.test(c(sum(single_gene_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT ## collapse all variants with MAC<10
      acat_var_pvalue[1]=binom_test$p.value
      maf=mean(single_gene_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10]) # use average MAF https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300) # avoid weight to be 0 
        
      }
      
    }  # end of if (nrow(single_gene_data)>1) # case 2: multiple variants in one group
  ##################### 
    if (max(acat_var_pvalue)==1 & length(acat_var_pvalue)>1)
      acat_var_pvalue[acat_var_pvalue==1]=1-1/(length(acat_var_pvalue)) # adjust pvalues https://github.com/yaowuliu/ACAT
    if (max(acat_var_pvalue)==1 & length(acat_var_pvalue)==1)
      acat_var_pvalue[acat_var_pvalue==1]=1-1e-300
      
    acat_gene_pvalue[i]=ACAT(acat_var_pvalue, acat_var_weight)
    
  } # end of if (length(num_var_group)==1) # only one variant group
  
############
  if (length(num_var_group)>1) # multiple variant groups
  {
    acat_group_pvalue=numeric()
    for (g in 1: length(num_var_group)) # use ACAT within every variant group 
    {
     #g=1
     single_gene_single_group_data=single_gene_data %>% filter(Group_Index==var_group_index[g])
     
     acat_var_pvalue=numeric()  # within each variant group, use ACAT
     acat_var_weight=numeric()
     
     if (nrow(single_gene_single_group_data)==1) # case 1: only one variant in one group
    {
      binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband, single_gene_single_group_data$Untransmitted_proband), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
      acat_var_pvalue[1]=binom_test$p.value
      maf=single_gene_single_group_data$Gnomad_non_neuro_AF
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)  ##### use weight as in ACAT paper, avoid zero weights  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
    } # end of   if (nrow(single_gene_data)==1) # only one variant in one group
    
    if (nrow(single_gene_single_group_data)>1) # case 2: multiple variants in one group
    {
      minor_alele_count=numeric()
      for (j in 1:nrow(single_gene_single_group_data)) # count the total number of alleles for every variant 
        minor_alele_count[j]=single_gene_single_group_data$Transmitted_proband[j]+single_gene_single_group_data$Untransmitted_proband[j]
      
      minor_alele_count_lessthan10=which(minor_alele_count<10) # find variants with MAC<10
      #############
      if (length(minor_alele_count_lessthan10)<=1) # if only one or no variant with MAC<10, no need collapsing 
        for (j in 1:nrow(single_gene_single_group_data))
        {
         binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband[j], single_gene_single_group_data$Untransmitted_proband[j]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
      acat_var_pvalue[j]=binom_test$p.value  
        maf=single_gene_single_group_data$Gnomad_non_neuro_AF[j]
        acat_var_weight[j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
        }
    #############
      if (length(minor_alele_count_lessthan10)>1 & length(minor_alele_count_lessthan10)<nrow(single_gene_single_group_data)) # some, not all variants have MAC<10
      {
        binom_test=binom.test(c(sum(single_gene_single_group_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_single_group_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT ## collapse all variants with MAC<10
      acat_var_pvalue[1]=binom_test$p.value
      maf=mean(single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10])
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      minor_alele_count_morethan10=which(minor_alele_count>=10) # find variants with MAC>=10
      
      if (length(minor_alele_count_morethan10)==1)
      {
      binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband[minor_alele_count_morethan10], single_gene_single_group_data$Untransmitted_proband[minor_alele_count_morethan10]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
      acat_var_pvalue[2]=binom_test$p.value
      maf=single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10]
      acat_var_weight[2]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300)
      
      }
      if (length(minor_alele_count_morethan10)>1)  
        for (j in 1:length(minor_alele_count_morethan10))
        {
         binom_test=binom.test(c(single_gene_single_group_data$Transmitted_proband[minor_alele_count_morethan10[j]], single_gene_single_group_data$Untransmitted_proband[minor_alele_count_morethan10[j]]), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT
        acat_var_pvalue[1+j]=binom_test$p.value  
        maf=single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_morethan10[j]]
        acat_var_weight[1+j]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5,1e-300)
        }
      } # end of if (length(minor_alele_count_lessthan10)>1 & length(minor_alele_count_lessthan10)<nrow(single_gene_data)) # some, not all variants have MAC<10
      
       #############
      if (length(minor_alele_count_lessthan10)>1 & length(minor_alele_count_lessthan10)==nrow(single_gene_single_group_data)) #  all variants have MAC<10
      {
        binom_test=binom.test(c(sum(single_gene_single_group_data$Transmitted_proband[minor_alele_count_lessthan10]), sum(single_gene_single_group_data$Untransmitted_proband[minor_alele_count_lessthan10])), p=0.5, alternative = "greater") # calculate p value under alternative: T>UT ## collapse all variants with MAC<10
      acat_var_pvalue[1]=binom_test$p.value
      maf=mean(single_gene_single_group_data$Gnomad_non_neuro_AF[minor_alele_count_lessthan10]) # use average MAF https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6407498/pdf/main.pdf
      acat_var_weight[1]=max(dbeta(maf, a1, a2)*(maf*(1-maf))^0.5, 1e-300) # avoid weight to be 0 
        
      }
      
    }  # end of if (nrow(single_gene_data)>1) # case 2: multiple variants in one group
  ##################### 
     if (max(acat_var_pvalue)==1 & length(acat_var_pvalue)>1)
      acat_var_pvalue[acat_var_pvalue==1]=1-1/(length(acat_var_pvalue))
      if (max(acat_var_pvalue)==1 & length(acat_var_pvalue)==1)  # avoid enforcing singular p values to be 0
        acat_var_pvalue[acat_var_pvalue==1]=1-1e-300
     
   acat_group_pvalue[g]=ACAT(acat_var_pvalue, acat_var_weight)  
      
    } # end of for (g in length(num_var_group))
    
    if (max(acat_group_pvalue)==1 & length(acat_group_pvalue)>1)
      acat_group_pvalue[acat_group_pvalue==1]=1-1/(length(acat_group_pvalue))
    if (max(acat_group_pvalue)==1 & length(acat_group_pvalue)==1)
      acat_group_pvalue[acat_group_pvalue==1]=1-1e-300
    
   acat_gene_pvalue[i]=ACAT(acat_group_pvalue) # use ACAT to combine group level p values with equal weights 
    
    
  } # end of if (length(num_var_group)>1) # multiple variant groups
  
  
} # end of   for (i in 1:length(interest_gene))


#  write.csv(data.frame(Gene=interest_gene, ACAT_pvalue=acat_gene_pvalue), file="../output/KyleData/AllGene_6vargroups_ACAT_pvalue_0403_2023.csv")
 # write.csv(combine_001, file="../output/KyleData/BF.proband.AllGene_6vargroups_burden_counts_binom_oneside_pvalues_MPC_larger_than2_AF0.01_1208_2022.csv")


```


## qqplot of p values 

```{r, echo=F, message=F, warning=F}
staar_pvalue=as_tibble(read.csv("../output/KyleData/AllGene_6vargroups_ACAT_pvalue_0403_2023.csv"))
theo <- (ppoints(length(staar_pvalue$ACAT_pvalue)))
  #qqplot(-log(theo, base=10), -log(Burden.pvalue, base=10), xlab=expression(paste("Theoretical ",-log[10], "(p-values)")), ylab=expression(paste("Observed ", -log[10], "(p-values)")), main="(A) Burden", frame=F, cex.lab=0.8, pch=20)
  #abline(0,1,col='red', lwd=2)

#################  use ggplot draw qq plot 
sx <- sort(-log(theo, base=10)); sy <- sort(-log(staar_pvalue$ACAT_pvalue, base=10))
lenx <- length(sx)
leny <- length(sy)
if (leny < lenx)sx <- approx(1L:lenx, sx, n = leny)$y
if (leny > lenx)sy <- approx(1L:leny, sy, n = lenx)$y

burden=tibble(sx=sx, sy=sy)
 ggplot(burden) + geom_point(aes(x=sx, y=sy))+
   theme_classic()+
   xlab(expression(paste("Theoretical ",-log[10], "(p-values)")))+
   ylab(expression(paste("Observed ", -log[10], "(p-values)")))+
   theme(plot.title = element_text(hjust = 0.5, size=10,face="bold"))+  #center the title+
   ggtitle("")+
   geom_abline(intercept = 0, slope = 1, color="red", 
                 linetype="solid", size=1.5)+
theme(text = element_text(size=10))
 #qqplot_burden_LoF
```

for STAAR analysis 

* apply AF filter $AF<5\%$

* within each of 6 variant groups, use ACAT-V to combine single variant p values of binomial test using weights in the paper $w_{i, ACAT-V}=Beta(MAF_i, a1=1, a2=25)*\sqrt{MAF_i(1-MAF_i)}$. Variants with MAC<10 are collapsed for burden test, with weight $w_{0, ACAT-V}$. 

* use ACAT-O to combine 6 group level ACAT p values in the last step with equal weight as in the paper, as STAAR p value for each gene.  


## top genes by STAAR 


```{r, echo=F, message=F, warning=F}

gene_set=ASC_new_sample_GI %>% filter(Gnomad_non_neuro_AF<0.05) %>% select(Variant, Gene, Transmitted_proband, Untransmitted_proband, Group_Index) %>% drop_na()

ACAT_pvalues=as_tibble(read.csv("../output/KyleData/AllGene_6vargroups_ACAT_pvalue_0403_2023.csv"))
ACAT_pvalues_sorted=ACAT_pvalues%>%arrange(ACAT_pvalue)
#length(intersect(BF_proband_AllGene_sorted$Gene[1:100], ASD_102_Gene))
SFARI.gene=read.csv("../data/GeneSet/SFARI-Gene_genes_06-20-2019release_07-15-2019export.csv", header=T)

num_gene_to_display=100
gene_to_display=ACAT_pvalues_sorted$Gene[1:num_gene_to_display]
Transmitted_proband=numeric()
Untransmitted_proband=numeric()
OR=numeric(); STAAR_pvalue=numeric()
SFARI_score=rep("NA", num_gene_to_display)

for (i in 1:length(gene_to_display))
{
  Transmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Transmitted_proband) %>% pull)
  
  Untransmitted_proband[i]=sum(gene_set %>% filter(Gene==gene_to_display[i]) %>% select(Untransmitted_proband) %>% pull)
  
  burden_test=fisher.test(matrix(c(Transmitted_proband[i],num.family, Untransmitted_proband[i], num.family), nrow=2))
OR[i]=burden_test$estimate
STAAR_pvalue[i]=ACAT_pvalues_sorted$ACAT_pvalue[i]

if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])>0) 
 SFARI_score[i]=SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),]$gene.score
#if(nrow(SFARI.gene[which(SFARI.gene$gene.symbol==gene_to_display[i]),])==0) 
# SFARI_score[i]="NA"

}


data.frame(Gene=gene_to_display, SFARI_score=SFARI_score, Trans=Transmitted_proband, Untrans=Untransmitted_proband, OR=round(OR, 4), STAAR_pvalue=round(STAAR_pvalue,5))%>%
datatable(extensions = 'Buttons',
            options = list(dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


```{r, echo=F, warning=F, message=F}
ACAT_pvalues=as_tibble(read.csv("../output/KyleData/AllGene_6vargroups_ACAT_pvalue_0403_2023.csv"))

sort_pvalue=ACAT_pvalues%>% arrange(ACAT_pvalue)
topgene_number=20
topgenes=as.character(sort_pvalue$Gene[1:topgene_number])
all.gene=as.character(sort_pvalue$Gene)
topmirage.percent=numeric(); all.gene.percent=numeric()
pvalue=numeric()
for ( i in 1:length(Gene_Set))
{
  overlap1=length(intersect(topgenes, Gene_Set_List[[i]]))
  topmirage.percent[i]=overlap1/topgene_number
  
  overlap2=length(intersect(all.gene, Gene_Set_List[[i]]))
  all.gene.percent[i]=overlap2/length(all.gene)
  
  pvalue[i]=fisher.test(matrix(c(overlap1, overlap2, topgene_number, length(all.gene)), nrow=2))$p.value
  
}
topmirage.enrichment.tidy=data.frame(class=c(rep("Top20 Genes by STAAR p value",length(Gene_Set)), rep("All Genes", length(Gene_Set))), group=rep(Gene_Set,2), percent=c(topmirage.percent, all.gene.percent))

topmirage_enrichment2=ggplot(topmirage.enrichment.tidy, aes(group, percent)) + 
   geom_bar(aes(fill = class), stat = "identity", position = "dodge")+
   theme_classic()+
   xlab("")+ylab("Proportion of putative ASD genes")+
  theme(axis.text.y = element_text(size = 10))+ 
   ylim(c(0, 0.2))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face="bold"))+
   ggtitle("Enrichment of top 20 ASD genes by ACAT p values")+
   #theme(plot.title = element_text(size=12))
    theme(legend.title = element_blank())+
   geom_signif(annotations ="1.000", y_position = 0.015  ,xmin=0.7, xmax=1.2, size=1, vjust=-0.5, textsize=3.5)+ # textsize is for text over the bar; size is for width of lines 
   geom_signif(annotations ="1.000", y_position = 0.05  ,xmin=1.7, xmax=2.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="1.000", y_position = 0.045  ,xmin=2.7, xmax=3.2, size=1, vjust=-0.5, textsize=3.5)+
   geom_signif(annotations ="0.395", y_position = 0.08  ,xmin=3.7, xmax=4.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.008,xmin=4.7, xmax=5.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.015,xmin=5.7, xmax=6.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.018,xmin=6.7, xmax=7.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.04,xmin=7.7, xmax=8.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="1.000", y_position = 0.048,xmin=8.7, xmax=9.2, size=1, vjust=-0.5, textsize=3.5)+
  geom_signif(annotations ="0.409", y_position = 0.1,xmin=9.7, xmax=10.2, size=1, vjust=-0.5, textsize=3.5)+
    theme(legend.position="bottom")+
   theme(axis.text.x = element_text(angle = 15, vjust = 0.5, size=13))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
 theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
 topmirage_enrichment2

 #figure=ggarrange(topmirage_enrichment1, topmirage_enrichment2, nrow=1)
#annotate_figure(figure,
#                top = text_grob("AF<0.01", color = "red", face = "bold", size = 14),
#                bottom = text_grob("Datasource: \n ToothGrowth data set", color = "blue",
#                                   hjust = 1, x = 1, face = "italic", size = 10),
#                left = text_grob("Tooth length", color = "green", rot = 90),
#                right = "I'm done, thanks :-)!",
#                fig.lab = "Figure 1", fig.lab.face = "bold"
#)

```


