---
title: "Variant set (VS) analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r load packages, results='hide', include=FALSE, warning=F, message=F}
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
#source("..\\code\\MIRAGE_burden_only_variant_level.R")
```

```{r, echo=F}
loadRData <- function(fileName){
#loads an RData file, and returns it
    load(fileName)
    get(ls()[ls() != "fileName"])
}
```


```{r defined function, echo=F, results='hide'}

wrap.it <- function(x, len)
{
  sapply(x, function(y) paste(strwrap(y, len),
                              collapse = "\n"),
         USE.NAMES = FALSE)
}


# Call this function with a list or vector
wrap.labels <- function(x, len)
{
  if (is.list(x))
  {
    lapply(x, wrap.it, len)
  } else {
    wrap.it(x, len)
    }
}

test.func=function(evid, Data, N1, N0) # given evid, and sample size, perform the burden analysis
{
evid.data=Data[Data$ID %in% evid,]
count=c(sum(evid.data$No.case), sum(evid.data$No.contr))
Time=c(N1, N0)
pois.test=poisson.test(count, Time, r=1, alternative="greater")
return(result=list(odds.ratio=pois.test$estimate, p.value=pois.test$p.value, rate.case=count[1]/N1, rate.contr=count[2]/N0))
}
```

We create a number of variant sets by combining gene sets and exon information stratified by allale frequency (AF) cutoffs.  




```{r, results='hide', echo=F, eval=F, warning=F, message=F}
###############################################################
# (1) this chunk is used for variant set analysis by burden and mirage-VS
# (2) definitions of var.fea; exon.fea and gene.set are in ASD.Rmd
# (3) to run function multi.group.func.for.variant, first need to load  # source("..\\code\\MIRAGE_burden_only_variant_level.R")
###############################################################

comb.exon.cutoff=exon.cutoff;
#sub.com=c(exon.fea, "LoF", "Prby Damaging", "CriticalExon", "Consensus")
sub.com=c(exon.fea, var.fea)
comb.fea=paste(rep(sub.com, each=length(gene.set)), gene.set)
AF.cutoff=c("MAF<1e-2", "MAF<1e-3", "MAF<1e-4", "Private"); colname=c("OR", "p.value", "rate.ca", "rate.co")
comb.summ=matrix(nrow=length(comb.fea), ncol=4*length(AF.cutoff))
rownames(comb.summ)=comb.fea; colnames(comb.summ)=paste(rep(AF.cutoff, each=length(colname)), colname, sep="~")
comb.summ.MIRAGE=comb.summ
#################  record burden and mirage for each cate split from combined feature ###########
Burden.cate=list()
MIRAGE.pvalue=list()
MIRAGE.para.est=list()
MIRAGE.cate.index=list()
nn=0
all.evid=list(); ii=0; all.evid.fea=paste(rep(comb.fea, each=length(AF.cutoff)), AF.cutoff)
signal1.evid=list(); jj1=0
signal2.evid=list(); jj2=0
signal3.evid=list(); jj3=0
signal4.evid=list(); jj4=0
for (i in 14:length(sub.com))
{# i=1 
  cat(i, "th subcom of ", length(sub.com), " is running", "\n")
  
  if (i<=length(comb.exon.cutoff))    # this is for exon 
  for (j in 1:length(gene.set))
  { # j=1
   for (k in 1:(length(AF.cutoff)-1)) 
  { #k=1
    comb.evid=intersect(intersect(evid.exon[[i]], gene.evid[[13+j]]), var.evid[[k]]) 
    pois.test=test.func(comb.evid, var.data, N1, N0)
    comb.summ[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    ii=ii+1
    all.evid[[ii]]=comb.evid
    ###################### MIRAGE burden
    cand.data=Anno.Data[which(Anno.Data$ID %in% comb.evid),]
    gene.data=eight.partition(cand.data)
    overlap.data=gene.data[gene.data$ID %in% comb.evid,]
    order.overlap.data=overlap.data[order(overlap.data$group.index, decreasing=F),]
    psbl.index=unique(order.overlap.data$group.index); actu.num.group=length(psbl.index)
    delta.init=runif(1); beta.init=runif(actu.num.group)
    order.overlap.data$original.group.index=order.overlap.data$group.index
    
    if (nrow(order.overlap.data)>0)
  {  
    burden.matrix=matrix(nrow=actu.num.group, ncol=4)  # this is burden for every category split from combined feature
    colnames(burden.matrix)=c("OR", "p.value", "rate.case", "rate.contr")
    for (jj in 1:actu.num.group)
    {    
      order.overlap.data$group.index[order.overlap.data$group.index==psbl.index[jj]]=jj # re-index the group labels
      pois.test=test.func(order.overlap.data[order.overlap.data$original.group.index==psbl.index[jj],]$ID, var.data, N1, N0)
      burden.matrix[jj,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    }
    
   para.est=multi.group.func.for.variant(order.overlap.data, N1, N0, gamma.mean=3, sigma=2, delta=0.2, beta.init, actu.num.group)
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, para.est$pvalue[length(para.est$pvalue)], pois.test$rate.case, pois.test$rate.contr)
   nn=nn+1
   MIRAGE.pvalue[[nn]]=para.est$cate.pvalue
   MIRAGE.para.est[[nn]]=para.est$beta.est
   MIRAGE.cate.index[[nn]]=psbl.index
   Burden.cate[[nn]]=burden.matrix
   }
  if (nrow(order.overlap.data)==0)  
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=NA
    

    ######################
    
    if (pois.test$p.value<0.05)
     {
      if (k==1)
         {
          jj1=jj1+1
          signal1.evid[[jj1]]=comb.evid
         }
         if (k==2)
         {
          jj2=jj2+1
          signal2.evid[[jj2]]=comb.evid
         }
         if (k==3)
         {
          jj3=jj3+1
          signal3.evid[[jj3]]=comb.evid
         }
      }
   }  
    k=length(AF.cutoff)
    comb.evid=intersect(intersect(evid.exon[[i]], gene.evid[[13+j]]), var.evid[[11]]) 
    pois.test=test.func(comb.evid, var.data, N1, N0)
    comb.summ[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    ii=ii+1
    all.evid[[ii]]=comb.evid 
    
    ###################### MIRAGE burden
    cand.data=Anno.Data[which(Anno.Data$ID %in% comb.evid),]
    gene.data=eight.partition(cand.data)
    overlap.data=gene.data[gene.data$ID %in% comb.evid,]
    order.overlap.data=overlap.data[order(overlap.data$group.index, decreasing=F),]
    psbl.index=unique(order.overlap.data$group.index); actu.num.group=length(psbl.index)
    delta.init=runif(1); beta.init=runif(actu.num.group)
    order.overlap.data$original.group.index=order.overlap.data$group.index
    
    if (nrow(order.overlap.data)>0)
  {  
    burden.matrix=matrix(nrow=actu.num.group, ncol=4)  # this is burden for every category split from combined feature
    colnames(burden.matrix)=c("OR", "p.value", "rate.case", "rate.contr")
    for (jj in 1:actu.num.group)
    {    
      order.overlap.data$group.index[order.overlap.data$group.index==psbl.index[jj]]=jj # re-index the group labels
      pois.test=test.func(order.overlap.data[order.overlap.data$original.group.index==psbl.index[jj],]$ID, var.data, N1, N0)
      burden.matrix[jj,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    }
    #  delta=runif(1)
   para.est=multi.group.func.for.variant(order.overlap.data, N1, N0, gamma.mean=3, sigma=2, delta=0.2, beta.init, actu.num.group)
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, para.est$pvalue[length(para.est$pvalue)], pois.test$rate.case, pois.test$rate.contr)
   nn=nn+1
   MIRAGE.pvalue[[nn]]=para.est$cate.pvalue
   MIRAGE.para.est[[nn]]=para.est$beta.est
   MIRAGE.cate.index[[nn]]=psbl.index
   Burden.cate[[nn]]=burden.matrix
   }
  if (nrow(order.overlap.data)==0)  
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=NA 
    
    
    
    if (pois.test$p.value<0.05)
    {
      jj4=jj4+1
      signal4.evid[[jj4]]=comb.evid
      }
  }
  
  if (sub.com[i] %in% var.fea==T)   # this is for variant feature 
    for (j in 1:length(gene.set))
    {    
     for (k in 1:(length(AF.cutoff)-1))  
    {
    comb.evid=intersect(intersect(var.evid[[which(sub.com[i]==var.fea)]], gene.evid[[13+j]]), var.evid[[k]]) 
    pois.test=test.func(comb.evid, var.data, N1, N0)
    comb.summ[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    ii=ii+1
    all.evid[[ii]]=comb.evid
    
    ###################### MIRAGE burden
    cand.data=Anno.Data[which(Anno.Data$ID %in% comb.evid),]
    gene.data=eight.partition(cand.data)
    overlap.data=gene.data[gene.data$ID %in% comb.evid,]
    order.overlap.data=overlap.data[order(overlap.data$group.index, decreasing=F),]
    psbl.index=unique(order.overlap.data$group.index); actu.num.group=length(psbl.index)
    delta.init=runif(1); beta.init=runif(actu.num.group)
    order.overlap.data$original.group.index=order.overlap.data$group.index
   
  if (nrow(order.overlap.data)>0)
  {     
   burden.matrix=matrix(nrow=actu.num.group, ncol=4)  # this is burden for every category split from combined feature
    colnames(burden.matrix)=c("OR", "p.value", "rate.case", "rate.contr")
    for (jj in 1:actu.num.group)
    {    
      order.overlap.data$group.index[order.overlap.data$group.index==psbl.index[jj]]=jj # re-index the group labels
      pois.test=test.func(order.overlap.data[order.overlap.data$original.group.index==psbl.index[jj],]$ID, var.data, N1, N0)
      burden.matrix[jj,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    }
    #  delta=runif(1)
   para.est=multi.group.func.for.variant(order.overlap.data, N1, N0, gamma.mean=3, sigma=2, delta=0.2, beta.init, actu.num.group)
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, para.est$pvalue[length(para.est$pvalue)], pois.test$rate.case, pois.test$rate.contr)
   nn=nn+1
   MIRAGE.pvalue[[nn]]=para.est$cate.pvalue
   MIRAGE.para.est[[nn]]=para.est$beta.est
   MIRAGE.cate.index[[nn]]=psbl.index
   Burden.cate[[nn]]=burden.matrix
   }
  if (nrow(order.overlap.data)==0)  
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=NA
    
    
    if (pois.test$p.value<0.05)
     {
      if (k==1)
         {
          jj1=jj1+1
          signal1.evid[[jj1]]=comb.evid
         }
         if (k==2)
         {
          jj2=jj2+1
          signal2.evid[[jj2]]=comb.evid
         }
         if (k==3)
         {
          jj3=jj3+1
          signal3.evid[[jj3]]=comb.evid
         }
      }
     } 
     k=length(AF.cutoff)
    comb.evid=intersect(intersect(var.evid[[which(sub.com[i]==var.fea)]], gene.evid[[13+j]]), var.evid[[11]]) 
    pois.test=test.func(comb.evid, var.data, N1, N0)
    comb.summ[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    ii=ii+1
    all.evid[[ii]]=comb.evid 
    
    ###################### MIRAGE burden
    cand.data=Anno.Data[which(Anno.Data$ID %in% comb.evid),]
    gene.data=eight.partition(cand.data)
    overlap.data=gene.data[gene.data$ID %in% comb.evid,]
    order.overlap.data=overlap.data[order(overlap.data$group.index, decreasing=F),]
    psbl.index=unique(order.overlap.data$group.index); actu.num.group=length(psbl.index)
    delta.init=runif(1); beta.init=runif(actu.num.group)
    order.overlap.data$original.group.index=order.overlap.data$group.index
    
    if (nrow(order.overlap.data)>0)
  {  
    burden.matrix=matrix(nrow=actu.num.group, ncol=4)  # this is burden for every category split from combined feature
    colnames(burden.matrix)=c("OR", "p.value", "rate.case", "rate.contr")
    for (jj in 1:actu.num.group)
    {    
      order.overlap.data$group.index[order.overlap.data$group.index==psbl.index[jj]]=jj # re-index the group labels
      pois.test=test.func(order.overlap.data[order.overlap.data$original.group.index==psbl.index[jj],]$ID, var.data, N1, N0)
      burden.matrix[jj,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    }
   
   para.est=multi.group.func.for.variant(order.overlap.data, N1, N0, gamma.mean=3, sigma=2, delta=0.2, beta.init, actu.num.group)
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, para.est$pvalue[length(para.est$pvalue)], pois.test$rate.case, pois.test$rate.contr)
   nn=nn+1
   MIRAGE.pvalue[[nn]]=para.est$cate.pvalue
   MIRAGE.para.est[[nn]]=para.est$beta.est
   MIRAGE.cate.index[[nn]]=psbl.index
   Burden.cate[[nn]]=burden.matrix
   }
  if (nrow(order.overlap.data)==0)  
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=NA
    
    
    
    
    if (pois.test$p.value<0.05)
    {
      jj4=jj4+1
      signal4.evid[[jj4]]=comb.evid
      }
    }
  
  if (sub.com[i]=="CriticalExon")  # this is for critical exon
    for (j in 1:length(gene.set))
    {    
     for (k in 1:(length(AF.cutoff)-1))  
    {
    comb.evid=intersect(intersect(critical.exon, gene.evid[[13+j]]), var.evid[[k]]) 
    pois.test=test.func(comb.evid, var.data, N1, N0)
    comb.summ[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    ii=ii+1
    all.evid[[ii]]=comb.evid
    
    ###################### MIRAGE burden
    cand.data=Anno.Data[which(Anno.Data$ID %in% comb.evid),]
    gene.data=eight.partition(cand.data)
    overlap.data=gene.data[gene.data$ID %in% comb.evid,]
    order.overlap.data=overlap.data[order(overlap.data$group.index, decreasing=F),]
    psbl.index=unique(order.overlap.data$group.index); actu.num.group=length(psbl.index)
    delta.init=runif(1); beta.init=runif(actu.num.group)
    order.overlap.data$original.group.index=order.overlap.data$group.index
    
    if (nrow(order.overlap.data)>0)
  {  
    burden.matrix=matrix(nrow=actu.num.group, ncol=4)  # this is burden for every category split from combined feature
    colnames(burden.matrix)=c("OR", "p.value", "rate.case", "rate.contr")
    for (jj in 1:actu.num.group)
    {    
      order.overlap.data$group.index[order.overlap.data$group.index==psbl.index[jj]]=jj # re-index the group labels
      pois.test=test.func(order.overlap.data[order.overlap.data$original.group.index==psbl.index[jj],]$ID, var.data, N1, N0)
      burden.matrix[jj,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    }
    #  delta=runif(1)
   para.est=multi.group.func.for.variant(order.overlap.data, N1, N0, gamma.mean=3, sigma=2, delta=0.2, beta.init, actu.num.group)
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, para.est$pvalue[length(para.est$pvalue)], pois.test$rate.case, pois.test$rate.contr)
   nn=nn+1
   MIRAGE.pvalue[[nn]]=para.est$cate.pvalue
   MIRAGE.para.est[[nn]]=para.est$beta.est
   MIRAGE.cate.index[[nn]]=psbl.index
   Burden.cate[[nn]]=burden.matrix
   }
  if (nrow(order.overlap.data)==0)  
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=NA
    
  
    
    
    if (pois.test$p.value<0.05)
     {
      if (k==1)
         {
          jj1=jj1+1
          signal1.evid[[jj1]]=comb.evid
         }
         if (k==2)
         {
          jj2=jj2+1
          signal2.evid[[jj2]]=comb.evid
         }
         if (k==3)
         {
          jj3=jj3+1
          signal3.evid[[jj3]]=comb.evid
         }
      }
     } 
    k=length(AF.cutoff)
    comb.evid=intersect(intersect(critical.exon, gene.evid[[13+j]]), var.evid[[11]]) 
    pois.test=test.func(comb.evid, var.data, N1, N0)
    comb.summ[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    ii=ii+1
    all.evid[[ii]]=comb.evid 
    
    ###################### MIRAGE burden
    cand.data=Anno.Data[which(Anno.Data$ID %in% comb.evid),]
    gene.data=eight.partition(cand.data)
    overlap.data=gene.data[gene.data$ID %in% comb.evid,]
    order.overlap.data=overlap.data[order(overlap.data$group.index, decreasing=F),]
    psbl.index=unique(order.overlap.data$group.index); actu.num.group=length(psbl.index)
    delta.init=runif(1); beta.init=runif(actu.num.group)
    order.overlap.data$original.group.index=order.overlap.data$group.index
    
     if (nrow(order.overlap.data)>0)
  {     
    burden.matrix=matrix(nrow=actu.num.group, ncol=4)  # this is burden for every category split from combined feature
    colnames(burden.matrix)=c("OR", "p.value", "rate.case", "rate.contr")
    for (jj in 1:actu.num.group)
    {    
      order.overlap.data$group.index[order.overlap.data$group.index==psbl.index[jj]]=jj # re-index the group labels
      pois.test=test.func(order.overlap.data[order.overlap.data$original.group.index==psbl.index[jj],]$ID, var.data, N1, N0)
      burden.matrix[jj,]=c(pois.test$odds.ratio, pois.test$p.value, pois.test$rate.case, pois.test$rate.contr)
    }
    #  delta=runif(1)

   para.est=multi.group.func.for.variant(order.overlap.data, N1, N0, gamma.mean=3, sigma=2, delta=0.2, beta.init, actu.num.group)
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=c(pois.test$odds.ratio, para.est$pvalue[length(para.est$pvalue)], pois.test$rate.case, pois.test$rate.contr)
   nn=nn+1
   MIRAGE.pvalue[[nn]]=para.est$cate.pvalue
   MIRAGE.para.est[[nn]]=para.est$beta.est
   MIRAGE.cate.index[[nn]]=psbl.index
   Burden.cate[[nn]]=burden.matrix
   }
  if (nrow(order.overlap.data)==0)  
   comb.summ.MIRAGE[(i-1)*length(gene.set)+j,((k-1)*4+1):(k*4)]=NA 
    
    
    
    if (pois.test$p.value<0.05)
    {
      jj4=jj4+1
      signal4.evid[[jj4]]=comb.evid
      }
    }
   
}  
#save(comb.summ, comb.summ.MIRAGE, MIRAGE.cate.index, MIRAGE.para.est, MIRAGE.pvalue, Burden.cate, file="..\\output\\CombinedFeature\\Burden.MIRAGE.for.Combined.Feature.RData")
```


```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.partition.RData")
Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF; 1%<AF<5%", "2"="LoF; AF<1%", "3" = "Damaging; 1%<AF<5%","4"= "Damaging; 0.1%<FA<1%", "5" ="Damaging; AF<0.1%", "6" ="Non-damaging; 1%<AF<5%", "7" ="Non-damaging; 0.1%<AF<1%", "8" ="Non-damaging; AF<0.1%")


################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.pvalue.lessthan.0.05.csv")
###################

g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  xlim(c(0,4))+ylim(c(0,5.5))+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
theme(plot.title = element_text(hjust = 0.5, size=10)) +# center the title 
theme(legend.title=element_blank())

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,5.5))+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
  theme(legend.text=element_text(size=1))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom")  # multiple panels share one legend 
#dev.off()




```

For about 480 combined features, perform burden test and MIRAGE-vs test.  use BH (Benjamini & Hochberg (1995)) multiple correcttion to adjust p values and blue line are with FDR 0.05.  




old version which is combining exon.score, AF and gene.set



```{r, echo=F,warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.RData")
#Burden.result=tibble(OR=as.numeric(comb.summ[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ[,c(2,6,10,14)]))
#g1=ggplot(Burden.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden test")

#MIRAGE.result=tibble(OR=as.numeric(comb.summ.MIRAGE[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ.MIRAGE[,c(2,6,10,14)]))
#g2=ggplot(MIRAGE.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("MIRAGE")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF; 1%<AF<5%", "2"="LoF; AF<1%", "3" = "Damaging; 1%<AF<5%","4"= "Damaging; 0.1%<AF<1%", "5" ="Damaging; AF<0.1%", "6" ="Non-damaging; 1%<AF<5%", "7" ="Non-damaging; 0.1%<AF<1%", "8" ="Non-damaging; AF<0.1%")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.pvalue.lessthan.0.05.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))

g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  xlim(c(0,4))+ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
theme(plot.title = element_text(hjust = 0.5, size=10)) +# center the title 
theme(legend.title=element_blank())

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
  theme(legend.text=element_text(size=1))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom")  # multiple panels share one legend 
#dev.off()

############# generate figure in research profile 
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/enrichment.pdf")
#ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
#  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
#  geom_point()+
#  xlab(expression(paste(hat(eta))))+
#  ylab(expression(paste("-log" [10], "(p value)")))+
#  ylim(c(0,y.max))+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
#  ggtitle("")+
#  theme_classic()+
#  theme(plot.title = element_text(hjust = 0.5, size=10))+  # center the title 
  #labs(color='variant group')+   # define legend title 
#  theme(legend.title=element_blank())+
#  theme(legend.text=element_text(size=7))
#dev.off()
#############

```


### Redefined version 

below  is redefined version to avoid redundency 



```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.redefined.version.RData")
#Burden.result=tibble(OR=as.numeric(comb.summ[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ[,c(2,6,10,14)]))
#g1=ggplot(Burden.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden test")

#MIRAGE.result=tibble(OR=as.numeric(comb.summ.MIRAGE[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ.MIRAGE[,c(2,6,10,14)]))
#g2=ggplot(MIRAGE.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("MIRAGE")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF; 1%<AF<5%", "2"="LoF; AF<1%", "3" = "Damaging; 1%<AF<5%","4"= "Damaging; 0.1%<FA<1%", "5" ="Damaging; AF<0.1%", "6" ="Non-damaging; 1%<AF<5%", "7" ="Non-damaging; 0.1%<AF<1%", "8" ="Non-damaging; AF<0.1%")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.pvalue.lessthan.0.05.redefined.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))

g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  xlim(c(0,4))+ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
theme(plot.title = element_text(hjust = 0.5, size=10)) +# center the title 
theme(legend.title=element_blank())

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=10))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
  theme(legend.text=element_text(size=1))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom")  # multiple panels share one legend 
#dev.off()
```



### 720 combined features


720 combined features are generated by  damaging(4)+AF(3)+gene.set(10)+exon(6). 
solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/1757(total number of tests). 

```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.v2.RData")
#Burden.result=tibble(OR=as.numeric(comb.summ[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ[,c(2,6,10,14)]))
#g1=ggplot(Burden.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden test")

#MIRAGE.result=tibble(OR=as.numeric(comb.summ.MIRAGE[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ.MIRAGE[,c(2,6,10,14)]))
#g2=ggplot(MIRAGE.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("MIRAGE")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF; 1%<AF<5%", "2"="LoF; AF<1%", "3" = "Damaging; 1%<AF<5%","4"= "Damaging; 0.1%<AF<1%", "5" ="Damaging; AF<0.1%", "6" ="Non-damaging; 1%<AF<5%", "7" ="Non-damaging; 0.1%<AF<1%", "8" ="Non-damaging; AF<0.1%")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.damaging.pvalue.lessthan.0.05.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=1757
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=30))+
theme(text = element_text(size=35))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 10))) # symbol size in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
theme(text = element_text(size=35))

  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 
#dev.off()
```


### 720 combined features (2)

Below is damaging(4, replace consensus with SIFT)+AF(3)+gene.set(10)+exon(6)=720 combined features


solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/720(total number of tests). 

there are four variant categories and each combined feature belongs to one categories, depending on where the most variants are from. 

```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.v3.RData")
#Burden.result=tibble(OR=as.numeric(comb.summ[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ[,c(2,6,10,14)]))
#g1=ggplot(Burden.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden test")

#MIRAGE.result=tibble(OR=as.numeric(comb.summ.MIRAGE[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ.MIRAGE[,c(2,6,10,14)]))
#g2=ggplot(MIRAGE.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("MIRAGE")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="other LoF sets", "2"="LoF; AF<1%", "3" = "Damaging; AF<0.1%","4"= "other missense sets")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.damaging.pvalue.lessthan.0.05.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=720
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
  theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=20))+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5))) # symbol size in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  #theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())#+
#theme(text = element_text(size=35))


  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 
#dev.off()
```


### 120 Combined features  

Below is damaging(4, replace consensus with SIFT)+AF(3)+gene.set(10)=120 combined features

solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/120(total number of tests). 

there are four variant categories and each combined feature belongs to one categories, depending on where the most variants are from. 

```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.v4.RData")
#Burden.result=tibble(OR=as.numeric(comb.summ[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ[,c(2,6,10,14)]))
#g1=ggplot(Burden.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden test")

#MIRAGE.result=tibble(OR=as.numeric(comb.summ.MIRAGE[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ.MIRAGE[,c(2,6,10,14)]))
#g2=ggplot(MIRAGE.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("MIRAGE")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="other LoF sets", "2"="LoF; AF<1%", "3" = "Damaging; AF<0.1%","4"= "other missense sets")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.damaging.pvalue.lessthan.0.05.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=120
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
  theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=20))+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5))) # symbol size in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  #theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
theme(text = element_text(size=20))


  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 
#dev.off()
```



### 120 combined features (2)

Below is damaging(4, replace consensus with SIFT)+AF(3 disjoint intervals)+gene.set(10)=120 combined features

solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/120(total number of tests). 
AF is using three cutoffs $10^{-2}, 10^{-3}, 10^{-4}$ disjoint. three variant groups: LoF; missense AF<0.1%; missense AF>0.1%

```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.v7.RData")
#Burden.result=tibble(OR=as.numeric(comb.summ[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ[,c(2,6,10,14)]))
#g1=ggplot(Burden.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden test")

#MIRAGE.result=tibble(OR=as.numeric(comb.summ.MIRAGE[,c(1, 5, 9, 13)]), pvalue=as.numeric(comb.summ.MIRAGE[,c(2,6,10,14)]))
#g2=ggplot(MIRAGE.result, aes(x=OR, y=-log(pvalue, base=10)))+
#  geom_point()+
#  geom_hline(yintercept=-log(0.05, base=10), linetype="dashed", color = "red")+
#  xlab("Odds ratio")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("MIRAGE")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF", "2"="Missense; AF>0.1%", "3" = "Missense; AF<0.1%")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(Burden.MIRAGE.combine.clear, file="../output/CombinedFeature/significant.combined.features.ofgene.set.disjoint.AF.missense.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=length(MIRAGE.pvalue)
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
  theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=20))+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5)))+ # symbol size in legend 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  #theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
theme(text = element_text(size=20))


  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 
#dev.off()
```


```{r, echo=F}
Burden.MIRAGE.combine.clear.separate=Burden.MIRAGE.combine.clear %>% separate(combine.feature, c("var.effect", "gene.set", "maf"), sep="_")
data.for.table=tibble(Variant.effect=Burden.MIRAGE.combine.clear.separate$var.effect, Gene.set=Burden.MIRAGE.combine.clear.separate$gene.set, MAF=Burden.MIRAGE.combine.clear.separate$maf, No.case=Burden.MIRAGE.combine.clear.separate$No.var.case, No.control=Burden.MIRAGE.combine.clear.separate$No.var.control, OR.burden=round(Burden.MIRAGE.combine.clear.separate$Burden.OR,4), Pvalue.burden=round(Burden.MIRAGE.combine.clear.separate$Burden.pvalue,4), eta.mirage=round(Burden.MIRAGE.combine.clear.separate$MIRAGE.risk.prop,4),pvalue.mirage=round(Burden.MIRAGE.combine.clear.separate$MIRAGE.pvalue,4))
#kable(data.for.table, caption = "combined feature")
```


### 120 combined features (3)


Below is damaging(4, replace consensus with SIFT)+AF(3 disjoint intervals)+gene.set(10)=120 combined features

solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/120(total number of tests). 
AF is using three cutoffs $10^{-2}, 10^{-3}, 10^{-4}$ disjoint. two variant groups: LoF and missense 

```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.v6.RData")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF", "2"="Missense")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(Burden.MIRAGE.combine.clear, file="../output/CombinedFeature/combined.features.of.gene.set.disjoint.AF.damaging.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=length(MIRAGE.pvalue)
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
  theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=20))+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5))) # symbol size in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  #theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
theme(text = element_text(size=20))


  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="bottom") 
#dev.off()
```


### 150 combined features 

Below is damaging(5, replace consensus with SIFT, add non-damaging)+AF(3 disjoint intervals)+gene.set(10)=150 combined features

solid blue line is FDR 0.05 and dashed red line is Bonferroni correction p = 0.05/150(total number of tests). 
AF is using three cutoffs $10^{-2}, 10^{-3}, 10^{-4}$ disjoint. three variant groups: LoF; missense AF<0.1%; missense AF>0.1%

```{r, echo=F, warning=F, message=F}
rm("Burden.cate", "MIRAGE.cate.index", "MIRAGE.para.est", "MIRAGE.pvalue")
load("../output/CombinedFeature/Burden.MIRAGE.for.Combined.exon.geneset.AF.partition.old.version.v8.RData")


Burden.OR=numeric(); Burden.pvalue=numeric(); No.var.case=numeric(); No.var.control=numeric()
feature.name=numeric(); cate.index=numeric()
for (ii in 1:length(Burden.cate))
{
  Burden.OR=c(Burden.OR, Burden.cate[[ii]][,1])
  Burden.pvalue=c(Burden.pvalue, Burden.cate[[ii]][,2])
  No.var.case=c(No.var.case, Burden.cate[[ii]][,3]*4315)
  No.var.control=c(No.var.control, Burden.cate[[ii]][,4]*4315)
  feature.name=c(feature.name, rownames(Burden.cate[[ii]]))
  cate.index=c(cate.index, MIRAGE.cate.index[[ii]])
}     

############### multiple correction for p values
MIRAGE.pvalue=unlist(MIRAGE.pvalue)
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()
 
##############

Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=unlist(MIRAGE.para.est), MIRAGE.pvalue=unlist(MIRAGE.pvalue), cate.index=unlist(MIRAGE.cate.index), No.var.case=No.var.case, No.var.control=No.var.control, combine.feature=feature.name)
Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA
#Burden.MIRAGE.combine.clear <- as_tibble(data.matrix(Burden.MIRAGE.combine)[is.finite(rowSums(data.matrix(Burden.MIRAGE.combine))),])
Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine
Burden.MIRAGE.combine.clear$cate.index=Burden.MIRAGE.combine.clear$cate.index %>% recode("1"="LoF", "2"="Missense; AF>0.1%", "3" = "Missense; AF<0.1%")



################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(Burden.MIRAGE.combine.clear, file="../output/CombinedFeature/significant.combined.features.ofgene.set.disjoint.AF.missense.csv")
###################
y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=length(MIRAGE.pvalue)
g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
  theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
theme(legend.text=element_text(size=20))+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5)))+ # symbol size in legend 
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index),size=5))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  #theme(plot.title = element_text(hjust = 0.5, size=35))+  # center the title 
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())+
theme(text = element_text(size=20))




damaging.nondamaging.burden=tibble(damaging.burden=Burden.MIRAGE.combine.clear$Burden.pvalue[1:30], damaging.burden.feature=Burden.MIRAGE.combine.clear$combine.feature[1:30], nondamaging.burden=Burden.MIRAGE.combine.clear$Burden.pvalue[118:147], nondamaging.burden.feature=Burden.MIRAGE.combine.clear$combine.feature[118:147])
g3=ggplot(damaging.nondamaging.burden, aes(x=-log(nondamaging.burden, base=10), y=-log(damaging.burden, base=10), size=5))+
  geom_point()+
  xlab(expression(paste("-log" [10], " (p value) (non-damaging)")))+
  ylab(expression(paste("-log" [10], " (p value) (damaging)")))+
 ggtitle("Burden")+
  theme_classic()+
theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  geom_abline(intercept = 0, slope = 1)+
guides(size=FALSE)+ 
theme(text = element_text(size=20))


damaging.nondamaging.mirage.vs=tibble(damaging.mirage.vs=Burden.MIRAGE.combine.clear$MIRAGE.pvalue[1:30], damaging.mirage.vs.feature=Burden.MIRAGE.combine.clear$combine.feature[1:30], nondamaging.mirage.vs=Burden.MIRAGE.combine.clear$MIRAGE.pvalue[118:147], nondamaging.mirage.vs.feature=Burden.MIRAGE.combine.clear$combine.feature[118:147])
g4=ggplot(damaging.nondamaging.mirage.vs, aes(x=-log(nondamaging.mirage.vs, base=10), y=-log(damaging.mirage.vs, base=10), size=5))+
  geom_point()+
  xlab(expression(paste("-log" [10], " (p value) (non-damaging)")))+
  ylab(expression(paste("-log" [10], " (p value) (damaging)")))+
   xlim(c(0,6))+
ggtitle("MIRAGE-VS")+
  theme_classic()+
theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  geom_abline(intercept = 0, slope = 1)+
guides(size=FALSE)+ 
theme(text = element_text(size=20))



damaging.nondamaging.mirage.vs.tidy=tibble(pvalue=c(damaging.nondamaging.mirage.vs$damaging.mirage.vs, damaging.nondamaging.mirage.vs$nondamaging.mirage.vs), class=c(rep("damaing", nrow(damaging.nondamaging.mirage.vs)), rep("non-damaging", nrow(damaging.nondamaging.mirage.vs))))

g5=ggplot(damaging.nondamaging.mirage.vs.tidy, aes(x=-log(pvalue, base=10), fill=class, size=5))+
  #geom_histogram(aes(y=..density..), alpha=0.5,position='identity',binwidth=0.5)+
  geom_histogram(aes(y=(..count..)/sum(..count..)), position="dodge2")+
  #geom_histogram(position="dodge2")+
  theme_classic()+
  xlab(expression(paste("-log" [10], "(p value)")))+
  ylab("Frequency")+
theme(text = element_text(size=20))+
guides(size=FALSE)+  # suppress size legend
theme(legend.spacing.x = unit(1.0, 'cm')) # add space between symbols in legend 
#g5

####### look at variant groups with FDR<0.05 damaging vs non-damaging
damaging.vs.fdr.index=which(damaging.nondamaging.mirage.vs.tidy$pvalue[1:30]<mirage.pvalue.fdr.0.05.cutoff)
nondamaging.vs.fdr.index=damaging.vs.fdr.index+30
damaging.nondamaging.mirage.vs.tidy.fdr=tibble(var.group=c(damaging.nondamaging.mirage.vs$damaging.mirage.vs.feature[damaging.vs.fdr.index], damaging.nondamaging.mirage.vs$damaging.mirage.vs.feature[damaging.vs.fdr.index]), pvalue=c(damaging.nondamaging.mirage.vs.tidy$pvalue[damaging.vs.fdr.index],damaging.nondamaging.mirage.vs.tidy$pvalue[nondamaging.vs.fdr.index]), class=c(rep("damaging", length(damaging.vs.fdr.index)), rep("non.damaging", length(damaging.vs.fdr.index))))

g6=ggplot(damaging.nondamaging.mirage.vs.tidy.fdr, aes(y=-log(pvalue, base=10), x=var.group, fill=class))+
geom_bar(stat="identity", position=position_dodge())+
theme_classic()+
xlab("")+
ylab(expression(paste("-log" [10], " (p value)")))+
theme(plot.title = element_text(hjust = 0.5, size=7))+  #center the title+
theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=10))+
theme(text = element_text(size=10))#+
#  coord_flip()
#g6
  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange( g1, g2, g5,   common.legend = TRUE, legend="right") 

g6
```

This table shows the results by burden and mirage-VS for 150 combined features. 

```{r, echo=F}
Burden.MIRAGE.combine.clear.separate=Burden.MIRAGE.combine.clear %>% separate(combine.feature, c("var.effect", "gene.set", "maf"), sep="_")
data.for.table=tibble(Variant.effect=Burden.MIRAGE.combine.clear.separate$var.effect, Gene.set=Burden.MIRAGE.combine.clear.separate$gene.set, MAF=Burden.MIRAGE.combine.clear.separate$maf, No.case=Burden.MIRAGE.combine.clear.separate$No.var.case, No.control=Burden.MIRAGE.combine.clear.separate$No.var.control, OR.burden=round(Burden.MIRAGE.combine.clear.separate$Burden.OR,4), Pvalue.burden=round(Burden.MIRAGE.combine.clear.separate$Burden.pvalue,4), eta.mirage=round(Burden.MIRAGE.combine.clear.separate$MIRAGE.risk.prop,4),pvalue.mirage=round(Burden.MIRAGE.combine.clear.separate$MIRAGE.pvalue,4))

data.for.table.sort=data.for.table %>% arrange(pvalue.mirage)
kable(data.for.table.sort, caption = "Summary of combined features")%>%
kable_styling() %>%
scroll_box(height = "500px")
```
