---
title: "Variant set (VS) analysis for transmitted variants in ASC cell paper"
output: html_document
---

```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
#library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
```



```{r, echo=F, cache=T, warning=F, message=F}
# read into data
part1=as_tibble(read.table("../../AutismDatafromASC_Cellpaper/ASC_v15_TDT_GQPL25.hAB30.results.2018-04-01.part1.tsv.gz",sep = '\t', header = TRUE, fill = TRUE))
part2=as_tibble(read.table("../../AutismDatafromASC_Cellpaper/ASC_v15_TDT_GQPL25.hAB30.results.2018-04-01.part2.tsv.gz",sep = '\t', header = TRUE, fill = TRUE))
cell_trans_data=full_join(part1, part2)
```

## Allele frequency distribution


it seems the data is already filtered with $AF>10^{-3}$. 

```{r, echo=F, cache=T, warning=F, message=F}
## add Exac Allele frequency to the data
avinput.hg19_exac03_dropped=as_tibble(read.table("../../AutismDatafromASC_Cellpaper/avinput.hg19_exac03_dropped"))
ASC_ExacAF=tibble(Variant=paste(avinput.hg19_exac03_dropped$V3, avinput.hg19_exac03_dropped$V4, avinput.hg19_exac03_dropped$V6, avinput.hg19_exac03_dropped$V7, sep=":"), ExacAF=avinput.hg19_exac03_dropped$V2)
cell_trans_data_AF=left_join(cell_trans_data, ASC_ExacAF)
cell_trans_data_AF$ExacAF[is.na(cell_trans_data_AF$ExacAF)==T]=0 # fill NA AF with 0
hist(cell_trans_data_AF$ExacAF)
summary(cell_trans_data_AF$ExacAF)
```

## Use 3 AF bins 


### variant counts in each variant group 

Use same 6 variants groups in cell paper, stratify each variant group with AF cutoff, $AF>10^{-4}, 10^{-4}>AF>10^{-5}, AF<10^{-5}$.


```{r, echo=F, warning=F, message=F}
## partition vriants into disjoint groups 
variant_groups=list()
variant_groups[[1]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.0001) # AF>10^{-4}
variant_groups[[2]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.00001 & ExacAF<=0.0001)
variant_groups[[3]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter( ExacAF<=0.00001)

variant_groups[[4]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.0001)
variant_groups[[5]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.00001 & ExacAF<=0.0001)
variant_groups[[6]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter( ExacAF<=0.00001)

variant_groups[[7]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter(ExacAF>0.0001)
variant_groups[[8]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter(ExacAF>0.00001 & ExacAF<=0.0001)
variant_groups[[9]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter( ExacAF<=0.00001)

variant_groups[[10]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter(ExacAF>0.0001)
variant_groups[[11]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter(ExacAF>0.00001 & ExacAF<=0.0001)
variant_groups[[12]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter( ExacAF<=0.00001)

variant_groups[[13]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(ExacAF>0.0001)
variant_groups[[14]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(ExacAF>0.00001 & ExacAF<=0.0001)
variant_groups[[15]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(  ExacAF<=0.00001)

variant_groups[[16]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter(ExacAF>0.0001)
variant_groups[[17]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter(ExacAF>0.00001 & ExacAF<=0.0001)
variant_groups[[18]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter( ExacAF<=0.00001)

count_variant_groups=matrix(nrow=18, ncol=2)
colnames(count_variant_groups)=c("transmitted", "untransmitted")
rownames(count_variant_groups)=paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=3), c("0.0001<AF", "0.00001<AF<=0.0001", "AF<=0.00001"), sep="&")
names(variant_groups)=paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=3), c("0.0001<AF", "0.00001<AF<=0.0001", "AF<=0.00001"), sep="&")
for (i in 1:length(variant_groups))
  if (nrow(variant_groups[[i]])>0)
    count_variant_groups[i,]=c(variant_groups[[i]]%>% summarise(Transmitted_proband)%>%sum(na.rm=T), variant_groups[[i]]%>% summarise(Untransmitted_proband)%>%sum(na.rm=T))
#count_variant_groups=count_variant_groups[complete.cases(count_variant_groups) & rowSums(count_variant_groups[,]) != 0, ]  
dim(count_variant_groups)
#kable(count_variant_groups, align = "lccrr", format = "html")
count_variant_groups
```


```{r burdern analysis-gene level, results='hide', echo=F, warning=F, message=F}
# read into gene set 
GeneDB=src_sqlite(path="C:\\Shengtong\\Research\\rare-var\\gene.list.db", create=F)
#GeneDB=src_sqlite(path="../../gene.list.db", create=F)
gene_cate1=data.frame(collect(tbl(GeneDB, "SFARI_HighConf")))
gene_cate2=data.frame(collect(tbl(GeneDB, "SFARI_StrongCand")))
gene_cate3=data.frame(collect(tbl(GeneDB, "SFARI_cate3_gene")))
gene_cate4=data.frame(collect(tbl(GeneDB, "SFARI_cate4_gene")))
gene_cate5=data.frame(collect(tbl(GeneDB, "SFARI_cate5_gene")))
gene_cate6=data.frame(collect(tbl(GeneDB, "SFARI_cate6_gene")))
gene_cateS=data.frame(collect(tbl(GeneDB, "SFARI_cateS_gene")))
TADAGene=data.frame(collect(tbl(GeneDB, "TADAGenelist")))
Qlessthan5percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.05]
Qlessthan20percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.2]
Qlessthan30percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.3]
Qlessthan40percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.4]
Qlessthan50percentgene=TADAGene$TadaName[TADAGene$qvalue.combined<0.5]
Qlargerthan90percentgene=TADAGene$TadaName[TADAGene$qvalue.combined>0.9]
###### ID gene ######
ID_Gene=as.character(data.frame(collect(tbl(GeneDB, "Pinto14AJHG_IDgene")))$GeneID)
#### high confidence gene 
High_Confidence_Gene=union(union(gene_cate1$GeneID, gene_cate2$GeneID),Qlessthan5percentgene)
#### moderate confidence gene 
Moderate_Confidence_Gene=setdiff(union(union(gene_cate3$GeneID, gene_cateS$GeneID), Qlessthan30percentgene),Qlessthan5percentgene)
#### psd gene 
purcell.genelist=data.frame(collect(tbl(GeneDB, "Purcell2014_genelist"))) ## PSD gene, SCZdenovo gene 
PSD_Gene=purcell.genelist$Gene_symbol[purcell.genelist$PSD=="Y"]
#### FMRP gene 
FMRP_Gene=purcell.genelist$Gene_symbol[purcell.genelist$FMRP.target=="Y"]
#### Autism KB gene 
ASD_KB_Gene=as.character(data.frame(collect(tbl(GeneDB, "AutismKB_gene")))$GeneID)
#### constraint gene 
Constraint_Gene=data.frame(collect(tbl(GeneDB, "Samocha_2014NG_constraintgene")))$gene
### RVIS gene 
RVIS.Allgene=data.frame(collect(tbl(GeneDB, "RVIS_gene")))
RVIS_Gene=RVIS.Allgene$GeneID[RVIS.Allgene$RVIS.percentile<5] # top 5% gene
#### Haplo insufficient gene 
Haploinsufficient_Gene=as.character(data.frame(collect(tbl(GeneDB, "Petrovski_plosgen_haploinsuff_gene")))$GeneID)
#### SCZ data 
SCZ_Gene=purcell.genelist$Gene_symbol
Gene_Set=c("ID gene","High", "Mod", "PSD", "FMRP", "AutismKB", "constraint gene", "RVIS", "Haploinsuff", "SCZ gene")
Gene_Set_List=list()
Gene_Set_List[[1]]=ID_Gene
Gene_Set_List[[2]]=High_Confidence_Gene
Gene_Set_List[[3]]=Moderate_Confidence_Gene
Gene_Set_List[[4]]=PSD_Gene
Gene_Set_List[[5]]=FMRP_Gene
Gene_Set_List[[6]]=ASD_KB_Gene
Gene_Set_List[[7]]=Constraint_Gene
Gene_Set_List[[8]]=RVIS_Gene
Gene_Set_List[[9]]=Haploinsufficient_Gene
Gene_Set_List[[10]]=SCZ_Gene
```

```{r, echo=F, eval=F, warning=F, message=F}
# run mirage-vs on combined features 
######### It takes about 30  minutes to finish running 
Combine_Feature=paste(rep(names(variant_groups), each=length(Gene_Set)), Gene_Set, sep="&")
Burden_MIRAGE_VS=matrix(nrow=length(Combine_Feature), ncol=6)
rownames(Burden_MIRAGE_VS)=Combine_Feature
colnames(Burden_MIRAGE_VS)=c("Transmitted", "Untransmitted", "Burden_OR", "Burden_Pval", "MIRAGE-VS_eta", "MIRAGE-VS_Pval")
for (i in 1:length(variant_groups))
  if (nrow(variant_groups[[i]])>0)
    for (j in 1:length(Gene_Set))
  { 
      cat("i is", i, "is running", "\n")
      
   # i=18; j=10  
    Feature_Data=variant_groups[[i]]%>% filter(GENE_NAME %in% Gene_Set_List[[j]])%>% select(Transmitted_proband, Untransmitted_proband) %>% add_column(Group_Index=1)%>% drop_na()
    
    if (nrow(Feature_Data)>0 & sum(Feature_Data$Transmitted_proband, Feature_Data$Untransmitted_proband)>0)
    {  
     Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),1:2]=c(sum(Feature_Data$Transmitted_proband),sum(Feature_Data$Untransmitted_proband))
 
    pois.test=poisson.test(Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),1:2], c(5869, 5869), r=1, alternative="greater")
 Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),3:4]=c(pois.test$estimate,pois.test$p.value)
 
    vs.mirage=mirage_vs(data.frame(Feature_Data),n1=5869,n2=5869)
 Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
 
    }
  
    }
#write.csv(Burden_MIRAGE_VS, file="../output/CellDataASC/Variant_Set_6Varcate_3AFCutoff_10Geneset.csv")
```


###  Burden vs MIRAGE-VS



```{r, echo=F, warning=F, message=F}
Burden_MIRAGE_VS=as_tibble(read.csv("../output/CellDataASC/Variant_Set_6Varcate_3AFCutoff_10Geneset.csv")) %>% add_column(Group_Index=rep(paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=3), c("0.0001<AF", "0.00001<AF<=0.0001", "AF<=0.00001"), sep="&"), each=10))%>% drop_na()


Burden.OR=Burden_MIRAGE_VS %>% select(Burden_OR) %>% pull()
Burden.pvalue=Burden_MIRAGE_VS %>% select(Burden_Pval) %>% pull()
No.var.case=Burden_MIRAGE_VS %>% select(Transmitted)%>% pull()
No.var.control=Burden_MIRAGE_VS %>% select(Untransmitted)%>% pull()
############### multiple correction for p values

MIRAGE.pvalue=Burden_MIRAGE_VS %>% select(MIRAGE.VS_Pval)%>% pull()
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()


##############

MIRAGE.risk.prop=Burden_MIRAGE_VS%>% select(MIRAGE.VS_eta)%>%pull()
Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=MIRAGE.risk.prop, MIRAGE.pvalue=MIRAGE.pvalue, No.var.case=No.var.case, No.var.control=No.var.control)
#Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA

Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine %>% add_column(cate.index=Burden_MIRAGE_VS$Group_Index)




################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.damaging.pvalue.lessthan.0.05.csv")
###################

y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=138
################### this argument setting is too large, could be used in paper 
#g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
#  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
#  geom_point()+
  #xlim(c(0,4))+
 # ylim(c(0,y.max))+
#  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
#  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
 # xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden")+
#  theme_classic()+
#   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
#theme(legend.title=element_blank())+
#theme(legend.text=element_text(size=30))+
#theme(text = element_text(size=35))+
#guides(size=FALSE)+  # suppress size legend 
#guides(shape = guide_legend(override.aes = list(size = 10))) # symbol size in legend 
########################### this argument setting is in website 
  g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
#theme(legend.text=element_text(size=30))+
#theme(text = element_text(size=35))+
guides(size=FALSE)+   # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5))) # symbol size in legend 
#  g1
##################################  

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())#+
#  g2
#theme(text = element_text(size=35))

  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="right") 
#dev.off()
```


## Use 2 AF bins 


### variant counts in each variant group 

Use same 6 variants groups in cell paper, stratify each variant group with AF cutoff, $AF>10^{-4}, AF< 10^{-4}$.


```{r, echo=F, warning=F, message=F}
## partition vriants into disjoint groups 
variant_groups=list()
variant_groups[[1]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.0001) # AF>10^{-4}
variant_groups[[2]]=cell_trans_data_AF %>% filter(pLI>=0.995) %>% filter(PTV==T)%>% filter(ExacAF<0.0001)


variant_groups[[3]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter(ExacAF>0.0001)
variant_groups[[4]]=cell_trans_data_AF %>% filter(pLI>=0.5 & pLI<0.995) %>% filter(PTV==T)%>% filter(ExacAF<=0.0001)

variant_groups[[5]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% filter(ExacAF>0.0001)
variant_groups[[6]]=cell_trans_data_AF %>% filter(pLI<0.5) %>% filter(PTV==T)%>% 
filter(ExacAF<=0.0001)

variant_groups[[7]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter(ExacAF>0.0001)
variant_groups[[8]]=cell_trans_data_AF %>%filter(MPC>=2)%>% filter(ExacAF<=0.0001)

variant_groups[[9]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(ExacAF>0.0001)
variant_groups[[10]]=cell_trans_data_AF %>% filter(MPC<2 & MPC>=1)%>% filter(ExacAF<=0.0001)

variant_groups[[11]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter(ExacAF>0.0001)
variant_groups[[12]]=cell_trans_data_AF %>% filter(MPC<1)%>% filter(ExacAF<=0.0001)

count_variant_groups=matrix(nrow=12, ncol=2)
colnames(count_variant_groups)=c("transmitted", "untransmitted")
rownames(count_variant_groups)=paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=2), c("AF>0.0001", "AF<=0.0001"), sep="&")
names(variant_groups)=paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=2), c("AF>0.0001", "AF<=0.0001"), sep="&")
for (i in 1:length(variant_groups))
  if (nrow(variant_groups[[i]])>0)
    count_variant_groups[i,]=c(variant_groups[[i]]%>% summarise(Transmitted_proband)%>%sum(na.rm=T), variant_groups[[i]]%>% summarise(Untransmitted_proband)%>%sum(na.rm=T))
#count_variant_groups=count_variant_groups[complete.cases(count_variant_groups) & rowSums(count_variant_groups[,]) != 0, ]  
dim(count_variant_groups)
#kable(count_variant_groups, align = "lccrr", format = "html")
count_variant_groups
```

```{r, echo=F, eval=F, warning=F, message=F}
# run mirage-vs on defined combined features 
######### It takes about 10  minutes to finish running 
Combine_Feature=paste(rep(names(variant_groups), each=length(Gene_Set)), Gene_Set, sep="&")
Burden_MIRAGE_VS=matrix(nrow=length(Combine_Feature), ncol=6)
rownames(Burden_MIRAGE_VS)=Combine_Feature
colnames(Burden_MIRAGE_VS)=c("Transmitted", "Untransmitted", "Burden_OR", "Burden_Pval", "MIRAGE-VS_eta", "MIRAGE-VS_Pval")
for (i in 1:length(variant_groups))
  if (nrow(variant_groups[[i]])>0)
    for (j in 1:length(Gene_Set))
  { 
      cat("i is", i, "is running", "\n")
      
   # i=18; j=10  
    Feature_Data=variant_groups[[i]]%>% filter(GENE_NAME %in% Gene_Set_List[[j]])%>% select(Transmitted_proband, Untransmitted_proband) %>% add_column(Group_Index=1)%>% drop_na()
    
    if (nrow(Feature_Data)>0 & sum(Feature_Data$Transmitted_proband, Feature_Data$Untransmitted_proband)>0)
    {  
     Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),1:2]=c(sum(Feature_Data$Transmitted_proband),sum(Feature_Data$Untransmitted_proband))
 
    pois.test=poisson.test(Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),1:2], c(5869, 5869), r=1, alternative="greater")
 Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),3:4]=c(pois.test$estimate,pois.test$p.value)
 
    vs.mirage=mirage_vs(data.frame(Feature_Data),n1=5869,n2=5869)
 Burden_MIRAGE_VS[((i-1)*length(Gene_Set)+j),5:6]=c(vs.mirage$eta.est, vs.mirage$eta.pvalue)
 
    }
  
    }
#write.csv(Burden_MIRAGE_VS, file="../output/CellDataASC/Variant_Set_6Varcate_2AFCutoff_10Geneset.csv")
```



###  Burden vs MIRAGE-VS



```{r, echo=F, warning=F, message=F}
Burden_MIRAGE_VS=as_tibble(read.csv("../output/CellDataASC/Variant_Set_6Varcate_2AFCutoff_10Geneset.csv")) %>% add_column(Group_Index=rep(paste(rep(c("PLI>=0.995", "0.5<=pLI<0.995", "pLI<0.5", "MPC>=2", "1<=MPC<2", "0<=MPC<1"),each=2), c("AF>0.0001", "AF<=0.0001"), sep="&"), each=10))%>% drop_na()


Burden.OR=Burden_MIRAGE_VS %>% select(Burden_OR) %>% pull()
Burden.pvalue=Burden_MIRAGE_VS %>% select(Burden_Pval) %>% pull()
No.var.case=Burden_MIRAGE_VS %>% select(Transmitted)%>% pull()
No.var.control=Burden_MIRAGE_VS %>% select(Untransmitted)%>% pull()
############### multiple correction for p values

MIRAGE.pvalue=Burden_MIRAGE_VS %>% select(MIRAGE.VS_Pval)%>% pull()
Burden.MIRAGE.pvalue=cbind(Burden.pvalue, MIRAGE.pvalue)
Burden.pvalue.adjust=p.adjust(Burden.pvalue, method="BH", n=length(Burden.pvalue))
MIRAGE.pvalue.adjust=p.adjust(MIRAGE.pvalue, method="BH", n=length(MIRAGE.pvalue))
Burden.MIRAGE.pvalue.adjust.pvalue=tibble(Burden.pvalue=Burden.pvalue, MIRAGE.pvalue=MIRAGE.pvalue, Burden.adjust.pvalue=Burden.pvalue.adjust, MIRAGE.adjust.pvalue=MIRAGE.pvalue.adjust)
burden.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(Burden.adjust.pvalue<0.05) %>% summarize(max.burden.pvalue=max(Burden.pvalue)) %>% pull()
mirage.pvalue.fdr.0.05.cutoff=Burden.MIRAGE.pvalue.adjust.pvalue%>% filter(MIRAGE.adjust.pvalue<0.05) %>% summarize(max.mirage.pvalue=max(MIRAGE.pvalue)) %>% pull()


##############

MIRAGE.risk.prop=Burden_MIRAGE_VS%>% select(MIRAGE.VS_eta)%>%pull()
Burden.MIRAGE.combine=tibble(Burden.OR=Burden.OR, Burden.pvalue=Burden.pvalue, MIRAGE.risk.prop=MIRAGE.risk.prop, MIRAGE.pvalue=MIRAGE.pvalue, No.var.case=No.var.case, No.var.control=No.var.control)
#Burden.MIRAGE.combine <- drop_na(Burden.MIRAGE.combine) # remove rows with NA

Burden.MIRAGE.combine.clear=Burden.MIRAGE.combine %>% add_column(cate.index=Burden_MIRAGE_VS$Group_Index)




################### save significant combined features with p values less than 0.05
significant.combined.features=Burden.MIRAGE.combine.clear %>% filter(Burden.pvalue<0.05 | MIRAGE.pvalue<0.05) %>% arrange(MIRAGE.pvalue)
#write.csv(significant.combined.features, file="../output/CombinedFeature/significant.combined.features.of.Exon.gene.set.AF.damaging.pvalue.lessthan.0.05.csv")
###################

y.max=max(-log(Burden.pvalue, base=10), -log(MIRAGE.pvalue, base=10))
num.test=138
################### this argument setting is too large, could be used in paper 
#g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index), size=15, stroke = 0.5))+
#  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
#  geom_point()+
  #xlim(c(0,4))+
 # ylim(c(0,y.max))+
#  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
#  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
 # xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
#  ggtitle("Burden")+
#  theme_classic()+
#   theme(legend.position = "none")+  # no legend 
#theme(plot.title = element_text(hjust = 0.5, size=35)) +# center the title 
#theme(legend.title=element_blank())+
#theme(legend.text=element_text(size=30))+
#theme(text = element_text(size=35))+
#guides(size=FALSE)+  # suppress size legend 
#guides(shape = guide_legend(override.aes = list(size = 10))) # symbol size in legend 
########################### this argument setting is in website 
  g1=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=Burden.OR, y=-log(Burden.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+
  geom_point()+
  #xlim(c(0,4))+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(burden.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  xlab("OR")+ylab(expression(paste("-log" [10], "(p value)")))+
  ggtitle("Burden")+
  theme_classic()+
   theme(legend.position = "none")+  # no legend 
theme(plot.title = element_text(hjust = 0.5)) +# center the title 
theme(legend.title=element_blank())+
#theme(legend.text=element_text(size=30))+
#theme(text = element_text(size=35))+
guides(size=FALSE)+   # suppress size legend 
guides(shape = guide_legend(override.aes = list(size = 5))) # symbol size in legend 
#  g1
##################################  

g2=ggplot(as_tibble(Burden.MIRAGE.combine.clear), aes(x=MIRAGE.risk.prop, y=-log(MIRAGE.pvalue, base=10), color=as.factor(cate.index), shape=as.factor(cate.index)))+
  scale_shape_manual(values=1:nlevels(as.factor(Burden.MIRAGE.combine.clear$cate.index)))+  # manually define shapes if more than 6 groups 
  geom_point()+
  xlab(expression(paste(hat(eta))))+
  ylab("")+
  ylim(c(0,y.max))+
  geom_hline(yintercept=-log(0.05/num.test, base=10), linetype="dashed", color = "red")+
  geom_hline(yintercept=-log(mirage.pvalue.fdr.0.05.cutoff, base=10), linetype="solid", color = "blue")+
  ggtitle("MIRAGE-VS")+
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5))+  # center the title 
  #labs(color='variant group')+   # define legend title 
  theme(legend.title=element_blank())#+
#  g2
#theme(text = element_text(size=35))

  #theme(legend.text=element_text(size=10))
  #theme(legend.position = c(0.8, 4),
  #        legend.direction = "vertical")
#+
  #theme(legend.position="bottom")
#pdf("C:/Users/han24/OneDrive - UWM/rare-var/Figure/Supp/Fig-Burden_MIRAGE-vs_CombinedFeature_Split.pdf")
ggarrange(g1, g2,nrow=1, common.legend = TRUE, legend="right") 
#dev.off()
```


## Old sample vs Cell data 


To compare old sample and cell data, look at LoF +AF<1E-4+10 gene set and how they compare with each other. 

Old sample: LoF: the union of  "stopgain", "frameshift substitution", "splicing", "stoploss" 

cell sample: LoF: protein truncating variants (PTV) with pLI>=0.5


### LoF variants in different gene set 



```{r, echo=F, cache=T, warning=F, message=F}
### old sample
########## read into data
All.Anno.Data=as_tibble(read.table("../../AnnotatedTrans.txt", header=T))
N1=4315; N0=4315
All.Anno.Data[All.Anno.Data =="."] <- NA
All.Anno.Data$ExacAF[is.na(All.Anno.Data$ExacAF)]=0 # set AF of NA to zero 
Anno.Data=All.Anno.Data[which(All.Anno.Data$ExacAF<0.05 & All.Anno.Data$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonymous SNV
Anno.Data=All.Anno.Data[which(All.Anno.Data$Annotation!="synonymous SNV"),] # use AF cutoff and exclude synonumous SNV
```

```{r, echo=F, warning=F, message=F}
######### obtain LOF variants with AF<1-E4+10 gene set in old sample
LoF.def=c("stopgain", "frameshift substitution", "splicing", "stoploss")
old_variant_set=Anno.Data%>%filter(Annotation %in% LoF.def & ExacAF<0.0001)%>%select(ID, Gene, No.case, No.contr) %>% drop_na() # LoF variant with AF<1E-4 in old sample
transmitted_old_sample=numeric()
nontransmitted_old_sample=numeric()
for ( i in 1:10)
 {
  Feature_Data=old_variant_set%>%filter(Gene %in% Gene_Set_List[[i]])
  if (nrow(Feature_Data)>0)
  { 
    transmitted_old_sample[i]=Feature_Data%>%select(No.case)%>%sum()
    nontransmitted_old_sample[i]=Feature_Data%>%select(No.contr)%>%sum()
  }
  }

LoF_old_sample=tibble(Gene=rep(Gene_Set,2), status=rep(c("transmitted", "nontransmitted"), each=10), variant_count=c(transmitted_old_sample, nontransmitted_old_sample))

g_old_sample=ggplot(LoF_old_sample, aes(x = Gene, y = variant_count, fill = status)) +
  geom_col(position = "dodge")+
  theme_classic()+
  ylim(c(0,1000))+
  ylab("Variant count")+xlab("Gene set")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))+
theme(text = element_text(size=10))+
ggtitle("old sample")+
  theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title


```

```{r, echo=F, warning=F, message=F}
######### obtain LOF variants with AF<1-E4+10 gene set in cell data
cell_variant_set=cell_trans_data_AF %>% filter(pLI>=0.5) %>% filter(PTV==T)%>% filter(ExacAF<0.0001) %>% select(Variant, GENE_NAME, Transmitted_proband, Untransmitted_proband)  # LoF variant in cell data with AF<1-10-4
transmitted_cell_sample=numeric()
nontransmitted_cell_sample=numeric()
for ( i in 1:10)
 {
  Feature_Data=cell_variant_set%>%filter(GENE_NAME %in% Gene_Set_List[[i]])
  if (nrow(Feature_Data)>0)
  { 
    transmitted_cell_sample[i]=Feature_Data%>%select(Transmitted_proband)%>%sum()
    nontransmitted_cell_sample[i]=Feature_Data%>%select(Untransmitted_proband)%>%sum()
  }
  }

LoF_cell_sample=tibble(Gene=rep(Gene_Set,2), status=rep(c("transmitted", "nontransmitted"), each=10), variant_count=c(transmitted_cell_sample, nontransmitted_cell_sample))

g_cell_sample=ggplot(LoF_cell_sample, aes(x = Gene, y = variant_count, fill = status)) +
  geom_col(position = position_dodge())+
  #geom_bar(stat="identity")+
  theme_classic()+
  ylim(c(0,1000))+
  ylab("Variant count")+xlab("Gene set")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, size=10))+
theme(text = element_text(size=10))+
ggtitle("cell sample")+
  theme(plot.title = element_text(hjust = 0.5, size=10)) # center the title

ggarrange( g_old_sample, g_cell_sample, common.legend = TRUE, legend="right") 
```


### LoF variants in different FMRP genes 

LoF variants in 50 FMRP genes. 

```{r, echo=F}
### display LoF variant distribution between two samples for specific gene set 
######################################################
## Gene_Set[7]: FMRP gene 
variant_gene_old_sample=old_variant_set %>% filter(Gene %in% Gene_Set_List[[7]])%>% rename(GENE_NAME=Gene) 
variant_gene_cell_sample=cell_variant_set %>% filter(GENE_NAME %in% Gene_Set_List[[7]])
variant_gene_old_new_sample_combine=full_join(variant_gene_old_sample, variant_gene_cell_sample, by="GENE_NAME") %>% select(GENE_NAME, No.case, No.contr, Transmitted_proband, Untransmitted_proband)
####### fill NA with 0
variant_gene_old_new_sample_combine$No.case=variant_gene_old_new_sample_combine$No.case%>%replace_na(0)
variant_gene_old_new_sample_combine$No.contr=variant_gene_old_new_sample_combine$No.contr%>%replace_na(0)
variant_gene_old_new_sample_combine$Transmitted_proband=variant_gene_old_new_sample_combine$Transmitted_proband%>%replace_na(0)
variant_gene_old_new_sample_combine$Untransmitted_proband=variant_gene_old_new_sample_combine$Untransmitted_proband%>%replace_na(0)
########### count variants for every gene 
unique_gene=unique(variant_gene_old_new_sample_combine$GENE_NAME)[1:50]
unique_gene_case=numeric()
unique_gene_control=numeric()
unique_gene_transmitted_proband=numeric()
unique_gene_untransmitted_proband=numeric()
for (i in 1:length(unique_gene))
{
 unique_gene_case[i]=variant_gene_old_new_sample_combine %>% filter(GENE_NAME %in% unique_gene[i]) %>% select(No.case)%>%sum()
 unique_gene_control[i]=variant_gene_old_new_sample_combine %>% filter(GENE_NAME %in% unique_gene[i]) %>% select(No.contr)%>%sum()
unique_gene_transmitted_proband[i]=variant_gene_old_new_sample_combine %>% filter(GENE_NAME %in% unique_gene[i]) %>% select(Transmitted_proband)%>%sum()  
unique_gene_untransmitted_proband[i]=variant_gene_old_new_sample_combine %>% filter(GENE_NAME %in% unique_gene[i]) %>% select(Untransmitted_proband)%>%sum()
}
unique_gene_variant_count=tibble(Gene=unique_gene, No.case=unique_gene_case, No.control=unique_gene_control, Transmitted_proband=unique_gene_transmitted_proband, Untransmitted_proband=unique_gene_untransmitted_proband)
###########################
transmitted_variant=tibble(Gene=c(as.character(unique_gene), as.character(unique_gene)), count=c(unique_gene_variant_count$No.case, unique_gene_variant_count$Transmitted_proband), sample=c(rep("old", nrow(unique_gene_variant_count)), rep("cell", nrow(unique_gene_variant_count))))


g1=ggplot(transmitted_variant, aes(x=Gene, y=count, shape=sample, color=sample))+ 
 geom_point()+
  ylab("Transmitted variant counts")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=7))
g1

untransmitted_variant=tibble(Gene=c(as.character(unique_gene), as.character(unique_gene)), count=c(unique_gene_variant_count$No.control, unique_gene_variant_count$Untransmitted_proband), sample=c(rep("old", nrow(unique_gene_variant_count)), rep("cell", nrow(unique_gene_variant_count))))


g2=ggplot(untransmitted_variant, aes(x=Gene, y=count, shape=sample, color=sample))+ 
 geom_point()+
  ylab("Untransmitted variant counts")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=7))
g2

```

